<<sol.comp, echo=FALSE, cache=TRUE, results=hide>>=

what <- c("rname", "strand", "pos", "qwidth", "seq")
param <- ScanBamParam(what = what)

bam <- scanBam("/home/ele/Data/nlaIII-tags/UW08F_tags.bam", param=param)

nTags <- length(bam[[1]]$seq)

counts <- as.data.frame(table(bam[[1]]$rname))
names(counts) <- c("contig", "solexa.tags")

nMapped <- sum(counts$solexa.tags)
nUniq <- length(unique(bam[[1]]$seq))

nUMapped <- nTags-nMapped

lst <- lapply(names(bam[[1]]), function(elt) {
  do.call(c, unname(lapply(bam, "[[", elt)))
})
names(lst) <- names(bam[[1]])
df <- do.call("DataFrame", lst)

nUniqUMapped <- length(unique(df[is.na(df$rname), "seq"]))

## tag.seq <- sapply(as.list(bam[[1]]$seq), toString)

counts <- merge(counts, countsTable, by.x="contig", by.y="row.names")
counts$all.reads <- apply(countsTable, 1, sum)

#better use
# plotmatrix
## exp_plot <- ggplot(counts, aes(y=solexa.tags+1, x=all.reads+1)) +
##   geom_point(alpha=0.3) + geom_smooth(method="lm") + scale_x_log10() +
##   scale_y_log10()

contig.df <- merge(contig.df, counts)

cor.count.tab <- round(cor(contig.df[,
                                     c("solexa.tags", "E1", "E2", "L2", "M", "T1", "T2", "all.reads")],
                           method="spearman"),4)


cor.count.tab.MN <- round(cor(contig.df[contig.df$category%in%"MN",
                                        c("solexa.tags", "E1", "E2", "L2",  "M", "T1", "T2", "all.reads")],
                              method="spearman"),4)


cor.count.tab.MN.AC <- round(cor(contig.df[contig.df$AcMN,
                                           c("solexa.tags", "E1", "E2", "L2",  "M", "T1", "T2", "all.reads")],
                                 method="spearman"),4)



de.eel <- contig.df[contig.df$contig%in%sig.eel.contigs,
                    c("contig", "Ac",
                      "E1", "E2", "L2", "M", "T1", "T2",
                      "Bm.annot", "Bm.bit", "nr.uniprot.annot", "nr.bit")]

de.mf <- contig.df[contig.df$contig%in%sig.mf.contigs,
                    c("contig", "Ac",
                      "E1", "E2", "L2", "M", "T1", "T2",
                      "Bm.annot", "Bm.bit", "nr.uniprot.annot", "nr.bit")]

## tubulin and actin are there other contigs with same annot not differing some not
tubulin <- contig.df[grepl("Tubulin ", contig.df$Bm.annot) |
                     grepl("Tubulin ", contig.df$nr.uniprot.annot),
                    c("contig", "Ac",
                      "E1", "E2", "L2", "M", "T1", "T2",
                      "Bm.annot", "Bm.bit", "nr.uniprot.annot", "nr.bit")]
tub.total <- colSums(tubulin[, c("E1", "E2", "L2", "M", "T1", "T2")])

actin <- contig.df[grepl("^Actin", contig.df$Bm.annot) |
                     grepl("^Actin ", contig.df$nr.uniprot.annot),
                    c("contig", "Ac",
                      "E1", "E2", "L2", "M", "T1", "T2",
                      "Bm.annot", "Bm.bit", "nr.uniprot.annot", "nr.bit")]
act.total <- colSums(actin[, c("E1", "E2", "L2", "M", "T1", "T2")])


cyto <- contig.df[grepl("Cytochrome c oxidase subunit", contig.df$Bm.annot) |
                     grepl("Cytochrome c oxidase subunit", contig.df$nr.uniprot.annot),
                    c("contig", "Ac",
                      "E1", "E2", "L2", "M", "T1", "T2",
                      "Bm.annot", "Bm.bit", "nr.uniprot.annot", "nr.bit")]

cyto.total <- colSums(cyto[, c("E1", "E2", "L2", "M", "T1", "T2")])

@ 
