%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}


\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\usepackage{/usr/share/R/texmf/tex/latex/Sweave}
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of the swimbladder-nematode
    Anguillicola crassus: Resources for an alien parasite}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel G Heitlinger\correspondingauthor$^{1,2}$%
       \email{Emanuel G Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Horst Taraschewski$^1$%
       \email{Horst Taraschewski- dc20@rz.uni-karlsruhe.de}%
       and Mark Blaxter$^2$%
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute 1, University of Karlsruhe,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} 
  \paragraph*{Results:} 
  \paragraph*{Conclusions:}
  Yeh!
  
\end{abstract}


      \ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

\textit{Anguillicola crassus} Kuwahara, Niimi et Itagaki, 1974
\cite{kuwahara_Niimi_Itagaki_1974} is a nematode feeding on blood in
the swimbladder of freshwater eels of the genus
\textit{Anguilla}. Originally endemic to East-Asian populations of the
Japanese Eel (\textit{Anguilla japonica}), \textit{A. crassus} has
attracted interest due to recent anthropogenic expansion of its
geographic- and host-range to Europe and the European eel
(\textit{Anguilla anguilla}). Soon after it had been recorded for the
first time in 1982 in North-West Germany \cite{fischer_teichwirt}, to
where it was most likely introduced by live-eel trade
\cite{koops_anguillicola-infestations_1989, koie_swimbladder_1991},
\textit{A. crassus} rapidly spread throughout populations of its newly
acquired host (for a review see \cite{kirk_impact_2003}). At the
present day it is found in all but the northernmost population of the
European eel in Iceland \cite{kristmundsson_parasite_2007}.

The impact of \textit{A. crassus} on the European eel has been a major
focus of research during the past decades. High prevalence of the
parasite of above 70\% (e.g. \cite{wrtz_distribution_1998}), as well
as high intesities of infections were reported, throughout the newly
colonized area \cite{lefebvre_anguillicolosis:_2004}.  Based on a
broad base of work on its epidemiology \textit{A. crassus} can be
regarded as a model for parasite introduction and spread
\cite{taraschewski_hosts_2007}.

As in the natural host in Asia prevalence and mean intesity of
infection are lower than in Europe \cite{mnderle_occurrence_2006},
high epidemiological parameters were attributed to the inadequate
immune-response of the European Eel
\cite{knopf_swimbladder_2006}. While the Japanese eel is capable of
killing larvae of the parasite after vaccination
\cite{knopf_vaccination_2008} or under high infection pressure
\cite{heitlinger_massive_2009}, only pathological effects such a
thikening of the swimbladder wall \cite{wrtz_histopathological_2000}
have been found in the European eel. Interestingly the differences in
the two host also affect the size and life-history of the worm: In
European eels the nematodes are bigger and develop and reproduce
faster \cite{knopf_differences_2004}.

The genus \textit{Anguillicola} holds a phylogenetic position basal to
the Spirurina (clade III \textit{sensu}
\cite{blaxter_molecular_1998}), one of 5 major clades of nematodes
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
exclusively exhibit a parasitic lifestyle and comprise important human
pathogens as well as prominent parasites of livestock (e.g. the
Filaroidea and Ascarididae). This phylogenetic position makes the
Anguillicoloidae an interesting system in the endeavor to understand
the emergence of parasitism in Spirurina and as an ``outgroup'' for
functional studies of parasitism in this clade. Some functionally
interesting genes in this respect are thought to be under diversifying
selection in an arms-race between host and
parasite\cite{zang_serine_2001}.

Recent advances in sequencing technology (often termed Next Generation
Sequencing; NGS), provide the opportunity for rapid and cost-effective
generation of genome-scale data. The longer read length of
454-sequencing \cite{pmid16056220} compared to other NGS technologies,
allows \textit{de novo} assembly of Expressed Sequence Tags (ESTs) in
organisms lacking previouse genomic or transcriptomic data (for a
comprehensive list of studies using this approach before Oct 2010 see
\cite{pmid20950480}). Such transcriptomic datasetes are still less
expensive than genomic data-sets in terms sequencing costs and
analytical needs.

The difference of the immune attack on \textit{A. crassus} in the two
different hosts provides an opportunity to investigate the parasite's
response to different ``immune environments'' on a transcriptomic
basis.

In this study we report assembly of raw reads from cDNA libraries of
L2 larvae, four female adults and one male adult of
\textit{A. crassus} into contigs (continuous sequence, representing a
mRNA-transcript) of a reference transcriptome. We describe screening
for xenobiotics and host-contamination, prediction of protein
sequences and functional annotation of these contigs. Furthermore
combining ORF prediction and identification of SNPs allowed inference
of diversifying selection.

% Aditionally we analysed gene expression, by counting of raw
% reads from two European worms and two Taiwanese worms mapped to the
% assembled refference. We tested both positively selected and
% differentially expressed genes for overrepresentation of GO-terms and
% KEGG-parthways, with the aim to test for an enrichment of terms and
% parthways involved in host-parasite interaction.

As shorter read-length but higher throughput of the Illumina-Solexa
platform provides superior means for gene expression analyis
\cite{pmid21627854}, the transcriptome assembly presented here is
planned to be used as a reference in a future study using
gene-expression tags (SuperSAGE \cite{pmid20967605}) or RNA-seq
\cite{pmid19015660}. To test the suitability of the presented
transcriptome we mapped a pilot-set of expression-tags. This analysis
is used as an additional estimate for the correctness of the hypotesis
that each contig in the assembly constitutes.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}


 \subsection*{Sequencing, trimming and pre-assembly screening}


In total 756363 raw sequencing reads were
trimmed and quality screened. After this trimming
585949 sequences were regarded valid, table 1
gives an overview of trimming statistics for different
libraries. Notably the variability in the number of sequences excluded
due to low quality and short quality-sequence in \texttt{seqclean} was
high indicating different properties of read-quality in the different
libraries. The 585949 reads regarded valid
after this trimming were further screened for rRNA and
host-contamination before assembly.





In total 232894 reads were discarded prior to assembly,
202823 due to \texttt{blast} hits (e-value cut of 1e-5,
hit longer than 80\% of the read-length and identity bigger 95\%) to
rRNA, 30071 to host-databases (same thresholds). Figure
(rRNA\_plots.png) and table 2 give an overview of this pre-assembly
rRNA and host-screening in different libraries. During annotoation of
the rRNA-database we constructed for \textit{A. crassus} (assembled
from reads screened against general rRNA-databases) we found some
contigs with high similarity 18S sequences from cercozoan and other
flaggelate protists over the majority of their length. Table 3 gives
an overview of the hits in NCBI-nt for these 11 contigs assembled from
rRNA sequences. The origin of the rRNA-reads assembled in these
contigs (in screening-database assembly, not the main assembly, where
they were screened out) showed that they originated nearly exclusively
from library L2.
 
Containing the highest number of raw reads, library E1 showed the
highest amount of low-quality and short sequences in trimming and a
high amount of rRNA and host-contamination, making it the library with
the lowest number of sequences used for assembly.

\subsection*{Assembly}


For assembly we used a method proposed by \cite{pmid20950480}: We
combined two assemblies merging their contigs. For a detailed
description of the assembly method see the supplementary methods
file. In total 13851 contigs were regarded good quality, as they
were supported by two assembly algorithms (in the remainder of the
manuscript ``the assembly'' refers to this ``second-order''-assembly
and quality contigs refers to the good-quality contig-category from
this). In additon an expanded dataset consisting of the afore
mentioned assembly plus the (3745) contigs only supported by
one of the ``first-order'' assemblies and (22591) singletons
was obtained (this is reffered to as the ``fullest assembly'', the
sequences in this dataset are referred to as Tentative Unique Genes;
TUGs). In the methods supplementary file we show that these datasets
are considered the most parsimonious (having the smallest size) for
their quality (covering the largest amount of sequence in reference
transcritomes). In the high quality assembly parsimony and low
redundancy is prioritized, in the full assembly completeness.


The mean per-base coverage of the TUGs was 12.84 and the mean
per-base unique coverage was 4.62. The mean per-base coverage
of the quality contigs was lower with 10.98, but the unique
coverage higher with 6.84. This indicates a higher amount
of redundancy in the full assebmly set compared to the good-quality
set. The distribution of mean per base coverages for single contigs or
TUGs is given in Figure (coverage\_per\_contig.png)

Figure (coverage\_plots) shows the relationship between, read-number
in the assemblies and coverage.

\subsection*{Marking possible host- and xenobiotic contamination}



To elucidate the taxonimic origin of the assembled sequences, we used
extensive \texttt{blast}-queries.

\subsubsection*{Host-contamination approach}

In this evaluation we labeled all sequences hitting (e-value
cut-off of 1e-5) the same databases used for screening prior to
assembly (eel-mRNA, eelrRNA, \textit{A. crassus}rRNA) with the
respective database-hit, if the sequence did not produce a better hit
(based on bit-score comparison) against the nucleotide version of
nempep4 \cite{pmid21550347}.  

To make this evaluation more specific we only sequences hitting whith
the over 50\% of their length with an identity >70\% to label the
TUGs as ``contamination'' accordig to the respective database-hit.

Two kinds sequences were labeled ``valid'' \textit{A. crassus}
sequences: Valid due to evidence of nematode origin (hitting nempep4),
and valid due to absence of evidence of host-origin.

Table 3 gives the numbers of TUGs labeled as possible host or rRNA
contamination. The 38371 TUGs regarded to be of
\textit{A. crassus}-origin give a positive indication of the origin of
sequence data. However TUGs hitting \textit{A. crassus}-rRNA still
have a high coverage, indicating a noticeable loss of raw-read data
(e.g. used in mapping) to additional rRNA-coverage. In contrast
TUGs labeled to be of host-origin had a slightly lower coverage than
TUGs labeled valid, indicating an only moderate amount of raw-data
being ``lost'' to this kind of off-target data.

\subsubsection*{Off target data approach}

To elucidate possible xenobiotic sequences (comprising unexpected
off-target data) we searched our complete dataset against both NCBI-nt
and NCBI-nr. Figure (phylum\_plots) gives an overview of the number of
top-hits to different phyla sorted by kingdom.

For every study of the biology of \textit{A. crassus} based on the
presented dataset we advise to use the taxonomic evaluation and
host-contamination screening at latest at the stage of interpretation
of results to avoid misleading conclusions based on differences in
off target data.

For example a study excluding putative host-derived sequence prior to
analysis, would have 36166 TUGs or 12796
quality-contigs left for analysis, based on choosing only sequence not
labeled ``Chordata'' (from nt- and nr-searches) and labeled ``valid''
or ``valid-nempep'' (from the evaluation of host-contamination). A
second analysis aiming to exclude all data suspicious to be not of
\textit{A. crassus}-origin would choose 31893 TUGs or
10999 quality-contigs, by using sequence labeled
``Nematoda'' or ``no hit'' (from nt- and nr-searches) and ``valid'' or
``valid-nempep''. We use such restricted dataset like in the second
example in later analysis to verify that results are not induced by a
possible contamination backround.

\subsection*{Evolutionary conservation}


Based on taxonomically classified \texttt{blast}-results described
above we also determined evolutionary at 2 thresholds (bitscore of 50
and 80; roughly equivalent with an e-value of e-6 and e-10
respectively) at three taxonomic levels: We categorized a sequence if
it had a hit to ``nematoda'' and a second phylum as conserved across
the phylum level and to ``metazoa'' and a second kindom as conserved
across the kingdom level. As a third taxonomic level we defined the
Spirurina (Clade III \textit{sensu}
\cite{blaxter_molecular_1998}). Conversely from these data
``conserved'', ``novel in metazoa'', ``novel in nematoda'' and ``novel
in clade III'' categories could be computed. See table 5 for a summary
of these categories for the two thresholds.


\subsection*{Protein prediction}

%% read P4EST output processed with
%% coordinates_from_p4e.pl


For in total
39625 TUGs
a protein was predicted. A summary of protein prediction using the
different methods implemented in prot4EST is given in table 6.
 
The full open reading frame was obtained for in total
414 TUGs, while for
3304 the 5' end and for
10178 the 3’ end was complete.  A
predicted ORF was defined at complete if it was starting with an
``atg''-start-codon, having at least 3 non-coding bases 5' of the
start-codon and having a ``tag'' (2122
observations), ``taa'' (4989) or ``tga''
(3026) stop-codon 3' of the ORF.

In addition to the protein-sequence used for annotation we obtained a
corrected nucleotide-sequence using the nucleotide equivalent of the
protein-prediction. This was necessary as \texttt{prot4Est} joins
high-scoring pairs from \texttt{blast}-searces introducing gaps (or
unknown fill-bases respectively) relative to the raw nucleotide
sequence if needed to obtain the correct frame for the ORF. In
15988 TUGs the corrected sequence with the
imputed ORF was slightly changed compared to the raw sequence.
 
\subsection*{SNP analysis}
\label{sec:snp-analysis}

%% SNP calling from VARSCAN output






In order to improve SNP-calling we screened a set of 13518
candidate-SNPs predicted on in total 15718866 bases
(1412806 bases with a minimum coverage of more then 8-fold
available for SNP-calling). We first excluded SNPs predicted to be
present as more than 2 alleles and to undetermined (``N'') base in the
reference retaining 13042 SNPs. The ratio of transitions
(8663) to transversion (4379) in
this set was 1.98 (ti/tv).

To further screen those SNPs we utilized observations made in
combination with protein prediction, i.e. with the inferred
open-reading frame. As noted above SNPs at unknown fill-bases had to
be disregarded, but the imputation of a continuous corrected ORF had
the advantage of making the coordinates for the ORF straight forward
to use.

8309 of the raw SNPs were predicted to be inside
an ORF, 2713 at a first positon in a codon,
2172 on a second base and
3424 on a third base in a codon.

As expected ti/tv inside ORFs was with
2.43 higher than outside ORFs
with 1.42.

To asses the effect of a mutation at a given base we used an idea of
Mark Blaxter (unpublished, some information at \cite{Marks_ontology}),
to classify every base as to its ``response'' to mutation. We used
this classification to compute the number of synonymous and
non-synonymous sites in a given TUG. In total 8847926
non-synonymous and 2739223 sites were found in ORFs, of
these 858749.67 non-synonymous sites and
259935.33 synonymous sites were covered at least 8-fold,
and thus available for SNP-calling.

For the raw set of candidate-SNPs the ratio of synonymous
polymorphisms per synonymous site to non-synonymous polymorphisms per
non-synonymous site (dn/ds) was 0.45.

To improve SNP-calling with respect to the well-known homopolymer
issues in 454-sequencing\cite{pmid21685085}, we analysed the effect of
exclusion of SNPs in, or close to, homopolymer regions. We observed
changes in ti/tv and in dn/ds when SNPs were discareded due to
different sizes thresholds for homoploymer-runs and different
proximity thresholds (see figure snp\_ex\_parameter\_plots.png). Based
on this we decided to exclude SNPs with a homopolymer-run as long as
or longer than 4 bases inside a window of 11 bases (5 to bases to the
right, 5 to the left) around the SNP.

After these screening steps based on a sequence-features we
investigated the effect of data volume and mapping on the remaining
candidate SNPs. Based on a slight overrepresentation of SNPs on first-
and second-positions in the ORF and on a little bigger than average
excess of non-synonymous polymorphisms in SNPs with a percentage of
the minority allele smaller than 7\% (see figure
snp\_pos\_eff\_plots.png a and b) we examined exclusion of SNPs at
this percentage threshold. Evaluation of an exclusion of SNPs based on
coverage at the SNP-site, did not seem necessary (see figure
snp\_pos\_eff\_plots.png c and d).

Table 7 gives on overview of how the basic SNP-statistics described
above changed with screening of the candidate SNPs. The change of
ti/tv back to lower values, when SNPs were screened based on a 7\%
coverage threshold, left the benefit of this screening-step
questionable.

However, calculating dn/ds on a per contig base, the screening based
on percentage threshold of the minority-allele showed it's benefits:
figure dn\_ds\_scales.png b shows how for the unscreened SNP-set dn/ds
scaled with the coverage of a contig. This correlation was not longer
present if the percentage screening was used (see figure
dn\_ds\_scales.png d, the linear model had a slightly negative
non-significant slope). For both the screened and unscreened sets of
SNPs there was a significant slope for the number of SNPs in a contig
predicting dn/ds (see figure dn\_ds\_scales.png a and c).

Figure dens\_dn\_ds.png gives the distribution of per-contig dn/ds for
the fully screened set of SNPs. The final numbers of SNPs per
kilo-base was 4.44,
7.87 synonymous SNPs per 1000 synonymous
bases and 2.43 non-synonymous SNPs per
1000 non-synonymous bases. For a total of
980 TUGs
(858 high
  quality contigs) a value for dn/ds could be obtained, because at
  least one synonymous SNP was found.

  It should be noted that the overall dn/ds value of
  0.32 is different from the mean dn/ds value of contigs
  0.23, as in the first the
  complete length of the sequence (also from TUGs without synonymous
  SNP predicted) can be included.

\subsection*{Annotation}

%%% annotation.Rnw --- 

%% Author: emanuelheitlinger@gmail.com



\begin{Schunk}
\begin{Soutput}
[1] 1
\end{Soutput}
\end{Schunk}



We obtained annotations with GO-terms for 9569, with EC-numbers
for 3741 and with KEGG-pathways for 5990 TUGs using
the \texttt{Blast}-based program annot8r.

Figure (annotataionVenn.tiff) gives an overview of the
10274 TUGs annotations were found for with
different methods.


We compared GO-annotations for high-level GO-slim terms to the
annotations obtained the same way for all \textit{B. malayi}-proteins:
Our transcriptome for \textit{A. crassus} shows a remarkably similar
distribution of GO-terms to the full set of annotations for the genome
of the related nematode (see figure go\_bm\_com.png).

In addition we inferred presence of signal peptide cleavage sites in
the predicted protein sequence: \texttt{SignalP} \cite{pmid21959131}
predicted 1060 signal peptide cleavage sites and
83 signal peptides with a transmembrane signature.

\subsection*{Consolidated results}







The following set of analyses, represetnt enrichment analysis where
the ``gene-universe'' to test against has to be defined. Possible
off-target sequences were excluded prior to analysis. Testing was
performed against a universe of only TUGs we were sure to be of
\textit{A. crassus}-origin.

\subsubsection*{Signal-positives have higher dn/ds}

TUGs predicted to contain signal peptide cleavage sites by the
neural-networks method in \texttt{SignalP} have higher dn/ds values
than TUGs without signal peptide cleavage sites (p =
0.055; one sided U-test; see also
figure sigp\_dn\_ds.png).

\subsubsection*{Enrichment of GO-categories in high dn/ds}

We defined TUGs with a dn.ds higher than 0.5 as positively selected
and tested each node-term in the ontology for a over-representation in
this set. Tables X give an overview of enriched terms for different
parts of the ontology.

The terms ``amino acid transmembrane transporter activity'' and
``peptidase activity'' were among the overrepresented terms for
``molecular function''. Underrepresented were on the other hand terms
associated with ribosomal proteins and transcription (visible in
``molecular function'' and ``cellular compartment'') .

TODO: BP and CC over-representation. 

To make sure these inferences were not biased by redundancy in the
fullest data-set we repeated the analysis using only the good-quality
contigs as gene-universe. The results showed to be consistent with
those obtained for the fullest data-set (data not shown).

\subsubsection*{Novel in cladeIII have elevated dn/ds}

Figure conservation\_dn\_ds.png shows differences in dn/ds across the
categories defined for evolutionary conservation. At a bitscore
threshold of 80 sequences novel in clade III had a significantly
higher dn/ds than other sequences (p =
0.038; two sided U-test). At a
bitscore thershold of 50 results were non-significant but showed the
same trend.

\subsubsection*{Sequences novel in nematodes are enriched for
  Signal-positives}

Figure signal\_novel.png gives the proportions of
\texttt{SignalP}-predictions for each category of evolutionary
conservation. Generally - across bit-score thresholds - sequences
novel in nematodes contained the highest proportion of
signal-positives.

\subsection*{Differential expression}
                

Using methods developed for sequencing data, we analyzed
gene-expression inferred from mapping. Of the 341285 reads
mapping to the fullest assembly 264440 mapped uniquely
(with their best hit) and were counted on a per library base.

\subsection*{Comparison with tag-sequencing pilot data-set}


5096312 of 6201930 (559824 unique) NlaIII-tags
mapped to the fullest assembly. Only 1105618
(317782 unique) tags did not map to any sequence in the
fullest assembly.

Table 9a gives correlations coefficients between tag-counts and
454-libraries. Correlations-coefficients between 454-libraries were
generally low, indicating a high proportion of noise or biological
differences between samples. Correlation between expression-tags and
454-read counts were even lower. However when only analyzing counts to
good-quality contigs, correlation coefficients improved both between
libraries and between 454-libraries and solexa-tags (see table 9b). No
further improvements were made, when counts were limited to contigs
surely \textit{A. crassus} (see table 9c). Correlations between
library T2 and other 454-libraries, as well as with solexa-tag counts
were lower than between other libraries.

To gain power in statistical analysis we limited the set of
gene-objects analyzed for differential expression to the good-quality
contigs.

\subsubsection*{Differential expression between male and female worms}

Despite the lack of replicates for male worms we were able to identify
25 sequences being significantly
over-expressed in male worms. In fact all these TUGs were nearly
exclusively expressed in males. 

\subsubsection*{Differential expression between adults and L2-larvae}

For the L2-library we changed our approach and used gene-expression
analysis rather to highlight the off-target data in this library. For
this reason we used counts for the fullest assembly.

479 sequences being expressed exclusively
in L2 library were strongly enriched in sequences being labeled as
possible off-target data in taxonomic classification. From these
sequences only 57 had best hits to metazoa and only
6 to nematoda.

\subsubsection*{Differential expression between worms from the European
  and Japanese eel}

None of the TUGs in the present evaluation showed significant
differential expression between worms from the European and Japanese
Eel. Diagnostic plots provided by \texttt{DESeq} made clear, that both
depth of sequencing and number of replicates have to be higher
contrasting these conditions.

However, comparing expression-analysis on the full data-set to analysis
limited to the high quality of reliable \textit{A. crassus}-contigs it
was clear that the quality-data-set reduces within-condition variance
and results were closer to significance: The lowest adjusted p-values
for the cleaned data-set were around 0.4, while on the full data-set
only adjusted p-values above 0.8 could be obtained.

\subsection*{Secreted immunomodulatory molecules}

We further highlight a list of genes with similarities to known
immunumodulators. Should I???


\section*{Discussion}

We are providing transcriptome-data for the parasite
\textit{A. crassus}, enabling a broad spectrum of molecular research
on this ecologically and economically important species.

We emphasize the importance of screening for xenobiotics. We consider
this aspect important in any deep transcriptome project. First the
depth of sequencing is leading to the generation of large amounts of
off-target data from a ``metatranscriptomic community'' associated with a
target organism. Second due to the abundance of laboratory
contamination and the possibility of cross-contamination if libraries
are sequenced only on a subset of a picotiter-plate (i.e. without the
use of barcodes distinguishing between samples \cite{pmid20137071})
non-biological contamination can be introduced. 
However, in the context of a parasite (or an infected host) the
screening for off-target data and contamination becomes even more
important: Correct inference of biological origin for a given contig
constitutes a prerequisite for most downstream analysis or the
interpretation of results.

Cross-contamination from different compartments of a picolitre-plate
was ruled out by our sequence provider, using Multiplex Indexes (MID)
for one library and similarity searches to neighboring lanes for the
other libraries.

For the remaining off-target and contamination problem we archived
separation of sequences in two steps, one before assembly, one
afterward. Both screening-steps had to rely solely on sequence
comparison. The screening-step before assembly has to employ lower
stringency as sequence comparisons on sequence as short as reads are
less informative than on longer contig-sequence. In our case of
\textit{A. crassus}, neither of the two host species has genomic data
available for use in similarity searches. A publicly available
transcriptome-data-set for European eel \cite{pmid21080939} in addition
to a unpublished data-set for the same species was augmented with a
data-set generated from the Japanese eel sequenced for the purpose of
screening \textit{A. crassus}-sequences in the present project. The
pre-assembly screening had the rationale of facilitating the assembly
process reducing the amount of divergent sequence from two
host-species and the amount of extensively covered rRNA sequence. In
our sequencing we were not able to reproducibly alleviate the rRNA
coverage. This has probably been due to the fact that extraction of
total-RNA from worms filled with host blood resulted in low amounts of
starting material, and amplification using standard kits did not allow
to reproducibly alleviate rRNA abundance. As the same problems
existed in preparation of liver tissue of the host species it seems
likely that the blood of eels contains substances limiting the success
of specific amplification protocols. In fact it is known that
compounds like hemoglobin can inhibit PCR reactions \cite{pmid9327537}
and reverse transcription \cite{pmid16109794}.

Although raw reads with rRNA hits were screened out prior to assembly,
it was still possible to gain insights from these off-target data, as
we assembled and annotated screening databases. Some of the rRNA data
especially from the L2 library showed high similarity to flagellate
eukaryotes. It could be possibly derived from an unknown protist
living in the swimbladder of eels (possibly as a commensal of
\textit{A. crassus}), from where the L2 larvae for RNA-preparation were
washed out. This seems worth further investigation, especially as it
has been controversial whether encapsulated objects in the swimbladder
of eels could be attributed solely to \textit{A. crassus}
\cite{heitlinger_massive_2009} or to opportunist coinfections. 

A second examination of sequence origin was obtained after
assembly. It can employ higher stringency do to the longer sequence
length in an assembled data-set and because we did not discard any
off-target data, but marked TUGs for careful selection prior to
special analysis or differential interpretation afterward. Such
attempts to employ taxonomic screening were used before in a
transcriptome projects for the garter snake \cite{pmid21138572} and a
study on lake sturgeon even evaluated horizontal gene-transfer, when
xenobiotic sequences were found \cite{pmid20386959} (with a negative
result). A study describing a custom pipeline for
transcriptome-assembly from pyrosequencing reads \cite{pmid20034392}
suggested the use of \texttt{EST3} \cite{pmid17218127}, to infer
sequence-origin based on nucleotide frequency. However, we were not
able to use this approach successfully, probably due to the fact that
xenobiotic sequence in our data-set stems from multiple sources with
different gc-content and codon-usage.

Data from our L2 larvae library showed its anomaly later in
gene-expression analysis, when off-target xenobiotic data was found to
be responsible for the differences to other libraries. This makes this
off-target data an interesting starting point for future investigation
of the species community living in infected swimbladders of eels.

Compared to other NGS transcriptome sequencing projects, our assembly
approach generated a smaller number of contigs. Projects using the
\texttt{mira} assembler often report substantially more contigs for
data-sets of similar size (see e.g. \cite{pmid21364769}), comparable
to the \texttt{mira} sub-assembly in our approach. We demonstrated that
our assembly approach generated a set less redundant and more complete
sequences compared to the two ``out of the box'' approaches tested.

Protein prediction showed a trend towards completeness and elevated
coverage on the 3’ end of transcripts, as a result of RNA preparation
using oligo-dT primers, an effect that seems to be ubiquitous in deep
transcriptome sequencing projects (e.g. \cite{pmid20331785}). The low
number of complete ORFs and especially the low number of start-codons
(and corresponding Methionine amino acids) in predicted proteins seems
cumbersome but cross-validation in combination with SNP-calling
demonstrated the overall correctness of predicted proteins.

We were able to demonstrate, that screening of SNPs in or adjacent to
homopolymer regions ``improved'' overall measurements on SNP-quality:

First the ratio of transitions to transversions (ti/tv) increased.
Such an increase is explainable by the removal of ``noise'' associated
with common homopolymer-errors \cite{pmid21685085}. Assuming that
errors would be independent of transversion-transition bias erroneous
SNPs would have a ti/tv of 0.5 and thereby lower the overall
value. Other explanations for these observations are hard to find so
it can be concluded that removing noise from homopolymer
sequencing-error ti/tv increases.  The value of 2.38
(1.82 outside,
2.74 inside ORFs) is in good agreement with
the overall ti/tv of humans (2.16 \cite{pmid21169219}) or
\textit{Drosophila} (2.07 \cite{pmid21143862}). 

The ratio of non-synonymous SNPs per non-synonymous site to synonymous
SNPs per synonymous site (dn/ds) decreased with removal of SNPs
adjacent to homopolymer regions from 0.45 to
0.32 after full screening. Similar to ti/tv it the most
plausible explanation is the removal of error, as unbiased error would
lead to a dn/ds of 1. While dn/ds is not unproblematic to interpret
within populations \cite{pmid19081788}, assuming negative (purifying)
selection on most protein-coding genes lower values seem more
plausible, also in comparison with other studies (see further text).

We used a threshold value for the minority allele of 7\% for exclusion
of SNPs, this corresponds to the ca. 10 ``haploid equivalents'' (5
individual worms plus an negligible amount of L2 larvae - in the L2
library and within the female adult worms - bearing possibly
additional diversity). It is hard to explain, that ti/tv decreased in
this filtering step, while dn/ds still further decreased.

The benefit of this screening was mainly a reduction of non-synonymous
SNPs in high coverage contigs. When it was applied dn/ds did not
scale with coverage. Working with an estimate of dn/ds independent of
coverage, efforts to control for sampling a biased by sampling depth
(i.e. coverage) like developed \cite{pmid18590545} and used
\cite{pmid20478048} could be avoided.  

When the whole of coding sequences are studied, of which only a small
subset of sites can be under diversifying selection, dn/ds of ~0.5 has
been suggested as threshold for assuming diversifying selection
\cite{pmid15579698} instead of the classical threshold of 1
\cite{pmid6449605}. In the transcripts from the female reproductive
tract of \textit{Drosophila} dn/ds was 0.15 \cite{pmid15579698} and in
the 0.21 male reproductive tract \cite{pmid11404480} (although for ESTs
specific to the male accessory gland were shown to have a higher dn/ds
of 0.47). Pyrosequencing studies found dn/ds to be between 0.13 and
0.27 (depending on tissue type genes were mainly expressed in) in the
Zebra finch transcriptome \cite{pmid20359325}, 0.12 in the
transcriptome of \textit{Tigriopus californicus} \cite{pmid21199025}
and 0.3 in the parasitic nematode \textit{Ancylostoma canium}
\cite{pmid20470405}. In comparison with these results even our
estimate after screening seems high (although it should be noted, that
the latter tree studies report a mean dn/ds over contigs - the
\textit{A. canium} doesn't make clear what exactly is reported - and
therefore the value has to be compared to our mean dn/ds over contigs
of 0.23) and further
investigation using deeper sequencing of more individuals on the
solexa GAII platform will be used to fully exclude the possibility of
this result being induced by sequencing error. Moreover such an
experiment should try to test that divergence between populations is
leading to positive selection on only the possibly diverging European
populations. For such a study the set of SNPs found here are
invaluable, as it can be used to define a gold standard set of SNPs
found with both technologies.

We were able to obtain high-quality annotations for a large set of
TUGs. Comparison with protein sequence derived from \textit{B. malayi}
showed a a remarkable degree of agreement regarding the occurrence of
terms. This implies, that our transcriptome-data-set is a
representative subset of a nematode-parasite
genome. 

Over-representation of GO-term in genes under diversifying selection
(at a threshold of dn/ds>0.5, as established above) highlighted many
interesting gene-products:

In the molecular function category two amino acid transmembrane
transporters (``Contig5699'' and ``Contig866'') - the only contigs
with this annotation (or annotation, which is an offspring-term of
this) and a dn/ds obtained - were found to have a dn/ds>0.5. Such
transporter are thought to be important in the survival of parasites in
a host \cite{pmid18272295}.

Enrichment in the category ``peptidase activity'' highlighted twelve
peptidases (from 43 with a dn/ds obtained). All twelve have orthologs
in \textit{B. malayi} and \textit{C. elgans} and are conserved across
kingdoms. Despite their conservation peptidases are thought to have
have acquired new and prominent roles in host-parasite interaction
compared to free living organisms: In \textit{A. crassus} a
trypsin-like proteinase has been identified thought to be utilized by
the tissue-dwelling L3 stage to penetrate host tissue and an aspartyl
proteinase thought to be a digestive enzyme in adults
\cite{polzer_identification_1993}.

The under-representation of ribosomal proteins (term ``structural
constituent of ribosome'') in disruptively selected contigs is in good
agreement with the notion that ribosomal proteins are extremely
conserved across kingdoms \cite{pmid9664699} and should be under under
strong negative selection. 

The additional prediction of signal sites for cleavage allowed
interpretation and cross-validation of the results from SNP-calling:
The detection of signal-peptides secretion using \textit{in silico}
analysis of ESTs has been used to highlight candidate genes for
example in \textit{Nippostrongylus brasiliensis}
\cite{harcus_signal_2004} and in a large scale analysis across all
nematode \cite{nagaraj_needles_2008} ESTs. Proteomic analysis in
\textit{B. malayi} \cite{pmid19352421,pmid18958170} and
\textit{Heligmosomoides polygyrus} \cite{pmid21722761} was able to
find evidence for excretion for some of the protein-products and to
highlight additional candidate genes.

We found an elevated dn/ds for signal-positives. These result could be
explained follow the logic of signal-positives being more likely to be
secreted to the host-parasite interface and proteins involved in
host-parasite interaction being more likely to be under disruptive
selection. Signal-positive TUGs with high dn/ds constitute another set
of genes worth further examination in future studies.

TUGs predicted to be novel in the phylum nematoda contained the
highest proportion of signal-positives. A interpretation of this
findings could be a confirmation of a study on \textit{Nippostrongylus
  brasiliensis} \cite{harcus_signal_2004}, where signal positives were
reported as less conserved. In the present study we did not aim to
identify ``novelty to \textit{A. crassus}'' as we believe in a deep
sequencing project he absence of sequence similarity could be
attributed to erroneous sequence instead of true novelty, and thereby
blur analysis. However novelty in nematodes and to a lesser extend
novelty in Spirurina seems to support the notion, that - if not
diversified within nematoda to an extend leading to a complete loss of
similarity, like suggested in the mentioned study - signal positives
in nematodes could have taken a divergent evolutionary path from their
orthologs in other phyla.

It was within our expectation, that expression analysis failed to give
conclusive results, as the present data-set is not fully adequate for
this kind of analysis: First we did not include replicates for
libraries of male adults as well as for L2-larvae. Second one of the
replicates for female worms (library E1) resulted in a low amount of
sequence mappable to protein-coding (non-rRNA) genes.
However some of the results are still valuable:

DESeq was able to report genes significantly differing in expression
between male and female worms and between the L2 library and the all
other worms. This was possible for male worms as well as for
L2-larvae, were no replicated samples were obtained, due due the
special features of this package \cite{pmid20979621}. However only
over-expression in non-repeated samples can be detected, as obviously
lack of expression in one sample can't validate

Comparisons were lacking significance, as methods are designed for
deeper sequencing and more importantly more replicates would be
needed. Differences between the L2-library and other libraries were
mainly due to off-target data, and TUGs solely found in the L2 library
are ... 



%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}

  
%%%%%%%%%%%%%%%%%%
\section*{Methods}
\subsection*{Worm samples, RNA extraction, cDNA synthesis and
  Sequencing}
\textit{A. crassus} from Japanese eels were sampled from Kao-Ping river
and an adjacent aquaculture in Taiwan as described in
\cite{heitlinger_massive_2009}. Worms from the European eel were
sampled in Sniardwy Lake, Poland (53.751959N ,21.730957E) and from the
Linkenheimer Altrhein, Germany (49.0262N; 8.310556E). After
determination of the sex of adult nematodes, all worms were stored in
RNA-later (Quiagen, Hilden, Germany) until extraction of RNA. RNA was
extracted from:

\begin{itemize}
\item one worm from a  cultured Taiwanese eel (sample T1)
\item one worm from a wild Taiwanese eel (sample T2)
\item one worm from an eel from the German sampling site (E1)
\item one worm an eel from Polish sampling site (E2)
\end{itemize}

In addition RNA was extracted from L2-Larve from the German sampling
site (sample L2) and a from a male worm from the Taiwanese aquaculture
(sample M). RNA was reverse transcribed and amplified into cDNA using
the MINT-cDNA synthesis kit (Evrogen, Moscow, Russia).

For host-contamination screening a liver-sample from an uninfected
Japanese eel was prepared using the methods as described above for
\textit{A. crassus} samples.

A emulsion PCR was performed for each cDNA library according to the
manufacturer's potocol (Roche/454 Life Sciences). For library E1 a
Multiplex Index (MID) (Roche/454 Life Sciences) was used in
preparation of the sequencing adapter. The libraries were sequenced in
different runs of the Roche/454 Genome Sequencer FLX System: Library
T2 on an eighth plate of the instrument usning standard
``FLX-chemistry'', the remaining libraries on a eighth of a plate
using ``FLX-Titanium-chemistry''.

\subsection*{Trimming, quality control and assembly}

Raw sequences were extracted in fasta format (with the corresponding
qualities files) using sffinfo (Roche/454) and screened for adapter
sequences of the MINT-amplification-kit using cross-match \cite{PHRAP}
(with parameters -minscore 20 and -minmatch 10). Seqclean
\cite{tgicl_pertea} was used to screen poly-A-tails, low quality,
repetitive and short (<100 bases) sequences. In addition all reads
were \texttt{blast}ed (1e-5 -F F) against the following databases:

\begin{itemize}
\item a combined eel-mRNA database consisting of an assembly of
  sequences from the liver of the Japanese eel sequenced for this
  purpose (as described above), a sequence assembly of unpublished
  (sanger-) ESTs (made available to us by Gordon Cramb; University of
  St Andrews) and from EeelBase \cite{pmid21080939} a publically
  availble transcriptome database for the European eel.
\item a eel-rRNA database from a rRNA screening of the above and
  assembly together with publically available rRNA-sequences.
\item an \textit{A. crassus} rRNA-database from screening of our
  dataset against nematode-rRNA, and assembly of these rRNA
  reads. This database notably also contained xenobiotic rRNA
  sequences.
\end{itemize}

Reads mapping to one of the databases with more than 80\% of their
length and 95\% identity were removed from the dataset. Screenig and
trimming information was written back into sff-format using
\texttt{sfffile} (Roch/454).

We used an approach proposed by Kumar and Blaxter\cite{pmid20950480},
combining assemblies from the \texttt{mira} \cite{miraEST} and
\texttt{newbler} \cite{pmid16056220}. Briefly the two assemblies are
combined into one using Cap3\cite{Cap3_Huang} and only contigs
supported by both assemblers are regarded good quaility. For further
details see the supplementary methods.

\subsection*{Post assembly classification and taxonomic assesment}

After assembly contigs were assesed a second time for
host-contamination and other xenobiotics:

The contigs were \texttt{blast}ed (with a cut-off 1e-5) against the
same databases used prior to assembly (Eel-mRNA, Eel-rRNA,
\textit{A. crassus}-rRNA and additionally against the nucleotide
version of nempep4 \cite{parkinson_nembase:resource_2004,
  pmid21550347}, determining the best hit across databases. These best
hits across databases were screened and only such hits involving more
then 50\% of the 

Additionally \texttt{blast} (\texttt{blastn} e-value cut-off 1e-5)
against NCBI-nt and (\texttt{blastx} e-value cut-off 1e-5) against
NCBI-nt was used to determine taxon-membership of the top hit at the
family, phylum and kingdom rank.

\subsection*{Protein prediction and annotation}

Proteins were predicted using the \texttt{Prot4EST} (version 3.0b)
\cite{wasmuth_prot4est:_2004}: First \texttt{blast} searches against a
rRNA-database, a mitochondrial database and against
uniref100\cite{pmid18836194} were preformed. Then results were used to
predict proteins directly (joining single high scoring pairs, and
thereby intorducing gaps and ambiguous bases if needed). Secondly
using the codon-usage from \texttt{blast}-predictions a simulated
transcriptome was generated, reverse translating the
\textit{B. malayi} proteom, as training-data-set for
\texttt{ESTscan}’s\cite{estscan} hidden Markov models.  If both
\texttt{blast}-based prediction and \texttt{ESTscan} failed, simply
the longest ORF is inferred.

\texttt{Blast}-based annotations were inferred using Annot8r (version
1.1.1) \cite{schmid_annot8r:_2008}: Searches were performed against
all sequences in uniref100 \cite{pmid18836194} being annotated with
GO-terms, EC-numbers and KEGG-parthways. Up to 10 (possibly
contradictory) annotations based on a bitscore cut-off of 55 were
obtained for each annotated database.  For comparison annotations were
obtained the same way for all \textit{Brugia malayi} proteins in
uniref100.

% Additionally domain-based annotation was
% obtained using InterProScan (iprscan command-line tool, version 4.6)
% \cite{pmid11590104}.

\texttt{SignalP V4.0} \cite{pmid21959131} was used to predict signal
peptide cleavage sites and signal anchor signatures.

\subsection*{SNP analysis}

As protein-prediction inferres gaps (e.g from sequencing errors) to
predict the most likely protein, not only start- and end-coordinates
of open reading frames (ORFs) had to be extracted from the output of
\texttt{Prot4EST}. We did this in a custom \texttt{perl}-script using
a \texttt{blast}-search with the nucleotide equivalent of the protein
as query and the raw sequence as subject. We obtain the
hit-coordinates as ORF-coordinates and imputed the
\texttt{blast}-query as corrected ORF-sequencs.

We mapped the raw reads against the the complete unigene set, with the
imputed sequences for those contigs with proteins predicted, using
ssaha2 (with parameters -kmer 13 -skip 3 -seeds 6 -score 100 -cmatch
10 -ckmer 6 -output sam -best 1).

\texttt{pileup}-files were produced using \texttt{samtools}
\cite{journals/bioinformatics/LiHWFRHMAD09}, discarding sequences
mapping to multiple regions with the best hit. VarScan
\cite{pmid19542151} (pileup2snp) was used with default parameters on
\texttt{pileup}-files. This output was further screened as described
in the results part of the manuscript.

% \subsection*{Respiration analysis}

% We extracted contigs annotated with ``GO:0009060 aerobic respiration''
% and ``GO:0009061 anaerobic respiration''

\subsection*{Gene-expression analysis}

For NlaIII-tag-sequencing total RNA was prepared as described above
from a worm from the Polish sampling site. A sequence-tag libray was
created following the protocol supplied by Illmina for this method.
Briefly after synthesis of cDNA on oligo(dt)-beads, this cDNA is
digested with the enzyme NlaIII (restriction site ``CATG''). After
ligation of an adaptor containing its restriction site the enzyme MmeI
cuts 17 bases downstream of its binding site generating a sequence tag
of in total 21 bases.

For 454 reads, read counts were obtained from the mapping to imputed
sequence described above. Tag-sequences were mapped using
\texttt{BWA}\cite{pmid19451168}. And read counts extracted using
\texttt{Samtools}.

The R-package DESeq\cite{pmid20979621} was used to normalize for
library-size and analyse statistical significance of differential
expression. Spearman correlation-coefficients were calculated for raw
counts.

\subsection*{General coding methods}

The bulk of analysis (unless otherwise cited) presented in this paper
was carried out in R \cite{R_project} using custom scripts. We used a
method provided in the R-packages
Sweave\cite{lmucs-papers:Leisch:2002} and Weaver\cite{weaver} for
``reproducible research'' combining R and \TeX code in a single
file. All intermediate data files needed to compile the present
manuscript from data-sources are provided upon request. For
visualistaion we used the R-packages lattice\cite{lattice-book} and
ggplot2\cite{ggplot-book}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Competing interests}
The authors declare no competing interests.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgments}
\ifthenelse{\boolean{publ}}{\small}{} We thank Stephen Bridgett from
gene-pool sequencing service for general help with raw data and for
cross-contamination screening of libraries.

The work of EGH is funded by
Volkswagen Foundation, "F\"{o}rderinitiative Evolutionsbiologie".
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article} % Style BST file
  \bibliography{/home/ele/bibtex/master,/home/ele/bibtex/master2}
} % Bibliography file (usually '*.bib' )

%%%%%%%%%%%


\ifthenelse{\boolean{publ}}{\end{multicols}}{}
\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \subsection*{Figure 1 - rRNA-screening statistics}
 


  \subsection*{Figure 2 - Contig-coverage}

  \subsection*{Figure 3 - }

  \subsection*{Figure 4 - }

\subsection*{Figure 5 - }

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
  \subsection*{Table 1 - Trimming statistics}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrrrr}
  \hline
 & raw\_reads & short & lowq & dust & shortq & valid \\ 
  \hline
T2 & 116366 &  46 & 24361 &  50 & 3226 & 88683 \\ 
  T1 & 99482 &  44 & 6589 &  13 & 1301 & 91535 \\ 
  L2 & 112718 &  55 & 10571 & 147 & 4880 & 97065 \\ 
  M & 106726 &  44 & 10761 & 225 & 4454 & 91242 \\ 
  E1 & 209325 &  49 & 49798 &  15 & 42882 & 116581 \\ 
  E2 & 111746 & 163 & 8478 &  20 & 2242 & 100843 \\ 
  total & 756363 & 401 & 110558 & 470 & 58985 & 585949 \\ 
   \hline
\end{tabular}
\subsection*{Table 2 - Pre-assembly screening statistics}
% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrr}
  \hline
 & AcrRNA & eelmRNA & eelrRNA & valid \\ 
  \hline
E1 & 76403 & 4835 & 13112 & 22231 \\ 
  E2 & 11213 & 3613 &  69 & 85948 \\ 
  L2 & 35940 & 1220 & 1603 & 58302 \\ 
  M & 31351 & 1187 & 418 & 58286 \\ 
  T1 & 24929 & 7475 & 514 & 58617 \\ 
  T2 & 7233 & 11741 &  38 & 69671 \\ 
   \hline
\end{tabular}
\subsection*{Table 3 - \texttt{Blast}-hits to protozoan rRNA in
  pre-assembly screening}
% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{lll}
  \hline
sequence.identifier & sequence.identitiy & hsp.length \\ 
  \hline
gi$|$299836113$|$gb$|$GU290110.1$|$ & 99.67 & 599 \\ 
  gi$|$261259658$|$emb$|$FN393299.1$|$ & 98.42 & 310 \\ 
  gi$|$219524834$|$gb$|$EU709197.1$|$ & 98.84 & 515 \\ 
  gi$|$225216791$|$gb$|$FJ176706.1$|$ & 90.02 & 610 \\ 
  gi$|$238617605$|$gb$|$FJ973380.1$|$ & 96.26 & 985 \\ 
  gi$|$323320595$|$gb$|$HQ918172.1$|$ & 97.36 & 606 \\ 
  gi$|$269993998$|$dbj$|$AB520736.1$|$ & 100 & 555 \\ 
  gi$|$224996440$|$gb$|$FJ654272.1$|$ & 98.63 & 145 \\ 
  gi$|$161015540$|$gb$|$EF577167.1$|$ & 99.87 & 759 \\ 
  gi$|$294831542$|$dbj$|$AB526843.1$|$ & 96.35 & 657 \\ 
  gi$|$225216791$|$gb$|$FJ176706.1$|$ & 97.68 & 257 \\ 
   \hline
\end{tabular}
\subsection*{Table 4 - Post-assembly host-screening }
  
% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrrr}
  \hline
 & AcrRNA & eelmRNA & eelrRNA & valid\_nempep & valid\_no\_hit \\ 
  \hline
number & 604 & 1162 & 50 & 1254 & 37117 \\ 
  mean coverage & 3.28 & 1.80 & 2.00 & 6.67 & 2.44 \\ 
   \hline
\end{tabular}
\subsection*{Table 5 - Evolutionary conservation }
% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrr}
  \hline
 & conserved & novel.in.metazoa & novel.in.nematoda & novel.in.clade3 \\ 
  \hline
bit.threshold.50 & 7741 & 1720 & 1769 & 1523 \\ 
  bit.threshold.80 & 4715 & 1402 & 1686 & 1695 \\ 
   \hline
\end{tabular}
\subsection*{Table 6 - Protein prediction statistics}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrr}
  \hline
 & p4e-$>$BLAST-similarity & p4e-$>$ESTScan & p4e-$>$LongestORF & no-prediction \\ 
  \hline
plus strand & 9701 & 8005 & 6393 & 562 \\ 
  minus strand & 4813 & 5368 & 5345 &   0 \\ 
   \hline
\end{tabular}

\subsection*{Table 7 - SNP summry statistics}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rlllllllll}
  \hline
 &   &   & pos & in & codon &   & ti/tv  &   &   \\ 
  & No.SNPs & in.ORF & 1 & 2 & 3 & overall & ins.orf & outs.orf & dn.ds \\ 
   \hline
raw & 13042 & 8309 & 2713 & 2172 & 3424 & 1.98 & 2.43 & 1.42 & 0.45 \\ 
  h.screened & 9523 & 6395 & 2058 & 1604 & 2733 & 2.86 & 3.32 & 2.16 & 0.39 \\ 
  p.screened & 6276 & 4226 & 1368 & 925 & 1933 & 2.38 & 2.74 & 1.82 & 0.32 \\ 
  \end{tabular}

\subsection*{Table 8 - GO-terms in positively selected}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{longtable}{rrll}
 Count & Size & Term & direction \\ 
  \hline
 12 &  43 & peptidase activity & Over \\ 
    2 &   2 & L-amino acid transmembrane transporter activity & Over \\ 
   44 & 269 & catalytic activity & Over \\ 
    3 &   6 & ribonucleoprotein binding & Over \\ 
    7 &  26 & ATPase activity & Over \\ 
   21 & 113 & hydrolase activity & Over \\ 
    2 &   3 & carboxylic acid transmembrane transporter activity & Over \\ 
    2 &   3 & testosterone dehydrogenase activity & Over \\ 
    2 &   3 & organic acid transmembrane transporter activity & Over \\ 
    2 &   3 & oxidoreduction-driven active transmembrane transporter activity & Over \\ 
    2 &   3 & ribonuclease activity & Over \\ 
    2 &   3 & amino acid transmembrane transporter activity & Over \\ 
    2 &   3 & testosterone dehydrogenase (NAD+) activity & Over \\ 
    2 &   3 & amine transmembrane transporter activity & Over \\ 
    1 &  44 & structural constituent of ribosome & Under \\ 
   \hline
  3 &   3 & branched chain family amino acid metabolic process & Over \\ 
    3 &   3 & branched chain family amino acid catabolic process & Over \\ 
   10 &  32 & brain development & Over \\ 
    7 &  19 & positive regulation of cell cycle process & Over \\ 
    4 &   7 & spermatid differentiation & Over \\ 
    4 &   7 & spermatid development & Over \\ 
    6 &  15 & response to starvation & Over \\ 
    3 &   4 & positive regulation of mitosis & Over \\ 
    3 &   4 & positive regulation of nuclear division & Over \\ 
   11 &  40 & central nervous system development & Over \\ 
   13 &  52 & regulation of cell cycle & Over \\ 
   11 &  42 & cellular amino acid metabolic process & Over \\ 
    5 &  12 & autophagy & Over \\ 
   10 &  37 & regulation of cell cycle process & Over \\ 
    8 &  27 & interphase & Over \\ 
    8 &  27 & interphase of mitotic cell cycle & Over \\ 
    2 &   2 & pentose metabolic process & Over \\ 
    2 &   2 & xylulose metabolic process & Over \\ 
    2 &   2 & response to disaccharide stimulus & Over \\ 
    2 &   2 & embryonic body morphogenesis & Over \\ 
    2 &   2 & response to sucrose stimulus & Over \\ 
    2 &   2 & L-amino acid transport & Over \\ 
    2 &   2 & NADP metabolic process & Over \\ 
   17 &  81 & apoptosis & Over \\ 
   16 &  76 & regulation of molecular function & Over \\ 
   27 & 151 & catabolic process & Over \\ 
   12 &  52 & cellular amine metabolic process & Over \\ 
   33 & 195 & response to stress & Over \\ 
    5 &  14 & mitotic cell cycle G1/S transition DNA damage checkpoint & Over \\ 
    5 &  14 & regulation of cellular amine metabolic process & Over \\ 
   10 &  41 & reproductive structure development & Over \\ 
   12 &  53 & amine metabolic process & Over \\ 
    8 &  30 & muscle organ development & Over \\ 
    7 &  25 & regulation of catabolic process & Over \\ 
    3 &   6 & centrosome organization & Over \\ 
    3 &   6 & RNA catabolic process & Over \\ 
    3 &   6 & microtubule organizing center organization & Over \\ 
    3 &   6 & positive regulation of organelle organization & Over \\ 
    5 &  15 & signal transduction in response to DNA damage & Over \\ 
    5 &  15 & mitotic cell cycle G1/S transition checkpoint & Over \\ 
    5 &  15 & G1/S transition checkpoint & Over \\ 
    5 &  15 & DNA damage response, signal transduction by p53 class mediator & Over \\ 
    5 &  15 & G1/S transition of mitotic cell cycle & Over \\ 
    5 &  15 & regulation of G1/S transition of mitotic cell cycle & Over \\ 
   16 &  80 & cell cycle phase & Over \\ 
   11 &  49 & mRNA metabolic process & Over \\ 
   18 &  94 & nervous system development & Over \\ 
    4 &  11 & skeletal muscle organ development & Over \\ 
    4 &  11 & imaginal disc development & Over \\ 
   14 &  69 & regulation of apoptosis & Over \\ 
   14 &  69 & regulation of programmed cell death & Over \\ 
    2 &   3 & positive regulation of mitotic metaphase/anaphase transition & Over \\ 
    2 &   3 & negative regulation of reproductive process & Over \\ 
    2 &   3 & germ cell migration & Over \\ 
    2 &   3 & centrosome duplication & Over \\ 
    2 &   3 & centrosome separation & Over \\ 
    2 &   3 & protein tetramerization & Over \\ 
    2 &   3 & protein homotetramerization & Over \\ 
    2 &   3 & mitotic centrosome separation & Over \\ 
    2 &   3 & regulation of the force of heart contraction & Over \\ 
    2 &   3 & spliceosomal conformational changes to generate catalytic conformation & Over \\ 
    2 &   3 & nuclear mRNA cis splicing, via spliceosome & Over \\ 
    2 &   3 & amino acid transport & Over \\ 
   19 & 103 & cell cycle & Over \\ 
    8 &  33 & ATP synthesis coupled electron transport & Over \\ 
    8 &  33 & mitochondrial ATP synthesis coupled electron transport & Over \\ 
    8 &  33 & regulation of mitotic cell cycle & Over \\ 
   17 &  90 & programmed cell death & Over \\ 
    3 &   7 & regulation of neurotransmitter levels & Over \\ 
    3 &   7 & cellular response to starvation & Over \\ 
   18 &  97 & cell death & Over \\ 
   18 &  97 & death & Over \\ 
   13 &  64 & macromolecule catabolic process & Over \\ 
    6 &  22 & response to protein stimulus & Over \\ 
   15 & 188 & gene expression & Under \\ 
    1 &  46 & cellular protein complex disassembly & Under \\ 
    1 &  46 & macromolecular complex disassembly & Under \\ 
    1 &  46 & cellular macromolecular complex disassembly & Under \\ 
    1 &  46 & protein complex disassembly & Under \\ 
    1 &  45 & viral genome expression & Under \\ 
    1 &  45 & viral transcription & Under \\ 
    1 &  45 & pancreas development & Under \\ 
    1 &  45 & endocrine pancreas development & Under \\ 
    1 &  45 & endocrine system development & Under \\ 
    4 &  79 & translation & Under \\ 
    1 &  44 & translational termination & Under \\ 
    8 & 118 & transcription & Under \\ 
    2 &  55 & cellular component disassembly at cellular level & Under \\ 
    2 &  55 & cellular component disassembly & Under \\ 
   21 & 225 & biosynthetic process & Under \\ 
    0 &  25 & positive regulation of intracellular protein kinase cascade & Under \\ 
   14 & 165 & cellular macromolecule biosynthetic process & Under \\ 
    0 &  24 & oocyte differentiation & Under \\ 
   21 & 222 & cellular biosynthetic process & Under \\ 
    1 &  37 & oogenesis & Under \\ 
    0 &  23 & oocyte development & Under \\ 
    0 &  23 & cation transport & Under \\ 
    2 &  47 & viral infectious cycle & Under \\ 
    2 &  47 & viral reproductive process & Under \\ 
   15 & 168 & macromolecule biosynthetic process & Under \\ 
    1 &  35 & positive regulation of response to stimulus & Under \\ 
    0 &  22 & positive regulation of MAPKKK cascade & Under \\ 
   \hline
  4 &   7 & small nuclear ribonucleoprotein complex & Over \\ 
    2 &   2 & Cajal body & Over \\ 
    2 &   2 & U4/U6 x U5 tri-snRNP complex & Over \\ 
    2 &   2 & U5 snRNP & Over \\ 
    3 &   6 & nuclear speck & Over \\ 
   27 & 152 & mitochondrion & Over \\ 
    5 &  15 & nuclear body & Over \\ 
    2 &   3 & clathrin sculpted vesicle & Over \\ 
    2 &   3 & basement membrane & Over \\ 
    2 &   3 & plant-type cell wall & Over \\ 
    2 &   3 & plasma membrane respiratory chain complex I & Over \\ 
    2 &   3 & plasma membrane respiratory chain & Over \\ 
    0 &  29 & large ribosomal subunit & Under \\ 
    0 &  27 & cytosolic large ribosomal subunit & Under \\ 
   27 & 266 & nucleus & Under \\ 
   17 & 185 & non-membrane-bounded organelle & Under \\ 
   17 & 185 & intracellular non-membrane-bounded organelle & Under \\ 
    4 &  65 & nucleolus & Under \\ 
   \hline
\hline
\end{longtable}
\subsection*{Table 9 - Correlation between read-counts in
  454-libraries and solexa-tags}

\newpage

\subsubsection*{Table 9 a - analysing all TUGs}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrrrrrr}
  \hline
 & solexa.tags & E1 & E2 & L2 & M & T1 & T2 & all.reads \\ 
  \hline
solexa.tags & 1.000 & 0.257 & 0.356 & -0.165 & 0.320 & 0.233 & 0.127 & 0.315 \\ 
  E1 & 0.257 & 1.000 & 0.154 & -0.076 & 0.229 & 0.145 & 0.071 & 0.254 \\ 
  E2 & 0.356 & 0.154 & 1.000 & -0.246 & 0.126 & 0.134 & 0.090 & 0.295 \\ 
  L2 & -0.165 & -0.076 & -0.246 & 1.000 & -0.181 & -0.237 & -0.266 & 0.127 \\ 
  M & 0.320 & 0.229 & 0.126 & -0.181 & 1.000 & 0.077 & 0.016 & 0.278 \\ 
  T1 & 0.233 & 0.145 & 0.134 & -0.237 & 0.077 & 1.000 & 0.029 & 0.210 \\ 
  T2 & 0.127 & 0.071 & 0.090 & -0.266 & 0.016 & 0.029 & 1.000 & 0.350 \\ 
  all.reads & 0.315 & 0.254 & 0.295 & 0.127 & 0.278 & 0.210 & 0.350 & 1.000 \\ 
   \hline
\end{tabular}\subsubsection*{Table 9 b - analysing good-category contigs only}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrrrrrr}
  \hline
 & solexa.tags & E1 & E2 & L2 & M & T1 & T2 & all.reads \\ 
  \hline
solexa.tags & 1.000 & 0.371 & 0.528 & -0.196 & 0.450 & 0.393 & 0.199 & 0.385 \\ 
  E1 & 0.371 & 1.000 & 0.324 & -0.064 & 0.366 & 0.307 & 0.172 & 0.312 \\ 
  E2 & 0.528 & 0.324 & 1.000 & -0.280 & 0.324 & 0.411 & 0.197 & 0.373 \\ 
  L2 & -0.196 & -0.064 & -0.280 & 1.000 & -0.191 & -0.242 & -0.358 & 0.084 \\ 
  M & 0.450 & 0.366 & 0.324 & -0.191 & 1.000 & 0.264 & 0.083 & 0.347 \\ 
  T1 & 0.393 & 0.307 & 0.411 & -0.242 & 0.264 & 1.000 & 0.156 & 0.324 \\ 
  T2 & 0.199 & 0.172 & 0.197 & -0.358 & 0.083 & 0.156 & 1.000 & 0.437 \\ 
  all.reads & 0.385 & 0.312 & 0.373 & 0.084 & 0.347 & 0.324 & 0.437 & 1.000 \\ 
   \hline
\end{tabular}
\subsubsection*{Table 9 c - analysing good-category contigs surely from \textit{A. crassus} only}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Thu Oct 13 17:35:12 2011
\begin{tabular}{rrrrrrrrr}
  \hline
 & solexa.tags & E1 & E2 & L2 & M & T1 & T2 & all.reads \\ 
  \hline
solexa.tags & 1.000 & 0.373 & 0.524 & -0.123 & 0.438 & 0.411 & 0.168 & 0.430 \\ 
  E1 & 0.373 & 1.000 & 0.311 & 0.000 & 0.346 & 0.313 & 0.160 & 0.341 \\ 
  E2 & 0.524 & 0.311 & 1.000 & -0.186 & 0.271 & 0.421 & 0.148 & 0.434 \\ 
  L2 & -0.123 & 0.000 & -0.186 & 1.000 & -0.106 & -0.140 & -0.258 & 0.001 \\ 
  M & 0.438 & 0.346 & 0.271 & -0.106 & 1.000 & 0.249 & 0.030 & 0.384 \\ 
  T1 & 0.411 & 0.313 & 0.421 & -0.140 & 0.249 & 1.000 & 0.151 & 0.397 \\ 
  T2 & 0.168 & 0.160 & 0.148 & -0.258 & 0.030 & 0.151 & 1.000 & 0.505 \\ 
  all.reads & 0.430 & 0.341 & 0.434 & 0.001 & 0.384 & 0.397 & 0.505 & 1.000 \\ 
   \hline
\end{tabular}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Additional Files}

File A\_crassus\_contigs\_full.csv lists all data computed on the contig
level, including sequences (raw, coding, imputed). File
A\_crassus\_contigs\_readable.csv lists only the metadata not including
sequences.

 

\end{bmcformat}
\end{document}
