%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}


\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of the swimbladder-nematode
    Anguillicola crassus: Resources for an alien parasite}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel G Heitlinger\correspondingauthor$^{1,2}$%
       \email{Emanuel G Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Stephen Bridgett$^{3}$%
       \email{Stephen Bridgett- sbridget@staffmail.ed.ac.uk}%
       Anna Montazam$^{3}$%
       \email{Anna Montazam- Anna.Montazam@ed.ac.uk}%
       Horst Taraschewski$^1$%
       \email{Horst Taraschewski- dc20@rz.uni-karlsruhe.de}%
       and Mark Blaxter$^2$%
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute 1, University of Karlsruhe,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
        \iid(3)The GenePool Sequencing Service, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} 
  \paragraph*{Results:} 
  \paragraph*{Conclusions:}
  Yeh!
  
\end{abstract}


      \ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

The nematode Anguillicola crassus Kuwahara, Niimi et Itagaki, 1974
\cite{kuwahara_Niimi_Itagaki_1974} is a parasite of freshwater eels of
the genus Anguilla, and adults localise to the swim bladder where they
feed on blood. Larvae are transmitted via crustacean intermediate
hosts. Originally endemic to East-Asian populations of the Japanese
eel (Anguilla japonica), A. crassus has attracted interest due to
recent anthropogenic expansion of its geographic and host ranges to
Europe and the European eel (Anguilla anguilla). Recorded for the
first time in 1982 in North-West Germany \cite{fischer_teichwirt},
where it was most likely introduced through the live-eel trade
\cite{koops_anguillicola-infestations_1989, koie_swimbladder_1991},
A. crassus has spread rapidly through populations of its newly
acquired host \cite{kirk_impact_2003}. At the present day it is found
in all An. anguilla populations except those in Iceland
\cite{kristmundsson_parasite_2007}. A. crassus can be regarded as a
model for invasive parasite introduction and spread
\cite{taraschewski_hosts_2007}.

A. crassus has a major impact on An. anguilla populations. In its
natural host in Asia infection prevalence and mean intensity of
infection are lower than in Europe \cite{mnderle_occurrence_2006},
where high prevalence (above 70\% \cite{wrtz_distribution_1998}) and
high infection intesities have been reported throughout the newly
colonized area \cite{lefebvre_anguillicolosis:_2004}. The virulence of
A .crassus in this new host has been attributed to an inadequate
immune response in An. anguilla \cite{knopf_swimbladder_2006}. While
the An.  japonica is capable of killing larvae of the parasite after
vaccination \cite{knopf_vaccination_2008} or under high infection
pressure \cite{heitlinger_massive_2009}, responses in An. anguilla
have hallmarks of pathology, including thickening of the swim bladder
wall \cite{wrtz_histopathological_2000}.  Interestingly host also
affects the adult size and life-history of the nematodes: In European
eels the nematodes are bigger and develop and reproduce faster
\cite{knopf_differences_2004}.


The genus Anguillicola is placed in the nematode suborder Spirurina
(clade III sensu \cite{blaxter_molecular_1998})
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
are exclusively parasitic and include important human pathogens (the
causative agents of filariases and ascariasis) as well as prominent
veterinary parasites. Molecular phylogenetic analyses place
Anguillicola in a clade of spirurine nematodes (Spirurina B of
[Laetsch et al submitted]) that have an freshwater or marine
intermediate host, but infect a wide range of carnivorous definitive
hosts. Spirurina B is sister to the main Spirurina C, including the
agents of filariases and ascariasis), and thus A. crassus may be used
as an outgroup taxon to understand the evolution of parasitic
phenotypes in these species.

Recent advances in sequencing technology (often termed Next Generation
Sequencing; NGS), provide the opportunity for rapid and cost-effective
generation of genome-scale data. The Roche 454 platform
\cite{pmid16056220} offers longer reads than other NGS technologies,
and thus is suited to de novo assembly of genome-scale data in
previously understudied species. Roche 454 data has particular
application in transcriptomics \cite{pmid20950480}. The difference in
the biology of A. crassus in An. japonica (coevolved) and An. anguilla
(recently captured) eel hosts likely results from an interaction
between different host and parasite responses, underpinned by
definitive differences in host genetics, and possible genetic
differentiation between the invading European and endemic Asian
parasites. As part of aprogramme to understand the invasiveness of
A. crassus in An. anguilla, we are investigating differences in gene
expression and genetic distinction between invading European and
endemic Asian A. crassus exposed to the two different host
species. Here we report on the generation of a reference transcriptome
for A. crassus based on Roche 454 data, and explore patterns of gene
expression and diversity.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}

<<load.libs, echo=FALSE, results=hide>>=
library(VennDiagram)
library(xtable)
library(reshape)
library(ggplot2)

library(GSEABase)
library(GOstats)
library(GO.db)

library(limma)
library(XML)
library(coin)
library(multcomp)

library(DESeq)
library(ShortRead)
library(biomaRt)

library(BioIDMapper)

library(Rhh)

source("/home/ele/thesis/454/common_R_functions.R")
@ 

\subsection*{Sampling A. crassus}

One female worm and one male worm were sampled from an aquaculture
with height infection loads in Taiwan. An additonal female worm was
sampled from a stream with low infection pressure adjacent to the
aquaculture. All these worms were parsitising endemic
\textit{An. japonica}. A female worm and pool of L2 larval stages were
sampled from \textit{An. anguilla} in the river Rhein, one female worm
from a lake in Poland. All adult worms were filled with large amounts
of host-bood, therefore we anticipated abundand host-contamination in
sequencing and decided to sequence a liver sample of an unifected
\textit{An. japonica} for screening.


 \subsection*{Sequencing, trimming and pre-assembly screening}

<<trim, echo=FALSE, results=hide, cache=TRUE>>=
trimdat <- readLines("/home/ele/Data/454/trimmed/trimming.stats")
trimfile <-  grep("Trimming sff file *", trimdat, value=TRUE)
trimfile <- sapply(strsplit(trimfile, " |,"), function (x) basename(x[4]))
lib <- gsub(".sff'", "", trimfile)

raw_reads<- grep("Sequences analyzed:.*", trimdat, value=TRUE)
raw_reads <- as.numeric(gsub("Sequences analyzed: *", "", raw_reads))
short <- grep("by 'short':.*", trimdat, value=TRUE)
short <- as.numeric(gsub(" *by 'short': *", "", short))
lowq <- grep("by 'low_qual':.*", trimdat, value=TRUE)
lowq <- as.numeric(gsub(" *by 'low_qual': *", "", lowq))
dust <- grep("by 'dust':.*", trimdat, value=TRUE)
dust <- as.numeric(gsub(" *by 'dust': *", "", dust))
shortq <- grep("by 'shortq':.*", trimdat, value=TRUE)
shortq <- as.numeric(gsub(" *by 'shortq': *", "", shortq))
valid <- raw_reads-(short+lowq+dust+shortq)

TRIM <- data.frame(row.names=lib, raw_reads, short, lowq, dust, shortq, valid)
TRIM["total",] <- colSums(TRIM)

## subsitute the publication style library abrevations
rownames(TRIM) <- gsub("179F", "T1", rownames(TRIM))
rownames(TRIM) <- gsub("10F", "T2", rownames(TRIM))
rownames(TRIM) <- gsub("KS4F", "E1", rownames(TRIM))
rownames(TRIM) <- gsub("UW07F", "E2", rownames(TRIM))
rownames(TRIM) <- gsub("L2R3", "L2", rownames(TRIM))
rownames(TRIM) <- gsub("M175", "M", rownames(TRIM))

@ 

<<read.libs, echo=FALSE, results=hide, cache=TRUE>>=

RE <- list()
for (l in (c("10F", "179F", "M175", "L2R3", "KS4F", "UW07F"))){
   command <- paste("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/trimmed/",
                    l, ".sff.trimmed.sff;", sep="")
   connect <- pipe(command)
   RE[l] <- list(read.sequences(connect))
   closeAllConnections()
}
names <- gsub(" .*$", "", unlist(lapply(RE, names)))
lib <- melt(RE)

## subsitute the publication style library abrevations
lib$L1 <- gsub("179F", "T1", lib$L1)
lib$L1 <- gsub("10F", "T2", lib$L1)
lib$L1 <- gsub("KS4F", "E1", lib$L1)
lib$L1 <- gsub("UW07F", "E2", lib$L1)
lib$L1 <- gsub("L2R3", "L2", lib$L1)
lib$L1 <- gsub("M175", "M", lib$L1)

RE <- cbind(read=names, lib)
names(RE)[2:3] <- c("seq", "lib")
RE$seq <- as.character(RE$seq)
RE$length <- nchar(RE$seq)

eel.all <- read.sequences(pipe("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/Li68_assembly/sff/Li68.sff"))

eel.clean <- read.sequences(pipe("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/Li68_assembly/sff/Li68.sff.trimmed.sff"))

eel.rRNA.contigs <- read.sequences("/home/ele/Data/454/Li68_assembly/Aj_liver_rRNA.fasta")

eel.contigs <- read.sequences("/home/ele/Data/454/Li68_assembly/Aj_liver.fasta")

@ 

\SweaveInput{../contamination/pre_assembly.Rnw}

A total of \Sexpr{TRIM["total","raw_reads"]} raw sequencing reads were
generated for \textit{A. crassus} (Table 1). These were trimmed for
base call quality, and filtered by length to give
\Sexpr{TRIM["total","valid"]} high-quality reads (spanning
\Sexpr{sum(pre.screen.tab$valid.span)#$} bases). In the eel data-set
from \Sexpr{length(eel.all)} raw reads \Sexpr{length(eel.clean)} were
assembled after basic quality screening.

We then screened the \textit{A. crassus} reads for contamination by
host (\Sexpr{sum(pre.screen.tab$eelmRNA)#$} matched previously
sequenced eel genes in our own \textit{An. anguilla} 454
transcriptome, which was partitioned in \Sexpr{length(eel.contigs)}
mRNA and \Sexpr{length(eel.rRNA.contigs)} rRNA TUGs after
the nematode (\Sexpr{sum(pre.screen.tab$AcrRNA)#$} reads matched large
or small subunit nuclear or mitochondrial ribosomal RNA sequences of
A. crassus) (Table 1). In addition to fish mRNAs, we identified (and
removed) \Sexpr{pre.screen.tab[["L2","Cercozoa"]]} reads in the
library derived from the L2 nematodes that had significant similarity
to cercozoan (likely parasite) ribosomal RNA genes (Table 1).

\subsection*{Assembly}

<<load.meth, echo=FALSE>>=
load("/home/ele/thesis/454/Method-assembly/Method.Rdata")

nMN <- nrow(contig.df[contig.df$category%in%"MN",])

nbad.c <- nrow(contig.df[contig.df$category%in%
                         c("M_n","N_n","N_1","M_1"),])

nSing <- nrow(contig.df[contig.df$category%in%0,])
@


We assembled the remaining \Sexpr{length(raw.fasta)} reads (spanning
\Sexpr{nBases} bases) using the combined assembler strategy
\cite{pmid20950480} and Roche 454 GSassembler (version 2.6) and MIRA
(version 3.21) \cite{miraEST}. From this we derived \Sexpr{nMN}
contigs that were supported by both assembly algorithms,
\Sexpr{nbad.c} contigs only supported by one of the assembly
algorithms and \Sexpr{nSing} singletons that were not assembled by
either approach (Table 2). When scored by matches to known genes, the
contigs supported by both assemblers are of the highest credibility,
and this set is thus termed the high credibility assembly
(highCA). Those with evidence from only one assembler and the
singletons are of lower credibility (lowCA). These datasets are the
most parsimonious (having the smallest size) for their quality
(covering the largest amount of sequence in reference
transcritomes). In the highCA parsimony and low redundancy is
prioritized, while in the complete assembly (highCA plus lowCA)
completeness is proiritized. The 40187 sequences (contig consensuses
and singletons) in the complete assembly are referred to below as
tentatively unique genes (TUGs).

<<mean.cov, echo=FALSE, cache=TRUE>>=

cov.mn <- tapply(con.pile$coverage,
                 con.pile$contig%in%contig.df[contig.df$category=="MN", "contig"],
                 mean)
                            
cov.mn.u <- tapply(con.pile.uniq$uniq_coverage,
                   con.pile.uniq$contig%in%contig.df[contig.df$category=="MN", "contig"],
                   mean)

cov.nu.total <- mean(con.pile$coverage)
cov.u.total <- mean(con.pile.uniq$uniq_coverage)


@ 

\SweaveInput{../contamination/post_assembly.Rnw}

We screened the complete assembly for residual host contamination, and
identified
\Sexpr{length(contig.df$seq.origin%in%c("eelmRNA","Chordata"))#$}
  TUGs that had higher, significant similarity to eel (and chordate)
  sequences (our 454 ESTs and EMBLBank Chordata proteins) than to
  nematode sequences \cite{pmid21550347}.

  Given our prior identification of cercozoan ribosomal RNAs, we also
  screened the complete assembly for contamination with other
  transcriptomes, and found \Sexpr{other.king["Fungi"]} TUGs with hits
  to fungi (e.g Ajellomycetaceae, 53 hits),
  \Sexpr{other.king["Viridiplantae"]} TUGs whith hits to plants and
  \Sexpr{other.king["undef"]} hits to Protists (e.g. Trypanosomatidae,
  26 hits and Vahlkampfiidae, 38 hits), Bacteria (mostly
  Proteobacteria, 484 hits) and Viruses (see also additional figure
  phylum\_plots.png).

  No hits were found to Wolbachia or related Bacteria known as
  symbionts of Ecdyosozoans.

  Our assembly thus has \Sexpr{nrow(contig.df[contig.df$Ac,])#$} TUGs,
  spanning \Sexpr{sum(nchar(contig.df[contig.df$Ac,"seq"]))#$} bases
  (of which \Sexpr{nrow(contig.df[contig.df$AcMN,])#$} are
  highCA-derived, and span
  \Sexpr{sum(nchar(contig.df[contig.df$Ac,"seq"]))#$} bases) that are
  likely to derive from of \textit{A. crassus}.

\subsection*{Protein prediction}
\SweaveInput{../conservation/conservation.Rnw}
\SweaveInput{../protein_prediction/imputed_from_p4EST.Rnw}
\SweaveInput{../snp_analysis/snp_analysis.Rnw}
\SweaveInput{../snp_analysis/snp_by_population.Rnw}
\SweaveInput{../annotation/annotation.Rnw}
\SweaveInput{../annotation/read_signalp.Rnw}
\SweaveInput{../annotation/brugia-compare.Rnw}


For
\Sexpr{nrow(contig.df[contig.df$Ac&contig.df$method!="no-prediction",])}
TUGs a protein was predicted using prot4EST
\cite{wasmuth_prot4est:_2004} (Table 2). The full open reading frame
was obtained in
\Sexpr{nrow(contig.df[contig.df$Ac&imputed$full.length,])} TUGs, while
while for \Sexpr{nrow(contig.df[contig.df$Ac&imputed$full.5p,])} the
5’ end and for \Sexpr{nrow(contig.df[contig.df$Ac&imputed$full.3p,])}
the 3' end was complete. In \Sexpr{same.same[["FALSE"]]} TUGs the
corrected sequence with the imputed ORF was slightly changed compared
to the raw sequence.

\subsection*{Annotation}

We obtained basic annotations with orthologous sequences from
\textit{C. elgeans} for \Sexpr{length(all.ce.hit)} TUGs, from
\textit{B. malayi} for \Sexpr{length(all.bm.hit)} TUGs, from nempep
\cite{parkinson_nembase:resource_2004, pmid21550347} for
\Sexpr{n.nempep} TUGs and whith uni-prot proteins for
\Sexpr{n.uniprot} TUGs.

We used annot8r \cite{schmid_annot8r:_2008} to assign gene ontology
(GO) terms for \Sexpr{nGO} TUGs, Enzyme Commission (EC) numbers for
\Sexpr{nEC} TUGs and Kyoto Encyclopaedia of Genes and Genomes (KEGG)
pathway annotations for \Sexpr{nKEGG} TUGs (Table 2). Additionally
\Sexpr{nIPR} highCA derived contigs were anntotated with GO terms
through Interpro Scan \cite{pmid11590104}. Nearly one third
(\Sexpr{nallannotations}) of the \textit{A. crassus} TUGs were
annotated with at least one identifier, and \Sexpr{all.identifier} had
GO, EC and KEGG annotations (Figure 1).

We compared our \textit{A. crassus} GO annotations for high-level
GO-slim terms to the annotations (obtained the same way) for the
complete proteome of the filarial nematode \textit{B. malayi} and the
complete proteome of \textit{C. elegans}. 

Correlation shows the occurence of terms for the partial transcriptome
of \textit{A. crassus} to be more similar to the proteome of
\textit{B. malayi} (\Sexpr{round(Go.rep.Ac.Bm,2)}; Spearman
correlation coefficient) than to the proteome of \textit{C. elegans}
(\Sexpr{round(Go.rep.Ac.Ce,2)}). Also the tow model-nematode compared
ot each other (\Sexpr{round(Go.rep.Bm.Ce,2)}) are less similar in the
occurence of therm.

We inferred presence of signal peptide cleavage sites in the predicted
protein sequence using SignalP \cite{pmid21959131}. We predicted
\Sexpr{nsigP} signal peptide cleavage sites and \Sexpr{nsigPtm} signal
peptides with a transmembrane signature. Again these predictions are
more similar to predictions using the same methods for the proteome
\textit{B. malayi} (\Sexpr{nsigP.Bm} signal peptide cleavage sites and
\Sexpr{nsigPtm.Bm} with transmembrane anchor) than for the proteome of
\textit{C. elegans} (\Sexpr{nsigP.Ce} signal peptide cleavage sites
and \Sexpr{nsigPtm.Ce} with transmembrane anchor).

\subsection*{Evolutionary conservation}

Evolutionary conservation The A. crassus TUGs were classified as
conserved, conserved in Metazoa, conserved in Nematoda, conserved in
Spirurina or novel to A. crassus by comparing them to public databases
and using two BLAST bit-score cutoffs to define relatedness.



****Now describe the data, and reference table 3, add novel in
A. crassus****
 
\subsection*{Identification of single nucleotide polymorphisms}

We called single nucleotide polymorphisms (SNPs) on the 1,412,806
bases of the TUGs that had coverage of more then 8-fold available
using VARScan \cite{pmid19542151}. We excluded SNPs predicted to have
more than 2 alleles or that mapped to an undertermined (N) base in the
reference, and retained 13042 SNPs. The ratio of transitions (ti;
8663) to transversion (tv; 4379) in this set was 1.98. Using the
prot4EST prtedictions and the corrected sequences, 8309 of the SNPs
were predicted to be inside an ORF, with 2713 at codon first positons,
2172 at second positions and 3424 at third positions. As expected
ti/tv inside ORFs (2.43) was higher than outside ORFs (1.42). The
ratio of synonymous polymorphisms per synonymous site to
non-synonymous polymorphisms per non-synonymous site (dn/ds) was 0.45.
We filtered these SNPs to exclude those that might be associated with
analytical bias. As Roche 454 sequences have well-known systematic
errors associated with homopolymeric nucleotide sequences
\cite{pmid21685085}, we analysed the effect of exclusion of SNPs in,
or close to, homopolymer regions. We observed changes in ti/tv and in
dn/ds when SNPs were discarded using different size thresholds for
homoploymer runs and proximity thresholds (see figure
snp\_ex\_parameter\_plots.png). Based on this we decided to exclude SNPs
with a homopolymer-run as long as or longer than 4 bases inside a
window of 11 bases (5 to bases to the right, 5 to the left) around the
SNP. We also observed a relationship between TUG dn/ds and TUG
coverage, associated with the presence of sites with low abundance
minority alleles (less than 7\% of the allele calls), suggesting that
some of these may be errors.  Removing low abundance minority allele
SNPs from the set removed this effect.  Our filtered SNP dataset
includes \Sexpr{nrow(VARqp)} SNPs. We retained 4.44 SNPs per kb of
contig sequence, with 7.87 synonymous SNPs per 1000 synonymous bases
and 2.43 non-synonymous SNPs per 1000 non-synonymous bases. A mean
dn/ds of 0.32 was calculated for the 980 TUGs (858 highCA and lowCA
contigs) containing at least one synonymous SNP.



\subsection*{Polymorphisms associated with biological processes}

\SweaveInput{../snp_analysis/overrepresentation.Rnw}

We consolidated our annotation and polymorphism analyses by examining
correlations between nonsynonymous variability and particular
classifications. Signal peptide containing proteins have been shown to
have higher rates of evolution than cytosolic proteins in a number of
nematode species. In \textit{A. crassus}, TUGs predicted to contain
signal peptide cleavage sites in SignalP had higher dn/ds values than
TUGs without signal peptide cleavage sites (p =
\Sexpr{round(u.test[["p.value"]],3)}; one sided U-test; Figure
sigp\_dn\_ds.png). Proteins predicted to be novel to the Nematoda were
significantly enriched in signal peptide annotation.

(Figure signal\_novel.png ***why is nematoda on the outside? should
order by breadth of conservation***).

Positive selection can be inferred from dn/ds analyses, and we defined
TUGs with a dn/ds higher than 0.5 as positively selected. We
identified over- and under-represented GO ontology terms associated
with these putatively positively selected genes (Table 3). Within the
molecular function category, ``peptidase activity'' was the most
significantly overrepresented term and had twelve TUGs supporting the
overrrepresentation. The highlighted twelve peptidases annotated with
eleven unique orthologs in \textit{C. elegans} and \textit{B. malayi}.
The term ``structural constituent of ribosome'' was underrepresented.
The biological process category provides less usable information for a
nematode (highlighting e.g. brain or pancreas development). The same
is true for the cellular compartment, however unterrepresented terms
can connected to ribosomal proteins, validating the analysis for the
molecular function category.

At a bit score threshold of 80 sequences novel in clade III had a
significantly higher dn/ds than other sequences (p = 0.038;
\Sexpr{round(u.novel.80[["p.value"]],3)}; two sided U-test). At a
bitscore thershold of 50 results were non-significant but showed the
same trend (Figure conservation\_dn\_ds.png).

\subsection*{Differential expression}
                
\SweaveInput{../gene_expression/gene_expression.Rnw}

sing methods developed for sequencing data, we analyzed
gene-expression inferred from mapping. Of the \Sexpr{n454.map} reads
mapping to the fullest assembly \Sexpr{n454.Umap} mapped uniquely
(with their best hit) and were counted on a per library base.

\subsection*{Comparison with tag-sequencing pilot data-set}

%% \SweaveInput{../solexa_compare/solexa_compare.Rnw}

\Sexpr{nMapped} of \Sexpr{nTags} (\Sexpr{nUniq} unique) NlaIII-tags
mapped to the fullest assembly. Only \Sexpr{nUMapped}
(\Sexpr{nUniqUMapped} unique) tags did not map to any sequence in the
fullest assembly.

Table 9a gives correlations coefficients between tag-counts and
454-libraries. Correlations-coefficients between 454-libraries were
generally low, indicating a high proportion of noise or biological
differences between samples. Correlation between expression-tags and
454-read counts were even lower. However when only analyzing counts to
good-quality contigs, correlation coefficients improved both between
libraries and between 454-libraries and solexa-tags (see table 9b). No
further improvements were made, when counts were limited to contigs
surely \textit{A. crassus} (see table 9c). Correlations between
library T2 and other 454-libraries, as well as with solexa-tag counts
were lower than between other libraries.

To gain power in statistical analysis we limited the set of
gene-objects analyzed for differential expression to the good-quality
contigs.

\subsubsection*{Differential expression between male and female worms}

Despite the lack of replicates for male worms we were able to identify
\Sexpr{length(sig.mf.contigs)} sequences being significantly
over-expressed in male worms. In fact all these TUGs were nearly
exclusively expressed in males. 

\subsubsection*{Differential expression between adults and L2-larvae}

For the L2-library we changed our approach and used gene-expression
analysis rather to highlight the off-target data in this library. For
this reason we used counts for the fullest assembly.

\Sexpr{length(sig.ad.contigs)} sequences being expressed exclusively
in L2 library were strongly enriched in sequences being labeled as
possible off-target data in taxonomic classification. From these
sequences only \Sexpr{l2.met.hits} had best hits to metazoa and only
\Sexpr{l2.nem.hits} to nematoda.

\subsubsection*{Differential expression between worms from the European
  and Japanese eel}

None of the TUGs in the present evaluation showed significant
differential expression between worms from the European and Japanese
Eel. Diagnostic plots provided by \texttt{DESeq} made clear, that both
depth of sequencing and number of replicates have to be higher
contrasting these conditions.

However, comparing expression-analysis on the full data-set to analysis
limited to the high quality of reliable \textit{A. crassus}-contigs it
was clear that the quality-data-set reduces within-condition variance
and results were closer to significance: The lowest adjusted p-values
for the cleaned data-set were around 0.4, while on the full data-set
only adjusted p-values above 0.8 could be obtained.


<<save, echo=FALSE>>=
## write the MN.peptide sequence for iprscan prediction
MN.pep.fasta <- as.character(contig.df[contig.df$AcMN, "Ac.pep"])
names(MN.pep.fasta) <- as.character(contig.df[contig.df$AcMN, "contig"])
write.sequence(MN.pep.fasta, "/home/ele/thesis/454/MN.pep.fasta")

## for a more readable version of contig.df
readable <- !names(contig.df)%in%c("seq", "imp", "ontology", "coding", "Ac.pep")
write.csv(contig.df, "../A_crassus_contigs_full.csv")
write.csv(contig.df[,readable], "../A_crassus_contigs_readable.csv")

## for summary statistics a non-Ac reduced dataset
c.df <- contig.df[contig.df$Ac, ]

save.image("/home/ele/thesis/454/paper/paper.Rdata")
@ 

\section*{Discussion}

We have generated a de novo transcriptome for A. crassus an important
invasive parasite that threatens wild stocks of the European eel
An. anguilla. These data enable a broad spectrum of molecular research
on this ecologically and economically important parasite. As
A. crassus lives in close association with its host, we have used
exhaustive filtering to attempt to remove all host-derived, and
host-associated organism-derived contamination from the data. To do
this we have also generated a transcriptome dataset from the
definitive host An. japonica. The non-nematode, non-eel data
identified, particularly in the L2 sample, showed highest identity to
flagellate protists, which may have been parasitising the eel (or the
nematode). Encapsulated objects observed in eel swim bladder walls
\cite{heitlinger_massive_2009} could be due solely to immune attrition
of A. crassus larvae or to other coinfections.

A second examination of sequence origin was performed after assembly,
employing higher stringency cutoffs.  Similar taxonomic screening was
used in a garter snake transcriptome project \cite{pmid21138572}, and
an analysis of lake sturgeon tested and rejected hypotheses of
horizontal gene-transfer when xenobiont sequences were identified
\cite{pmid20386959}. A custom pipeline for transcriptome assembly from
pyrosequencing reads \cite{pmid20034392} proposed the use of EST3
\cite{pmid17218127} to infer sequence origin based simply on
nucleotide frequency. We were not able to use this approach
successfully, probably due to the fact that xenobiont sequences in our
data set derive from multiple sources with different GC content and
codon usage.

Compared to other NGS transcriptome sequencing projects
[references???], the combined assembly approach generated a smaller
number of contigs that had lower redundancy and higher
completeness. Projects using the mira assembler often report
substantially greater numbers of contigs for datasets of similar size
(see e.g. \cite{pmid21364769}), comparable to the mira sub-assembly in
our approach. The use of oligo(dT) to capture mRNAs probably explains
the bias towards 3' end completeness and a relative lack of true
initiation codons in our protein prediction. This bias is
near-ubiquitous in deep transcriptome sequencing projects
(e.g. \cite{pmid20331785}).

We generated transcriptome data from multiple A. crassus of Taiwanese
and European origin, and identified SNPs both within and between
populations. Screening of SNPs in or adjacent to homopolymer regions
improved overall measurements of SNP quality. The ratio of transitions
to transversions (ti/tv) increased. Such an increase is explained by
the removal of “noise” associated with common homopolymer errors
[27]. The value of 2.38 (1.82 outside, 2.74 inside ORFs) is in good
agreement with the overall ti/tv of humans (2.16 \cite{pmid21169219})
or Drosophila (2.07 \cite{pmid21143862}). The ratio of non-synonymous
SNPs per non-synonymous site to synonymous SNPs per synonymous site
(dn/ds) decreased with removal of SNPs adjacent to homopolymer regions
from 0.45 to 0.32 after full screening. The most plausible explanation
is the removal of error, as unbiased error would lead to a dn/ds of
1. While dn/ds is not unproblematic to interpret within populations
\cite{pmid19081788}, the assumption of negative (purifying) selection
on most protein-coding genes makes lower mean values seem more
plausible. We used a threshold value for the minority allele of 7\%
for exclusion of SNPs, based on an estimate that approximately 10
haploid equivalents were sampled (5 individual worms plus an
negligible contribution from L2 larvae in the L2 library and within
the female adult worms). The benefit of this screening was mainly a
reduction of non-synonymous SNPs in high coverage contigs, and a
removal of the dependence of dn/ds on coverage. Working with an
estimate of dn/ds independent of coverage, efforts to control for
sampling a biased by sampling depth (i.e. coverage; see
\cite{pmid18590545} and \cite{pmid20478048} ) could be avoided.


 


***
When the whole of coding sequences are studied, of which only a small
subset of sites can be under diversifying selection, dn/ds of ~0.5 has
been suggested as threshold for assuming diversifying selection
\cite{pmid15579698} instead of the classical threshold of 1
\cite{pmid6449605}. In the transcripts from the female reproductive
tract of \textit{Drosophila} dn/ds was 0.15 \cite{pmid15579698} and in
the 0.21 male reproductive tract \cite{pmid11404480} (although for ESTs
specific to the male accessory gland were shown to have a higher dn/ds
of 0.47). Pyrosequencing studies found dn/ds to be between 0.13 and
0.27 (depending on tissue type genes were mainly expressed in) in the
Zebra finch transcriptome \cite{pmid20359325}, 0.12 in the
transcriptome of \textit{Tigriopus californicus} \cite{pmid21199025}
and 0.3 in the parasitic nematode \textit{Ancylostoma canium}
\cite{pmid20470405}. In comparison with these results even our
estimate after screening seems high (although it should be noted, that
the latter tree studies report a mean dn/ds over contigs - the
\textit{A. canium} doesn't make clear what exactly is reported - and
therefore the value has to be compared to our mean dn/ds over contigs
of \Sexpr{round(mean(contig.df$dn.ds,na.rm=TRUE),2)#$}) and further
investigation using deeper sequencing of more individuals on the
solexa GAII platform will be used to fully exclude the possibility of
this result being induced by sequencing error. Moreover such an
experiment should try to test that divergence between populations is
leading to positive selection on only the possibly diverging European
populations. For such a study the set of SNPs found here are
invaluable, as it can be used to define a gold standard set of SNPs
found with both technologies.

We were able to obtain high-quality annotations for a large set of
TUGs. Comparison with protein sequence derived from \textit{B. malayi}
showed a a remarkable degree of agreement regarding the occurrence of
terms. This implies, that our transcriptome-data-set is a
representative subset of a nematode-parasite
genome. 

Over-representation of GO-term in genes under diversifying selection
(at a threshold of dn/ds>0.5, as established above) highlighted many
interesting gene-products:

In the molecular function category two amino acid transmembrane
transporters (``Contig5699'' and ``Contig866'') - the only contigs
with this annotation (or annotation, which is an offspring-term of
this) and a dn/ds obtained - were found to have a dn/ds>0.5. Such
transporter are thought to be important in the survival of parasites in
a host \cite{pmid18272295}.

Enrichment in the category ``peptidase activity'' highlighted twelve
peptidases (from 43 with a dn/ds obtained). All twelve have orthologs
in \textit{B. malayi} and \textit{C. elgans} and are conserved across
kingdoms. Despite their conservation peptidases are thought to have
have acquired new and prominent roles in host-parasite interaction
compared to free living organisms: In \textit{A. crassus} a
trypsin-like proteinase has been identified thought to be utilized by
the tissue-dwelling L3 stage to penetrate host tissue and an aspartyl
proteinase thought to be a digestive enzyme in adults
\cite{polzer_identification_1993}.

The under-representation of ribosomal proteins (term ``structural
constituent of ribosome'') in disruptively selected contigs is in good
agreement with the notion that ribosomal proteins are extremely
conserved across kingdoms \cite{pmid9664699} and should be under under
strong negative selection. 

The additional prediction of signal sites for cleavage allowed
interpretation and cross-validation of the results from SNP-calling:
The detection of signal-peptides secretion using \textit{in silico}
analysis of ESTs has been used to highlight candidate genes for
example in \textit{Nippostrongylus brasiliensis}
\cite{harcus_signal_2004} and in a large scale analysis across all
nematode \cite{nagaraj_needles_2008} ESTs. Proteomic analysis in
\textit{B. malayi} \cite{pmid19352421,pmid18958170} and
\textit{Heligmosomoides polygyrus} \cite{pmid21722761} was able to
find evidence for excretion for some of the protein-products and to
highlight additional candidate genes.

We found an elevated dn/ds for signal-positives. These result could be
explained follow the logic of signal-positives being more likely to be
secreted to the host-parasite interface and proteins involved in
host-parasite interaction being more likely to be under disruptive
selection. Signal-positive TUGs with high dn/ds constitute another set
of genes worth further examination in future studies.

TUGs predicted to be novel in the phylum nematoda contained the
highest proportion of signal-positives. A interpretation of this
findings could be a confirmation of a study on \textit{Nippostrongylus
  brasiliensis} \cite{harcus_signal_2004}, where signal positives were
reported as less conserved. In the present study we did not aim to
identify ``novelty to \textit{A. crassus}'' as we believe in a deep
sequencing project he absence of sequence similarity could be
attributed to erroneous sequence instead of true novelty, and thereby
blur analysis. However novelty in nematodes and to a lesser extend
novelty in Spirurina seems to support the notion, that - if not
diversified within nematoda to an extend leading to a complete loss of
similarity, like suggested in the mentioned study - signal positives
in nematodes could have taken a divergent evolutionary path from their
orthologs in other phyla.

It was within our expectation, that expression analysis failed to give
conclusive results, as the present data-set is not fully adequate for
this kind of analysis: First we did not include replicates for
libraries of male adults as well as for L2-larvae. Second one of the
replicates for female worms (library E1) resulted in a low amount of
sequence mappable to protein-coding (non-rRNA) genes.
However some of the results are still valuable:

DESeq was able to report genes significantly differing in expression
between male and female worms and between the L2 library and the all
other worms. This was possible for male worms as well as for
L2-larvae, were no replicated samples were obtained, due due the
special features of this package \cite{pmid20979621}. However only
over-expression in non-repeated samples can be detected, as obviously
lack of expression in one sample can't validate

Comparisons were lacking significance, as methods are designed for
deeper sequencing and more importantly more replicates would be
needed. Differences between the L2-library and other libraries were
mainly due to off-target data, and TUGs solely found in the L2 library
are ... 



%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}

  
%%%%%%%%%%%%%%%%%%
\section*{Methods}


\subsection*{Nematode samples, RNA extraction, cDNA synthesis and Sequencing}

\textit{A. crassus} from \textit{JAn. japonica} were sampled from
Kao-Ping river and an adjacent aquaculture in Taiwan as described in
\cite{heitlinger_massive_2009}. Worms from An. anguilla were sampled
in Sniardwy Lake, Poland (53.751959N, 21.730957E) and from the
Linkenheimer Altrhein, Germany (49.0262N, 8.310556E). After
determination of the sex of adult nematodes, they were stored in
RNA-later (Quiagen, Hilden, Germany) until extraction of RNA. RNA was
extracted from individual adult male and female nematodes and from a
population of L2 larvae (Table 1). RNA was reverse transcribed and
amplified into cDNA using the MINT-cDNA synthesis kit (Evrogen,
Moscow, Russia).  For host contamination screening a liver-sample from
an uninfected A. japonica was also processed.  Emulsion PCR was
performed for each cDNA library according to the manufacturer’s
potocols (Roche/454 Life Sciences), and sequenced on a Roche 454
Genome Sequencer FLX. All samples were sequenced using the FLX
Titanium chemistry, except for the taiwanese female sample T2, which
was sequenced using FLX standard chemistry, to generate between 99,000
and 209,000 raw reads. For the L2 larval library, which had a larger
number of non-A. crassus, non-An. anguilla reads, we confirmed that
these data were not laboratory contaminants by screening Roche 454
data produced on the same run in independent sequencing lanes.


\subsection*{Trimming, quality control and assembly}

Raw sequences were extracted in fasta format (with the corresponding
qualities files) using sffinfo (Roche/454) and screened for adapter
sequences of the MINT-amplification-kit using cross-match \cite{PHRAP}
(with parameters -minscore 20 and -minmatch 10). Seqclean
\cite{tgicl_pertea} was used to identify and remove poly-A-tails, low
quality, repetitive and short (<100 base) sequences. All reads were
compared to a set of screening databases using BLAST (expect value
cutoff E<1e-5, low complexity filtering turned off: -F F). The
databases used were (a) a host sequence database comprising an
assembly of the An. japonica Roche 454 data, an unpublished assembly
of An. anguilla Sanger dideoxy sequencesd expressed sequence tags
(made available to us by Gordon Cramb, University of St Andrews) and
transcripts from from EeelBase \cite{pmid21080939} a publically
availble transcriptome database for the European eel; (b) a database
of ribosomal RNA (rRNA) sequences from eel species derived from our
Roche 454 data and EMBL-Bank; and (c) a database of rRNA sequences
identified in our A. crassus data by comparing the reads to known
nematode rRNAs from EMBL-Bank. This last database notably also
contained xenobiont rRNA sequences. Reads with matches to one of these
databases over more than 80\% of their length and with greater than
95\% identity were removed from the dataset. Screening and trimming
information was written back into sff-format using sfffile (Roche
454). The filtered and trimmed data were assembled using the combined
assembly approach \cite{pmid20950480}, combining assemblies from the
mira \cite{miraEST} and newbler \cite{pmid16056220}. ****Give the
details here and we will trim the text later **** . The two assemblies
were combined into one using Cap3 \cite{Cap3_Huang} at default
settings and contigs labeled by whether they derived from both
assemblies or one assembly only.

\subsection*{Post-assembly classification and taxonomic assignment of
  contigs}

After assembly contigs were assessed a second time for host and other
contamination by comparing them (using BLAST) to the three databases
defined above, and also to nembase4, a nematode transcriptome database
derived from whole genome sequencing and EST assemblies
\cite{parkinson_nembase:resource_2004, pmid21550347}. For each contig,
the highest-scoring match was recorded as long as it spanned more than
50\% of the contig. We also compared the contigs to the NCBI
non-redundant nucleotide (NCBI-nt) and protein (NCBI-nr) databases,
recording the taxonomy of all best matches with expect values better
than 1e-05.

\subsection*{Protein prediction and annotation}


Protein translations were predicted from the contigs using prot4EST
(version 3.0b) \cite{wasmuth_prot4est:_2004}. Proteins were predicted
either by joining single high scoring segment pairs (HSPs) from a
BLAST search of uniref100 \cite{pmid18836194}, or by ESTscan
\cite{estscan}, using a training data the \textit{Brugia malayi}
complete proteome back-translated using a codon usage table derived
from the BLAST HSPs, or, if the first two methods failed, simply the
longest ORF in the contig. For contigs where the proein prediction
required insertion or deletion of bases in the original sequence, we
also imputed an edited sequence for each affected contig. Annotations
with Gene Ontology (GO), Enzyme Commission (EC) and Kyoto
Encyclopaedia of Genes and Genomes (KEGG) terms were inferred for
these proteins using Annot8r (version 1.1.1)
\cite{schmid_annot8r:_2008}, using the annotated sequences available
in uniref100 \cite{pmid18836194}. Up to 10 annotations based on a
BLAST similarity bitscore cut-off of 55 were obtained for each
annotation set. The complete \textit{B. malayi} proteome (as present
in uniref100) and the complete \textit{C. elegans} proteome (as
present in wormbase v.220) were also annotated in the same
way. SignalP V4.0 \cite{pmid21959131} was used to predict signal
peptide cleavage sites and signal anchor signatures.

\subsection*{Single nucleotide polymorphism analysis}

We mapped the raw reads against the the complete set of contigs,
replacing imputed sequences for originals where relevant, using ssaha2
(with parameters -kmer 13 -skip 3 -seeds 6 -score 100 -cmatch 10
-ckmer 6 -output sam -best 1). From the ssaha2 output, pileup-files
were produced using samtools
\cite{journals/bioinformatics/LiHWFRHMAD09}, discarding reads mapping
to multiple regions. VarScan \cite{pmid19542151} (pileup2snp) was used
with default parameters on pileup-files to output lists of single
nucleotide polymorphisms (SNPs) and their locations.

% \subsection*{Respiration analysis}

% We extracted contigs annotated with ``GO:0009060 aerobic respiration''
% and ``GO:0009061 anaerobic respiration''

\subsection*{Gene-expression analysis}

For Roche 454 data, read counts for each transcript were obtained from
the mapping to imputed sequence performed for SNP
analyses. Tag-sequences were mapped using BWA \cite{pmid19451168}. And
read counts extracted using Samtools
\cite{journals/bioinformatics/LiHWFRHMAD09}. For deepSAGE
NlaIII-tag-sequencing, total RNA was prepared as described above from
a female nematode from the Polish sampling site. A deepSAGE library
was constructed following the protocol supplied by Illumina. Briefly
after synthesis of cDNA on oligo(dT)-beads, cDNA was digested with the
NlaIII (recognition site CATG), and the oligo(dT)-anchored 3' ends of
mRNAs retained. After ligation of an adaptor containing an MmeI
restriction site, the type II enzyme MmeI was used to cut 17 bases
from the 3' end fragment, genrating a 21 base tag, expected to be
unique for most mRNAs. The R-package DESeq \cite{pmid20979621} was
used to normalize for library size and analyse statistical
significance of differential expression of both Roche 454 and deepSAGE
data. Spearman correlation coefficients were calculated for raw
counts.

\subsection*{General coding methods}

The bulk of analysis (unless otherwise cited) presented in this paper
was carried out in R \cite{R_project} using custom scripts. We used a
method provided in the R-packages
Sweave\cite{lmucs-papers:Leisch:2002} and Weaver\cite{weaver} for
``reproducible research'' combining R and \TeX code in a single
file. All intermediate data files needed to compile the present
manuscript from data-sources are provided upon request. For
visualistaion we used the R-packages lattice\cite{lattice-book} and
ggplot2\cite{ggplot-book}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Competing interests}
The authors declare no competing interests.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgments}
\ifthenelse{\boolean{publ}}{\small}{}

The work of EGH is funded by
Volkswagen Foundation, "F\"{o}rderinitiative Evolutionsbiologie".
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article} % Style BST file
  \bibliography{/home/ele/bibtex/master,/home/ele/bibtex/master2}
} % Bibliography file (usually '*.bib' )

%%%%%%%%%%%


\ifthenelse{\boolean{publ}}{\end{multicols}}{}
\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}

  \subsection*{Figure 1 - rRNA-screening statistics}
 
  Annotations obtained for Gene Ontology (GO), Enzyme Commission (EC)
  and Kyoto Encyclopaedia of Genes and Genomes (KEGG) terms through
  Annot8r\cite{schmid_annot8r:_2008} for all TUGs (a) and for higCA
  derived contigs (b). The latter includes annotations obtained with
  interpro scan \cite{pmid11590104}.

  \subsection*{Figure 2 - Contig-coverage}

 
\subsection*{Figure 3 -Sequences novel in nematodes are enriched for
  Signal-positives}

Figure 3 gives the proportions of \texttt{SignalP}-predictions for
each category of evolutionary conservation. Generally - across
bit-score thresholds - sequences novel in nematodes contained the
highest proportion of signal-positives.

  \subsection*{Figure 4 - }

\subsection*{Figure 5 - }

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
  \subsection*{Table 1 - Sampling, trimming and pre-assembly screening}

<<echo=FALSE, results=tex>>=
## print(xtable(TRIM, display=rep("d", times=7)), floating=FALSE)
trim_mark <- transform(TRIM, lowqal = short+lowq+dust+shortq)[1:6, ]
trim_mark <- trim_mark[,c(1,7)]

trim_mark <- merge(pre.screen.tab,
                   trim_mark, by="row.names")

trim_mark$life.st <- c("adult f", "adult f", "L2 lavae",
                       "adult m", "adult f", "adult f")

trim_mark$source.p <- c("Europe R", "Europe P", "Europe R",
                       "Asia C", "Asia C", "Asia W")

trim_mark[7,] <- c("Eel", "", "", "", "", length(eel.clean), sum(nchar(eel.clean)),
                   length(eel.all), length(eel.all)-length(eel.clean), "liver", "Taiwan")

trim_mark <- trim_mark[, c(1, 10, 11,  8, 9, 2, 4, 5, 3, 6, 7 )]

names(trim_mark)[c(1, 4)] <- c("library", "raw.reads")

write.csv(trim_mark, "/home/ele/thesis/454/paper/table1.csv")

trim_mark.tab <- xtable(trim_mark)
print(trim_mark.tab, floating=FALSE, include.rownames=FALSE)
@   

\subsection*{Table 2 - assembly classification and summarized contig stats}

<<echo=FALSE, results=tex>>=
T2 <- as.data.frame.array(tapply(contig.df$contig, contig.df$category%in%"MN", length))
rownames(T2) <- c("lowCA", "highCA")
names(T2) <- "total.contigs"
T2$rRNA.contigs <- table(contig.df$category%in%"MN", contig.df$seq.origin%in%"AcrRNA")[, 2]
T2$fish.contigs <- table(contig.df$category%in%"MN",
                         contig.df$seq.origin%in%c("eelmRNA", "Chordata"))[, 2]
T2$xeno.contigs <- table(contig.df$category%in%"MN",
                         !contig.df$seq.origin%in%c("Nematoda", "No_hit",
                                                  "eelmRNA", "Chordata", "AcrRNA"))[, 1]
T2$remaining.contigs <- table(contig.df$category%in%"MN", contig.df$Ac)[, 2]
T2$remaining.span <- tapply(contig.df$seq,
                            contig.df$category%in%"MN",
                            function (x) sum(nchar(as.character(x))))
T2$non.u.cov <- cov.mn
T2$cov <- cov.mn.u

T2$p4e <- table(c.df$category%in%"MN", gsub("p4e->", "", c.df$method))
T2$full.3p <- table(c.df$category%in%"MN", c.df$full.3p)[,2]
T2$full.5p <- table(c.df$category%in%"MN", c.df$full.5p)[,2]
T2$full.l <- table(c.df$category%in%"MN", c.df$full.length)[,2]

T2$GO <-  table(c.df$category%in%"MN", c.df$contig%in%GO.annot$pept_id)[, 2]
T2$EC <-  table(c.df$category%in%"MN", c.df$contig%in%EC.annot$pept_id)[, 2]
T2$KEGG <-  table(c.df$category%in%"MN", c.df$contig%in%KEGG.annot$pept_id)[, 2]
T2$IPR <-  table(c.df$category%in%"MN", c.df$contig%in%IPR.annot$pept_id)[, 2]

T2$nem.blast <- table(c.df$category%in%"MN", c.df$phylum.nr%in%"Nematoda")[, 2]

T2$any.blast <- table(c.df$category%in%"MN", c.df$phylum.nr%in%"No_hit")[,1]

T2 <- as.data.frame(t(T2))
T2 <- transform(T2, combined=lowCA+highCA)

T2["non.u.cov", "combined"] <- cov.nu.total
T2["cov", "combined"] <- cov.u.total

dig <- t(matrix(c(rep(0,24), rep(c(0, 3 , 3, 3), 2), rep(0,48)), nrow=4))
T2.tab <- xtable(T2, digits=dig)

print(T2.tab)
@ 

Tabel describing summary statistics for contigs from different
assembly-categories given in colums as highCA = high credibility
assembly; lowCA = low credibility assembly, CA = complete assembly.

Rows indicate summary statistics: total.contigs = numbers of total
contigs, fish.contigs = number of contigs hitting eel-mRNA or Chordata
in NCBI-nr or NCBI-nt (screened out), xeno.contigs = number of contigs
with best hit (NCBI-nr and NCBI-nt) to non-eukaryote (screened out),
remaining.contigs = number of contigs remaining after this screening,
remaining.span = total length of remaining contigs, non.u.cov =
non-unique mean base coverage of contigs, cov = unique mean base
coverage of contigs, p4e.X = number protein predictions derived in
p4e, where X describes the method of prediction (see Methods), full.3p
= number of contigs complete at 3', full.5p = number of contigs
complete at 5', GO = number of contigs with GO-anntotation, KEGG =
number of contigs with KEGG-anntotation, EC = number of contigs with
EC-anntotation, nem.blast = number of contigs with blast-hit to
nematode in nr, any.blast = number of contigs with blast-hit to
non-nematode (eukaryote non chordate) sequence in NCBI-nr.

\subsection*{Table 3 - Evolutionary conservation }
<<echo=FALSE, results=tex>>=
nov.tab <- rbind(summary.factor(contig.df$novel.50), 
                 summary.factor(contig.df$novel.80))[, c(1,3,4,2)]
rownames(nov.tab) <- c("bit.threshold.50", "bit.threshold.80")
print(xtable(nov.tab), floating=FALSE)

@ 

\subsection*{Table 4 - Over- and under-representation of GO-terms in
  positively selected}

GO-terms over- or under-represented (direction) in contigs putatively
under positive selection. Horizontal lines seratate categories of the
GO-ontology. First category is molecular function, then biological
process last cellular compartment. P values (Pval) for over- or
under-representation are given along with the number of positively
selected contigs (Count; dn/ds $>$ 0.5) and the number of contigs with
this annotation for wich a dn/ds was obtained (Size) and the
description of the GO-term (Term).

<<echo=FALSE, results=tex>>=
mf <- mf.dn.ds[, c(2, 5:8)]
bp <- bp.dn.ds[, c(2, 5:8)]
cc <- cc.dn.ds[, c(2, 5:8)]
nhlines <- c(0, nrow(mf), nrow(mf) + nrow(bp), nrow(mf) + nrow(bp) + nrow(cc))

onto.tab <- xtable(rbind(mf, bp, cc))

align(onto.tab) <- "rrrrp{4cm}r"

print(onto.tab, tabular.environment="longtable",
      floating=FALSE,
      hline.after=nhlines, include.rownames=FALSE)
@ 



\subsection*{Table 9 - Correlation between read-counts in
  454-libraries and solexa-tags}

\newpage

\subsubsection*{Table 9 a - analysing all TUGs}

<<echo=FALSE, results=tex>>=

print(xtable(cor.count.tab, digits=3), floating=FALSE)

@ 
\subsubsection*{Table 9 b - analysing good-category contigs only}

<<echo=FALSE, results=tex>>=

print(xtable(cor.count.tab.MN, digits=3), floating=FALSE)

@ 

\subsubsection*{Table 9 c - analysing good-category contigs surely from \textit{A. crassus} only}

<<echo=FALSE, results=tex>>=

print(xtable(cor.count.tab.MN.AC, digits=3), floating=FALSE)

@ 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Additional Files}

File A\_crassus\_contigs\_full.csv lists all data computed on the contig
level, including sequences (raw, coding, imputed). File
A\_crassus\_contigs\_readable.csv lists only the metadata not including
sequences.

\end{bmcformat}
\end{document}
