%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
\usepackage{rotating}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}


\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\usepackage{/home/ele/tools/R-devel/share/texmf/tex/latex/Sweave}
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of \textit{Anguillicola crassus} sampled by
    pyrosequencing}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel G Heitlinger\correspondingauthor$^{1,2}$% 
       \email{Emanuel G Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Stephen Bridgett$^{3}$% 
       \email{Stephen Bridgett- sbridget@staffmail.ed.ac.uk}%
       Anna Montazam$^{3}$% 
       \email{Anna Montazam- Anna.Montazam@ed.ac.uk}%
       Horst Taraschewski$^1$% 
       \email{Horst Taraschewski- dc20@rz.uni-karlsruhe.de}%
       and Mark Blaxter$^2$% 
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute 1, University of Karlsruhe,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
        \iid(3)The GenePool Sequencing Service, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} \textit{Anguillicola crassus} has been
  introduced from Asia, where it parasitises the Japanese eel
  \textit{Anguilla japonica}, to Europe 30 years ago. Here it
  parasitises the endangered, commercially exploited European eel
  \textit{Anguilla anguilla}, but differences in the parasite's
  phenotype and host relations (natural versus novel host) are not
  understood yet. Phylogenetics places \textit{A. crassus} at a key
  position for the emergence of parasitism, basal to one of the major
  clades of parasitic nematodes.
  \paragraph*{Results:}
  After extensive screening of 756,363 raw pyrosequencing reads, we
  assembled 353,055 into 11,372 contigs spanning 6,575,121 bases and
  additionally obtained 21,153 singletons and lower quality contigs
  spanning 6,157,974 bases. We obtained annotations for roughly 55\%
  of the contigs and roughly 30\% of the tentatively unique genes
  (TUGs) confirming the high quality of especially the contigs. We
  identified 5,112 high quality single nucleotide polymorphisms (SNPs)
  and suggest 199 of them as most suitable markers for
  population-genetic studies. The correlation between different
  analyses provided further insights and confirmed biologically
  relevant expectations: we found an overabundance of predicted signal
  peptide cleavage sites in sequence conserved in Nematoda and novel
  in \textit{A. crassus}, correlations between coding polymorphism and
  differential expression and between evolutionary conservation and
  presence of orthologs with lethal RNAi-phenotypes in
  \textit{C. elegans}. GO-term analysis identified an enrichment of
  peptidases and subunits of the respiratory chain for transcripts
  under positive selection. Enzymes for energy metabolism were also
  found enriched in genes differentially expressed between European
  and Asian \textit{A. crassus}.
  \paragraph*{Conclusions:}
  The transcriptome of \textit{A. crassus} is a basis for molecular
  research on this important species. It furthermore has the potential
  to provide unique insights into the evolution of parasitism in the
  Spirurina. We identified energy metabolism as a candidate phenotype
  for differences between European and Asian worms due to modification
  or even divergent evolution of gene expression.
\end{abstract}

\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

The nematode \textit{Anguillicola crassus} Kuwahara, Niimi et Itagaki,
1974 is a native parasite of the Japanese eel \textit{Anguilla
  japonica} \cite{kuwahara_Niimi_Itagaki_1974}. Adults localise to the
swim bladder where they feed on blood
\cite{polzer_identification_1993}. Larvae are transmitted via
crustacean intermediate hosts
\cite{de_charleroy_life_1990}. Originally endemic to East-Asian
populations of the Japanese eel (\textit{Anguilla japonica}),
\textit{A. crassus} has attracted interest due to recent anthropogenic
expansion of its geographic and host ranges to Europe and the European
eel (\textit{Anguilla anguilla}). Recorded for the first time in 1982
in North-West Germany \cite{fischer_teichwirt}, where it was most
likely introduced through live-eel trade
\cite{koops_anguillicola-infestations_1989, koie_swimbladder_1991},
\textit{A. crassus} has spread rapidly through populations of its
newly acquired host \cite{kirk_impact_2003}. At the present day it is
found in all \textit{An. anguilla} populations except those in Iceland
\cite{kristmundsson_parasite_2007}. \textit{A. crassus} can be
regarded as a model for invasive parasite introduction and spread
\cite{taraschewski_hosts_2007}.

In its colonised host prevalence and mean intensity of infection are
higher than in \textit{An. japonica} \cite{mnderle_occurrence_2006,
  lefebvre_anguillicolosis:_2004}, which is accompanied by a larger
body mass of adult worms, an earlier onset of reproduction and a
larger egg output \cite{knopf_differences_2004}. These modifications
of the lifecycle as well as the virulence of \textit{A. crassus} in
its new host have been attributed to an inadequate immune response in
\textit{An. anguilla} \cite{knopf_swimbladder_2006}. Only
\textit{An. japonica} is capable of killing histotropic larvae of the
parasite after vaccination \cite{knopf_vaccination_2008} or under high
infection pressure \cite{heitlinger_massive_2009}. Accordingly mainly
\textit{An. anguilla} is affected by pathology, including thickening
and inflammation of the swim bladder wall \cite{wurtz_tara_2000}.

The genus Anguillicola is placed in the nematode suborder Spirurina
(clade III \textit{sensu} \cite{blaxter_molecular_1998})
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
are exclusively parasitic and include important human pathogens (the
causative agents of filariasis and ascariasis) as well as prominent
veterinary parasites. Molecular phylogenetic analyses place
Anguillicola in a clade of spirurine nematodes (Spirurina B of
\cite{dl_py}) that have an freshwater or marine intermediate host, but
infect a wide range of carnivorous definitive hosts. Spirurina B is
sister to the main Spirurina C, including the agents of filariasis and
ascariasis), and thus \textit{A. crassus} may be used as an outgroup
taxon to understand the evolution of parasitic phenotypes in these
species.

Recent advances in sequencing technology (often termed Next Generation
Sequencing; NGS), provide the opportunity for rapid and cost-effective
generation of genome-scale data. The Roche 454 platform
\cite{pmid16056220} offers longer reads than other NGS technologies,
and thus is suited to de novo assembly of genome-scale data in
previously understudied species. Roche 454 data has particular
application in transcriptomics \cite{pmid20950480}. The difference in
the biology of \textit{A. crassus} in \textit{An. japonica}
(coevolved) and \textit{An. anguilla} (recently captured) eel hosts
likely results from an interaction between different host and parasite
responses, underpinned by definitive differences in host genetics, and
possible genetic differentiation between the invading European and
endemic Asian parasites. As part of a programme to understand the
invasiveness of \textit{A. crassus} in \textit{An. anguilla}, we are
investigating differences in gene expression and genetic distinction
between invading European and endemic Asian \textit{A. crassus}
exposed to the two different host species. Here we report on the
generation of a reference transcriptome for \textit{A. crassus} based
on Roche 454 data, and explore patterns of gene expression and
diversity.


%%%%%%%%%%%%%%%%%%
\section*{Methods}


\subsection*{Nematode samples, RNA extraction, cDNA synthesis and Sequencing}

\textit{A. crassus} from \textit{An. japonica} were sampled from
Kao-Ping river and an adjacent aquaculture in Taiwan as described in
\cite{heitlinger_massive_2009}. Worms from \textit{An. anguilla} were
sampled in Sniardwy Lake, Poland (53.751959N, 21.730957E) and from the
Linkenheimer Altrhein, Germany (49.0262N, 8.310556E). After
determination of the sex of adult nematodes, they were stored in
RNA-later (Quiagen, Hilden, Germany) until extraction of RNA. RNA was
extracted from individual adult male and female nematodes and from a
population of L2 larvae (Table 1). RNA was reverse transcribed and
amplified into cDNA using the MINT-cDNA synthesis kit (Evrogen,
Moscow, Russia). For host contamination screening a liver-sample from
an uninfected \textit{An. japonica} was also processed. Emulsion PCR
was performed for each cDNA library according to the manufacturerâ€™s
protocols (Roche/454 Life Sciences), and sequenced on a Roche 454
Genome Sequencer FLX.  Raw sequencing reads are archived under
study-accession number SRP010313 in the NCBI Sequence Read Archive
(SRA; http://www.ncbi.nlm.nih.gov/Traces/sra) \cite{pmid22140104}.

All samples were sequenced using the FLX Titanium chemistry, except
for the Taiwanese female sample T1, which was sequenced using FLX
standard chemistry, to generate between 99,000 and 209,000 raw
reads. For the L2 larval library, which had a larger number of
non-\textit{A. crassus}, non-\textit{Anguilla} reads, we confirmed
that these data were not laboratory contaminants by screening Roche
454 data produced on the same run in independent sequencing lanes.

\subsection*{Trimming, quality control and assembly}

Raw sequences were extracted in fasta-format (with the corresponding
qualities files) using sffinfo (Roche/454) and screened for adapter
sequences of the MINT-amplification-kit using cross-match \cite{PHRAP}
(with parameters -minscore 20 -minmatch 10). Seqclean
\cite{tgicl_pertea} was used to identify and remove poly-A-tails, low
quality, repetitive and short (<100 base) sequences. All reads were
compared to a set of screening databases using BLAST (expect value
cutoff E<1e-5, low complexity filtering turned off: -F F). The
databases used were (a) a host sequence database comprising an
assembly of the \textit{An. japonica} Roche 454 data, a unpublished
assembly of \textit{An. anguilla} Sanger dideoxy sequenced expressed
sequence tags (made available to us by Gordon Cramb, University of St
Andrews) and transcripts from EeelBase \cite{pmid21080939} a publicly
available transcriptome database for the European eel; (b) a database
of ribosomal RNA (rRNA) sequences from eel species derived from our
Roche 454 data and EMBL-Bank; and (c) a database of rRNA sequences
identified in our \textit{A. crassus} data by comparing the reads to
known nematode rRNAs from EMBL-Bank. This last database notably also
contained xenobiont rRNA sequences. Reads with matches to one of these
databases over more than 80\% of their length and with greater than
95\% identity were removed from the dataset. Screening and trimming
information was written back into sff-format using sfffile(Roche
454). The filtered and trimmed data were assembled using the combined
assembly approach \cite{pmid20950480}: two assemblies were generated,
one using Newbler v2.6 \cite{pmid16056220} (with parameters -cdna
-urt), the other using Mira v3.2.1 \cite{miraEST} (with parameters
--job=denovo,est,accurate,454). The resulting two assemblies were
combined into one using Cap3 \cite{Cap3_Huang} at default settings and
contigs were labeled by whether they derived from both assemblies or
one assembly only (for a detailed analysis of the assembly categories
see the supporting Methods file).

\subsection*{Post-assembly classification and taxonomic assignment of
  contigs}

After assembly contigs were assessed a second time for host and other
contamination by comparing them (using BLAST) to the three databases
defined above, and also to nembase4, a nematode transcriptome database
derived from whole genome sequencing and EST assemblies
\cite{parkinson_nembase:resource_2004, pmid21550347}. For each contig,
the highest-scoring match was recorded as long as it spanned more than
50\% of the contig. We also compared the contigs to the NCBI
non-redundant nucleotide (NCBI-nt) and protein (NCBI-nr) databases,
recording the taxonomy of all best matches with expect values better
than 1e-05. TUGs with a best hit to non-Metazoans and to Chordata
within Metazoa were additionally excluded from further analysis.

%% nrow(contig.df[!contig.df$Ac &
%% contig.df$contamination%in%"eelmRNA" & contig.df$phylum.nr%in%"Nematoda", ])
%% investigate if these (149) should not be included rather as
%% Ac-origin DONE!!! The eelmRNA hits are _all_ much better!

\subsection*{Protein prediction and annotation}

Protein translations were predicted from the contigs using prot4EST
(version 3.0b) \cite{wasmuth_prot4est:_2004}. Proteins were predicted
either by joining single high scoring segment pairs (HSPs) from a
BLAST search of uniref100 \cite{pmid18836194}, or by ESTscan
\cite{estscan}, using as training data the \textit{Brugia malayi}
complete proteome back-translated using a codon usage table derived
from the BLAST HSPs, or, if the first two methods failed, simply the
longest ORF in the contig. For contigs where the protein prediction
required insertion or deletion of bases in the original sequence, we
also imputed an edited sequence for each affected contig. Annotations
with Gene Ontology (GO), Enzyme Commission (EC) and Kyoto Encyclopedia
of Genes and Genomes (KEGG) terms were inferred for these proteins
using Annot8r (version 1.1.1) \cite{schmid_annot8r:_2008}, using the
annotated sequences available in uniref100 \cite{pmid18836194}. Up to
10 annotations based on a BLAST similarity bitscore cut-off of 55 were
obtained for each annotation set. The complete \textit{B. malayi}
proteome (as present in uniref100) and the complete
\textit{C. elegans} proteome (as present in wormbase v.220) were also
annotated in the same way. SignalP V4.0 \cite{pmid21959131} was used
to predict signal peptide cleavage sites and signal anchor signatures
for the \textit{A. crassus}-transcriptome and similarly again for the
proteomes of the tow model-worms.  Additionally InterProScan
\cite{pmid11590104} (command line utility iprscan version 4.6 with
options -cli -format raw -iprlookup -seqtype p -goterms) was used to
obtain domain based annotations for the high credibility assembly
(highCA) derived contigs.

We recorded the presence of a lethal RNAi-phenotype in the
\textit{C. elegans} ortholog of each TUG using the biomart-interface
\cite{pmid22083790} to wormbase v. 220 through the R-package biomaRt
\cite{pmid19617889}.

\subsection*{Single nucleotide polymorphism analysis}

We mapped the raw reads against the the complete set of contigs,
replacing imputed sequences for originals where relevant, using ssaha2
\cite{pmid11591649} (with parameters -kmer 13 -skip 3 -seeds 6 -score
100 -cmatch 10 -ckmer 6 -output sam -best 1). From the ssaha2 output,
pileup-files were produced using samtools
\cite{journals/bioinformatics/LiHWFRHMAD09}, discarding reads mapping
to multiple regions. VarScan \cite{pmid19542151} (pileup2snp) was used
with default parameters on pileup-files to output lists of single
nucleotide polymorphisms (SNPs) and their locations. For enrichment
analysis of GO-terms we used the R-package GOstats
\cite{pmid17098774}.

Using Samtools \cite{journals/bioinformatics/LiHWFRHMAD09} (mpileup
-u) and Vcftools \cite{pmid21653522} (view -gcv) we genotyped
individual libraries for the list of previously found overall
SNPs. Genotype-calls were accepted at a phred-scaled genotype quality
threshold of 10. In addition to the relative heterozygosity (number of
homozygous sites/number of heterozygous sites) we used the R package
Rhh \cite{pmid21565077} to calculate internal relatedness
\cite{pmid11571049}, homozygosity by loci \cite{pmid17107491} and
standardised heterozygosity \cite{coltman81j} from these data.

We confirmed the significance of heterozygote-heterozygote correlation
by analysing the mean and 95\% confidence intervals from 1000
bootstrap replicates estimated for all measurements.

\subsection*{Gene-expression analysis}

Read-counts were obtained from the bam-files generated also for
genotyping using the R-package Rsamtoools \cite{rsamtools}. Counts to
off target data and lowCA contigs were disregarded. Furthermore
contigs with less than 32 reads over all libraries were excluded from
analysis, to avoid inference based on too low overall expression
values. Because very low coverage from library E1 and L2 leading
highly variable normalised data, we excluded these libraries from
analysis.

% R-package DESeq \cite{pmid20979621} to assess statistical
% significance of differences in counts according to groups of
% libraries.

The statistic of Audic and Claverie \cite{pmid9331369} as implemented
in ideg6 \cite{pmid12429865} was used to contrast single
libraries. Differential expression between libraries from different sex
of worms was accepted for genes differing between all female
libraries E2, T1 and T2 versus the male (M) library ($p<$0.01) but not
within any of the female libraries at the same threshold. Differential
expression between libraries from European and Asian origin was
accepted for genes differing between libraries E2 versus T1 and T2
($p<$0.01) but not between T1 versus T2.

\subsection*{Over-representation analyses}

Prior to analysis of GO-term over-representation (based on dn/ds or
expression values) we used the R-package annotationDbi
\cite{AnnotationDbi} to obtain a full list of associations (also with
higher-level terms) from Annot8r-annotations. We then used the
R-package topGO \cite{topGO} to traverse the annotation-graph and
analyse each node in the annotation for over-representation of the
associated term in the focal gene-set compared to a appropriate
universal gene-set (all contigs with dn/ds values or all contigs
analysed for gene-expression) with the ``classic'' method and Fisher's
exact test. From the resulting tables we removed uninformative terms,
for which an ancestral term already was already in the table and no
additional counts supported overrepresentation.

We used Mann-Whitney u-tests to test the influence of factors on dn/ds
values, when multiple contrasts between groups (factors) were
investigated we used Nemenyi-Damico-Wolfe-Dunn tests. For
overrepresentation of one group (factor) in other groups (factors) we
used Fisher's exact test.

\subsection*{General coding methods}

The bulk of analysis (unless otherwise cited) presented in this paper
was carried out in R \cite{R_project} using custom scripts. We used a
method provided in the R-packages Sweave
\cite{lmucs-papers:Leisch:2002} and Weaver \cite{weaver} for
``reproducible research'' combining R and \LaTeX code in a single
file. All intermediate data files needed to compile the present
manuscript from data-sources are provided upon request. For
visualisation we used the R-packages ggplot2 \cite{ggplot-book} and
VennDiagram \cite{pmid21269502}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}


\subsection*{Sampling \textit{A. crassus}}

One female worm and one male worm were sampled from an aquaculture
with height infection loads in Taiwan. An additional female worm was
sampled from a stream with low infection pressure adjacent to the
aquaculture. All these worms were parasitising endemic
\textit{An. japonica}. A female worm and pool of L2 larval stages were
sampled from \textit{An. anguilla} in the river Rhine, one female worm
from a lake in Poland. All adult worms were filled with large amounts
of host-blood, therefore we anticipated abundant host-contamination in
sequencing data and decided to sequence a liver sample of an uninfected
\textit{An. japonica} for screening.

 \subsection*{Sequencing, trimming and pre-assembly screening}


