%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}


\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\usepackage{/usr/share/R/texmf/tex/latex/Sweave}
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of the swimbladder-nematode
    Anguillicola crassus: Resources for an alien parasite}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel G Heitlinger\correspondingauthor$^{1,2}$%
       \email{Emanuel G Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Stephen Bridgett$^{3}$%
       \email{Stephen Bridgett- sbridget@staffmail.ed.ac.uk}%
       Anna Montazam$^{3}$%
       \email{Anna Montazam- Anna.Montazam@ed.ac.uk}%
       Horst Taraschewski$^1$%
       \email{Horst Taraschewski- dc20@rz.uni-karlsruhe.de}%
       and Mark Blaxter$^2$%
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute 1, University of Karlsruhe,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
        \iid(3)The GenePool Sequencing Service, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} 
  \paragraph*{Results:} 
  \paragraph*{Conclusions:}
  Yeh!
  
\end{abstract}


      \ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

The nematode Anguillicola crassus Kuwahara, Niimi et Itagaki, 1974
\cite{kuwahara_Niimi_Itagaki_1974} is a parasite of freshwater eels of
the genus Anguilla, and adults localise to the swim bladder where they
feed on blood. Larvae are transmitted via crustacean intermediate
hosts \cite{de_charleroy_life_1990}. Originally endemic to East-Asian
populations of the Japanese eel (Anguilla japonica),
\textit{A. crassus} has attracted interest due to recent anthropogenic
expansion of its geographic and host ranges to Europe and the European
eel (Anguilla anguilla). Recorded for the first time in 1982 in
North-West Germany \cite{fischer_teichwirt}, where it was most likely
introduced through the live-eel trade
\cite{koops_anguillicola-infestations_1989, koie_swimbladder_1991},
\textit{A. crassus} has spread rapidly through populations of its
newly acquired host \cite{kirk_impact_2003}. At the present day it is
found in all \textit{An. anguilla} populations except those in Iceland
\cite{kristmundsson_parasite_2007}. \textit{A. crassus} can be
regarded as a model for invasive parasite introduction and spread
\cite{taraschewski_hosts_2007}.

\textit{A. crassus} has a major impact on \textit{An. anguilla} populations. In its
natural host in Asia infection prevalence and mean intensity of
infection are lower than in Europe \cite{mnderle_occurrence_2006},
where high prevalence (above 70\% \cite{wrtz_distribution_1998}) and
high infection intesities have been reported throughout the newly
colonized area \cite{lefebvre_anguillicolosis:_2004}. The virulence of
A .crassus in this new host has been attributed to an inadequate
immune response in \textit{An. anguilla} \cite{knopf_swimbladder_2006}. While
the An.  japonica is capable of killing larvae of the parasite after
vaccination \cite{knopf_vaccination_2008} or under high infection
pressure \cite{heitlinger_massive_2009}, responses in \textit{An. anguilla}
have hallmarks of pathology, including thickening of the swim bladder
wall \cite{wrtz_histopathological_2000}.  Interestingly host also
affects the adult size and life-history of the nematodes: In European
eels the nematodes are bigger and develop and reproduce faster
\cite{knopf_differences_2004}.

The genus Anguillicola is placed in the nematode suborder Spirurina
(clade III sensu \cite{blaxter_molecular_1998})
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
are exclusively parasitic and include important human pathogens (the
causative agents of filariases and ascariasis) as well as prominent
veterinary parasites. Molecular phylogenetic analyses place
Anguillicola in a clade of spirurine nematodes (Spirurina B of
[Laetsch et al submitted]) that have an freshwater or marine
intermediate host, but infect a wide range of carnivorous definitive
hosts. Spirurina B is sister to the main Spirurina C, including the
agents of filariases and ascariasis), and thus \textit{A. crassus} may be used
as an outgroup taxon to understand the evolution of parasitic
phenotypes in these species.

Recent advances in sequencing technology (often termed Next Generation
Sequencing; NGS), provide the opportunity for rapid and cost-effective
generation of genome-scale data. The Roche 454 platform
\cite{pmid16056220} offers longer reads than other NGS technologies,
and thus is suited to de novo assembly of genome-scale data in
previously understudied species. Roche 454 data has particular
application in transcriptomics \cite{pmid20950480}. The difference in
the biology of \textit{A. crassus} in \textit{An. japonica} (coevolved) and \textit{An. anguilla}
(recently captured) eel hosts likely results from an interaction
between different host and parasite responses, underpinned by
definitive differences in host genetics, and possible genetic
differentiation between the invading European and endemic Asian
parasites. As part of aprogramme to understand the invasiveness of
\textit{A. crassus} in \textit{An. anguilla}, we are investigating differences in gene
expression and genetic distinction between invading European and
endemic Asian \textit{A. crassus} exposed to the two different host
species. Here we report on the generation of a reference transcriptome
for \textit{A. crassus} based on Roche 454 data, and explore patterns of gene
expression and diversity.


%%%%%%%%%%%%%%%%%%
\section*{Methods}


\subsection*{Nematode samples, RNA extraction, cDNA synthesis and Sequencing}

\textit{A. crassus} from \textit{An. japonica} were sampled from
Kao-Ping river and an adjacent aquaculture in Taiwan as described in
\cite{heitlinger_massive_2009}. Worms from \textit{An. anguilla} were sampled
in Sniardwy Lake, Poland (53.751959N, 21.730957E) and from the
Linkenheimer Altrhein, Germany (49.0262N, 8.310556E). After
determination of the sex of adult nematodes, they were stored in
RNA-later (Quiagen, Hilden, Germany) until extraction of RNA. RNA was
extracted from individual adult male and female nematodes and from a
population of L2 larvae (Table 1). RNA was reverse transcribed and
amplified into cDNA using the MINT-cDNA synthesis kit (Evrogen,
Moscow, Russia).  For host contamination screening a liver-sample from
an uninfected \textit{An. japonica} was also processed.  Emulsion PCR
was performed for each cDNA library according to the manufacturer’s
potocols (Roche/454 Life Sciences), and sequenced on a Roche 454
Genome Sequencer FLX. All samples were sequenced using the FLX
Titanium chemistry, except for the taiwanese female sample T2, which
was sequenced using FLX standard chemistry, to generate between 99,000
and 209,000 raw reads. For the L2 larval library, which had a larger
number of non-\textit{A. crassus}, non-\textit{Anguilla} reads, we
confirmed that these data were not laboratory contaminants by
screening Roche 454 data produced on the same run in independent
sequencing lanes.


\subsection*{Trimming, quality control and assembly}

Raw sequences were extracted in fasta format (with the corresponding
qualities files) using sffinfo (Roche/454) and screened for adapter
sequences of the MINT-amplification-kit using cross-match \cite{PHRAP}
(with parameters -minscore 20 and -minmatch 10). \texttt{Seqclean}
\cite{tgicl_pertea} was used to identify and remove poly-A-tails, low
quality, repetitive and short (<100 base) sequences. All reads were
compared to a set of screening databases using \texttt{BLAST} (expect
value cutoff E<1e-5, low complexity filtering turned off: -F F). The
databases used were (a) a host sequence database comprising an
assembly of the \textit{An. japonica} Roche 454 data, a unpublished
assembly of \textit{An. anguilla} Sanger dideoxy sequencesd expressed
sequence tags (made available to us by Gordon Cramb, University of St
Andrews) and transcripts from EeelBase \cite{pmid21080939} a
publically availble transcriptome database for the European eel; (b) a
database of ribosomal RNA (rRNA) sequences from eel species derived
from our Roche 454 data and EMBL-Bank; and (c) a database of rRNA
sequences identified in our \textit{A. crassus} data by comparing the
reads to known nematode rRNAs from EMBL-Bank. This last database
notably also contained xenobiont rRNA sequences. Reads with matches to
one of these databases over more than 80\% of their length and with
greater than 95\% identity were removed from the dataset. Screening
and trimming information was written back into sff-format using
sfffile (Roche 454). The filtered and trimmed data were assembled
using the combined assembly approach \cite{pmid20950480}: Two
assemblies were generated, one using \texttt{Newbler v2.6}
\cite{pmid16056220} (with parameters \texttt{-cdna -urt}), the other
using \texttt{Mira v3.2.1} \cite{miraEST} (with parameters
\texttt{--job=denovo,est,accurate,454}). The resulting two assemblies
were combined into one using \texttt{Cap3} \cite{Cap3_Huang} at default
settings and contigs were labeled by whether they derived from both
assemblies or one assembly only (for a detailed analysis of the
assembly categories see the supporting Methods file).

\subsection*{Post-assembly classification and taxonomic assignment of
  contigs}

After assembly contigs were assessed a second time for host and other
contamination by comparing them (using \texttt{BLAST}) to the three
databases defined above, and also to nembase4, a nematode
transcriptome database derived from whole genome sequencing and EST
assemblies \cite{parkinson_nembase:resource_2004, pmid21550347}. For
each contig, the highest-scoring match was recorded as long as it
spanned more than 50\% of the contig. We also compared the contigs to
the NCBI non-redundant nucleotide (NCBI-nt) and protein (NCBI-nr)
databases, recording the taxonomy of all best matches with expect
values better than 1e-05.

\subsection*{Protein prediction and annotation}

Protein translations were predicted from the contigs using
\texttt{prot4EST} (version 3.0b)
\cite{wasmuth_prot4est:_2004}. Proteins were predicted either by
joining single high scoring segment pairs (HSPs) from a BLAST search
of uniref100 \cite{pmid18836194}, or by \texttt{ ESTscan}
\cite{estscan}, using a training data the \textit{Brugia malayi}
complete proteome back-translated using a codon usage table derived
from the \texttt{BLAST} HSPs, or, if the first two methods failed,
simply the longest ORF in the contig. For contigs where the protein
prediction required insertion or deletion of bases in the original
sequence, we also imputed an edited sequence for each affected
contig. Annotations with Gene Ontology (GO), Enzyme Commission (EC)
and Kyoto Encyclopaedia of Genes and Genomes (KEGG) terms were
inferred for these proteins using Annot8r (version 1.1.1)
\cite{schmid_annot8r:_2008}, using the annotated sequences available
in uniref100 \cite{pmid18836194}. Up to 10 annotations based on a
BLAST similarity bitscore cut-off of 55 were obtained for each
annotation set. The complete \textit{B. malayi} proteome (as present
in uniref100) and the complete \textit{C. elegans} proteome (as
present in wormbase v.220) were also annotated in the same
way. SignalP V4.0 \cite{pmid21959131} was used to predict signal
peptide cleavage sites and signal anchor signatures for the
\textit{A. crassus}-transcriptome and similarly again for the proteoms
of the tow model-worms.  Additionally \texttt{InterProScan}
\cite{pmid11590104} (command line utility \texttt{iprscan} (version
4.6) with options \texttt{-cli -format raw -iprlookup -seqtype p
  -goterms}) was used to obtain domain based annotations for the high
credibility assembly (highCA) derived contigs.

We recorded the presence of a lethal rnai-phenotyp in the
\textit{C. elegans} ortholog of each TUG using the biomart
\cite{pmid22083790} interface to wormbase v. 220 through the R-package
biomaRt \cite{pmid19617889}.

\subsection*{Single nucleotide polymorphism analysis}

We mapped the raw reads against the the complete set of contigs,
replacing imputed sequences for originals where relevant, using
\texttt{ssaha2} \cite{pmid11591649} (with parameters \texttt{-kmer 13
  -skip 3 -seeds 6 -score 100 -cmatch 10 -ckmer 6 -output sam -best
  1}). From the ssaha2 output, pileup-files were produced using
samtools \cite{journals/bioinformatics/LiHWFRHMAD09}, discarding reads
mapping to multiple regions. \texttt{VarScan} \cite{pmid19542151}
(pileup2snp) was used with default parameters on pileup-files to
output lists of single nucleotide polymorphisms (SNPs) and their
locations.

Using \texttt{Samtools} \cite{journals/bioinformatics/LiHWFRHMAD09}
(mpileup -u) and \texttt{Vcftools}\cite{pmid21653522} (view -gcv) we
genotyped individual libraries for the list of previously found
overall SNPs. Genotype-calls were accepted at a phred-scaled genotype
quality threshold of 10. In addition to the relative heterozygosity
(number of homozygous sites/ number of heterozygous sites) we used the
R package Rhh \cite{pmid21565077} to calculate internal relatedness
\cite{pmid11571049}, homozygosity by loci \cite{pmid17107491} and
standardized heterozygosity \cite{coltman81j} from these data.

Using 1000 bootstrap replicates we confirmed the significance of
heterozygote-heterozygote correlation by analysing th lower bounds of
95\% confidence intervals from 1000 bootstrap replecates estimated for
all measurements.

\subsection*{Gene-expression analysis}




\subsection*{General coding methods}

The bulk of analysis (unless otherwise cited) presented in this paper
was carried out in R \cite{R_project} using custom scripts. We used a
method provided in the R-packages
Sweave\cite{lmucs-papers:Leisch:2002} and Weaver\cite{weaver} for
``reproducible research'' combining R and \LaTeX code in a single
file. All intermediate data files needed to compile the present
manuscript from data-sources are provided upon request. For
visualistaion we used the R-packages ggplot2 \cite{ggplot-book} and
VennDiagram \cite{pmid21269502}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}


\subsection*{Sampling \textit{A. crassus}}

One female worm and one male worm were sampled from an aquaculture
with height infection loads in Taiwan. An additonal female worm was
sampled from a stream with low infection pressure adjacent to the
aquaculture. All these worms were parsitising endemic
\textit{An. japonica}. A female worm and pool of L2 larval stages were
sampled from \textit{An. anguilla} in the river Rhein, one female worm
from a lake in Poland. All adult worms were filled with large amounts
of host-bood, therefore we anticipated abundant host-contamination in
sequencing data and decided to sequence a liver sample of an unifected
\textit{An. japonica} for screening.

 \subsection*{Sequencing, trimming and pre-assembly screening}






A total of 756363 raw sequencing reads were
generated for \textit{A. crassus} (Table 1). These were trimmed for
base call quality, and filtered by length to give
585949 high-quality reads (spanning
169863104 bases). In the eel data-set from
159370 raw reads 135072 were
assembled after basic quality screening.

We then screened the \textit{A. crassus} reads for contamination by
host (30071 matched previously
sequenced eel genes or our own \textit{An. japonica} 454
transcriptome, which had been assembled into
10639 mRNA
contigs. (181783 reads matched large or
small subunit nuclear or mitochondrial ribosomal RNA sequences of
\textit{A. crassus}) (Table 1). In addition to fish mRNAs, we
identified (and removed) 5286
reads in the library derived from the L2 nematodes that had
significant similarity to cercozoan (likely parasite) ribosomal RNA
genes (Table 1).

\subsection*{Assembly}


We assembled the remaining 353055 reads (spanning
100491819 bases) using the combined assembler strategy
\cite{pmid20950480} and Roche 454 GSassembler (version 2.6) and MIRA
(version 3.21) \cite{miraEST}. From this we derived 13851
contigs that were supported by both assembly algorithms,
3745 contigs only supported by one of the assembly
algorithms and 22591 singletons that were not assembled by
either approach (Table 2). When scored by matches to known genes, the
contigs supported by both assemblers are of the highest credibility,
and this set is thus termed the high credibility assembly
(highCA). Those with evidence from only one assembler and the
singletons are of lower credibility (lowCA). These datasets are the
most parsimonious (having the smallest size) for their quality
(covering the largest amount of sequence in reference
transcritomes). In the highCA parsimony and low redundancy is
prioritized, while in the complete assembly (highCA plus lowCA)
completeness is proiritized. The 40187 sequences (contig consensuses
and singletons) in the complete assembly are referred to below as
tentatively unique genes (TUGs).




We screened the complete assembly for residual host contamination, and
identified
40187
  TUGs that had higher, significant similarity to eel (and chordate)
  sequences (our 454 ESTs and EMBLBank Chordata proteins) than to
  nematode sequences \cite{pmid21550347}.

  Given our prior identification of cercozoan ribosomal RNAs, we also
  screened the complete assembly for contamination with other
  transcriptomes, and found 365 TUGs with hits
  to fungi (e.g Ajellomycetaceae, 53 hits),
  672 TUGs whith hits to plants and
  2002 hits to Protists (e.g. Trypanosomatidae,
  26 hits and Vahlkampfiidae, 38 hits), Bacteria (mostly
  Proteobacteria, 484 hits; see also additional figure
  phylum\_plots.png.

  No hits were found to Wolbachia or related Bacteria known as
  symbionts of Ecdyosozoans.

  Our assembly thus has 32518 TUGs,
  spanning 154052 bases
  (of which 11371 are
  highCA-derived, and span
  154052 bases) that are
  likely to derive from of \textit{A. crassus}.

\subsection*{Protein prediction}

%% read P4EST output processed with
%% coordinates_from_p4e.pl

%% SNP calling from VARSCAN output




%%% annotation.Rnw --- 

%% Author: emanuelheitlinger@gmail.com









For
32411
TUGs a protein was predicted using prot4EST
\cite{wasmuth_prot4est:_2004} (Table 2). The full open reading frame
was obtained in
353 TUGs, while
while for 2683 the
5’ end and for 8283
the 3' end was complete. In 13379 TUGs the
corrected sequence with the imputed ORF was slightly changed compared
to the raw sequence.

\subsection*{Annotation}

We obtained basic annotations with orthologous sequences from
\textit{C. elgeans} for
9554 TUGs,
from \textit{B. malayi} for
9662 TUGs,
from nempep \cite{parkinson_nembase:resource_2004, pmid21550347} for
11617
TUGs and whith uni-prot proteins for
11113 TUGs.

We used annot8r \cite{schmid_annot8r:_2008} to assign gene ontology
(GO) terms for 6509 TUGs, Enzyme Commission (EC) numbers for
2458 TUGs and Kyoto Encyclopaedia of Genes and Genomes (KEGG)
pathway annotations for 3844 TUGs (Table 2). Additionally
5125 highCA derived contigs were anntotated with GO terms
through Interpro Scan \cite{pmid11590104}. Nearly one third
(6987) of the \textit{A. crassus} TUGs were
annotated with at least one identifier, and 1829 had
GO, EC and KEGG annotations (Figure 1).

We compared our \textit{A. crassus} GO annotations for high-level
GO-slim terms to the annotations (obtained the same way) for the
complete proteome of the filarial nematode \textit{B. malayi} and the
complete proteome of \textit{C. elegans} (Figure 2).

Correlation shows the occurence of terms for the partial transcriptome
of \textit{A. crassus} to be more similar to the proteome of
\textit{B. malayi} (0.95; Spearman
correlation coefficient) than to the proteome of \textit{C. elegans}
(0.9). Also the tow model-nematode compared
ot each other (0.91) are less similar in the
occurence of thems than the two parasites.

We inferred presence of signal peptide cleavage sites in the predicted
protein sequence using SignalP \cite{pmid21959131}. We predicted
920 signal peptide cleavage sites and 65 signal
peptides with a transmembrane signature. Again these predictions are
more similar to predictions using the same methods for the proteome
\textit{B. malayi} (742 signal peptide cleavage sites and
41 with transmembrane anchor) than for the proteome of
\textit{C. elegans} (4273 signal peptide cleavage sites
and 154 with transmembrane anchor).

We inferred the presence of a lethal rnai phenotype in the orthologous
annotation of \textit{C. elegans}. For 257
TUGs a non-lethal phenotype was inferred for
6029 TUGs a lethal phenotype.

\subsection*{Evolutionary conservation}

\textit{A. crassus} TUGs were classified as conserved, conserved in
Metazoa, conserved in Nematoda, conserved in Spirurina or novel to
\textit{A. crassus} by comparing them to public databases and using
two \texttt{BLAST} bit-score cutoffs to define relatedness (Table 3).

Roughly a third and a quarter of the higCA derived contigs were
categorized as conserved across kingdoms at a bitscore threshold of 50
and 80, respectively. Roughly half or 3/5 of the these contigs were
identified as novel in \textit{A. crassus}.

The remaining higCA contigs spread across interediate
relatedness-levels. More sequences were categorzed as novel at the
phylum level (Nematoda) compared to kingdom and cladeIII level and the
nuber of contigs at interediate relatedness-levels was roughly
consistent for the two bitscore theresholds.

The latter points about intermediate conservation levels were also
true, when all TUGs were analysed. The numbers of TUGs categorized at
these intermediate levels roughly doubled. In contrast, the porportin
of additional conserved lowCA TUGs is small compared to additional
TUGs categorized as novel in \textit{A. crassus}, mirroring the higher
amount of erronous sequence.

Proteins predicted to be novel to Nematoda and novel in
\textit{A. crassus} were significantly enriched in signal peptide
annotation compared to conserved proteins, proteins novel in Metazoa
and novel in CladeIII (Fisher's exact test p$<$0.001 ; Figure 5) .

The proportion of lethal rnai-phenotypes was significantly higher for
orthologs of conserved TUGs
(94.65\%) than for othologs of
TUGs not conserved (97.23\%) across
kingdoms (p$<$0.001, Fisher's exact test).

\subsection*{Identification of single nucleotide polymorphisms}

We called single nucleotide polymorphisms (SNPs) on the
1099419 bases of the TUGs that had coverage of more then
8-fold available using VARScan \cite{pmid19542151}. We excluded SNPs
predicted to have more than 2 alleles or that mapped to an
undertermined (N) base in the reference, and retained
10458 SNPs. The ratio of transitions (ti;
6890) to transversion (tv;
3568) in this set was
1.93 . Using the prot4EST
prtedictions and the corrected sequences, 7153 of
the SNPs were predicted to be inside an ORF, with
2310 at codon first positons, 1819
at second positions and 3024 at third positions. As
expected ti/tv inside ORFs (tsv.raw.orf[2,3]) was higher than outside
ORFs (tsv.raw.orf[1,3]). The ratio of synonymous polymorphisms per
synonymous site to non-synonymous polymorphisms per non-synonymous
site (dn/ds) was 0.42. We filtered these SNPs to exclude
those that might be associated with analytical bias. As Roche 454
sequences have well-known systematic errors associated with
homopolymeric nucleotide sequences \cite{pmid21685085}, we analysed
the effect of exclusion of SNPs in, or close to, homopolymer
regions. We observed changes in ti/tv and in dn/ds when SNPs were
discarded using different size thresholds for homoploymer runs and
proximity thresholds (see Figure 3). Based on this we decided to
exclude SNPs with a homopolymer-run as long as or longer than 4 bases
inside a window of 11 bases (5 to bases to the right, 5 to the left)
around the SNP. We also observed a relationship between TUG dn/ds and
TUG coverage, associated with the presence of sites with low abundance
minority alleles (less than 7\% of the allele calls), suggesting that
some of these may be errors.  Removing low abundance minority allele
SNPs from the set removed this effect (Figure 4).  Our filtered SNP
dataset includes 5112 SNPs. We retained
4.65 SNPs per kb of contig sequence, with
8.37 synonymous SNPs per 1000 synonymous bases and
8.37 non-synonymous SNPs per 1000 non-synonymous
bases. A mean dn/ds of 0.231 was
calculated for the 859 TUGs
(762 highCA-derived contigs)
containing at least one synonymous SNP.

\subsection*{Polymorphisms associated with biological processes}

We consolidated our annotation and polymorphism analyses by examining
correlations between nonsynonymous variability and particular
classifications.

Signal peptide containing proteins have been shown to have higher
rates of evolution than cytosolic proteins in a number of nematode
species. In \textit{A. crassus}, TUGs predicted to contain signal
peptide cleavage sites in SignalP showed a trend towards higher dn/ds
values than TUGs without signal peptide cleavage sites (p =
0.074; two sided
Mann-Whitney-test) %; Figure sigp\_dn\_ds.png.

Positive selection can be inferred from dn/ds analyses, and we defined
TUGs with a dn/ds higher than 0.5 as positively selected. We
identified over- and under-represented GO ontology terms associated
with these putatively positively selected genes (Table 4). Within the
molecular function category, ``peptidase activity'' was the most
significantly overrepresented term and had twelve TUGs supporting the
overrrepresentation. The highlighted twelve peptidases annotated with
eleven unique orthologs in \textit{C. elegans} and \textit{B. malayi}.
The term ``structural constituent of ribosome'' was underrepresented.

While the biological process and cellular compartment categories
provide less information for a nematode (highlighting e.g. brain or
pancreas development), unterrepresented terms in both were connected
to ribosomal proteins, validating the analysis for the molecular
function category.

Other overrepresented terms abundant over categories pointed ot
subunits of the respriratory chain e.g. ``heme-copper terminal oxidase
activity'' and ``cytochrome-c oxidase activity'' in molecular function
and ``mitochondrion'' in cellular compartment.

At both bitscore thresholds contigs novel in clade III and novel in
\textit{A. crassu} had a significantly higher dn/ds than other contigs
(novel.in.metazoa - novel.in.Ac, 0.005 and 0.015; novel.in.nematoda -
novel.in.Ac, 0.005 and 0.002; novel.in.nematoda - novel.in.clade3,
0.207 and 0.045; comparison, p-value from bitscore of 50 and p-value
from bitscore of 80, Nemenyi-Damico-Wolfe-Dunn test, given only for
significant comparisons; Figure 6).

Orthologs of \textit{C. elegans} transcripts with lethal
rnai-phenotype are expected to evolve under stronger selective
constraints. Indeed the values of dn/ds showed a non-significant trend
towards lower values in TUGs with orthologs with a lethal phenotype
comapred to a non-lethal phenotypes
(p=0.138, two-sided U-test).

\subsection*{SNP markers for single worms}

We used \texttt{Samtools}\cite{journals/bioinformatics/LiHWFRHMAD09}
and \texttt{Vcftools}\cite{pmid21653522} to call genotypes in single
worms (adult sequencing libraries). This resulted in 199
informative sites in 152 contigs, where two alleles
were found in at least one assured genotype at least in one of the
worms.

Internal relatedness \cite{pmid11571049}, homozygosity by loci
\cite{pmid17107491} and standardized heterozygosity \cite{coltman81j}
were all highlighting the Taiwanese worm from the wild population
(sample T1) as the most and the European worm from Poland (sample E2)
as the least heterozygous individual. The other worms had intermediate
values between these two extremes.

We confirmed the genome-wide significance of these estimates using
heterozygosity-hetorozygosizty correlation \cite{pmid21565077}. These
tests confirmed the usability of the 199 SNP-markers for
analysis of inbreeding in \textit{A. crassus} using a higher number of
genotyped individuals ($\mu$ = 0.78, $ci_l$=0.444; $\mu$ = 0.86 and
$ci_l$ = 0.596; $\mu$ = 0.87 and $ci_l$= 0.632; mean and lower bound of
95\% confidence intervals from 1000 bootstrap replicates for intermal
realtedness, homozygosity by loci and standardized heterozygosity).

\subsection*{Differential expression}
                

Using methods developed for sequencing data in the R-packeage DESeq
\cite{pmid20979621}, we analyzed gene-expression inferred from
mapping. Of the {length(raw.fasta)} reads 252388
(71.49\%) mapped uniquely
(with their best hit) to the fullest assembly (including the contigs
of non \textit{A. crassus} as a filter to avoid false mapping and
excluding those afterwards).

22 Contigs were clearly overexpressed at adjusted p-values $<$ 0.001
in the male worm compared to the felmale worms.
7 of these were
annotatated as ``Major sperm protein'', all with different orthologs
in \textit{B. malayi} and
5
with different orthologs in \textit{C. elegans}. Other annotation were
``putative P40'' for ``Contig2545'' and ``Contig94'', ``MFP2'' for
``Contig437'', ``KH domain containing protein'' for ``Contig96'',
``Protein kinase domain containing protein'' for ``Contig96'',
``Protein-tyrosine phosphatase containing protein'' for ``Contig382''
and ``PDZ domain containing protein'' for ``Contig106''. For all these
contigs read-sums for orthologous groups in the two model nematodes
also showed significant overexpression in the same direction.

An exception was ``Contig194'' (``Phosphoenolpyruvate
carboxykinase''), for which the sums of othrolgous-counts for both
model-nematodes did not show a significant overexpression in the male.


Comparing libraries from European worms to Taiwanese worms, two
contigs were observed at adjusted p-values $<$ 0.001: ``Contig5250''
was only expressed in library E2 and was annotated as an ortholog of
alpha tubulin in both \textit{C. elegans} and \textit{Brugia
  malayi}. This annotation was shared with
18 and 16
  contigs respectively. Contrary to the identified contig, the sums of
  the read counts for this othologous groups showed a non-significant
  trend towards higher expression in the the Asian
  libraries. ``Contig13931''was only expressed in sample E1 and showed
  high similarity to \textit{Ascaris lumbricoides} small nuclear RNAs.

  Two contigs were identified as differentially expressed at adjusted
  p-values below 0.2 and were expressed highly in library E2:
  ``Contig5320'' shared annotation as ortholog of ``Cuticle collagen
  7'' in both model-nematodes with
  5 contigs.
    Counts for the orthologous groups showed an opposite trend of
    overexpression in the in the Taiwanes libraries.

    This was followed by a set of
    9 contigs
    overexpressed at exactly 20\% false discovery rate. They were all
    found only in Taiwanese libraries and the ortholog-groups showed a
    tendency towards expression . Three of these are worth further
    notice: ``Contig5164'' annotated as ``galactoside binding lectin''
    was not only expressed in one of the libraries, but in
    both. ``Contig110'' (``cysteine protease'') and ``Contig6355''
    (``plasminogen'') had



\section*{Discussion}

We have generated a de novo transcriptome for \textit{A. crassus} an
important invasive parasite that threatens wild stocks of the European
eel \textit{An. anguilla}. These data enable a broad spectrum of
molecular research on this ecologically and economically important
parasite. As \textit{A. crassus} lives in close association with its
host, we have used exhaustive filtering to attempt to remove all
host-derived, and host-associated organism-derived contamination from
the data. To do this we have also generated a transcriptome dataset
from the definitive host \textit{An. japonica}. The non-nematode,
non-eel data identified, particularly in the L2 sample, showed highest
identity to flagellate protists, which may have been parasitising the
eel (or the nematode). Encapsulated objects observed in eel swim
bladder walls \cite{heitlinger_massive_2009} could be due solely to
immune attrition of \textit{A. crassus} larvae or to other
coinfections.

A second examination of sequence origin was performed after assembly,
employing higher stringency cutoffs.  Similar taxonomic screening was
used in a garter snake transcriptome project \cite{pmid21138572}, and
an analysis of lake sturgeon tested and rejected hypotheses of
horizontal gene-transfer when xenobiont sequences were identified
\cite{pmid20386959}. A custom pipeline for transcriptome assembly from
pyrosequencing reads \cite{pmid20034392} proposed the use of EST3
\cite{pmid17218127} to infer sequence origin based simply on
nucleotide frequency. We were not able to use this approach
successfully, probably due to the fact that xenobiont sequences in our
data set derive from multiple sources with different GC content and
codon usage.

Compared to other NGS transcriptome sequencing projects
[!!!references???], the combined assembly approach generated a smaller
number of contigs that had lower redundancy and higher
completeness. Projects using the mira assembler often report
substantially greater numbers of contigs for datasets of similar size
(see e.g. \cite{pmid21364769}), comparable to the mira sub-assembly in
our approach. The use of oligo(dT) to capture mRNAs probably explains
the bias towards 3' end completeness and a relative lack of true
initiation codons in our protein prediction. This bias is
near-ubiquitous in deep transcriptome sequencing projects
(e.g. \cite{pmid20331785}).

***START: I moved the chunck below up (from a paragraph you did not
correct last time) to have annotation before SNPs

We were able to obtain high-quality annotations for a large set of
TUGs: For 40\% of the complete assebly and 60\% of our highCA assembly
\texttt{BLAST}-based annotations could be obtained. 45\% of the
contigs in the highCA assembly were additionally decorated with
domain-based annotations through \texttt{InterProScan}
\cite{pmid11590104}.

Comparison with complete protein sequence from the genomes of
\textit{B. malayi} and \textit{C. elegans} showed a remarkable degree
of agreement regarding the occurrence of terms in the two parasitic
worms. This agreement was higher than with the free living nematode
\textit{C. elegans} and even the two genome-sequencing-derived
proteoms showed less agreement with each other than the filarial
parasite with our data-set. This implies, that our transcriptome is
truely a representative partial genome
\cite{parkinson_partigene--constructing_2004} of a prasitic nematode.

Analysis of conservation identified more sequence novel in Nematode
than in the eukaryote kingdom or in Clade III this is in agreement
with prevalence of genic novelty in the Nematoda
\cite{wasmuth_extent_2008}. Furthermore the basal postition of
\textit{A. crassus} in CladeIII could be leading to novelty in the
clade not being shared with \textit{A. crassus}.

TUGs predicted to be novel in the phylum Nematoda and novel to
\textit{A. crassus} contained the highest proportion of
signal-positives. This confirms observations made in a study on
\textit{Nippostrongylus brasiliensis} \cite{harcus_signal_2004}, where
signal positives were reported as less conserved. Interestingly
enrichment of signal sequence beararing TUGs in our dataset was
constrained to sequences novel in nematodes and \textit{A. crassus}
(i.e. not to the level of CladeIII). This may be explained, in two
differenet ways involving the basal position of \textit{A. crassus}:
First the signal positives shared with all nematodes could be
conserved molecules not excreted by parasites. A different classs of
secreted/excreted molecules with prominent role in host parasite
interactions would not have arisen early in the evolution of
parasitism in cladeIII - or be too fast-evolving - and thus be
detected as specific to deeper sub-clades (i.e. to \textit{A. crassus}
in our data-set). A second explanation would be, that orthologs of
excreted parasite-specific genes could be among those shared whith
other nematodes and the fewer shared with clade III implying a
predisposition to parasitism outside of the Spirurina or even the
convergent evolution of secreted molecules in other parasitic
nematodes.

***End: I moved the above chunck up to have annotation before SNPs

We generated transcriptome data from multiple \textit{A. crassus} of
Taiwanese and European origin, and identified SNPs both within and
between populations. Screening of SNPs in or adjacent to homopolymer
regions improved overall measurements of SNP quality. The ratio of
transitions to transversions (ti/tv) increased. Such an increase is
explained by the removal of “noise” associated with common homopolymer
errors \cite{pmid21685085}. The value of raw.ts.tv[,3]
(tsv.raw.orf[1,3] outside, tsv.raw.orf[2,3] inside ORFs) is in good
agreement with the overall ti/tv of humans (2.16 \cite{pmid21169219})
or \textit{Drosophila} (2.07 \cite{pmid21143862}). The ratio of
non-synonymous SNPs per non-synonymous site to synonymous SNPs per
synonymous site (dn/ds) decreased with removal of SNPs adjacent to
homopolymer regions from 0.42 to
0.231 after full screening. The most
plausible explanation is the removal of error, as unbiased error would
lead to a dn/ds of 1. While dn/ds is not unproblematic to interpret
within populations \cite{pmid19081788}, the assumption of negative
(purifying) selection on most protein-coding genes makes lower mean
values seem more plausible. We used a threshold value for the minority
allele of 7\% for exclusion of SNPs, based on an estimate that
approximately 10 haploid equivalents were sampled (5 individual worms
plus an negligible contribution from L2 larvae in the L2 library and
within the female adult worms). The benefit of this screening was
mainly a reduction of non-synonymous SNPs in high coverage contigs,
and a removal of the dependence of dn/ds on coverage. Working with an
estimate of dn/ds independent of coverage, efforts to control for
sampling biased by depth (i.e. coverage; see \cite{pmid18590545} and
\cite{pmid20478048}) could be avoided.

*** you corrected up to her last time

Also in comparison with published intra-species values of dn/ds our
final estimate of seems plausible: in transcripts from the female
reproductive tract of \textit{Drosophila} dn/ds was 0.15
\cite{pmid15579698} and 0.21 in the male reproductive tract
\cite{pmid11404480} (although for ESTs specific to the male accessory
gland were shown to have a higher dn/ds of 0.47). A pyrosequencing
study in the parasitic nematode \textit{Ancylostoma canium}
\cite{pmid20470405} reported dn/ds of 0.3.

When the whole of coding sequences are studied, of which only a small
subset of sites can be under diversifying selection, dn/ds of ~0.5 has
been suggested as threshold for assuming diversifying selection
\cite{pmid15579698} instead of the classical threshold of 1
\cite{pmid6449605}. The use of this threshold for positive selection
led to the identification of over-represented of GO-term highlighting
very interesting transcripts:

Twelve peptidases under positive selection (from 43 with a dn/ds
obtained) ment an ennrichment in the category. All twelve have
orthologs in \textit{B. malayi} and \textit{C. elgans} and are
conserved across kingdoms. Despite their conservation peptidases are
thought to have acquired new and prominent roles in host-parasite
interaction compared to free living organisms: In \textit{A. crassus}
a trypsin-like proteinase has been identified thought to be utilized
by the tissue-dwelling L3 stage to penetrate host tissue and an
aspartyl proteinase thought to be a digestive enzyme in adults
\cite{polzer_identification_1993}. The tweleve proteinases under
positive selection could be the targets of the adaptive immunity
developed against \textit{A. crassus} \cite{knopf_migratory_2008,
  knopf_vaccination_2008}, which is often only elicited against
subtypes of larvae \cite{molnar_caps}.

The under-representation of ribosomal proteins (term ``structural
constituent of ribosome'') in positive selected contigs is in good
agreement with the notion that ribosomal proteins are extremely
conserved across kingdoms \cite{pmid9664699} and should be under under
strong negative selection.

Genotyping of individual worms identified a set of 199
SNPs with highest credibility and a high information content for
population-genetic studies. Levels of genome-wide heterozygosity found
for the 5 adult worms examined in our study are in agreement with a
microsattelite data showing reduced heterozygosity in European
populations of \textit{A. crassus}. 


*** Gene expression:

It was within our expectation, that expression analysis failed to give
conclusive results, as the present data-set is not fully adequate for
this kind of analysis: First we did not include replicates for
libraries of male adults as well as for L2-larvae. Second one of the
replicates for female worms (library E1) resulted in a low amount of
sequence mappable to protein-coding (non-rRNA) genes.
However some of the results are still valuable:

DESeq was able to report genes significantly differing in expression
between male and female worms and between the L2 library and the all
other worms. This was possible for male worms as well as for
L2-larvae, were no replicated samples were obtained, due due the
special features of this package \cite{pmid20979621}. However only
over-expression in non-repeated samples can be detected, as obviously
lack of expression in one sample can't validate

Comparisons were lacking significance, as methods are designed for
deeper sequencing and more importantly more replicates would be
needed. Differences between the L2-library and other libraries were
mainly due to off-target data, and TUGs solely found in the L2 library
are ... 


Despite the lack of replicates for male worms we were able to identify
xxx sequences being significantly over-expressed in male worms. In
fact all these TUGs were nearly exclusively expressed in males.



%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}

  
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Competing interests}
The authors declare no competing interests.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgments}
\ifthenelse{\boolean{publ}}{\small}{}

The work of EGH is funded by
Volkswagen Foundation, "F\"{o}rderinitiative Evolutionsbiologie".
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article} % Style BST file
  \bibliography{/home/ele/bibtex/master,/home/ele/bibtex/master2}
} % Bibliography file (usually '*.bib' )

%%%%%%%%%%%


\ifthenelse{\boolean{publ}}{\end{multicols}}{}
\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}

\subsection*{Figure 1 - Number of contigs annotated with different
  methods}
 
Number of annotations obtained for Gene Ontology (GO), Enzyme
Commission (EC) and Kyoto Encyclopaedia of Genes and Genomes (KEGG)
terms through Annot8r\cite{schmid_annot8r:_2008} for all TUGs (a) and
for higCA derived contigs (b). The latter includes additional
domain-based annotations obtained with interpro scan
\cite{pmid11590104}.


\subsection*{Figure 2 - Comparing high level GO-slim annotations}

For Gene Ontology (GO) categories molecular function, cellular
compartment and biological process the number of terms in high level
GO-slim categories is given as obtained through Annot8r
\cite{schmid_annot8r:_2008}. 

\subsection*{Figure 3 - Changes in ti/tv and dn/ds due to exclusion of
  homopolymer-runs}

\subsection*{Figure 4 - SNP calling and SNP categories}
 
Overabundance of SNPs at (a) codon-position two and of (c)
non-synynomous SNPs for low percentages of the minority allele. (b)
Significant positive correlation of coverage and dn/ds before removing
these SNPs at a threshold of 7\% ($p<$ 0.001, $R^2=$
0.015 and afterwards
($R^2<$0.001,
$p=$0.211 (d).


\subsection*{Figure 5 - Positive selection and evolutionary
  conservation}




\subsection*{Figure 6 - Enrichment of Signal-positives for categories
  of evolutionary conservations}

Proportions of \texttt{SignalP}-predictions for each category of
evolutionary conservation. Generally - across bit-score thresholds -
TUGS novel in nematodes and in \textit{A. crassus} the highest
proportion of signal-positives.


\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
  \subsection*{Table 1 - Sampling, trimming and pre-assembly screening}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Mon Nov 21 09:59:09 2011
\begin{tabular}{lllllllllll}
  \hline
library & life.st & source.p & raw.reads & lowqal & AcrRNA & eelmRNA & eelrRNA & Cercozoa & valid & valid.span \\ 
  \hline
E1 & adult f & Europe R & 209325 & 92744 & 76403 & 4835 & 13112 & 0 & 22231 & 7167338 \\ 
  E2 & adult f & Europe P & 111746 & 10903 & 11213 & 3613 & 69 & 0 & 85948 & 24046225 \\ 
  L2 & L2 lavae & Europe R & 112718 & 15653 & 30654 & 1220 & 1603 & 5286 & 58302 & 16661548 \\ 
  M & adult m & Asia C & 106726 & 15484 & 31351 & 1187 & 418 & 0 & 58286 & 17424408 \\ 
  T1 & adult f & Asia C & 99482 & 7947 & 24929 & 7475 & 514 & 0 & 58617 & 14443123 \\ 
  T2 & adult f & Asia W & 116366 & 27683 & 7233 & 11741 & 38 & 0 & 69671 & 20749177 \\ 
  Eel & liver & Taiwan & 159370 & 24298 &  &  &  &  & 135072 & 34482916 \\ 
   \hline
\end{tabular}
\subsection*{Table 2 - assembly classification and contig statistics}

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Mon Nov 21 09:59:09 2011
\begin{table}[ht]
\begin{center}
\begin{tabular}{rrrr}
  \hline
 & lowCA & highCA & combined \\ 
  \hline
total.contigs & 26336 & 13851 & 40187 \\ 
  rRNA.contigs & 556 & 34 & 590 \\ 
  fish.contigs & 1776 & 708 & 2484 \\ 
  xeno.contigs & 2332 & 742 & 3074 \\ 
  remaining.contigs & 21147 & 11371 & 32518 \\ 
  remaining.span & 8095986 & 7971550 & 16067536 \\ 
  non.u.cov & 15 & 11 & 13 \\ 
  cov & 2.443 & 6.838 & 4.624 \\ 
  p4e.BLAST-similarity & 4356.000 & 5663.000 & 10019.000 \\ 
  p4e.ESTScan & 8324 & 3597 & 11921 \\ 
  p4e.LongestORF & 8347 & 2085 & 10432 \\ 
  p4e.no-prediction & 93 & 14 & 107 \\ 
  full.3p & 5906 & 2714 & 8620 \\ 
  full.5p & 1484 & 1270 & 2754 \\ 
  full.l & 104 & 185 & 289 \\ 
  GO & 2635 & 3874 & 6509 \\ 
  EC & 966 & 1492 & 2458 \\ 
  KEGG & 1608 & 2236 & 3844 \\ 
  IPR & 0 & 7557 & 7557 \\ 
  nem.blast & 4868 & 5820 & 10688 \\ 
  any.blast & 5106 & 6007 & 11113 \\ 
   \hline
\end{tabular}
\end{center}
\end{table}
Summary statistics for contigs from different assembly-categories
given in colums as highCA = high credibility assembly; lowCA = low
credibility assembly, CA = complete assembly.

Rows indicate summary statistics: total.contigs = numbers of total
contigs, fish.contigs = number of contigs hitting eel-mRNA or Chordata
in NCBI-nr or NCBI-nt (screened out), xeno.contigs = number of contigs
with best hit (NCBI-nr and NCBI-nt) to non-eukaryote (screened out),
remaining.contigs = number of contigs remaining after this screening,
remaining.span = total length of remaining contigs, non.u.cov =
non-unique mean base coverage of contigs, cov = unique mean base
coverage of contigs, p4e.``X'' = number protein predictions derived in
p4e, where ``X'' describes the method of prediction (see Methods),
full.3p = number of contigs complete at 3', full.5p = number of
contigs complete at 5', GO = number of contigs with GO-anntotation,
KEGG = number of contigs with KEGG-anntotation, EC = number of contigs
with EC-anntotation, nem.blast = number of contigs with blast-hit to
nematode in nr, any.blast = number of contigs with blast-hit to
non-nematode (eukaryote non chordate) sequence in NCBI-nr.

\subsection*{Table 3 - Evolutionary conservation }
% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Mon Nov 21 09:59:09 2011
\begin{tabular}{rrrrrr}
  \hline
 & conserved & novel.in.metazoa & novel.in.nematoda & novel.in.clade3 & novel.in.Ac \\ 
  \hline
bit.50.all & 5604 & 1713 & 2173 & 1485 & 21543 \\ 
  bit.80.all & 3506 & 1382 & 2014 & 1525 & 24091 \\ 
  bit.50.highCA & 3479 & 875 & 1010 & 601 & 5406 \\ 
  bit.80.highCA & 2457 & 832 & 1084 & 716 & 6282 \\ 
   \hline
\end{tabular}
\subsection*{Table 4 - Over- and under-representation of GO-terms in
  positively selected}

GO-terms over- or under-represented (direction) in contigs putatively
under positive selection. Horizontal lines seratate categories of the
GO-ontology. First category is molecular function, then biological
process last cellular compartment. P values (Pval) for over- or
under-representation are given along with the number of positively
selected contigs (Count; dn/ds $>$ 0.5) and the number of contigs with
this annotation for wich a dn/ds was obtained (Size) and the
description of the GO-term (Term).

% latex table generated in R 2.13.0 by xtable 1.5-6 package
% Mon Nov 21 09:59:09 2011
\begin{longtable}{rrrp{7cm}r}
 Pvalue & Count & Size & Term & direction \\ 
  \hline
0.00 &  13 &  45 & peptidase activity & Over \\ 
  0.01 &   7 &  18 & heme-copper terminal oxidase activity & Over \\ 
  0.01 &   7 &  18 & oxidoreductase activity, acting on a heme group of donors & Over \\ 
  0.01 &   7 &  18 & oxidoreductase activity, acting on a heme group of donors, oxygen as acceptor & Over \\ 
  0.01 &   7 &  18 & cytochrome-c oxidase activity & Over \\ 
  0.01 &  49 & 283 & catalytic activity & Over \\ 
  0.02 &  13 &  52 & transmembrane transporter activity & Over \\ 
  0.02 &   9 &  31 & monovalent inorganic cation transmembrane transporter activity & Over \\ 
  0.02 &   2 &   2 & L-amino acid transmembrane transporter activity & Over \\ 
  0.03 &   9 &  33 & inorganic cation transmembrane transporter activity & Over \\ 
  0.03 &  23 & 117 & hydrolase activity & Over \\ 
  0.03 &   8 &  29 & hydrogen ion transmembrane transporter activity & Over \\ 
  0.04 &   3 &   6 & ribonucleoprotein binding & Over \\ 
  0.04 &  13 &  58 & transporter activity & Over \\ 
  0.05 &  11 &  47 & substrate-specific transmembrane transporter activity & Over \\ 
  0.05 &  16 &  77 & oxidoreductase activity & Over \\ 
  0.00 &   1 &  53 & structural constituent of ribosome & Under \\ 
  0.03 &   7 &  93 & RNA binding & Under \\ 
  0.04 &   2 &  44 & transition metal ion binding & Under \\ 
  0.05 &   0 &  20 & protein binding transcription factor activity & Under \\ 
  0.05 &   0 &  20 & transcription factor binding transcription factor activity & Under \\ 
  0.05 &   0 &  20 & transcription cofactor activity & Under \\ 
   \hline
0.00 &  13 &  37 & brain development & Over \\ 
  0.00 &  14 &  45 & central nervous system development & Over \\ 
  0.00 &   6 &  12 & response to electrical stimulus & Over \\ 
  0.00 &   3 &   3 & branched chain family amino acid metabolic process & Over \\ 
  0.00 &   3 &   3 & branched chain family amino acid catabolic process & Over \\ 
  0.00 &  11 &  36 & ATP synthesis coupled electron transport & Over \\ 
  0.00 &  11 &  36 & mitochondrial ATP synthesis coupled electron transport & Over \\ 
  0.01 &   7 &  18 & mitochondrial electron transport, cytochrome c to oxygen & Over \\ 
  0.01 &  22 & 101 & nervous system development & Over \\ 
  0.01 &  11 &  38 & oxidative phosphorylation & Over \\ 
  0.01 &   6 &  15 & response to starvation & Over \\ 
  0.01 &  12 &  45 & cellular amino acid metabolic process & Over \\ 
  0.01 &   7 &  20 & positive regulation of cell cycle process & Over \\ 
  0.01 &  14 &  58 & amine metabolic process & Over \\ 
  0.01 &   4 &   8 & positive regulation of organelle organization & Over \\ 
  0.01 &   4 &   8 & spermatid development & Over \\ 
  0.01 &   4 &   8 & spermatid differentiation & Over \\ 
  0.01 &   5 &  12 & hindbrain development & Over \\ 
  0.01 &   5 &  12 & cerebellum development & Over \\ 
  0.01 &   5 &  12 & metencephalon development & Over \\ 
  0.01 &   5 &  12 & response to methylmercury & Over \\ 
  0.01 &   5 &  12 & autophagy & Over \\ 
  0.02 &  36 & 203 & response to stress & Over \\ 
  0.02 &   2 &   2 & embryonic body morphogenesis & Over \\ 
  0.02 &   2 &   2 & xylulose metabolic process & Over \\ 
  0.02 &   2 &   2 & L-amino acid transport & Over \\ 
  0.02 &   2 &   2 & neuromuscular process controlling balance & Over \\ 
  0.02 &   2 &   2 & response to sucrose stimulus & Over \\ 
  0.02 &   2 &   2 & NADP metabolic process & Over \\ 
  0.02 &   2 &   2 & response to disaccharide stimulus & Over \\ 
  0.02 &   2 &   2 & pentose metabolic process & Over \\ 
  0.02 &  15 &  66 & behavior & Over \\ 
  0.02 &   8 &  27 & interphase & Over \\ 
  0.02 &   8 &  27 & interphase of mitotic cell cycle & Over \\ 
  0.02 &  11 &  43 & electron transport chain & Over \\ 
  0.02 &  11 &  43 & respiratory electron transport chain & Over \\ 
  0.02 &  29 & 156 & catabolic process & Over \\ 
  0.02 &   3 &   5 & positive regulation of mitosis & Over \\ 
  0.02 &   3 &   5 & positive regulation of nuclear division & Over \\ 
  0.02 &  13 &  56 & cellular amine metabolic process & Over \\ 
  0.02 &  20 &  99 & aging & Over \\ 
  0.02 &  10 &  39 & regulation of cell cycle process & Over \\ 
  0.03 &  17 &  81 & apoptosis & Over \\ 
  0.03 &  16 &  75 & regulation of molecular function & Over \\ 
  0.03 &  13 &  57 & regulation of cell cycle & Over \\ 
  0.03 &   5 &  14 & mitotic cell cycle G1/S transition DNA damage checkpoint & Over \\ 
  0.03 &   5 &  14 & sleep & Over \\ 
  0.03 &   4 &  10 & cellular amino acid catabolic process & Over \\ 
  0.03 &  10 &  41 & reproductive structure development & Over \\ 
  0.03 &   3 &   6 & microtubule organizing center organization & Over \\ 
  0.03 &   3 &   6 & RNA catabolic process & Over \\ 
  0.03 &   3 &   6 & centrosome organization & Over \\ 
  0.03 &   8 &  30 & muscle organ development & Over \\ 
  0.04 &  11 &  47 & cellular respiration & Over \\ 
  0.04 &  13 &  59 & energy derivation by oxidation of organic compounds & Over \\ 
  0.04 &   7 &  25 & regulation of catabolic process & Over \\ 
  0.04 &   5 &  15 & signal transduction in response to DNA damage & Over \\ 
  0.04 &   5 &  15 & G1/S transition of mitotic cell cycle & Over \\ 
  0.04 &   5 &  15 & regulation of G1/S transition of mitotic cell cycle & Over \\ 
  0.04 &   5 &  15 & mitotic cell cycle G1/S transition checkpoint & Over \\ 
  0.04 &   5 &  15 & G1/S transition checkpoint & Over \\ 
  0.04 &   5 &  15 & DNA damage response, signal transduction by p53 class mediator & Over \\ 
  0.04 &   5 &  15 & regulation of cellular amine metabolic process & Over \\ 
  0.04 &   6 &  20 & response to copper ion & Over \\ 
  0.04 &  24 & 131 & cellular catabolic process & Over \\ 
  0.05 &   4 &  11 & imaginal disc development & Over \\ 
  0.05 &   4 &  11 & amine catabolic process & Over \\ 
  0.05 &   4 &  11 & skeletal muscle organ development & Over \\ 
  0.05 &  11 &  49 & mRNA metabolic process & Over \\ 
  0.05 &   2 &   3 & nuclear mRNA cis splicing, via spliceosome & Over \\ 
  0.05 &   2 &   3 & germ cell migration & Over \\ 
  0.05 &   2 &   3 & positive regulation of mitotic metaphase/anaphase transition & Over \\ 
  0.05 &   2 &   3 & mitotic centrosome separation & Over \\ 
  0.05 &   2 &   3 & oligosaccharide catabolic process & Over \\ 
  0.05 &   2 &   3 & spliceosomal conformational changes to generate catalytic conformation & Over \\ 
  0.05 &   2 &   3 & amino acid transport & Over \\ 
  0.05 &   2 &   3 & negative regulation of reproductive process & Over \\ 
  0.05 &   2 &   3 & centrosome duplication & Over \\ 
  0.05 &   2 &   3 & centrosome separation & Over \\ 
  0.05 &   2 &   3 & protein tetramerization & Over \\ 
  0.05 &   2 &   3 & protein homotetramerization & Over \\ 
  0.00 &  15 & 201 & gene expression & Under \\ 
  0.00 &   1 &  57 & cellular protein complex disassembly & Under \\ 
  0.00 &   1 &  57 & macromolecular complex disassembly & Under \\ 
  0.00 &   1 &  57 & protein complex disassembly & Under \\ 
  0.00 &   1 &  57 & cellular macromolecular complex disassembly & Under \\ 
  0.00 &   1 &  55 & pancreas development & Under \\ 
  0.00 &   1 &  55 & endocrine pancreas development & Under \\ 
  0.00 &   1 &  55 & endocrine system development & Under \\ 
  0.00 &   1 &  55 & viral genome expression & Under \\ 
  0.00 &   1 &  55 & viral transcription & Under \\ 
  0.00 &   8 & 131 & transcription & Under \\ 
  0.00 &   1 &  54 & translational termination & Under \\ 
  0.00 &   4 &  89 & translation & Under \\ 
  0.00 &   2 &  66 & cellular component disassembly & Under \\ 
  0.00 &   2 &  66 & cellular component disassembly at cellular level & Under \\ 
  0.01 &  14 & 178 & cellular macromolecule biosynthetic process & Under \\ 
  0.01 &  22 & 243 & biosynthetic process & Under \\ 
  0.01 &  22 & 240 & cellular biosynthetic process & Under \\ 
  0.01 &  15 & 181 & macromolecule biosynthetic process & Under \\ 
  0.01 &   2 &  57 & viral reproductive process & Under \\ 
  0.01 &   2 &  57 & viral infectious cycle & Under \\ 
  0.02 &   0 &  26 & positive regulation of intracellular protein kinase cascade & Under \\ 
  0.03 &   1 &  38 & positive regulation of response to stimulus & Under \\ 
  0.03 &   0 &  24 & oocyte differentiation & Under \\ 
  0.03 &   0 &  23 & oocyte development & Under \\ 
  0.03 &   0 &  23 & cation transport & Under \\ 
  0.04 &   0 &  22 & positive regulation of MAPKKK cascade & Under \\ 
  0.05 &  24 & 234 & growth & Under \\ 
   \hline
0.01 &   4 &   7 & small nuclear ribonucleoprotein complex & Over \\ 
  0.01 &  31 & 164 & mitochondrion & Over \\ 
  0.02 &   2 &   2 & Cajal body & Over \\ 
  0.02 &   2 &   2 & U5 snRNP & Over \\ 
  0.02 &   2 &   2 & U4/U6 x U5 tri-snRNP complex & Over \\ 
  0.03 &  17 &  80 & mitochondrial part & Over \\ 
  0.04 &   3 &   6 & nuclear speck & Over \\ 
  0.04 &   5 &  15 & nuclear body & Over \\ 
  0.04 &  14 &  65 & mitochondrial membrane & Over \\ 
  0.05 &  14 &  66 & mitochondrial envelope & Over \\ 
  0.05 &   2 &   3 & clathrin sculpted vesicle & Over \\ 
  0.05 &   2 &   3 & plasma membrane respiratory chain complex I & Over \\ 
  0.05 &   2 &   3 & plasma membrane respiratory chain & Over \\ 
  0.05 &   2 &   3 & basement membrane & Over \\ 
  0.05 &   2 &   3 & plant-type cell wall & Over \\ 
  0.00 &   0 &  37 & large ribosomal subunit & Under \\ 
  0.01 &   0 &  35 & cytosolic large ribosomal subunit & Under \\ 
  0.01 &  28 & 280 & nucleus & Under \\ 
  0.02 &  19 & 201 & non-membrane-bounded organelle & Under \\ 
  0.02 &  19 & 201 & intracellular non-membrane-bounded organelle & Under \\ 
  0.02 &   4 &  71 & nucleolus & Under \\ 
  0.02 &   3 &  60 & cytosolic ribosome & Under \\ 
  0.02 &   1 &  38 & plastid & Under \\ 
  0.03 &   4 &  68 & cytosolic part & Under \\ 
  0.03 &   1 &  36 & chloroplast & Under \\ 
  0.05 &   5 &  73 & ribosome & Under \\ 
   \hline
\hline
\end{longtable}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Additional Files}

File A\_crassus\_contigs\_full.csv lists all data computed on the contig
level, including sequences (raw, coding, imputed). File
A\_crassus\_contigs\_readable.csv lists only the metadata not including
sequences.

\end{bmcformat}
\end{document}
