%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
\usepackage{rotating}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}


\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of the swimbladder-nematode
    \textit{Anguillicola crassus}: Resources for an alien parasite}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel G Heitlinger\correspondingauthor$^{1,2}$%
       \email{Emanuel G Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Stephen Bridgett$^{3}$%
       \email{Stephen Bridgett- sbridget@staffmail.ed.ac.uk}%
       Anna Montazam$^{3}$%
       \email{Anna Montazam- Anna.Montazam@ed.ac.uk}%
       Horst Taraschewski$^1$%
       \email{Horst Taraschewski- dc20@rz.uni-karlsruhe.de}%
       and Mark Blaxter$^2$%
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute 1, University of Karlsruhe,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
        \iid(3)The GenePool Sequencing Service, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} \textit{Anguillicola crassus} is an
  ecologically, economically and evolutionary interesting nematode. It
  has been introduced from Asia, where it parasitises the Japanese eel
  \textit{Angilla japonica}, to Europe ~30 years ago. Today it infects
  stocks of the endangered, commercially exploited European eel
  \textit{Anguilla anguilla}, permitting and necessitating research in
  a newly established host-parasite system. Furthermore phylogenetics
  places \textit{A. crassus} at a key position for the emergence of
  parasitism, basal to one of the major clades of parasitic nematodes.
  \paragraph*{Results:} 
  After extensive screening of 756.363 raw pyrosequencing reads, we
  assembled 353.055 into 11.371 contigs spanning 7.971.550 bases and
  additionally obtained 21.147 singleton and lower quality contigs
  spanning 8.095.986 bases. We obtained annotations for ca. 60\% of
  the contigs and 40\% of the tentatively unique genes (TUGs)
  confirming the high quality of especially the contigs.  We
  identified 5112 high quality single nucleotide polymorphisms (SNPs)
  and suggest 199 of them as most suitable markers for
  population-genetic studies. The correlation between different
  analyses provided further insights and confirmed biologicaly
  relevant expectations: we found an overabundance of predicted signal
  peptide cleavage sites in sequence conserved in Nematoda and novel
  in \textit{A. crassus}, correlations between coding polymorphism and
  differential expression, between coding polymorphism and peptide
  cleavage sites and between conservation and presence of ortholgs
  with lethal rnai-phenotypes in \textit{C. elegans}. GO-term analysis
  identified an enrichment of peptidases and subunits of the
  respiratory chain for transcritps under positive selection. Enzymes
  for energy metabolism were also found enriched in genes
  differentially expressed between European and Asian
  \textit{A. crassus}.
  \paragraph*{Conclusions:}
  The transcriptome of \textit{A. crassus} is a basis for molecular
  research on this important species. It furthermore provides unique
  insights into the evolution of parasitism in the Spirurina. Energy
  metabolism differs between European and Asian worms due to
  modification in the different host environment or divergent
  evolution of gene expression.
\end{abstract}

\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

The nematode Anguillicola crassus Kuwahara, Niimi et Itagaki, 1974
\cite{kuwahara_Niimi_Itagaki_1974} is a parasite of freshwater eels of
the genus Anguilla, and adults localise to the swim bladder where they
feed on blood. Larvae are transmitted via crustacean intermediate
hosts \cite{de_charleroy_life_1990}. Originally endemic to East-Asian
populations of the Japanese eel (Anguilla japonica),
\textit{A. crassus} has attracted interest due to recent anthropogenic
expansion of its geographic and host ranges to Europe and the European
eel (Anguilla anguilla). Recorded for the first time in 1982 in
North-West Germany \cite{fischer_teichwirt}, where it was most likely
introduced through the live-eel trade
\cite{koops_anguillicola-infestations_1989, koie_swimbladder_1991},
\textit{A. crassus} has spread rapidly through populations of its
newly acquired host \cite{kirk_impact_2003}. At the present day it is
found in all \textit{An. anguilla} populations except those in Iceland
\cite{kristmundsson_parasite_2007}. \textit{A. crassus} can be
regarded as a model for invasive parasite introduction and spread
\cite{taraschewski_hosts_2007}.

\textit{A. crassus} has a major impact on \textit{An. anguilla} populations. In its
natural host in Asia infection prevalence and mean intensity of
infection are lower than in Europe \cite{mnderle_occurrence_2006},
where high prevalence (above 70\% \cite{wrtz_distribution_1998}) and
high infection intesities have been reported throughout the newly
colonized area \cite{lefebvre_anguillicolosis:_2004}. The virulence of
A .crassus in this new host has been attributed to an inadequate
immune response in \textit{An. anguilla} \cite{knopf_swimbladder_2006}. While
the An.  japonica is capable of killing larvae of the parasite after
vaccination \cite{knopf_vaccination_2008} or under high infection
pressure \cite{heitlinger_massive_2009}, responses in \textit{An. anguilla}
have hallmarks of pathology, including thickening of the swim bladder
wall \cite{wurtz_tara_2000}.  Interestingly host also
affects the adult size and life-history of the nematodes: In European
eels the nematodes are bigger and develop and reproduce faster
\cite{knopf_differences_2004}.

The genus Anguillicola is placed in the nematode suborder Spirurina
(clade III sensu \cite{blaxter_molecular_1998})
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
are exclusively parasitic and include important human pathogens (the
causative agents of filariases and ascariasis) as well as prominent
veterinary parasites. Molecular phylogenetic analyses place
Anguillicola in a clade of spirurine nematodes (Spirurina B of
[Laetsch et al submitted]) that have an freshwater or marine
intermediate host, but infect a wide range of carnivorous definitive
hosts. Spirurina B is sister to the main Spirurina C, including the
agents of filariases and ascariasis), and thus \textit{A. crassus} may
be used as an outgroup taxon to understand the evolution of parasitic
phenotypes in these species.

Recent advances in sequencing technology (often termed Next Generation
Sequencing; NGS), provide the opportunity for rapid and cost-effective
generation of genome-scale data. The Roche 454 platform
\cite{pmid16056220} offers longer reads than other NGS technologies,
and thus is suited to de novo assembly of genome-scale data in
previously understudied species. Roche 454 data has particular
application in transcriptomics \cite{pmid20950480}. The difference in
the biology of \textit{A. crassus} in \textit{An. japonica}
(coevolved) and \textit{An. anguilla} (recently captured) eel hosts
likely results from an interaction between different host and parasite
responses, underpinned by definitive differences in host genetics, and
possible genetic differentiation between the invading European and
endemic Asian parasites. As part of a programme to understand the
invasiveness of \textit{A. crassus} in \textit{An. anguilla}, we are
investigating differences in gene expression and genetic distinction
between invading European and endemic Asian \textit{A. crassus}
exposed to the two different host species. Here we report on the
generation of a reference transcriptome for \textit{A. crassus} based
on Roche 454 data, and explore patterns of gene expression and
diversity.


%%%%%%%%%%%%%%%%%%
\section*{Methods}


\subsection*{Nematode samples, RNA extraction, cDNA synthesis and Sequencing}

\textit{A. crassus} from \textit{An. japonica} were sampled from
Kao-Ping river and an adjacent aquaculture in Taiwan as described in
\cite{heitlinger_massive_2009}. Worms from \textit{An. anguilla} were sampled
in Sniardwy Lake, Poland (53.751959N, 21.730957E) and from the
Linkenheimer Altrhein, Germany (49.0262N, 8.310556E). After
determination of the sex of adult nematodes, they were stored in
RNA-later (Quiagen, Hilden, Germany) until extraction of RNA. RNA was
extracted from individual adult male and female nematodes and from a
population of L2 larvae (Table 1). RNA was reverse transcribed and
amplified into cDNA using the MINT-cDNA synthesis kit (Evrogen,
Moscow, Russia).  For host contamination screening a liver-sample from
an uninfected \textit{An. japonica} was also processed.  Emulsion PCR
was performed for each cDNA library according to the manufacturer’s
protocols (Roche/454 Life Sciences), and sequenced on a Roche 454
Genome Sequencer FLX. All samples were sequenced using the FLX
Titanium chemistry, except for the taiwanese female sample T2, which
was sequenced using FLX standard chemistry, to generate between 99,000
and 209,000 raw reads. For the L2 larval library, which had a larger
number of non-\textit{A. crassus}, non-\textit{Anguilla} reads, we
confirmed that these data were not laboratory contaminants by
screening Roche 454 data produced on the same run in independent
sequencing lanes.


\subsection*{Trimming, quality control and assembly}

Raw sequences were extracted in \texttt{fasta}-format (with the
corresponding qualities files) using \texttt{sffinfo} (Roche/454) and
screened for adapter sequences of the MINT-amplification-kit using
\texttt{cross-match} \cite{PHRAP} (with parameters \texttt{-minscore
  20 -minmatch 10}). \texttt{Seqclean} \cite{tgicl_pertea} was used to
identify and remove poly-A-tails, low quality, repetitive and short
(<100 base) sequences. All reads were compared to a set of screening
databases using \texttt{BLAST} (expect value cutoff E<1e-5, low
complexity filtering turned off: -F F). The databases used were (a) a
host sequence database comprising an assembly of the
\textit{An. japonica} Roche 454 data, a unpublished assembly of
\textit{An. anguilla} Sanger dideoxy sequenced expressed sequence tags
(made available to us by Gordon Cramb, University of St Andrews) and
transcripts from EeelBase \cite{pmid21080939} a publicly available
transcriptome database for the European eel; (b) a database of
ribosomal RNA (rRNA) sequences from eel species derived from our Roche
454 data and EMBL-Bank; and (c) a database of rRNA sequences
identified in our \textit{A. crassus} data by comparing the reads to
known nematode rRNAs from EMBL-Bank. This last database notably also
contained xenobiont rRNA sequences. Reads with matches to one of these
databases over more than 80\% of their length and with greater than
95\% identity were removed from the dataset. Screening and trimming
information was written back into sff-format using \texttt{sfffile}
(Roche 454). The filtered and trimmed data were assembled using the
combined assembly approach \cite{pmid20950480}: Two assemblies were
generated, one using \texttt{Newbler v2.6} \cite{pmid16056220} (with
parameters \texttt{-cdna -urt}), the other using \texttt{Mira v3.2.1}
\cite{miraEST} (with parameters
\texttt{--job=denovo,est,accurate,454}). The resulting two assemblies
were combined into one using \texttt{Cap3} \cite{Cap3_Huang} at
default settings and contigs were labeled by whether they derived from
both assemblies or one assembly only (for a detailed analysis of the
assembly categories see the supporting Methods file).

\subsection*{Post-assembly classification and taxonomic assignment of
  contigs}

After assembly contigs were assessed a second time for host and other
contamination by comparing them (using \texttt{BLAST}) to the three
databases defined above, and also to nembase4, a nematode
transcriptome database derived from whole genome sequencing and EST
assemblies \cite{parkinson_nembase:resource_2004, pmid21550347}. For
each contig, the highest-scoring match was recorded as long as it
spanned more than 50\% of the contig. We also compared the contigs to
the NCBI non-redundant nucleotide (NCBI-nt) and protein (NCBI-nr)
databases, recording the taxonomy of all best matches with expect
values better than 1e-05. TUGs with a best hit to non-Metazoans and to
Chordata within Metazoa were additionally excluded from further
analysis.

%% nrow(contig.df[!contig.df$Ac &
%% contig.df$contamination%in%"eelmRNA" & contig.df$phylum.nr%in%"Nematoda", ])
%% investigate if these (149) should not be included rather as
%% Ac-origin DONE!!! The eelmRNA hits are _all_ much better!

\subsection*{Protein prediction and annotation}

Protein translations were predicted from the contigs using
\texttt{prot4EST} (version 3.0b)
\cite{wasmuth_prot4est:_2004}. Proteins were predicted either by
joining single high scoring segment pairs (HSPs) from a \texttt{BLAST}
search of uniref100 \cite{pmid18836194}, or by \texttt{ESTscan}
\cite{estscan}, using as training data the \textit{Brugia malayi}
complete proteome back-translated using a codon usage table derived
from the \texttt{BLAST} HSPs, or, if the first two methods failed,
simply the longest ORF in the contig. For contigs where the protein
prediction required insertion or deletion of bases in the original
sequence, we also imputed an edited sequence for each affected
contig. Annotations with Gene Ontology (GO), Enzyme Commission (EC)
and Kyoto Encyclopedia of Genes and Genomes (KEGG) terms were inferred
for these proteins using \texttt{Annot8r} (version 1.1.1)
\cite{schmid_annot8r:_2008}, using the annotated sequences available
in uniref100 \cite{pmid18836194}. Up to 10 annotations based on a
\texttt{BLAST} similarity bitscore cut-off of 55 were obtained for
each annotation set. The complete \textit{B. malayi} proteome (as
present in uniref100) and the complete \textit{C. elegans} proteome
(as present in wormbase v.220) were also annotated in the same
way. \texttt{SignalP V4.0} \cite{pmid21959131} was used to predict
signal peptide cleavage sites and signal anchor signatures for the
\textit{A. crassus}-transcriptome and similarly again for the
proteomes of the tow model-worms.  Additionally \texttt{InterProScan}
\cite{pmid11590104} (command line utility \texttt{iprscan} (version
4.6) with options \texttt{-cli -format raw -iprlookup -seqtype p
  -goterms}) was used to obtain domain based annotations for the high
credibility assembly (highCA) derived contigs.

We recorded the presence of a lethal rnai-phenotype in the
\textit{C. elegans} ortholog of each TUG using the biomart-interface
\cite{pmid22083790} to wormbase v. 220 through the R-package
\texttt{biomaRt} \cite{pmid19617889}.

\subsection*{Single nucleotide polymorphism analysis}

We mapped the raw reads against the the complete set of contigs,
replacing imputed sequences for originals where relevant, using
\texttt{ssaha2} \cite{pmid11591649} (with parameters \texttt{-kmer 13
  -skip 3 -seeds 6 -score 100 -cmatch 10 -ckmer 6 -output sam -best
  1}). From the \texttt{ssaha2} output, pileup-files were produced
using \texttt{samtools} \cite{journals/bioinformatics/LiHWFRHMAD09},
discarding reads mapping to multiple regions. \texttt{VarScan}
\cite{pmid19542151} (\texttt{pileup2snp}) was used with default
parameters on pileup-files to output lists of single nucleotide
polymorphisms (SNPs) and their locations. For enrichment analysis of
GO-terms we used the R-package \texttt{GOstats} \cite{pmid17098774}.

Using \texttt{Samtools} \cite{journals/bioinformatics/LiHWFRHMAD09}
(\texttt{mpileup -u}) and \texttt{Vcftools} \cite{pmid21653522}
(\texttt{view -gcv}) we genotyped individual libraries for the list of
previously found overall SNPs. Genotype-calls were accepted at a
phred-scaled genotype quality threshold of 10. In addition to the
relative heterozygosity (number of homozygous sites/number of
heterozygous sites) we used the R package \texttt{Rhh}
\cite{pmid21565077} to calculate internal relatedness
\cite{pmid11571049}, homozygosity by loci \cite{pmid17107491} and
standardized heterozygosity \cite{coltman81j} from these data.

Using 1000 bootstrap replicates we confirmed the significance of
heterozygote-heterozygote correlation by analyzing the mean and 95\%
confidence intervals from 1000 bootstrap replicates estimated for all
measurements.

\subsection*{Gene-expression analysis}

Read-counts were obtained from the \texttt{bam}-files generated also
for genotyping using the R-package \texttt{Rsamtoools}
\cite{rsamtools}. Counts to off target data and lowCA contigs were
disregarded. Furthermore contigs with less than 32 reads over all
libraries were excluded from analysis, to avoid inference based on too
low overall experssion values. Because very low coverage from library
E1 and L2 leading highly variable normalized data, we excluded these
libraries from analysis.

% R-package \texttt{DESeq} \cite{pmid20979621} to assess statistical
% significance of differences in counts according to groups of
% libraries.

The statistic of Audic and Claverie \cite{pmid9331369} as implemented
in \texttt{ideg6} \cite{pmid12429865} was used to contrast single
libries. Differential expression between libraries from different sex
of worms was accepted for genes differeing between all female
libraries E2, T1 and T2 versus the male (M) library ($p<$0.01) but not
within any of the female libraries at the same threshold. Differential
expression between libraries from European and Asian origin was
accepted for genes differeing between libraries E2 versus T1 and T2
($p<$0.01) but not between T1 versus T2.

\subsection*{Over-representation analyses}

Prior to analysis of GO-term over-representation (based on dn/ds or
expression values) we used the R-package \texttt{annotationDbi}
\cite{AnnotationDbi} to obtain a full list of associations (also with
higer-level terms) from \texttt{annot8r}-annotations. We then used the
R-package \texttt{topGO} \cite{topGO} to traverse the annotation-graph
and analyse each node in the annotation for over-representation of the
associated term in the focal gene-set compared to a appropriate
universal gene-set (all contigs with dn/ds values or all contigs
analysed for gene-exprssion) with the ``classic'' method and Fisher's
exact test.

We used non-paramentric Mann-Whitney u-tests to test the influence of
factors on dn/ds values, when multiple contrasts between groups
(facotrs) were investigated we used Nemenyi-Damico-Wolfe-Dunn
tests. For overrepresentation of one group (factor) in other groups
(factors) we used Fisher's exact test.

\subsection*{General coding methods}

The bulk of analysis (unless otherwise cited) presented in this paper
was carried out in \texttt{R} \cite{R_project} using custom
scripts. We used a method provided in the R-packages \texttt{Sweave}
\cite{lmucs-papers:Leisch:2002} and \texttt{Weaver} \cite{weaver} for
``reproducible research'' combining \texttt{R} and \LaTeX code in a
single file. All intermediate data files needed to compile the present
manuscript from data-sources are provided upon request. For
visualisation we used the R-packages \texttt{ggplot2}
\cite{ggplot-book} and \texttt{VennDiagram} \cite{pmid21269502}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}

<<load.libs, echo=FALSE, results=hide>>=
library(VennDiagram)
library(xtable)
library(reshape)
library(ggplot2)

library(AnnotationDbi)
library(topGO)

library(limma)
library(XML)
library(coin)
library(multcomp)

library(DESeq)
library(edgeR)
library(ShortRead)
library(biomaRt)

library(BioIDMapper)

library(Rhh)
library(Heatplus)

source("/home/ele/thesis/454/common_R_functions.R")
@ 

\subsection*{Sampling \textit{A. crassus}}

One female worm and one male worm were sampled from an aquaculture
with height infection loads in Taiwan. An additional female worm was
sampled from a stream with low infection pressure adjacent to the
aquaculture. All these worms were parasitising endemic
\textit{An. japonica}. A female worm and pool of L2 larval stages were
sampled from \textit{An. anguilla} in the river Rhine, one female worm
from a lake in Poland. All adult worms were filled with large amounts
of host-blood, therefore we anticipated abundant host-contamination in
sequencing data and decided to sequence a liver sample of an uninfected
\textit{An. japonica} for screening.

 \subsection*{Sequencing, trimming and pre-assembly screening}

<<trim, echo=FALSE, results=hide, cache=TRUE>>=
trimdat <- readLines("/home/ele/Data/454/trimmed/trimming.stats")
trimfile <-  grep("Trimming sff file *", trimdat, value=TRUE)
trimfile <- sapply(strsplit(trimfile, " |,"), function (x) basename(x[4]))
lib <- gsub(".sff'", "", trimfile)

raw_reads<- grep("Sequences analyzed:.*", trimdat, value=TRUE)
raw_reads <- as.numeric(gsub("Sequences analyzed: *", "", raw_reads))
short <- grep("by 'short':.*", trimdat, value=TRUE)
short <- as.numeric(gsub(" *by 'short': *", "", short))
lowq <- grep("by 'low_qual':.*", trimdat, value=TRUE)
lowq <- as.numeric(gsub(" *by 'low_qual': *", "", lowq))
dust <- grep("by 'dust':.*", trimdat, value=TRUE)
dust <- as.numeric(gsub(" *by 'dust': *", "", dust))
shortq <- grep("by 'shortq':.*", trimdat, value=TRUE)
shortq <- as.numeric(gsub(" *by 'shortq': *", "", shortq))
valid <- raw_reads-(short+lowq+dust+shortq)

TRIM <- data.frame(row.names=lib, raw_reads, short, lowq, dust, shortq, valid)
TRIM["total",] <- colSums(TRIM)

## subsitute the publication style library abrevations
rownames(TRIM) <- gsub("179F", "T1", rownames(TRIM))
rownames(TRIM) <- gsub("10F", "T2", rownames(TRIM))
rownames(TRIM) <- gsub("KS4F", "E1", rownames(TRIM))
rownames(TRIM) <- gsub("UW07F", "E2", rownames(TRIM))
rownames(TRIM) <- gsub("L2R3", "L2", rownames(TRIM))
rownames(TRIM) <- gsub("M175", "M", rownames(TRIM))

@ 

<<read.libs, echo=FALSE, results=hide, cache=TRUE>>=

RE <- list()
for (l in (c("10F", "179F", "M175", "L2R3", "KS4F", "UW07F"))){
   command <- paste("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/trimmed/",
                    l, ".sff.trimmed.sff;", sep="")
   connect <- pipe(command)
   RE[l] <- list(read.sequences(connect))
   closeAllConnections()
}
names <- gsub(" .*$", "", unlist(lapply(RE, names)))
lib <- melt(RE)

## subsitute the publication style library abrevations
lib$L1 <- gsub("179F", "T1", lib$L1)
lib$L1 <- gsub("10F", "T2", lib$L1)
lib$L1 <- gsub("KS4F", "E1", lib$L1)
lib$L1 <- gsub("UW07F", "E2", lib$L1)
lib$L1 <- gsub("L2R3", "L2", lib$L1)
lib$L1 <- gsub("M175", "M", lib$L1)

RE <- cbind(read=names, lib)
names(RE)[2:3] <- c("seq", "lib")
RE$seq <- as.character(RE$seq)
RE$length <- nchar(RE$seq)

eel.all <- read.sequences(pipe("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/Li68_assembly/sff/Li68.sff"))

eel.clean <- read.sequences(pipe("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/Li68_assembly/sff/Li68.sff.trimmed.sff"))

eel.rRNA.contigs <- read.sequences("/home/ele/Data/454/Li68_assembly/Aj_liver_rRNA.fasta")

eel.contigs <- read.sequences("/home/ele/Data/454/Li68_assembly/Aj_liver.fasta")

@ 

\SweaveInput{../contamination/pre_assembly.Rnw}

A total of \Sexpr{TRIM["total","raw_reads"]} raw sequencing reads were
generated for \textit{A. crassus} (Table 1). These were trimmed for
base call quality, and filtered by length to give
\Sexpr{TRIM["total","valid"]} high-quality reads (spanning
\Sexpr{sum(nchar(R$seq))#$} bases). In the eel dataset from
\Sexpr{length(eel.all)} raw reads \Sexpr{length(eel.clean)} were
assembled after basic quality screening.

We then screened the \textit{A. crassus} reads for contamination by
host (\Sexpr{sum(pre.screen.tab$eelmRNA)#$} reads matched previously
sequenced eel genes or our own \textit{An. japonica} 454
transcriptome, which had been assembled into
\Sexpr{length(eel.contigs)} mRNA
contigs. \Sexpr{sum(pre.screen.tab$AcrRNA)#$} reads matched large or
small subunit nuclear or mitochondrial ribosomal RNA sequences of
\textit{A. crassus} (Table 1). In addition to fish mRNAs, we
identified (and removed) \Sexpr{pre.screen.tab[["L2","Cercozoa"]]}
reads in the library derived from the L2 nematodes that had
significant similarity to cercozoan (likely parasite) ribosomal RNA
genes (Table 1).

\subsection*{Assembly}

<<load.meth, echo=FALSE>>=
load("/home/ele/thesis/454/Method-assembly/Method.Rdata")

nMN <- nrow(contig.df[contig.df$category%in%"MN",])

nbad.c <- nrow(contig.df[contig.df$category%in%
                         c("M_n","N_n","N_1","M_1"),])

nSing <- nrow(contig.df[contig.df$category%in%0,])
@

We assembled the remaining \Sexpr{length(raw.fasta)} reads (spanning
\Sexpr{nBases} bases) using the combined assembler strategy
\cite{pmid20950480} and Roche 454 GSassembler (version 2.6) and MIRA
(version 3.21) \cite{miraEST}. From this we derived \Sexpr{nMN}
contigs that were supported by both assembly algorithms,
\Sexpr{nbad.c} contigs only supported by one of the assembly
algorithms and \Sexpr{nSing} singletons that were not assembled by
either approach (Table 2). When scored by matches to known genes, the
contigs supported by both assemblers are of the highest credibility,
and this set is thus termed the high credibility assembly
(highCA). Those with evidence from only one assembler and the
singletons are of lower credibility (lowCA). These datasets are the
most parsimonious (having the smallest size) for their quality
(covering the largest amount of sequence in reference
transcriptomes). In the highCA parsimony and low redundancy is
prioritized, while in the complete assembly (highCA plus lowCA)
completeness is prioritized. The 40187 sequences (contig consensuses
and singletons) in the complete assembly are referred to below as
tentatively unique genes (TUGs).

<<mean.cov, echo=FALSE, cache=TRUE>>=

cov.mn <- tapply(con.pile$coverage,
                 con.pile$contig%in%contig.df[contig.df$category=="MN", "contig"],
                 mean)
                            
cov.mn.u <- tapply(con.pile.uniq$uniq_coverage,
                   con.pile.uniq$contig%in%contig.df[contig.df$category=="MN", "contig"],
                   mean)

cov.nu.total <- mean(con.pile$coverage)
cov.u.total <- mean(con.pile.uniq$uniq_coverage)

@ 

\SweaveInput{../contamination/post_assembly.Rnw}

We screened the complete assembly for residual host contamination, and
identified \Sexpr{n.fish} TUGs that had higher, significant similarity
to eel (and chordate) sequences (our and publicly available 454 ESTs
and EMBLBank Chordata proteins) than to nematode sequences
\cite{pmid21550347}.

Given our prior identification of cercozoan ribosomal RNAs, we also
screened the complete assembly for contamination with other
transcriptomes.

1153 TUGs were found mapping to Eukaryota outside of the kingdoms
Metazoa, Fungi and Viridiplantae. These hits included a wide range of
Protists ranging from Apicomplexa (mainly Sarcocystidae, 28 hits and
Cryptosporidiidae 10 hits) over Bacillariophyta (diatoms, mainly
Phaeodactylaceae, 41 hits) and Phaeophyceae (brown algae, mainly
Ectocarpaceae, 180 hits) and Stramenopiles (Albuginaceae, 63 hits) to
Kinetoplasitda (Trypanosomatidae, 26 hits) and Heterolobosea
(Vahlkampfiidae, 38 hits).

Additionally we found \Sexpr{other.king[["Fungi"]]} TUGs with hits to
fungi (e.g Ajellomycetaceae, 53 hits) and
\Sexpr{other.king[["Viridiplantae"]]} TUGs with hits to plants.

Hits outside the Eukaryota were mainly to Bacteria (825 hits) and
within those mostly to members of the Proteobacteria (484 hits). No
hits were found to Wolbachia or related Bacteria known as symbionts of
nematodes and arthropods. 9 TUGs were hitting sequence from Viruses and
8 from Archaea.

We excluded all TUGs with best hits outside Metazoa and our assembly
thus has \Sexpr{nrow(contig.df[contig.df$Ac,])#$} TUGs, spanning
\Sexpr{sum(nchar(contig.df[contig.df$Ac,"seq"]))#$} bases (of which
\Sexpr{nrow(contig.df[contig.df$AcMN,])#$} are highCA-derived, and
span \Sexpr{sum(nchar(contig.df[contig.df$AcMN,"seq"]))#$} bases) that
are likely to derive from of \textit{A. crassus}.

\subsection*{Protein prediction}
\SweaveInput{../conservation/conservation.Rnw}
\SweaveInput{../protein_prediction/imputed_from_p4EST.Rnw}
\SweaveInput{../snp_analysis/snp_analysis.Rnw}
\SweaveInput{../snp_analysis/snp_by_population.Rnw}
\SweaveInput{../annotation/annotation.Rnw}
\SweaveInput{../annotation/read_signalp.Rnw}
\SweaveInput{../annotation/brugia-compare.Rnw}
\SweaveInput{../snp_analysis/overrepresentation_snp.Rnw}


For
\Sexpr{nrow(contig.df[contig.df$Ac&contig.df$method!="no-prediction",])}
TUGs a protein was predicted using prot4EST
\cite{wasmuth_prot4est:_2004} (Table 2). The full open reading frame
was obtained in
\Sexpr{nrow(contig.df[contig.df$Ac&imputed$full.length,])} TUGs, while
while for \Sexpr{nrow(contig.df[contig.df$Ac&imputed$full.5p,])} the
5’ end and for \Sexpr{nrow(contig.df[contig.df$Ac&imputed$full.3p,])}
the 3' end was complete. In \Sexpr{same.same[["FALSE"]]} TUGs the
corrected sequence with the imputed ORF was slightly changed compared
to the raw sequence.

\subsection*{Annotation}

We obtained basic annotations with orthologous sequences from
\textit{C. elegans} for
\Sexpr{nrow(contig.df[contig.df$Ac&!is.na(contig.df$Ce.hit),])} TUGs,
from \textit{B. malayi} for
\Sexpr{nrow(contig.df[contig.df$Ac&!is.na(contig.df$Bm.hit),])} TUGs,
from nempep \cite{parkinson_nembase:resource_2004, pmid21550347} for
\Sexpr{nrow(contig.df[contig.df$Ac&!is.na(contig.df$nempep.hit),])}
TUGs and with uniprot proteins for
\Sexpr{nrow(contig.df[contig.df$Ac&!is.na(contig.df$nr.bit),])} TUGs.

We used annot8r \cite{schmid_annot8r:_2008} to assign gene ontology
(GO) terms for \Sexpr{nGO} TUGs, Enzyme Commission (EC) numbers for
\Sexpr{nEC} TUGs and Kyoto Encyclopedia of Genes and Genomes (KEGG)
pathway annotations for \Sexpr{nKEGG} TUGs (Table 2). Additionally
\Sexpr{nIPR} highCA derived contigs were annotated with GO terms
through \texttt{InterProScan} \cite{pmid11590104}. Nearly one third
(\Sexpr{nallannotations}) of the \textit{A. crassus} TUGs were
annotated with at least one identifier, and \Sexpr{all.identifier} had
GO, EC and KEGG annotations (Figure 1).

We compared our \textit{A. crassus} GO annotations for high-level
GO-slim terms to the annotations (obtained the same way) for the
complete proteome of the filarial nematode \textit{B. malayi} and the
complete proteome of \textit{C. elegans} (Figure 2).

Correlation shows the occurrence of terms for the partial
transcriptome of \textit{A. crassus} to be more similar to the
proteome of \textit{B. malayi} (\Sexpr{round(Go.rep.Ac.Bm,2)};
Spearman correlation coefficient) than to the proteome of
\textit{C. elegans} (\Sexpr{round(Go.rep.Ac.Ce,2)}). Also the tow
model-nematode compared to each other (\Sexpr{round(Go.rep.Bm.Ce,2)})
are less similar in the occurrence of terms than the two parasites.

We inferred presence of signal peptide cleavage sites in the predicted
protein sequence using \texttt{SignalP} \cite{pmid21959131}. We predicted
\Sexpr{nsigP} signal peptide cleavage sites and \Sexpr{nsigPtm} signal
peptides with a transmembrane signature. Again these predictions are
more similar to predictions using the same methods for the proteome
\textit{B. malayi} (\Sexpr{nsigP.Bm} signal peptide cleavage sites and
\Sexpr{nsigPtm.Bm} with transmembrane anchor) than for the proteome of
\textit{C. elegans} (\Sexpr{nsigP.Ce} signal peptide cleavage sites
and \Sexpr{nsigPtm.Ce} with transmembrane anchor).

We inferred the presence of a lethal rnai phenotype in the orthologous
annotation of \textit{C. elegans}. For \Sexpr{rnai.contigs[["FALSE"]]}
TUGs a non-lethal phenotype was inferred, for
\Sexpr{rnai.contigs[["TRUE"]]} TUGs a lethal phenotype.

\subsection*{Evolutionary conservation}

\textit{A. crassus} TUGs were classified as conserved, conserved in
Metazoa, conserved in Nematoda, conserved in Spirurina or novel to
\textit{A. crassus} by comparing them to public databases and using
two \texttt{BLAST} bit-score cutoffs to define relatedness (Table 3).

Roughly a third and a quarter of the higCA derived contigs were
categorized as conserved across kingdoms at a bitscore threshold of 50
and 80, respectively. Roughly half or 3/5 of the these contigs were
identified as novel in \textit{A. crassus}.

The remaining higCA contigs spread across intermediate
relatedness-levels. More sequences were categorised as novel at the
phylum level (Nematoda) compared to kingdom and clade III level and the
number of contigs at intermediate relatedness-levels was roughly
consistent for the two bitscore thresholds.

The latter points about intermediate conservation levels were also
true, when all TUGs were analysed. The numbers of TUGs categorised at
these intermediate levels roughly doubled. In contrast, the proportion
of additional conserved lowCA TUGs is small compared to additional
TUGs categorised as novel in \textit{A. crassus}, mirroring the higher
amount of erroneous sequence.

Proteins predicted to be novel to Nematoda and novel in
\textit{A. crassus} were significantly enriched in signal peptide
annotation compared to conserved proteins, proteins novel in Metazoa
and novel in clade III (Fisher's exact test p$<$0.001 ; Figure 3).

The proportion of lethal rnai phenotypes was significantly higher for
orthologs of conserved TUGs
(\Sexpr{con.rnai["conserved","proportion"]}\%) than for orthologs of
TUGs not conserved (\Sexpr{con.rnai["other","proportion"]}\%) across
kingdoms (p$<$0.001, Fisher's exact test).

\subsection*{Identification of single nucleotide polymorphisms}

We called single nucleotide polymorphisms (SNPs) on the
\Sexpr{nCov8bases} bases of the TUGs that had coverage of more then
8-fold available using \texttt{VARScan} \cite{pmid19542151}. We
excluded SNPs predicted to have more than 2 alleles or that mapped to
an undetermined (N) base in the reference, and retained
\Sexpr{nrow(VAR)} SNPs. The ratio of transitions (ti;
\Sexpr{raw.ts.tv[["transitions"]]}) to transversion (tv;
\Sexpr{raw.ts.tv[["transversions"]]}) in this set was
\Sexpr{round(raw.ts.tv[["ratioTS.TV"]],2)}. Using the prot4EST
predictions and the corrected sequences, \Sexpr{nrawOrf[["TRUE"]]} of
the SNPs were predicted to be inside an ORF, with
\Sexpr{nrawFrame[[1]]} at codon first positions,
\Sexpr{nrawFrame[[2]]} at second positions and \Sexpr{nrawFrame[[3]]}
at third positions. As expected ti/tv inside ORFs
(\Sexpr{tsv.raw.orf[2,3]}) was higher than outside ORFs
(\Sexpr{tsv.raw.orf[1,3]}). The ratio of synonymous polymorphisms per
synonymous site to non-synonymous polymorphisms per non-synonymous
site (dn/ds) was \Sexpr{raw.dn.ds}. We filtered these SNPs to exclude
those that might be associated with analytical bias. As Roche 454
sequences have well-known systematic errors associated with
homopolymeric nucleotide sequences \cite{pmid21685085}, we analysed
the effect of exclusion of SNPs in, or close to, homopolymer
regions. We observed changes in ti/tv and in dn/ds when SNPs were
discarded using different size thresholds for homopolymer runs and
proximity thresholds (see Figure 4). Based on this we decided to
exclude SNPs with a homopolymer-run as long as or longer than 4 bases
inside a window of 11 bases (5 to bases to the right, 5 to the left)
around the SNP. We also observed a relationship between TUG dn/ds and
TUG coverage, associated with the presence of sites with low abundance
minority alleles (less than 7\% of the allele calls), suggesting that
some of these may be errors. Removing low abundance minority allele
SNPs from the set removed this effect (Figure 5).  Our filtered SNP
dataset includes \Sexpr{nrow(VARqp)} SNPs. We retained
\Sexpr{per.base} SNPs per kb of contig sequence, with
\Sexpr{s.per.s.base} synonymous SNPs per 1000 synonymous bases and
\Sexpr{n.per.n.base} non-synonymous SNPs per 1000 non-synonymous
bases. A mean dn/ds of \Sexpr{round(mean(dn.ds.df$dn.ds),3)#$} was
calculated for the \Sexpr{nrow(dn.ds.df)} TUGs
(\Sexpr{nrow(dn.ds.df[dn.ds.df$AcMN,])#$} highCA-derived contigs)
containing at least one synonymous SNP.

\subsection*{Polymorphisms associated with biological processes}

We consolidated our annotation and polymorphism analyses by examining
correlations between nonsynonymous variability and particular
classifications.

Signal peptide containing proteins have been shown to have higher
rates of evolution than cytosolic proteins in a number of nematode
species. In \textit{A. crassus}, TUGs predicted to contain signal
peptide cleavage sites in SignalP showed non-significant a trend
towards higher dn/ds values than TUGs without signal peptide cleavage
sites (p = \Sexpr{round(u.test[["p.value"]],3)}; two sided
Mann-Whitney-test; Figure 6).

Positive selection can be inferred from dn/ds analyses, and we defined
TUGs with a dn/ds higher than 0.5 as positively selected. We
identified over-represented GO ontology terms associated with these
putatively positively selected genes (Table 4; and additional figure
1). Within the molecular function category, ``peptidase activity'' was
the most significantly overrepresented term and had twelve TUGs
supporting the overrepresentation. The highlighted twelve peptidases
annotated with eleven unique orthologs in \textit{C. elegans} and
\textit{B. malayi}.

%% The term ``structural constituent of ribosome'' was
%% underrepresented.

% While the biological process and cellular compartment categories
% provide less information for a nematode (highlighting e.g. brain or
% pancreas development), underrepresented terms in both were connected
% to ribosomal proteins, validating the analysis for the molecular
% function category.

Other overrepresented terms abundant over categories pointed to
subunits of the respiratory chain e.g. ``heme-copper terminal oxidase
activity'' and ``cytochrome-c oxidase activity'' in molecular function
and ``mitochondrion'' in cellular compartment (see Table 6 and
additional figure 1).

At both bitscore thresholds contigs novel in clade III and novel in
\textit{A. crassus} had a significantly higher dn/ds than other
contigs (novel.in.metazoa - novel.in.Ac, 0.005 and 0.015;
novel.in.nematoda - novel.in.Ac, 0.005 and 0.002; novel.in.nematoda -
novel.in.clade3, 0.207 and 0.045; comparison, p-value from bitscore of
50 and p-value from bitscore of 80, Nemenyi-Damico-Wolfe-Dunn test,
given only for significant comparisons).

Orthologs of \textit{C. elegans} transcripts with lethal
rnai phenotype are expected to evolve under stronger selective
constraints. Indeed the values of dn/ds showed a non-significant trend
towards lower values in TUGs with orthologs with a lethal phenotype
compared to a non-lethal phenotypes
(p=\Sexpr{round(u.rnai$p.value,3)#$}, two-sided U-test).

\subsection*{SNP markers for single worms}

We used \texttt{Samtools}\cite{journals/bioinformatics/LiHWFRHMAD09}
and \texttt{Vcftools}\cite{pmid21653522} to call genotypes in single
worms (adult sequencing libraries). This resulted in \Sexpr{nrow(Rh)}
informative sites in \Sexpr{geno.contigs} contigs, where two alleles
were found in at least one assured genotype at least in one of the
worms.

Internal relatedness \cite{pmid11571049}, homozygosity by loci
\cite{pmid17107491} and standardised heterozygosity \cite{coltman81j}
were all highlighting the Taiwanese worm from the wild population
(sample T1) as the most and the European worm from Poland (sample E2)
as the least heterozygous individual. The other worms had intermediate
values between these two extremes.

We confirmed the genome-wide significance of these estimates using
heterozygosity-heterozygosity correlation \cite{pmid21565077}. These
tests confirmed the representativeness of the \Sexpr{nrow(Rh)}
SNP-markers for the whole genome in population genetic studies ($\mu$
= 0.78, $ci_l$=0.444; $\mu$ = 0.86 and $ci_l$ = 0.596; $\mu$ = 0.87
and $ci_l$= 0.632; mean and lower bound of 95\% confidence intervals
from 1000 bootstrap replicates for internal relatedness, homozygosity
by loci and standardised heterozygosity). Using a higher number of
genotyped individuals these markers would allow to asses the amount of
inbreeding in populations of \textit{A. crassus}..

 \subsection*{Differential expression}
                
\SweaveInput{../gene_expression/gene_expression.Rnw}

\SweaveInput{../gene_expression/overrepresentation_exp.Rnw}

We also analysed gene-expression inferred from mapping. Of the
\Sexpr{length(raw.fasta)} reads \Sexpr{n454.map}
(\Sexpr{round((n454.map/length(raw.fasta))*100,2)}\%) mapped uniquely
(with their best hit) to the fullest assembly (including the all
assembled contigs as a ``filter'' later removing screened out
sequences for analysis). The number of reads mapping is given for each
library Table 1, to get unbiased estimates of expression we removed
also all contigs with a coverage lower than 32 reads overall and thus
analysed \Sexpr{nrow(ideg6)} contigs.

\Sexpr{nrow(express.eel.sig[express.sex.sig$LogFC<0,])#$} contigs
showed an expression predominantly in the male library,
\Sexpr{nrow(express.eel.sig[express.eel.sig$LogFC>0,])#$} contigs in
the female
library. \Sexpr{nrow(express.eel.sig[express.eel.sig$LogFC>0,])#$}
contigs were primarily expressed in the libraries from Taiwan,
\Sexpr{nrow(express.eel.sig[express.eel.sig$LogFC<0,])#$} contigs in
the European library.

Overrepresentation of of GO-terms differentially expressed between the
male and female libraries highlighted especially ribosomal proteins
oxidoreductases and collagen processing enzymes as enriched (Table 6
a; additional figures 1). Ribosomal proteins were all overexpressed in the male library,
oxidoreductases and collagen processing enzymes were overexpressed
female libraries.

Overrepresentation of of GO-terms differentially expressed between
libraries from woms of European and Asian origin highlighted catalytic
activity especially related to metabolism (Table 6 b; additional
Figure 3).  Acyltransferase contigs were all upregulated in the
European libraries. However, the expression patterns for other contigs
connected to metabolsim did not show concerted up or down-regulation
(eg. for ``steroid biosynthetic process'' 2 contigs were downregulated
in the European library, 3 contigs upregulated).

Enrichment of signal-positives was not found in any category of
overexpressed genes. Differntially expressed genes also showed no
pattern of enrichement in conservation categories and no enrichment of
\textit{C. elegans} orthologs with lethal/non-lethal rnai-phenotypes.

Significantly elevated dn/ds was found for contigs differentially
expressed according to worm-origin (Fisher's exact test
p=\Sexpr{round(u.test.dn.ds.eel$p.value,3)#$}; also both up- or
downregulated were significant). Contigs overexpressed in the female
libraries showed elevated levels of dn/ds (Fisher's exact test
p=\Sexpr{round(u.test.dn.ds.female$p.value,3)#$}). In contrast male
overexpressed genes showed decreased levels of dn/ds (Fisher's exact
test p=\Sexpr{round(u.test.dn.ds.male$p.value,3)#$}).

<<save, echo=FALSE>>=
## write the MN.peptide sequence for iprscan prediction
MN.pep.fasta <- as.character(contig.df[contig.df$AcMN, "Ac.pep"])
names(MN.pep.fasta) <- as.character(contig.df[contig.df$AcMN, "contig"])
write.sequence(MN.pep.fasta, "/home/ele/thesis/454/MN.pep.fasta")

## for a more readable version of contig.df
readable <- !names(contig.df)%in%c("seq", "imp", "ontology", "coding", "Ac.pep")
write.csv(contig.df, "../paper/additional_table1a.csv",
row.names=FALSE)
write.csv(contig.df[,readable], "../paper/additional_table1b.csv",
row.names=FALSE)

write.csv(VARqp[, c("contig", "base", "from", "to",
"nfrom", "nto", "perc", "inORF",
"inFRAME", "effect")], "../paper/additional_table2.csv",
row.names=FALSE)

names(express.sex.sig) <- c("E2", "T1", "T2", "M", "LogFC")
write.csv(express.sex.sig, "../paper/additional_table3a.csv",
row.names=FALSE)

names(express.eel.sig) <- c("E2", "T1", "T2", "LogFC")
write.csv(express.eel.sig, "../paper/additional_table3b.csv",
row.names=FALSE)

## for summary statistics a non-Ac reduced dataset
c.df <- contig.df[contig.df$Ac, ]

save.image("/home/ele/thesis/454/paper/paper.Rdata")
@ 

\section*{Discussion}

We have generated a de novo transcriptome for \textit{A. crassus} an
important invasive parasite that threatens wild stocks of the European
eel \textit{An. anguilla}. These data enable a broad spectrum of
molecular research on this ecologically and economically important
parasite. As \textit{A. crassus} lives in close association with its
host, we have used exhaustive filtering to attempt to remove all
host-derived, and host-associated organism-derived contamination from
the data. To do this we have also generated a transcriptome dataset
from the definitive host \textit{An. japonica}. The non-nematode,
non-eel data identified, particularly in the L2 sample, showed highest
identity to flagellate protists, which may have been parasitising the
eel (or the nematode). Encapsulated objects observed in eel swim
bladder walls \cite{heitlinger_massive_2009} could be due solely to
immune attrition of \textit{A. crassus} larvae or to other
coinfections.

A second examination of sequence origin was performed after assembly,
employing higher stringency cutoffs.  Similar taxonomic screening was
used in a garter snake transcriptome project \cite{pmid21138572}, and
an analysis of lake sturgeon tested and rejected hypotheses of
horizontal gene-transfer when xenobiont sequences were identified
\cite{pmid20386959}. A custom pipeline for transcriptome assembly from
pyrosequencing reads \cite{pmid20034392} proposed the use of EST3
\cite{pmid17218127} to infer sequence origin based simply on
nucleotide frequency. We were not able to use this approach
successfully, probably due to the fact that xenobiont sequences in our
data set derive from multiple sources with different GC content and
codon usage.

Compared to other NGS transcriptome sequencing projects
\cite{pmid20478048}, the combined assembly approach generated a
smaller number of contigs that had lower redundancy and higher
completeness. Projects using the mira assembler often report
substantially greater numbers of contigs for datasets of similar size
(see e.g. \cite{pmid21364769}), comparable to the mira sub-assembly in
our approach. The use of oligo(dT) to capture mRNAs probably explains
the bias towards 3' end completeness and a relative lack of true
initiation codons in our protein prediction. This bias is
near-ubiquitous in deep transcriptome sequencing projects
(e.g. \cite{pmid20331785}).

***START: I moved the chunck below up (from a paragraph you did not
correct last time) to have annotation before SNPs

We were able to obtain high-quality annotations for a large set of
TUGs: For 40\% of the complete assembly and 60\% of our highCA assembly
\texttt{BLAST}-based annotations could be obtained. 45\% of the
contigs in the highCA assembly were additionally decorated with
domain-based annotations through \texttt{InterProScan}
\cite{pmid11590104}.

Comparison with complete protein sequence from the genomes of
\textit{B. malayi} and \textit{C. elegans} showed a remarkable degree
of agreement regarding the occurrence of terms in the two parasitic
worms. This agreement was higher than with the free living nematode
\textit{C. elegans} and even the two genome-sequencing-derived
proteomes showed less agreement with each other than the filarial
parasite with our dataset. This implies that our transcriptome is
truly a representative partial genome
\cite{parkinson_partigene--constructing_2004} of a parasitic nematode.

Analysis of conservation identified more sequence novel in Nematode
than in the eukaryote kingdom or in clade III this is in agreement
with prevalence of genic novelty in the Nematoda
\cite{wasmuth_extent_2008}. Furthermore the basal position of
\textit{A. crassus} in clade III could be leading to most novelty in
the clade not being shared with \textit{A. crassus}.

TUGs predicted to be novel in the phylum Nematoda and novel to
\textit{A. crassus} contained the highest proportion of
signal-positives. This confirms observations made in a study on
\textit{Nippostrongylus brasiliensis} \cite{harcus_signal_2004}, where
signal positives were reported as less conserved. Interestingly
enrichment of signal sequence bearing TUGs in our dataset was
constrained to sequences novel in nematodes and \textit{A. crassus}
(i.e. not to the level of clade III). This may be explained, whit two
different hypotheses involving the basal position of
\textit{A. crassus}: First the signal positives shared with all
nematodes could be conserved molecules not excreted by parasites. A
different class of secreted/excreted molecules with prominent role in
host parasite interactions would not have arisen early in the
evolution of parasitism in clade III - or be too fast-evolving - and
thus be detected as specific to deeper sub-clades (i.e. to
\textit{A. crassus} in our dataset). A second explanation would be,
that orthologs of excreted parasite-specific genes could be among
those shared with other nematodes and the fewer shared with clade III
implying a predisposition to parasitism outside of the Spirurina or
even the convergent evolution of secreted molecules in other parasitic
nematodes. However analysis of dn/ds (see below) across conservation
categories favours the first hypothesis, as it identifies a higher
amount of positive selection in TUGs novel to clade III and
\textit{A. crassus} than to nematodes.

***End: I moved the above chunck up to have annotation before SNPs

We generated transcriptome data from multiple \textit{A. crassus} of
Taiwanese and European origin, and identified SNPs both within and
between populations. Screening of SNPs in or adjacent to homopolymer
regions improved overall measurements of SNP quality. The ratio of
transitions to transversions (ti/tv) increased. Such an increase is
explained by the removal of “noise” associated with common homopolymer
errors \cite{pmid21685085}. The value of \Sexpr{raw.ts.tv[,3]}
(\Sexpr{tsv.raw.orf[1,3]} outside, \Sexpr{tsv.raw.orf[2,3]} inside
ORFs) is in good agreement with the overall ti/tv of humans (2.16
\cite{pmid21169219}) or \textit{Drosophila} (2.07
\cite{pmid21143862}). The ratio of non-synonymous SNPs per
non-synonymous site to synonymous SNPs per synonymous site (dn/ds)
decreased with removal of SNPs adjacent to homopolymer regions from
\Sexpr{raw.dn.ds} to \Sexpr{round(mean(dn.ds.df$dn.ds),3)#$} after
full screening. The most plausible explanation is the removal of
error, as unbiased error would lead to a dn/ds of 1. While dn/ds is
not unproblematic to interpret within populations \cite{pmid19081788},
the assumption of negative (purifying) selection on most
protein-coding genes makes lower mean values seem more plausible. We
used a threshold value for the minority allele of 7\% for exclusion of
SNPs, based on an estimate that approximately 10 haploid equivalents
were sampled (5 individual worms plus an negligible contribution from
L2 larvae in the L2 library and within the female adult worms). The
benefit of this screening was mainly a reduction of non-synonymous
SNPs in high coverage contigs, and a removal of the dependence of
dn/ds on coverage. Working with an estimate of dn/ds independent of
coverage, efforts to control for sampling biased by depth
(i.e. coverage; see \cite{pmid18590545} and \cite{pmid20478048}) could
be avoided.

*** you corrected up to her last time

Also in comparison with published intra-species values of dn/ds our
final estimate of seems plausible: in transcripts from the female
reproductive tract of \textit{Drosophila} dn/ds was 0.15
\cite{pmid15579698} and 0.21 in the male reproductive tract
\cite{pmid11404480} (although for ESTs specific to the male accessory
gland were shown to have a higher dn/ds of 0.47). A pyrosequencing
study in the parasitic nematode \textit{Ancylostoma canium}
\cite{pmid20470405} reported dn/ds of 0.3.

When the whole of coding sequences are studied, of which only a small
subset of sites can be under diversifying selection, dn/ds of ~0.5 has
been suggested as threshold for assuming diversifying selection
\cite{pmid15579698} instead of the classical threshold of 1
\cite{pmid6449605}. The use of this threshold for positive selection
led to the identification of over-represented of GO-term highlighting
very interesting transcripts:

Twelve peptidases under positive selection (from 43 with a dn/ds
obtained) meant an enrichment in the category. All twelve have
different orthologs in \textit{B. malayi} and \textit{C. elgans} and
are conserved across kingdoms. Despite their conservation peptidases
are thought to have acquired new and prominent roles in host-parasite
interaction compared to free living organisms: In \textit{A. crassus}
a trypsin-like proteinase has been identified thought to be utilised
by the tissue-dwelling L3 stage to penetrate host tissue and an
aspartyl proteinase thought to be a digestive enzyme in adults
\cite{polzer_identification_1993}. The twelve proteinases under
positive selection could be the targets of the adaptive immunity
developed against \textit{A. crassus} \cite{knopf_migratory_2008,
  knopf_vaccination_2008}, which is often only elicited against
subtypes of larvae \cite{molnar_caps}.

% The under-representation of ribosomal proteins (term ``structural
% constituent of ribosome'') in positive selected contigs is in good
% agreement with the notion that ribosomal proteins are extremely
% conserved across kingdoms \cite{pmid9664699} and should be under under
% strong negative selection.

Genotyping of individual worms identified a set of \Sexpr{nrow(Rh)}
SNPs with highest credibility and a high information content for
population-genetic studies. Levels of genome-wide heterozygosity found
for the 5 adult worms examined in our study are in agreement with
microsattelite data \cite{wielgoss_population_2008} showing reduced
heterozygosity in European populations of \textit{A. crassus}.

We employed methods to developed for the comparison of cDNA-libraries
to make inferrence about possible differental gene-expression
according to experimental groups (origin of sequencing-libraries)
\cite{pmid9331369}. Such approaches are widely used with
pyrosequencing-data (e.g. \cite{pmid20470405}). For the statistically
valid comparison of conditions however, the unit of replication whould
be the individual library and approaches respecting this fact would be
desirable. However we were not able to use the R-packages DESeq
\cite{pmid20979621} or edgeR \cite{pmid19910308} developed for count
data from deep sequencing (but more targeted towards rna-seq on the
solexa-plattform) as both repetition and throughput of our present
experiment were too low. As a result the differentially expressed
genes are by no means significant for the investigated conditions, but
just for the specific cDNA-libraries. With these reservations we
identified genes differentially expressed between libraries prepared
form worms of different sex and worms from different origin.

Genes over-expressed in male \textit{A. crassus} comprise major sperm
proteins well known for their high expression in nematode sperm
\cite{pmid15275275}. A surprise was the overexpression of ribosomal
proteins in the male library.  

That collagen processing enzymes are overexpressed in female worms,
filled with developing embyos and larvae, is in line with a
complicated regulation and modulation of collagen in nematode larval
development \cite{pmid10637627}.

The overexpression acetyl-CoA acetyltransferase in European woms are
interesting expecially because of the role of these enzymes in
fatty-acid $\beta$-oxidation in peroxisomes and mitochondria
\cite{pmid4721607}. Together with a change in steroid metabolism and
the enrichment of mitochondrially localized enzymes these are
suggestive of changes in energy metabolism of \textit{A. crassus} form
different origins. Possible explanations would include a change to
more or less aerobic processes in worms in Europe due to their bigger
size and/or increased availablilty of nutrients.

Contigs overexpressed in the female libraries showed elevated levels
of dn/ds but genes overexpressed in males decreased levels of
dn/ds. The first finding is unexpected, as overexpressed in female
libraries will also contain contigs related to larval development
(such as the collagen modifying enzymes discussed above), these larval
transcripts in turn are expected to be under purifying selection
because of pleiotropic effects of genes in early development
\cite{pmid15371532}. Also the second finding is in slight contrast to
pubished results for male specific traits and transcripts are often
showning hallmarks of positive selection
\cite{pmid15795858,pmid11404480}. . In \textit{Ancylostoma caninum}
however, female-specific transcipts showed an enrichment fof
``parasitism genes'' \cite{pmid20470405} and a possible expantion
would be a similar enrichment of positively selected parasitism
related in our dataset. For males the decreased dn/ds can be explained
by the by the high number of ribosomal proteins, which are all show
very low levels of dn/ds (that these proteins are found differentially
expressed remaines puzzling though), while single transcripts
e.g. major sperm protein (exressed in the male library only) showed
elevated dn/ds but did not level the overall effect. But this also has
a positve aspect: it is unlikely that correlation of differential
expression with positive selection results from mapping artefacts, as
all the ribosomal proteins identified overexpressed in males have very
low dn/ds.

Genes differentiall expressed according to worm-origin (in either
direction) showed significantly elevated levels of dn/ds. This is
interpretable as a correlation between sequence evolution and
phenotypic modification in different host-environments or even
correation between sequence evolution and evolution of
gene-expression. Thus, whether expression of these genes is modified
in different hosts or evolved rapidly in a contemporary divergence
between European and Asian populations of \textit{A. crassus}, is in
the center of a future research programm building on the reference
transcriptome presented here. For such an annalysis it is important to
disentangle the influence of the host and the nematode population in a
co-inoculation experiment. Such a project will also use the individual
worm as the level of replication for ``conditions'' (that is,
worm-population and host-species) to allow rigid hypothesis
testing. Based on the pilot evalutaion presented here differences in
these factors are expected overlap with differences in male vs. female
worms and the careful cross-examination of the above factors with
worm-sex is adviced.

Population genetic approaches using the SNP-markers presented here
directly or populations genomic approaches choosing to use the SNPS
found here as gold-standard in comparison with higher throughput
technology, constitute another field of future research on
\textit{A. crassus}. The maintenance or loss of variation in European
populations in or close to genes under general positive selection will
be of major interest in such projects.

%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}

The \textit{A. crassus} transcriptome provides a basis of molecular
research on this ecologically important species. It further allows
insight in the evolution of parasitism complementing the catalogue of
available transcriptomic data with a member of the Spirurina
phylogenetically distant to so far sequenced parasites in this clade.
Differences in energy metabolism between European and Asian
\textit{A. crassus} constitute a candidate phenotype relevant for
phenotypic modification or contemporary divergent evolution as well as
for the long term evolution of parasitism.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Competing interests}
The authors declare no competing interests.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}

EGH and MB conceived and designed the experiments. EGH carried out
bioinformatic analyses. SB assisted in bioinformatic analyses. AM
prepared sequencing libraries. HT provided close supervision
throughout. EGH and MB interpreted results and prepared the
manuscript. All authors have read and approved the final manuscript.

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgments}
\ifthenelse{\boolean{publ}}{\small}{}

The work of EGH is funded by Volkswagen Foundation,
"F\"{o}rderinitiative Evolutionsbiologie".
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article} % Style BST file
  \bibliography{/home/ele/bibtex/master}
} % Bibliography file (usually '*.bib' )

%%%%%%%%%%%


\ifthenelse{\boolean{publ}}{\end{multicols}}{}
\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}

\subsection*{Figure 1 - Number of contigs annotated with different
  methods}
 
Number of annotations obtained for Gene Ontology (GO), Enzyme
Commission (EC) and Kyoto Encyclopedia of Genes and Genomes (KEGG)
terms through \texttt{Annot8r} \cite{schmid_annot8r:_2008} for all
TUGs (a) and for higCA derived contigs (b). The latter includes
additional domain-based annotations obtained with
\texttt{InterProScan} \cite{pmid11590104}.

\subsection*{Figure 2 - Comparing high level GO-slim annotations}

For Gene Ontology (GO) categories molecular function, cellular
compartment and biological process the number of terms in high level
GO-slim categories is given as obtained through \texttt{Annot8r}
\cite{schmid_annot8r:_2008}.

\subsection*{Figure 3 - Enrichment of Signal-positives for categories
  of evolutionary conservations}

Proportions of \texttt{SignalP}-predictions for each category of
evolutionary conservation. Generally - across bit-score thresholds -
TUGS novel in nematodes and in \textit{A. crassus} have the highest
proportion of signal-positives.


\subsection*{Figure 4 - Changes in ti/tv and dn/ds due to exclusion of
  homopolymer-runs}

When SNPs in or adjacent to homopolymeric regions are removed changes
in ti/tv and dn/ds are observed: As the overall number of SNPs is
reduced both ratios change to more plausible values. Note the reversed
axis for dn/ds to plot these lower values to the right. For
homopolymer length $>$ 3 a linear trend for the total number of SNPs
and the two measurements is observed. A width of 11 for the screening
window provides most plausible values (suggesting specificity) while
still incorporating a high number of SNPs (sensitivity).

\subsection*{Figure 5 - SNP calling and SNP categories}
 
Overabundance of SNPs at (a) codon-position two and of (c)
non-synonymous SNPs for low percentages of the minority allele. (b)
Significant positive correlation of coverage and dn/ds before removing
these SNPs at a threshold of 7\% ($p<$ 0.001, $R^2=$
\Sexpr{round(cov.lm.raw$adj.r.squared,3)#$}) and (d) afterwards
($R^2<$0.001,
$p=$\Sexpr{round(cov.lm$coefficients[2,"Pr(>|t|)"],3)#$}).


\subsection*{Figure 6 - Positive selection and evolutionary
  conservation}

Box-plots for dn/ds in TUGs according to different categories of
evolutionary conservation. Significant comparisons are
novel.in.metazoa - novel.in.Ac (0.005 and 0.015), novel.in.nematoda -
novel.in.Ac (0.005 and 0.002), novel.in.nematoda - novel.in.clade3
(0.207 and 0.045; p-value for bitscore of 50 and 80,
Nemenyi-Damico-Wolfe-Dunn test).

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\subsection*{Table 1 - Sampling, trimming and pre-assembly screening,
  library statistics}

For libraries two sequencing libraries from European eels (E1 and E2)
one form L2-larvae (L2), one from male (M) and two from Eels in Taiwan
(T1 and T2) the following statistics are given. life.st = lifecycle
stage: f for female m for male. source.p = source population: R for
Rhine, P for Poland, C for cultured, W for wild. raw.reads = raw
number of sequencing reads obtained. lowqal = number of reads
discarded due to low quality or length in \textit{Seqclean}
\cite{tgicl_pertea}. AcrRNA = number of reads hitting
\textit{A. crassus}-rRNA (screened). eelmRNA = number of reads hitting
eel transcriptome-sequences (screened). eelrRNA = number of reads
hitting eel-rRNA genes (screened). Cercozoa = number of reads hitting
cercozoan rRNA (screened). valid = number of reads valid after
screening (assembled). valid.span = number of bases valid (assembled).
mapping.unique = number of reads mapping uniquely to the
assembly. mapping.Ac = number of reads mapping to the part of the
assembly considered \textit{A. crassus} origin (see post-assembly
screening). mapping.MN = number of reads mapping to the highCA-derived
part of the assembly (and also \textit{A. crassus} origin). over.32 =
number of reads mapping to contigs with overall coverage of more than
32 reads (considered in gene-expression analysis).

<<echo=FALSE, results=tex>>=
## print(xtable(TRIM, display=rep("d", times=7)), floating=FALSE)
trim_mark <- transform(TRIM, lowqal = short+lowq+dust+shortq)[1:6, ]
trim_mark <- trim_mark[,c(1,7)]

trim_mark <- merge(pre.screen.tab,
                   trim_mark, by="row.names")

trim_mark$life.st <- c("adult f", "adult f", "L2 lavae",
                       "adult m", "adult f", "adult f")

trim_mark$source.p <- c("Europe R", "Europe P", "Europe R",
                       "Asia C", "Asia C", "Asia W")

trim_mark[7,] <- c("Eel", "", "", "", "", length(eel.clean), sum(nchar(eel.clean)),
                   length(eel.all), length(eel.all)-length(eel.clean), "liver", "Taiwan")

trim_mark <- trim_mark[, c(1, 10, 11,  8, 9, 2, 4, 5, 3, 6, 7 )]
names(trim_mark)[c(1, 4)] <- c("library", "raw.reads")


trim_mark <- merge(trim_mark, colSums(all.counts), by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "mapping.unique"

sum.Ac <- colSums(all.counts[rownames(all.counts)%in%contig.df[contig.df$Ac, "contig"], ])
trim_mark <- merge(trim_mark, sum.Ac, by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "mapping.Ac"

sum.Ac.MN <- colSums(all.counts[rownames(all.counts)%in%contig.df[contig.df$AcMN, "contig"], ])
trim_mark <- merge(trim_mark, sum.Ac.MN, by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "mapping.MN"

sum.o.32 <- colSums(counts[rownames(counts)%in%contig.df[contig.df$AcMN, "contig"], ])
trim_mark <- merge(trim_mark, sum.o.32, by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "over.32"

write.csv(trim_mark, "/home/ele/thesis/454/paper/table1.csv")

trim_mark.tab <- xtable(t(trim_mark))
print(trim_mark.tab, floating=FALSE, include.colnames=FALSE, hline.after = c(0,1, nrow(trim_mark)))
@   

\subsection*{Table 2 - assembly classification and contig statistics}

Summary statistics for contigs from different assembly-categories
given in columns as highCA = high credibility assembly; lowCA = low
credibility assembly, combined = complete assembly.

Rows indicate summary statistics: total.contigs = numbers of total
contigs, fish.contigs = number of contigs hitting eel-mRNA or Chordata
in NCBI-nr or NCBI-nt (screened out), xeno.contigs = number of contigs
with best hit (NCBI-nr and NCBI-nt) to non-eukaryote (screened out),
remaining.contigs = number of contigs remaining after this screening,
remaining.span = total length of remaining contigs, non.u.cov =
non-unique mean base coverage of contigs, cov = unique mean base
coverage of contigs, p4e.``X'' = number protein predictions derived in
p4e, where ``X'' describes the method of prediction (see Methods),
full.3p = number of contigs complete at 3', full.5p = number of
contigs complete at 5', GO = number of contigs with GO-annotation,
KEGG = number of contigs with KEGG-annotation, EC = number of contigs
with EC-annotation, nem.blast = number of contigs with
\texttt{BLAST}-hit to nematode in nr, any.blast = number of contigs
with \texttt{BLAST}-hit to non-nematode (eukaryote non chordate)
sequence in NCBI-nr.


<<echo=FALSE, results=tex>>=
T2 <- as.data.frame.array(tapply(contig.df$contig, contig.df$category%in%"MN", length))
rownames(T2) <- c("lowCA", "highCA")
names(T2) <- "total.contigs"

T2$rRNA.contigs <- table(contig.df$category%in%"MN", contig.df$contamination%in%"AcrRNA" &
                         !(contig.df$phylum.nr%in%c("Chordata") |
                          contig.df$phylum.nt%in%c("Chordata")))[, 2]

T2$fish.contigs <- table(contig.df$category%in%"MN",
                         contig.df$contamination%in%c("eelmRNA", "eelrRNA" ) |
                         contig.df$phylum.nr%in%c("Chordata") |
                         contig.df$phylum.nt%in%c("Chordata"))[, 2]

T2$xeno.contigs <- table(contig.df$category%in%"MN",
                         !(contig.df$contamination%in%c("eelmRNA", "eelrRNA", "AcrRNA") |
                           contig.df$phylum.nr%in%c("Chordata") |
                           contig.df$phylum.nt%in%c("Chordata")) & 
                         !(contig.df$kingdom.nt%in%c("Metazoa", "No_hit") &
                           contig.df$kingdom.nr%in%c("Metazoa", "No_hit")))[,2]
 
T2$remaining.contigs <- table(contig.df$category%in%"MN", contig.df$Ac)[, 2]
T2$remaining.span <- tapply(contig.df$seq,
                            contig.df$category%in%"MN",
                            function (x) sum(nchar(as.character(x))))
T2$non.u.cov <- cov.mn
T2$cov <- cov.mn.u

T2$p4e <- table(c.df$category%in%"MN", gsub("p4e->", "", c.df$method))
T2$full.3p <- table(c.df$category%in%"MN", c.df$full.3p)[,2]
T2$full.5p <- table(c.df$category%in%"MN", c.df$full.5p)[,2]
T2$full.l <- table(c.df$category%in%"MN", c.df$full.length)[,2]

T2$GO <-  table(c.df$category%in%"MN", c.df$contig%in%GO.annot$pept_id)[, 2]
T2$EC <-  table(c.df$category%in%"MN", c.df$contig%in%EC.annot$pept_id)[, 2]
T2$KEGG <-  table(c.df$category%in%"MN", c.df$contig%in%KEGG.annot$pept_id)[, 2]
T2$IPR <-  table(c.df$category%in%"MN", c.df$contig%in%IPR.annot$pept_id)[, 2]

T2$nem.blast <- table(c.df$category%in%"MN", c.df$phylum.nr%in%"Nematoda")[, 2]

T2$any.blast <- table(c.df$category%in%"MN", c.df$phylum.nr%in%"No_hit")[,1]

T2 <- as.data.frame(t(T2))
T2 <- transform(T2, combined=lowCA+highCA)

T2["non.u.cov", "combined"] <- cov.nu.total
T2["cov", "combined"] <- cov.u.total

dig <- t(matrix(c(rep(0,24), rep(c(0, 3 , 3, 3), 2), rep(0,52)), nrow=4))
T2.tab <- xtable(T2, digits=dig)

print(T2.tab)
@ 

\subsection*{Table 3 - Evolutionary conservation and novelty}

The kingdom Metazoa (novel.in.metazoa), the phylum
Nematoda(novel.in.nematoda) and clade III (Spirurina;
novel.in.spirurina) were assessed for occurrences of
\texttt{BLAST}-hits at two different bitscore thresholds (50 = bit.50
and 80 = bit.80). TUGs without any hit at a given threshold were
categorized as novel in \textit{A. crssus} (novel.in.Ac).  Both
novelty and conservation can be derived from this (numbers for
conservation would be the cumulative sum of lower-level novelty).

<<echo=FALSE, results=tex>>=
nov.tab <- rbind(summary.factor(contig.df[contig.df$Ac, "novel.50"]), 
                 summary.factor(contig.df[contig.df$Ac, "novel.80"]))
nov.tab.MN <- rbind(summary.factor(contig.df[contig.df$AcMN, "novel.50"]), 
                 summary.factor(contig.df[contig.df$AcMN, "novel.80"]))
nov.tab <- rbind(nov.tab, nov.tab.MN)
rownames(nov.tab) <- c("bit.50.all", "bit.80.all", "bit.50.highCA", "bit.80.highCA" )
print(xtable(nov.tab), floating=FALSE)

@ 

\subsection*{Table 4 - Over-representation of GO-terms in positively
  selected}

Significantly (p<0.05) over-represented GO-terms in contigs putatively
under positive selection. Horizontal lines separate categories of the
GO-ontology. First category is molecular function, second biological
process, last cellular compartment. P-values (p.value) for
over-representation are given along with the number of positively
selected contigs (Significant; dn/ds $>$ 0.5) and the number of
contigs with this annotation for which a dn/ds was obtained
(Annotated) and the description of the GO-term (Term). For a graph of
incuced GO-terms see also addional figure 1.

<<echo=FALSE, results=tex>>=

GO.over.dn.ds.tab <- xtable(GO.over.dn.ds)

align(GO.over.dn.ds.tab) <- "llp{3cm}llll"

print(GO.over.dn.ds.tab, tabular.environment="longtable",
      floating=FALSE, 
      hline.after=c(0, GO.over.dn.ds.lines), include.rownames=FALSE)

@ 

\subsection{Table 5 - Measurements of multi-locus heterozygosity for
  single worms}

Genotyping for a set of \Sexpr{nrow(Rh)} SNPs, different measurements
were obtained to asses genome-wide heterozygosity. Measurements for
relative heterozygosity (rel.het; number of homozygous sites/ number
of heterozygous sites), internal relatedness (int.rel;
\cite{pmid11571049}), homozygosity by loci (ho.loci;
\cite{pmid17107491}) and standardized heterozygosity (std.het;
\cite{coltman81j}) are given.  All these measurements are pointing to
sample T1 (Taiwanese worm from a wild population) as the most
heterozygous and sample E2 (the European worm from Poland) as the
least heterozygous individual. Heterozygote-heterozygote correlation
\cite{pmid21565077} confirmed the genome-wide significance of these
markers.

<<echo=FALSE, results=tex>>=
print(xtable(het.table))
@ 

\subsection{Table 6 - Over-representation of GO-terms differentially
  expressed}

Significantly (p<0.05) over-represented GO-terms in contigs
differentially expressed between male and female worms (a) or between
European and Asian origin (b). Horizontal lines separate categories of
the GO-ontology. First category is molecular function, second
biological process, last cellular compartment. P-values (p.value) for
over-representation are given along with the number of differentially
expressed contigs (Significant) and the number of contigs with this
annotation analysed (Annotated) and the description of the GO-term
(Term). For a graph of
incuced GO-terms see also addional figures 1.\\

a)\\

<<echo=FALSE, results=tex>>=

GO.over.sex.tab <- xtable(GO.over.sex)

align(GO.over.sex.tab) <- "llp{3cm}llll"

print(GO.over.sex.tab, tabular.environment="longtable",
      floating=FALSE, 
      hline.after=c(0, GO.over.sex.lines), include.rownames=FALSE)

@ 

b)\\

<<echo=FALSE, results=tex>>=

GO.over.eel.tab <- xtable(GO.over.eel)

align(GO.over.eel.tab) <- "llp{3cm}llll"

print(GO.over.eel.tab, tabular.environment="longtable",
      floating=FALSE, 
      hline.after=c(0, GO.over.eel.lines), include.rownames=FALSE)

@ 


<<echo=FALSE, results=tex>>=
print(xtable(het.table))
@ 

\normalsize

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Additional Files}

\subsubsection*{Additional text}
The additional text describes the assembly process and evaluation of
assembly quality in further detail. This text also contains figures
and tables.

\subsubsection*{Additional tables}

Additional table 1 a lists all data computed on the contig level,
including sequences (raw, coding, imputed, protein) additional table 1
b lists only the metadata not including sequences.

Additional table 2 lists high quality SNPs.

Addtional tables 3 list congtigs differentially expressed between male
and female worms (a) and European and Asian worms (b).


\subsubsection*{Additional figures}
Additional figure 1: Subgraphs of the GO-ontology categories induced
by the top 10 terms identified as enriched in differetn sets of
genes. Boxes indicate the 10 most significant terms. Box color
represents the relative significance, ranging from dark red (most
significant) to light yellow (least significant). In each node the
categoy-identifier, a (eventually truncated) description of the term,
the significance for enrichment and the number of DE / total number of
annotated gene is given. Black arrows indicate a is ``is-a''
relationship. GO-ontolgy category and the set of genes analysed for
the enrchment are indicated in each figure.

\end{bmcformat}
\end{document}
