%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
\usepackage{rotating}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}

\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}

% Begin ...
\begin{document}
\begin{bmcformat}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of the invasive eel swimbladder nematode
    parasite \textit{Anguillicola crassus}}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel Heitlinger\correspondingauthor$^{1,2,4}$% 
       \email{Emanuel Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Stephen Bridgett$^{3}$% 
       \email{Stephen Bridgett- sbridget@staffmail.ed.ac.uk}%
       Anna Montazam$^{3}$% 
       \email{Anna Montazam- Anna.Montazam@ed.ac.uk}%
       Horst Taraschewski$^1$% 
       \email{Horst Taraschewski- horst.taraschewski@kit.edu}%
       and Mark Blaxter$^{2,3}$% 
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute, Karlsruhe Institute of Technology,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth Laboratories, The University of Edinburgh, The King's Buildings, Edinburgh EH9 3JT, UK\\
        \iid(3)The GenePool Genomics Facility, The Ashworth Laboratories, The University of Edinburgh, The King's Buildings, Edinburgh EH9 3JT, UK\\
        \iid(4) present address: Department for Molecular Parasitology, Institute for Biology, Humboldt University Berlin, Philippstrasse 13, Haus 14, Berlin, Germany
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} \textit{Anguillicola crassus} is an
  economically and ecologically important parasitic nematode of
  eels. The native range of \textit{A. crassus} is in East Asia, where
  it infects \textit{Anguilla japonica}, the Japanese
  eel. \textit{A. crassus} was introduced into European eels,
  \textit{Anguilla anguilla}, 30 years ago. The parasite is more
  pathogenic in its new host than in its native one, and is thought to
  threaten the endangered \textit{An. anguilla} across its range. The
  molecular bases for the increased pathogenicity of the nematodes in
  their new hosts is not known.
  \paragraph*{Results:} A reference transcriptome was assembled for
  \textit{A. crassus} from Roche 454 pyrosequencing data. Raw reads
  (756,363 total) from nematodes from \textit{An. japonica} and
  \textit{An. anguilla} hosts were filtered for likely host
  contaminants and ribosomal RNAs. The remaining 353,055 reads were
  assembled into 11,372 contigs of a high confidence assembly
  (spanning 6.6 Mb) and an additional 21,153 singletons and contigs of
  a lower confidence assembly (spanning an additional 6.2 Mb). Roughly
  55\% of the high confidence assembly contigs were annotated with
  domain- or protein sequence similarity derived functional
  information. Sequences conserved only in nematodes, or unique to
  \textit{A. crassus} were more likely to have secretory signal
  peptides. Thousands of high quality single nucleotide polymorphisms
  were identified, and coding polymorphism was correlated with
  differential expression between individual nematodes. Transcripts
  identified as being under positive selection were enriched in
  peptidases. Enzymes involved in energy metabolism were enriched in
  the set of genes differentially expressed between European and Asian
  \textit{A. crassus}.
  \paragraph*{Conclusions:}
  The reference transcriptome of \textit{A. crassus} is of high
  quality, and will serve as a basis for future work on the invasion
  biology of this important parasite. The polymorphisms identified
  will provide a key tool set for analysis of population structure and
  identification of genes likely to be involved in increased
  pathogenicity in European eel hosts. The identification of
  peptidases under positive selection is a first step in this
  programme.
\end{abstract}

\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

The nematode \textit{Anguillicola crassus} Kuwahara, Niimi et Itagaki,
1974 is a native parasite of the Japanese eel \textit{Anguilla
  japonica} \cite{kuwahara_Niimi_Itagaki_1974}. Adults localise to the
swim bladder where they feed on blood
\cite{polzer_identification_1993}. Larvae are transmitted via
crustacean intermediate hosts
\cite{de_charleroy_life_1990}. Originally endemic to East Asian
populations of \textit{An. japonica}, \textit{A. crassus} has
attracted interest due to recent anthropogenic expansion of its
geographic and host ranges to Europe and the European eel,
\textit{Anguilla anguilla}. \textit{A. crassus} was recorded for the
first time in Europe in North-West Germany in 1982
\cite{fischer_teichwirt}, where it was most likely introduced through
the live-eel trade \cite{koops_anguillicola-infestations_1989,
  koie_swimbladder_1991}. \textit{A. crassus} has subsequently spread
rapidly through populations of its newly acquired host
\cite{kirk_impact_2003}, and has been found in all
\textit{An. anguilla} populations except those in Iceland
\cite{kristmundsson_parasite_2007}. \textit{A. crassus} can thus be
regarded as a model for the introduction and spread of invasive
parasites \cite{taraschewski_hosts_2007}.

In \textit{An. anguilla}, both prevalence and mean intensity of
infection by \textit{A. crassus} are higher than in \textit{
  An. japonica} \cite{mnderle_occurrence_2006,
  lefebvre_anguillicolosis:_2004}. In \textit{An.  anguilla}
infections, the adult nematodes are larger, have an earlier onset of
reproduction, a greater egg output \cite{knopf_differences_2004} and
induce increased pathology, including thickening and inflammation of
the swim bladder wall \cite{wurtz_tara_2000}. It has been suggested
that the life history modifications and changed virulence observed in
\textit{A. crassus} in the new host are due to an inadequate immune
response in \textit{An. anguilla}
\cite{knopf_swimbladder_2006}. \textit{An. japonica} is capable of
killing histotropic larvae of the parasite after vaccination
\cite{knopf_vaccination_2008} or under high infection pressure
\cite{heitlinger_massive_2009}, but this does not happen in
\textit{A. anguilla}.

The genus \textit{Anguillicola} is placed in the nematode suborder
Spirurina (clade III \textit{sensu} \cite{blaxter_molecular_1998})
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
are exclusively parasitic and include important human pathogens (the
causative agents of filariasis and ascariasis) as well as prominent
veterinary parasites. Molecular phylogenetic analyses place
\textit{Anguillicola} in a clade of spirurine nematodes (Spirurina B
of \cite{dl_py}) that have a freshwater or marine intermediate host,
but infect a wide range of carnivorous definitive hosts. Spirurina B
is sister to the main Spirurina C, including the agents of filariasis
and ascariasis), and thus \textit{A. crassus} may be used as an
outgroup taxon to understand the evolution of parasitic phenotypes in
these species.

The differences in the biology of \textit{A. crassus} in
\textit{An. japonica} (coevolved) and \textit{An. anguilla} (recently
captured) eel hosts is likely to result from differential interactions
between host genetics and parasite genetics. While genetic differences
between the host species are expected, it is not known what part, if
any, genetic differentiation between the invading European and endemic
Asian parasites plays.  European \textit{A. crassus} are less
genetically variable than parasites taken from Asian hosts
\cite{wielgoss_population_2008}, reflecting the derived nature of the
invading populations and the likely population bottlenecks this
entailed. As part of a programme to understand the invasiveness of
\textit{A. crassus} in \textit{An. anguilla}, we are investigating
differences in gene expression and genetic distinction between
invading European and endemic Asian \textit{A. crassus} exposed to the
two host species.

Recent advances in sequencing technology (often termed next generation
sequencing) provide the opportunity for rapid and cost-effective
generation of genome-scale data. The Roche 454 platform
\cite{pmid16056220} is particularly suited to transcriptomics of
previously unstudied species \cite{pmid20950480}. Here we describe the
generation of a reference transcriptome for \textit{A. crassus} based
on Roche 454 data, and explore patterns of gene expression and
diversity within the nematode.

%%%%%%%%%%%%%%%%%%
\section*{Methods}


\subsection*{Nematode samples, RNA extraction, cDNA synthesis and Sequencing}

\textit{A. crassus} from \textit{An. japonica} were sampled from
Kao-Ping river and an adjacent aquaculture in Taiwan as described in
\cite{heitlinger_massive_2009}. Nematodes from \textit{An. anguilla}
were sampled from Sniardwy Lake, Poland and from the Linkenheimer
Altrhein, Germany. After determination of the sex of adult nematodes,
they were stored in RNA-later (Quiagen, Hilden, Germany) until
extraction of RNA. RNA was extracted from individual adult male and
female nematodes and from a population of second stage larvae (L2)
(Table 1). For host contamination screening a liver sample from an
uninfected \textit{An. japonica} was also processed. RNA was reverse
transcribed and amplified into cDNA using the MINT-cDNA synthesis kit
(Evrogen, Moscow, Russia). Emulsion PCR and library preparation were
performed for each cDNA library according to the manufacturer's
protocols (Roche/454 Life Sciences), and sequenced on a Roche 454
Genome Sequencer FLX.

Raw sequencing reads are archived under study-accession number
SRP010313 in the NCBI Sequence Read Archive (SRA;
http://www.ncbi.nlm.nih.gov/Traces/sra) \cite{pmid22140104}. All
samples were sequenced using the FLX Titanium chemistry, except for
the Taiwanese female sample T1, which was sequenced using FLX standard
chemistry, to generate between 99,000 and 209,000 raw reads per
sample. For the L2 library, which had a larger number of
non-\textit{A. crassus}, non-\textit{Anguilla} reads, we confirmed
that these data were not laboratory contaminants by screening Roche
454 data produced on the same run in independent sequencing lanes.

\subsection*{Trimming, quality control and assembly}

Raw sequences were extracted in FASTA format (with the corresponding
qualities files) using sffinfo (Roche/454) and screened for MINT
adapter sequences using cross-match \cite{PHRAP} (with parameters
-minscore 20 -minmatch 10). Seqclean \cite{tgicl_pertea} was used to
identify and remove poly-A-tails, low quality, low complexity and
short (<100 base) sequences. All reads were compared to a set of
screening databases using BLAST \cite{pmid2231712} (expect value
cutoff E<1e-5, low complexity filtering turned off: -F F). The
databases used were (a) a host sequence database comprising an
assembly of the \textit{An. japonica} Roche 454 data, a unpublished
assembly of \textit{An. anguilla} Sanger dideoxy sequenced expressed
sequence tags (made available to us by Gordon Cramb, University of St
Andrews) and transcripts from EeelBase \cite{pmid21080939}, a publicly
available transcriptome database for the European eel; (b) a database
of ribosomal RNA (rRNA) sequences from eel species derived from our
Roche 454 data and EMBL-Bank; and (c) a database of rRNA sequences
identified in our \textit{A. crassus} data by comparing the reads to
known nematode rRNAs from EMBL-Bank. This last database notably also
contained cobiont rRNA sequences. Reads with matches to one of these
databases over more than 80\% of their length and with greater than
95\% identity were removed from the dataset. Screening and trimming
information was written back into sff-format using sfffile (Roche
454). The filtered and trimmed data were assembled using the combined
assembly approach \cite{pmid20950480}: Two assemblies were generated,
one using Newbler v2.6 \cite{pmid16056220} (with parameters -cdna
-urt), the other using Mira v3.2.1 \cite{miraEST} (with parameters
--job=denovo,est,accurate,454). The resulting two assemblies were
combined into one using Cap3 \cite{Cap3_Huang} at default settings and
contigs were labeled by whether they derived from both assemblies
(high confidence assembly; highCA), or one assembly only (lowCA; for a
detailed analysis of the assembly categories see the supporting
Methods file). The superset of highCA contigs, lowCA contigs and the
remaining unassembled reads defines the set of tentatively unique
genes (TUGs).

\subsection*{Post-assembly classification and taxonomic assignment of
  contigs}

We rescreened the assembly for host and other contamination by
comparing it (using BLAST) to the three databases defined above, and
also to NEMBASE4, a nematode transcriptome database derived from whole
genome sequencing and EST assemblies
\cite{parkinson_nembase:resource_2004, pmid21550347}. For each contig,
the highest-scoring match was recorded, if it spanned more than 50\%
of the contig. We also compared the contigs to the NCBI non-redundant
nucleotide (NCBI-nt) and protein (NCBI-nr) databases, recording the
taxonomy of all best matches with expect values better than
1e-05. Sequences with a best hit to non-Metazoans or to Chordata
within Metazoa were excluded from further analysis.

%% nrow(contig.df[!contig.df$Ac &
%% contig.df$contamination%in%"eelmRNA" & contig.df$phylum.nr%in%"Nematoda", ])
%% investigate if these (149) should not be included rather as
%% Ac-origin DONE!!! The eelmRNA hits are _all_ much better!

\subsection*{Protein prediction and annotation}

Protein translations were predicted from the contigs using prot4EST
(version 3.0b) \cite{wasmuth_prot4est:_2004}. Proteins were predicted
either by joining single high scoring segment pairs (HSPs) from a
BLAST search of uniref100 \cite{pmid18836194}, or by ESTscan
\cite{estscan}, using as training data the \textit{Brugia malayi}
complete proteome \cite{ghedin_draft_2007} back-translated using a
codon usage table derived from the BLAST HSPs, or, if the first two
methods failed, simply the longest ORF in the contig. For contigs
where the protein prediction required insertion or deletion of bases
in the original sequence, we also imputed an edited sequence for each
affected contig. Annotations with Gene Ontology (GO), Enzyme
Commission (EC) and Kyoto Encyclopedia of Genes and Genomes (KEGG)
terms were inferred for these proteins using annot8r (version 1.1.1)
\cite{schmid_annot8r:_2008}, using the annotated sequences available
in uniref100 \cite{pmid18836194}. Up to 10 annotations based on a
BLAST similarity bitscore cut-off of 55 were obtained for each
annotation set. The complete \textit{B. malayi} proteome (as present
in uniref100) and the complete \textit{C. elegans} proteome (as
present in WormBase v.220) were also annotated in the same
way. SignalP V4.0 \cite{pmid21959131} was used to predict signal
peptide cleavage sites and signal anchor signatures for the
\textit{A. crassus} transcriptome and for the proteomes of the two
model nematodes. InterProScan \cite{pmid11590104} (command line
utility iprscan version 4.6 with options -cli -format raw -iprlookup
-seqtype p -goterms) was used to obtain domain annotations for the
highCA contigs. We recorded the presence of a lethal RNAi phenotype in
the \textit{C. elegans} ortholog of each TUG using the biomart
-interface \cite{pmid22083790} to WormBase v. 220 using the R package
biomaRt \cite{pmid19617889}.

\subsection*{Single nucleotide polymorphism analysis}

We mapped the raw reads to the complete set of contigs, replacing
imputed sequences for originals where relevant, using ssaha2 (with
parameters -kmer 13 -skip 3 -seeds 6 -score 100 -cmatch 10 -ckmer 6
-output sam -best 1) \cite{pmid11591649}. From the ssaha2 output,
pileup files were produced using samtools
\cite{journals/bioinformatics/LiHWFRHMAD09}, discarding reads mapping
to multiple regions. VarScan \cite{pmid19542151} (pileup2snp) was used
with default parameters on pileup files to output lists of single
nucleotide polymorphisms (SNPs) and their locations.


In the 10,496 SNPs thus defined, the ratio of transitions (ti; 6,908)
to transversion (tv; 3,588) was 1.93. From the prot4EST predictions,
7,189 of the SNPs were predicted to be inside an ORF, with 2,322 at
codon first positions, 1,832 at second positions and 3,035 at third
positions. As expected, ti/tv inside ORFs (2.39) was higher than
outside ORFs (1.25). The ratio of synonymous polymorphisms per
synonymous site to non-synonymous polymorphisms per non-synonymous
site in this unfiltered SNP set (dn/ds) was 0.45, rather high compared
to other analyses. Roche 454 sequences have well-known systematic
errors associated with homopolymeric nucleotide sequences [58], and
the effect of exclusion of SNPs in, or close to, homopolymer regions
was explored. When SNPs were discarded using different size thresholds
for homopolymer runs and proximity thresholds, the ti/tv and in dn/ds
ratios changed (Additional Figure 1). Based on this SNPs associated
with a homopolymer run longer than 3 bases within a window of 11 bases
(5 bases to the right, 5 to the left) around the SNP were
discarded. There was a relationship between TUG dn/ds and TUG
coverage, associated with the presence of sites with low abundance
minority alleles (less than 7\% of the allele calls), suggesting that
some of these may be errors. Removing low abundance minority allele
SNPs from the set removed this effect (Additional Figure 2). For
enrichment analysis of GO terms associated with positively selected
TUGs we used the R package GOstats \cite{pmid17098774}.

Using Samtools \cite{journals/bioinformatics/LiHWFRHMAD09} (mpileup
-u) and Vcftools \cite{pmid21653522} (view -gcv) we genotyped
individual libraries for each of the master list of SNPs. Genotype-
calls were accepted at a phred-scaled genotype quality threshold of
10. In addition to the relative heterozygosity (number of homozygous
sites/number of heterozygous sites) we used the R package Rhh
\cite{pmid21565077} to calculate internal relatedness
\cite{pmid11571049}, homozygosity by locus \cite{pmid17107491} and
standardised heterozygosity \cite{coltman81j} from these data. We
confirmed the significance of heterozygote-heterozygote correlation by
analysing the mean and 95\% confidence intervals from 1000 bootstrap
replicates estimated for all measurements.

\subsection*{Gene expression analysis}

Read-counts were obtained from the bam files generated for genotyping
using the R package Rsamtoools \cite{rsamtools}. LowCA contigs and
contigs with fewer than 32 reads over all libraries were excluded from
analysis. Libraries E1 and L2 had very low overall counts and thus we
excluded these libraries from analysis. The statistic of Audic and
Claverie \cite{pmid9331369} as implemented in ideg6
\cite{pmid12429865} was used to contrast single
libraries. Differential expression between libraries from male versus
female nematodes was accepted for genes that differed in expression
values between all the female libraries (E2, T1 and T2; see Table 1)
versus the male (M) library (p <0.01), but had no differential
expression within any of the female libraries at the same
threshold. Differential expression between libraries from nematodes of
European \textit{An. anguilla} and Taiwanese \textit{An. japonica}
origin was accepted for genes that differed in expression values
between library E2 and both libraries T1 and T2 (p <0.01), but showed
no differences between T1 and T2.

\subsection*{Overrepresentation analyses}

The R package annotationDbi \cite{AnnotationDbi} was used to obtain a
full list of associations (along with higher-level terms) from annot8r
annotations prior to analysis of GO term overrepresentation in gene
sets selected on the basis of dn/ds or expression values. The R
package topGO \cite{topGO} was used to traverse the annotation graph
and analyse each node term for overrepresentation in the focal gene
set compared to an appropriate universal gene set (all contigs with
dn/ds values or all contigs analysed for gene expression) with the
``classic'' method and Fisher's exact test. Terms for which an
offspring term was already in the table and no additional counts
supported overrepresentation were removed. Mann-Whitney u-tests were
used to test the influence of factors on dn/ds values. To investigate
multiple contrasts between groups (factors) Nemenyi-Damico-Wolfe-Dunn
tests were used, and for overrepresentation of one group (factor) in
other groups (factors) Fisher's exact test was used.

\subsection*{General coding methods}

The bulk of analysis (unless otherwise described) presented was
carried out in R \cite{R_project} using custom scripts. For
visualisation we used the R packages ggplot2 \cite{ggplot-book} and
VennDiagram \cite{pmid21269502}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}

<<load.libs, echo=FALSE, results=hide>>=
library(VennDiagram)
library(xtable)
library(reshape)
library(ggplot2)
library(plyr)

library(AnnotationDbi)
library(topGO)

library(limma)
library(XML)
library(coin)
library(multcomp)

library(DESeq)
library(edgeR)
library(ShortRead)
library(biomaRt)

## library(BioIDMapper)

library(Rhh)
library(Heatplus)

source("/home/ele/thesis/454/common_R_functions.R")
@ 

\subsection*{Sampling \textit{A. crassus}}


One female \textit{A. crassus} and one male \textit{A. crassus} were
sampled from an \textit{An. japonica} aquaculture with high infection
loads in Taiwan, and an additional female was sampled from an
\textit{An. japonica} caught in a stream with low infection pressure
adjacent to the aquaculture. A female nematode and pool of L2 were
sampled from \textit{An. anguilla} in the river Rhine, and one female
from \textit{A. anguilla} from a lake in Poland. All adult nematodes
were replete with host blood. To assist in downstream filtering of
host from nematode reads, we also sampled RNA from the liver of an
uninfected Taiwanese \textit{An. japonica}.


<<trim, echo=FALSE, results=hide, cache=TRUE>>=
trimdat <- readLines("/home/ele/Data/454/trimmed/trimming.stats")
trimfile <-  grep("Trimming sff file *", trimdat, value=TRUE)
trimfile <- sapply(strsplit(trimfile, " |,"), function (x) basename(x[4]))
lib <- gsub(".sff'", "", trimfile)

raw_reads<- grep("Sequences analyzed:.*", trimdat, value=TRUE)
raw_reads <- as.numeric(gsub("Sequences analyzed: *", "", raw_reads))
short <- grep("by 'short':.*", trimdat, value=TRUE)
short <- as.numeric(gsub(" *by 'short': *", "", short))
lowq <- grep("by 'low_qual':.*", trimdat, value=TRUE)
lowq <- as.numeric(gsub(" *by 'low_qual': *", "", lowq))
dust <- grep("by 'dust':.*", trimdat, value=TRUE)
dust <- as.numeric(gsub(" *by 'dust': *", "", dust))
shortq <- grep("by 'shortq':.*", trimdat, value=TRUE)
shortq <- as.numeric(gsub(" *by 'shortq': *", "", shortq))
valid <- raw_reads-(short+lowq+dust+shortq)

TRIM <- data.frame(row.names=lib, raw_reads, short, lowq, dust, shortq, valid)
TRIM["total",] <- colSums(TRIM)

## subsitute the publication style library abrevations
rownames(TRIM) <- gsub("179F", "T1", rownames(TRIM))
rownames(TRIM) <- gsub("10F", "T2", rownames(TRIM))
rownames(TRIM) <- gsub("KS4F", "E1", rownames(TRIM))
rownames(TRIM) <- gsub("UW07F", "E2", rownames(TRIM))
rownames(TRIM) <- gsub("L2R3", "L2", rownames(TRIM))
rownames(TRIM) <- gsub("M175", "M", rownames(TRIM))

@ 

<<read.libs, echo=FALSE, results=hide, cache=TRUE>>=

RE <- list()
for (l in (c("10F", "179F", "M175", "L2R3", "KS4F", "UW07F"))){
   command <- paste("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/trimmed/",
                    l, ".sff.trimmed.sff;", sep="")
   connect <- pipe(command)
   RE[l] <- list(read.sequences(connect))
   closeAllConnections()
}
names <- gsub(" .*$", "", unlist(lapply(RE, names)))
lib <- melt(RE)

## subsitute the publication style library abrevations
lib$L1 <- gsub("179F", "T1", lib$L1)
lib$L1 <- gsub("10F", "T2", lib$L1)
lib$L1 <- gsub("KS4F", "E1", lib$L1)
lib$L1 <- gsub("UW07F", "E2", lib$L1)
lib$L1 <- gsub("L2R3", "L2", lib$L1)
lib$L1 <- gsub("M175", "M", lib$L1)

RE <- cbind(read=names, lib)
names(RE)[2:3] <- c("seq", "lib")
RE$seq <- as.character(RE$seq)
RE$length <- nchar(RE$seq)

eel.all <- read.sequences(pipe("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/Li68_assembly/sff/Li68.sff"))

eel.clean <- read.sequences(pipe("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/Li68_assembly/sff/Li68.sff.trimmed.sff"))

eel.rRNA.contigs <- read.sequences("/home/ele/Data/454/Li68_assembly/Aj_liver_rRNA.fasta")

eel.contigs <- read.sequences("/home/ele/Data/454/Li68_assembly/Aj_liver.fasta")

@ 

\SweaveInput{../contamination/pre_assembly.Rnw}

\subsection*{Assembly and post-assembly screening}

<<load.meth, echo=FALSE>>=
load("../Supporting_information/Assembly.Rdata")

nMN <- nrow(contig.df[contig.df$category%in%"MN",])

nbad.c <- nrow(contig.df[contig.df$category%in%
                         c("M_n","N_n","N_1","M_1"),])

nSing <- nrow(contig.df[contig.df$category%in%0,])
@

A total of 756,363 raw sequencing reads were generated for
\textit{A. crassus} (Table 1). These were rigorously filtered (see
supporting infromation) and 353,055 remaining reads (spanning
100,491,819 bases) were assembled using the combined assembler
strategy \cite{pmid20950480}, employing Roche 454 gsAssembler (also
known as Newbler; version 2.6) and MIRA (version 3.21)
\cite{miraEST}. This coassembly will be included in future versions of
nembase (nembase5) and is available at
http://afterparty.bio.ed.ac.uk/study/show/1440745. It comprised
\Sexpr{cn(nMN)} contigs supported by both assembly algorithms,
\Sexpr{cn(nbad.c)} contigs supported by only one of the assembly
algorithms and \Sexpr{cn(nSing)} singletons that not assembled by
either program (Table 2). Contigs supported by both assemblers were
longer than those supported by only one assembler, and were more
likely to have a significant similarity to previously sequenced
protein coding genes than contigs assembled by only one of the
algorithms, or the remaining unassembled singletons. These constitute
the highCA, while those with evidence from only one assembler and the
singletons are the lowCA. These datasets were the most parsimonious
(having the smallest size) for their quality (covering the largest
amount of sequence in reference transcriptomes). In the highCA
parsimony and low redundancy was prioritised, while in the complete
assembly (highCA plus lowCA including singletons) completeness was
prioritised. The 40,187 sequences (contig consensuses and singletons)
in the complete assembly are referred to as tentatively unique genes
(TUGs).

<<mean.cov, echo=FALSE, cache=TRUE>>=

cov.mn <- tapply(con.pile$coverage,
                 con.pile$contig%in%contig.df[contig.df$category=="MN", "contig"],
                 mean)
                            
cov.mn.u <- tapply(con.pile.uniq$uniq_coverage,
                   con.pile.uniq$contig%in%contig.df[contig.df$category=="MN", "contig"],
                   mean)

cov.nu.total <- mean(con.pile$coverage)
cov.u.total <- mean(con.pile.uniq$uniq_coverage)

@ 

\SweaveInput{../contamination/post_assembly.Rnw}

We screened the complete assembly for remaining host contamination,
and identified \Sexpr{cn(n.fish)} TUGs that had significant, higher
similarity to eel (and/or chordate; EMBLBank Chordata proteins) than
to nematode sequences \cite{pmid21550347}. Given the identification of
cercozoan ribosomal RNAs in the L2 library, we also screened the
complete assembly for contamination with transcripts from other taxa.

1,153 TUGs were found with highest significant similarity to Eukaryota
outside of the kingdoms Metazoa, Fungi and Viridiplantae. These
contigs matched genes from a wide range of protists from Apicomplexa
(mainly Sarcocystidae, 28 hits and Cryptosporidiidae 10 hits),
Bacillariophyta (diatoms, mainly Phaeodactylaceae, 41 hits),
Phaeophyceae (brown algae, mainly Ectocarpaceae, 180 hits),
Stramenopiles (Albuginaceae, 63 hits), Kinetoplasitda
(Trypanosomatidae, 26 hits) and Heterolobosea (Vahlkampfidae, 38
hits). Additionally \Sexpr{cn(other.king[["Fungi"]])} TUGs had best,
significant matches to genes from fungi (e.g Ajellomycetaceae, 53
hits) and 585 TUGs had best, significant matches to genes from
plants. Outside the Eukaryota there were significant best matches to
Bacteria (825 TUGs; mostly to members of the Proteobacteria), Archaea
(8 TUGs) and viruses (9 TUGs). No TUGs had significant, best matches
to \textit{Wolbachia} or related Bacteria known as symbionts of
nematodes and arthropods. All TUGs with highest similarity to
sequences deriving from taxa outside Metazoa were excluded. The final,
screened \textit{A. crassus} assembly has
\Sexpr{cn(nrow(contig.df[contig.df$Ac,]))#$} TUGs, spanning
\Sexpr{cn(sum(nchar(as.character(contig.df[contig.df$Ac,"seq"]))))#$}
bases (of which \Sexpr{cn(nrow(contig.df[contig.df$AcMN,]))#$} are
highCA-derived, and span
\Sexpr{cn(sum(nchar(as.character(contig.df[contig.df$AcMN,"seq"]))))#$}
bases). All analyses reported below are based on this filtered
dataset.

\SweaveInput{../conservation/conservation.Rnw}
\SweaveInput{../protein_prediction/imputed_from_p4EST.Rnw}
\SweaveInput{../snp_analysis/snp_analysis.Rnw}
\SweaveInput{../snp_analysis/snp_by_population.Rnw}
\SweaveInput{../annotation/annotation.Rnw}
\SweaveInput{../annotation/read_signalp.Rnw}
\SweaveInput{../annotation/brugia-compare.Rnw}
\SweaveInput{../snp_analysis/overrepresentation_snp.Rnw}

\subsection*{Annotation}

For
\Sexpr{cn(nrow(contig.df[contig.df$Ac&contig.df$method!="no-prediction",]))}
of the \textit{A. crassus} TUGs a protein was predicted using prot4EST
\cite{wasmuth_prot4est:_2004} (Table 2). An apparently full-length
open reading frame (ORF) was identified in
\Sexpr{cn(nrow(contig.df[contig.df$Ac&imputed$full.length,]))} TUGs,
while for \Sexpr{cn(nrow(contig.df[contig.df$Ac&!imputed$full.5p,]))}
the 5' ends and for
\Sexpr{cn(nrow(contig.df[contig.df$Ac&!imputed$full.3p,]))} the 3'
ends were missing. In \Sexpr{cn(same.same[["FALSE"]])} TUGs the
corrected sequence with the imputed ORF was slightly changed compared
to the raw sequence by insertions or deletions necessary to obtain a
continuous reading frame. Using BLAST we determined that
\Sexpr{cn(nrow(contig.df[contig.df$Ac&!is.na(contig.df$Ce.hit),]))}
had significant similarity to \textit{C. elegans} proteins,
\Sexpr{cn(nrow(contig.df[contig.df$Ac&!is.na(contig.df$Bm.hit),]))}
TUGs matched \textit{B. malayi} proteins, and
\Sexpr{cn(nrow(contig.df[contig.df$Ac&!is.na(contig.df$nempep.hit),]))}
TUGs had matches in NEMPEP4 \cite{parkinson_nembase:resource_2004,
  pmid21550347}. Comparison to the UniProt reference identified
\Sexpr{cn(nrow(contig.df[contig.df$Ac&!is.na(contig.df$nr.bit),]))}
TUGs with significant similarities. We used annot8r
\cite{schmid_annot8r:_2008} to assign GO terms to \Sexpr{cn(nGO)}
TUGs, EC numbers for \Sexpr{cn(nEC)} TUGs and KEGG pathway annotations
for \Sexpr{cn(nKEGG)} TUGs (Table 2). Additionally \Sexpr{cn(nIPR)}
highCA contigs were annotated with GO terms through InterProScan
\cite{pmid11590104}. Nearly one third (\Sexpr{cn(nallannotations)}) of
the \textit{A. crassus} TUGs were annotated with at least one
identifier, and \Sexpr{cn(all.identifier)} had GO, EC and KEGG
annotations (Figure 1).

We compared our \textit{A. crassus} GO annotations for high-level
GO-slim terms to the annotations (obtained in the same way) for the
complete proteome of the spirurid filarial nematode \textit{B. malayi}
and the complete proteome of \textit{C. elegans} (Figure 2). The
occurrence of GO terms in the annotation of the partial transcriptome
of \textit{A. crassus} was more similar to that of the proteome of
\textit{B. malayi} (0.95; Spearman correlation coefficient) than to
the that of the proteome of \textit{C. elegans} (0.9).

Despite the lack of completeness at the 5' end suggested by peptide
prediction, just over 3\% of the TUGs were predicted to be secreted
(\Sexpr{cn(nsigP)} with signal peptide cleavage sites and
\Sexpr{cn(nsigPtm)} signal peptides with a transmembrane
signature). Again these predictions are more similar to predictions
using the same methods for the proteome of \textit{B. malayi} (742
signal peptide cleavage sites and 41 with transmembrane anchor) than
for the proteome of \textit{C. elegans} (4,273 signal peptide cleavage
sites and 154 with transmembrane anchor).

By comparison to RNAi phenotypes for \textit{C. elegans} genes
\cite{pmid12529635, pmid19910365} likely to be orthologous to
\textit{A. crassus} TUGs, 6,029 TUGs were inferred to be essential
(RNAi lethal phenotype in \textit{C. elegans}).

To explore the phylogenetic conservation of \textit{A. crassus} TUGs,
they were classified as conserved across kingdoms, conserved in
Metazoa, conserved in Nematoda, conserved in Spirurina or novel to
\textit{A. crassus} by comparing them to custom database subsets using
BLAST (Table 3). Using a relatively strict cutoff, a quarter of the
highCA contigs were conserved across kingdoms, and ~10\% were
apparently restricted to Nematoda. Nearly half of the highCA contigs
were novel to \textit{A. crassus}.

Similar patterns were observed for conservation assessed at different
stringency, and when assessed across all TUGs, except that a higher
proportion of all TUGs were apparently unique to \textit{A. crassus}.

Proteins predicted to be restricted to Nematoda and novel in
\textit{A. crassus} were significantly enriched in signal peptide
annotation compared to conserved proteins, proteins novel in Metazoa
and novel in Spirurina (Fisher's exact test p<0.001 ; Figure 3). The
proportion of lethal RNAi phenotypes was significantly higher for
\textit{C. elegans} presumed orthologs of TUGs conserved across
kingdoms (97.23\%) than for orthologs of TUGs not conserved across
kingdoms (94.59\%; p<0.001, Fisher's exact test).

\subsection*{Identification and analysis of single nucleotide
  polymorphisms}

Single nucleotide polymorphisms (SNPs) were called using VARScan
\cite{pmid19542151} on the \Sexpr{cn(nCov8bases)} bases of TUGs that
had coverage of more than 8-fold available. SNPs predicted to have
more than 2 alleles, or that mapped to an undetermined (N) base were
excluded, as were SNP likely to be due to base calling errors close to
homopolymer tracts and SNP calls resulting from apparent rare
variants. 

Our filtered SNP dataset includes 5,113 SNPs, with 4.65 SNPs per kb of
contig sequence. There were 7.95 synonymous SNPs per 1000 synonymous
bases and 2.44 non-synonymous SNPs per 1000 non synonymous bases. A
mean dn/ds of 0.244 was calculated for the 765 TUGs (all highCA
contigs) containing at least one synonymous SNP. Positive selection
can be inferred from high dn/ds ratios. Overrepresented GO ontology
terms associated with TUGs with dn/ds higher than 0.5 were identified
(Table 4; Additional Figure 10 a, b, c). Within the molecular function
category, ``peptidase activity'' was the most significantly
overrepresented term. Twelve of the thirteen high dn/ds TUGs annotated
as peptidases each had unique orthologs in \textit{C. elegans} and
\textit{B. malayi}. Other overrepresented categories identified
subunits of the respiratory chain: ``heme-copper terminal oxidase
activity'' and ``cytochrome-c oxidase activity'' in molecular function
and ``mitochondrion'' in cellular compartment. Contigs identified as
novel to Spirurina and novel in \textit{A. crassus} had a
significantly higher dn/ds than other contigs (Additional Figure 3).

Signal peptide containing proteins have been shown to have higher
rates of evolution than cytosolic proteins in a number of nematode
species. \textit{A. crassus} TUGs predicted to contain signal peptide
cleavage sites showed a non-significant trend towards higher dn/ds
values than TUGs without signal peptide cleavage sites (p =
\Sexpr{round(u.test[["p.value"]],3)}; two sided
Mann-Whitney-test). Orthologs of \textit{C. elegans} transcripts with
lethal RNAi phenotype are expected to evolve under stronger selective
constraints and the values of dn/ds showed a non-significant trend
towards lower values in TUGs with orthologs with a lethal phenotype
compared to a non-lethal phenotypes
(p=\Sexpr{round(u.rnai$p.value,3)#$}, two-sided U-test).\\

The genotypes of single adult nematodes were called using Samtools
\cite{journals/bioinformatics/LiHWFRHMAD09} and Vcftools
\cite{pmid21653522}, and \Sexpr{cn(nrow(Rh))} informative sites (where
two alleles were found in at least one assured genotype at least in
one of the nematodes) were identified in \Sexpr{cn(geno.contigs)}
contigs. Internal relatedness \cite{pmid11571049}, homozygosity by
loci \cite{pmid17107491} and standardised heterozygosity
\cite{coltman81j} all identified the Taiwanese nematode from
aquaculture (sample T1) as the most heterozygous and the European
nematode from Poland (sample E2) as the least heterozygous
individuals.

The genome-wide representativeness of these \Sexpr{cn(nrow(Rh))} SNP
markers for the whole genome in population genetic studies was
confirmed using heterozygosity-heterozygosity correlation
\cite{pmid21565077}: mean internal relatedness = 0.78, lower bound of
95\% confidence intervals from 1000 bootstrap replicates (cil) =
0.444; mean homozygosity by loci = 0.86, cil = 0.596; standardised
heterozygosity = 0.87, cil= 0.632.

 \subsection*{Differential gene expression}
                
\SweaveInput{../gene_expression/gene_expression.Rnw}
\SweaveInput{../gene_expression/overrepresentation_exp.Rnw}

Gene expression was inferred by the unique mapping of 252,388
(71.49\%) of the raw reads to the fullest assembly (including the all
assembled contigs as a "filter"; total contigs/all TUGs in Table
2). In analysis, non-\textit{A. crassus} contigs, and all contigs with
fewer than 32 reads overall were excluded. Thus
\Sexpr{cn(nrow(ideg6))} TUGs were analysed for differential expression
using ideg6 for normalisation and the statistic of Audic and Claverie
\cite{pmid9331369} for detection of differences. Of these TUGs, 54
showed expression predominantly in the male library, 56 TUGs were more
highly represented in the female library, 56 TUGs were primarily
expressed in the libraries from Taiwan, and 22 TUGs were
overrepresented in the European library.

Analysis of overrepresentation of of GO terms associated with TUGs
differentially expressed between male and female libraries identified
ribosomal proteins, oxidoreductases and collagen processing enzyme
terms (Table 6; Additional Figure 10 g, h, i). The ribosomal proteins were
all overexpressed in the male library, while the oxidoreductases and
collagen processing enzymes were overexpressed in female
libraries. Similar analysis of overrepresentation of of GO terms
associated with the TUGs differentially expressed between European
nematodes and Asian nematodes identified several terms of catalytic
activity related to metabolism (Table 7; Additional Figure 10
d, e, f). TUGs annotated as acyltransferase were upregulated in the
European libraries. However, the expression patterns for other TUGs
with overrepresented terms connected to metabolism did not show
concerted up or down-regulation. Thus for the term ``steroid
biosynthetic process'', 2 TUGs were downregulated and 3 contigs
upregulated in European nematodes. No enrichment of of signal peptide
positive TUGs, of TUG conservation categories, or TUGs with
\textit{C. elegans} orthologs with lethal or non-lethal
RNAi-phenotypes was identified. Significantly elevated dn/ds was found
for TUGs differentially expressed in European versus Asian nematodes
(Fisher's exact test p=0.007; also both up- or down-regulated were
significant). TUGs overexpressed in the female libraries showed
elevated levels of dn/ds (Fisher's exact test p=0.041), but contrast
male overexpressed genes showed decreased levels of dn/ds (Fisher's
exact test p=0.014).

<<save, echo=FALSE, results=hide>>=
## write the MN.peptide sequence for iprscan prediction
MN.pep.fasta <- as.character(contig.df[contig.df$AcMN, "Ac.pep"])
names(MN.pep.fasta) <- as.character(contig.df[contig.df$AcMN, "contig"])
write.sequence(MN.pep.fasta, "/home/ele/thesis/454/MN.pep.fasta")

## for a more readable version of contig.df
readable <- !names(contig.df)%in%c("seq", "imp", "ontology", "coding", "Ac.pep")
needed <- !names(contig.df)%in%c("ontology", "cov.ontology")

write.csv(contig.df[,needed], "../paper/A.crassus_transcriptome_contig_data.csv",
row.names=FALSE)

write.csv(VARqp[, c("contig", "base", "from", "to",
"nfrom", "nto", "perc", "inORF",
"inFRAME", "effect")], "../paper/A.crassus_transcriptome_screened_SNPs.csv",
row.names=FALSE)

write.single.genes.rev <- function(genes, folder) {
  sapply(genes,
         function (x) {
           s <- fullest.assembly[x]
           if (as.logical(as.character(contig.df[contig.df$contig%in%x, "mstrand"]))){
             s <- revcom(fullest.assembly[x])}
           write.sequence(s, paste(folder, x, ".fasta", sep=""))
         })
}

write.single.genes.rev(peptidases, "../snp_analysis/peptidases/")

names(express.sex.sig) <- c("E2", "T1", "T2", "M", "LogFC")
express.sex.sig <- merge(express.sex.sig,
                         contig.df[, c("contig", "nempep.annot")], by.x=0, by.y="contig")
express.sex.sig <- express.sex.sig[order(express.sex.sig$LogFC), ]
write.csv(express.sex.sig, "../paper/A.crassus_transcriptome_sexDE.csv",
row.names=TRUE)

top10.sex <- express.sex.sig[c(1:5, (nrow(express.sex.sig)-4):nrow(express.sex.sig)),
                         "Row.names"]

write.single.genes.rev(top10.sex, "../gene_expression/top10_sex/")

names(express.eel.sig) <- c("E2", "T1", "T2", "LogFC")
express.eel.sig <- merge(express.eel.sig,
                         contig.df[, c("contig", "nempep.annot")], by.x=0, by.y="contig")
express.eel.sig <- express.eel.sig[order(express.eel.sig$LogFC), ]
write.csv(express.eel.sig, "../paper/A.crassus_transcriptome_originDE.csv",
row.names=TRUE)

top10.eel <- express.eel.sig[c(1:5, (nrow(express.eel.sig)-4):nrow(express.eel.sig)),
                         "Row.names"]

write.single.genes.rev(top10.eel, "../gene_expression/top10_eel/")

## for summary statistics a non-Ac reduced dataset
c.df <- contig.df[contig.df$Ac, ] #$

save.image("/home/ele/thesis/454/paper/paper.Rdata")
@ 

\section*{Discussion}

We have generated a \textit{de novo} transcriptome for
\textit{A. crassus}, an important invasive parasite that threatens
wild stocks of the European eel \textit{An. anguilla}. These data will
enable a broad spectrum of molecular research on this ecologically
important and evolutionarily interesting parasite.

As \textit{A. crassus} lives in close association with its host, we
used exhaustive filtering to remove all host-derived, and
host-associated organism-derived contamination from the raw and
assembled data. We generated a transcriptome dataset from the
definitive host \textit{An. japonica} as part of this filtering
process. In addition to eel-derived transcripts, we also removed data
apparently derived from protists, particularly cercozoans, that may
have been co-parasites of the eels sampled. Similar taxonomic
screening of transcrioptome data has been shown to be important
previously \cite{pmid21138572}, particularly in rejection of
hypotheses of horizontal gene transfer into the focal species
\cite{pmid20386959}. We were not able to use base frequency and codon
usage based screening to identify contaminant data \cite{pmid20034392,
  pmid17218127} because contaminant sequences in our data derived from
multiple genomes.


We used a combined assembly approach \cite{pmid20950480} to generate a
transcriptome estimate that had low redundancy and high
completeness. Projects using single assemblers often report
substantially greater numbers of contigs for datasets of similar size
(see e.g. \cite{pmid21364769}). The 3 bias in the assembly likely
derivesd from the use of oligod(T) in mRNA capture and cDNA synthesis
and is near-ubiquitous in deep transcriptome sequencing projects (e.g.
\cite{pmid20331785}). The final \textit{A. crassus} TUG assembly
(32,418 contig consensuses) spans 12.7 Mb, and thus likely covers most
of the expected span of the transcriptome (the \textit{C. elegans}
transcriptome spans 30 Mb \cite{pmid9851916}, and the
\textit{B. malayi} transcriptome 14 Mb \cite{ghedin_draft_2007}),
albeit fragmented.

Comparison between free-living and parasitic nematode species can be
used to identify genes that may underpin adaptations for parastism
\cite{parkinson_transcriptomic_2004, wasmuth_extent_2008}. Annotations
were derived for a ~30\% of all TUGs, and over 50\% of the highCA
contigs using sequence similarity to known proteins. Domain
annotations were derived for 45\% of the highCA TUGs using
InterProScan \cite{pmid11590104}. Comparison with the complete
proteomes of \textit{B. malayi} and \textit{C. elegans} showed a
remarkable degree of congruence in annotation spectrum in the two
parasitic nematodes. This implies that the \textit{A. crassus}
transcriptome is a representative partial genome
\cite{parkinson_partigene--constructing_2004}. Using a
taxonomically-stratified analysis of BLAST similarities, we identified
more \textit{A. crassus} TUGs that apparently arose in the common
ancestor of Nematoda than arose in the last common ancestor of the
Spirurina. As \textit{A. crassus} is part of a lineage that arises
basally in Spirurina, the lack of genes associated with Sprirurina may
be due to phylogenetic distance obscuring relationships, particularly
if the genes underpinning parasitism are, as would be expected,
rapidly evolving. TUGs predicted to be part of gene families that
arose in the last common ancestor of Nematoda or to be novel to
\textit{A. crassus} contained the highest proportion of genes
predicted to have secretory signal peptides. This confirms
observations made in a \textit{Nippostrongylus brasiliensis}
\cite{harcus_signal_2004}, where secreted and surface proteins were
less conserved. Analysis of dn/ds (see below) across conservation
categories favors the hypothesis of rapid evolution in proteins with
more restricted phylogenetic origins.

Transcriptome data were generated from multiple individual
\textit{A. crassus} of Taiwanese and European origin. We identified
abundant SNPs both within and between populations, but noted aberrant
patterns in the ratio of transitions to transversions (ti/tv) and the
ratio of non-synonymous SNPs per non-synonymous site to synonymous
SNPs per synonymous site (dn/ds). Screening of SNPs in or adjacent to
homopolymer regions, removing noise'' associated with common
homopolymer errors \cite{pmid21685085}, improved overall measurements
of SNP quality, increased the ti/tv ratio to more closely resemble
that of canonical datasets, and resulted in a reduced, credible dn/ds
ratio distribution. The corrected ti/tv value of 1.93 (1.25 outside
and 2.39 inside ORFs) is in good agreement with the overall ti/tv of
\textit{Homo sapiens} (2.16 \cite{pmid21169219}) or \textit{Drosophila
  melanogaster} (2.07 \cite{pmid21143862}). The mean dn/ds ratio
decreased with removal of SNPs adjacent to homopolymer regions from
0.45 to 0.24. While interpretation of dn/ds ratios within populations
is not unproblematic \cite{pmid19081788}, the assumption of negative
(purifying) selection on most protein coding genes makes lower mean
values seem more plausible.


We applied a threshold value for the minority allele of 7\% for
exclusion of SNPs, as approximately 10 haploid equivalents were
sampled (5 individual nematodes plus negligible contributions from the
L2 library and offspring within the adult female nematodes). This
screening reduced the number of non-synonymous SNPs in high coverage
TUGs, removed the dependence of dn/ds on coverage, and removed the
need to control for sampling biased by depth (i.e. coverage; see
\cite{pmid18590545} and \cite{pmid20478048}).

The final dn/ds estimates seem plausible, as \textit{D. melanogaster}
female reproductive tract transcripts have dn/ds of 0.15
\cite{pmid15579698} and a Roche 454 transcriptomic analysis of the
parasitic nematode \textit{Ancylostoma caninum} reported dn/ds of 0.3
\cite{pmid20470405}. A dn/ds threshold (on coding sequence) of 0.5 has
been suggested as threshold for assuming positive selection
\cite{pmid15579698}. Using this we identified 144 TUGs that may be
under positive selection, thirteen peptidases were under positive
selection (out of 43 annotated), and the GO term peptidases was
significantly overrepresented in the set of positively selected
TUGs. Those thirteen peptidases are deeply conserved, as twelve had
unique orthologue pairs in \textit{B. malayi} and
\textit{C. elgans}. Peptidases have previously been proposed to have
acquired prominent roles in host-parasite interactions. An
\textit{A. crassus} trypsin-like proteinase may be utilised by the
tissue-dwelling third stage larvae to penetrate host tissue and an
aspartyl proteinase may be a blood meal digestive enzyme in adults
\cite{polzer_identification_1993}. The thirteen proteinases under
positive selection could be targets of adaptive immunity developed
against \textit{A. crassus} \cite{knopf_migratory_2008,
  knopf_vaccination_2008}, which is often only elicited against some
but not all larvae \cite{molnar_caps}.

A set of 199 high-credibility SNPs with high information content for
population genetic studies was identified by genotyping individual
nematodes. The low number of SNPs inferred reflects both the variance
in allele contribution introduced in transcriptomic data and the
stringency of the software used, which is targeted at higher
throughput genome sequence data \cite{pmid21653522x}. Nevertheless,
levels of genome-wide heterozygosity found for the five adult
nematodes examined are in agreement with existing microsattelite data
that show reduced heterozygosity in European populations of
\textit{A. crassus} \cite{wielgoss_population_2008}. The Polish female
nematode was the most highly inbred, while the nematode from the
cultured
\textit{An. japonica} from Taiwan was the most highly outbred.\\

While our experiment was not designed to identify differential
expression between conditions (due to low replication) we used methods
developed for comparison of cDNA libraries \cite{pmid9331369} to infer
differential gene expression according to the origin of the sequencing
libraries. This approach is widely used with 454 transcriptome data
(e.g. \cite{pmid20470405}). We can only tentatively infer differential
expression of a gene under different conditions (sex, origin) based on
identification of significantly differential expression between
libraries. Genes over-expressed in the male \textit{A. crassus}
included major sperm proteins [83], and, surprisingly, a suite of
ribosomal proteins. Collagen processing enzymes were overexpressed in
the female nematodes in line with modulation of collagen synthesis in
nematode embryonic development, and the ovoviviparity of this species
\cite{pmid10637627}. Acetyl-CoA acetyltransferase was identified as
overexpressed in European nematodes compared to the Asian
one. Acetyl-CoA acetyltransferases act in fatty-acid-oxidation in
peroxisomes and mitochondria \cite{pmid4721607}. Together with a
change in steroid metabolism and the enrichment of mitochondrially
localised enzymes these suggest changes in the energy metabolism of
\textit{A. crassus} from different origins. Possible explanations
could include a change to more or less aerobic processes in nematodes
in Europe due to their bigger size and/or increased availability of
nutrients.  TUGs overexpressed in the female libraries showed elevated
levels of dn/ds but genes overexpressed in males had decreased levels
of dn/ds. The first finding is unexpected, as genes overexpressed in
female libraries will also include TUGs related to larval development
(such as the collagen modifying enzymes discussed above), and these
larval transcripts in turn are expected to be under purifying
selection because of pleiotropic effects of genes in early development
\cite{pmid15371532}. The second contrasts with findings that male
specific traits and transcripts often show hallmarks of positive
selection \cite{pmid15795858, pmid11404480}. In \textit{A. caninum},
female-specific transcripts showed an enrichment of parasitism
genes'' \cite{pmid20470405} and a possible explanation would be a
similar enrichment of positively selected parasitism-related genes in
our dataset. For males the decreased dn/ds may be explained by the
high number of ribosomal protein-encoding TUGs, which all show very
low levels of dn/ds. That these TUGs were found to be differentially
expressed remains puzzling. Some male-overexpressed TUGs, such as that
encoding major sperm protein, showed elevated dn/ds. It is unlikely
that correlation of differential expression with positive selection
results from mapping artifacts, as all the ribosomal protein encoding
TUGs identified overexpressed in males have very low dn/ds.\\

Genes differentially expressed according to the geographic origin of
the nematodes showed significantly elevated levels of dn/ds. We
interpret this as reflecting a correlation between sequence evolution
and phenotypic modification in different host, environments or
correlation between sequence evolution and evolution of gene
expression. Whether expression of these genes is modified in different
hosts or evolved rapidly in the contemporary divergence between
European and Asian populations of \textit{A. crassus}, is one focus of
ongoing work building on the reference transcriptome presented
here. For such an analysis it will be important to disentangle the
influence of the host and the nematode population in common garden,
co-inoculation experiments.

\section*{Conclusions }

The \textit{A. crassus} transcriptome provides a basis for a new era
of molecular research on this ecologically important species. It will
aid not only analysis of the invasive biology of this parasite,
assisting in identifying the origins of invading populations as well
as the adaptations that may be being selected in the new European
host, but also in the investigation of the acquisition of parasitism
in the great clade of animal parasites, Spirurina. In particular,
positive selection of proteinases and differences in energy metabolism
between European and Asian \textit{A. crassus} constitute a candidate
phenotype relevant for phenotypic modification or contemporary
divergent evolution as well as for the long term evolution of
parasitism.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Competing interests}
The authors declare no competing interests.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}

EH, HT and MB conceived and designed the experiments. EH carried out
bioinformatic analyses. SB assisted in bioinformatic analyses. AM
prepared sequencing libraries. HT provided close supervision
throughout. EH and MB interpreted results and prepared the
manuscript. All authors have read and approved the final manuscript.

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgements}
\ifthenelse{\boolean{publ}}{\small}{}

This work has been made possible through a grant provided to EH by
Volkswagen Foundation, "F\"{o}rderinitiative Evolutionsbiologie". The
GenePool Genomics Facility is core funded by The School of Biological
Sciences, University of Edinburgh, the Darwin Trust of Edinburgh, the
UK Natural Environment Research Council (award reference R8/H10/56)
and the UK Medical Research Council (award reference G0900740). We are
grateful to Yun-San Han for his help collecting samples in Taiwan and
to Karim Gharbi for overseeing the project within the
GenePool. Timoth\'ee Cezard, Sujai Kumar and Graham Thomas gave
essential analytic and informatic support.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article} % Style BST file
  \bibliography{/home/ele/bibtex/master}
} % Bibliography file (usually '*.bib' )

%%%%%%%%%%%

\ifthenelse{\boolean{publ}}{\end{multicols}}{}
\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}

\subsection*{Figure 1 - Annotation of the Anguilicolla crassus transcriptome}
 

Number of annotated sequences in the transcriptome of
\textit{A. crssus} for all TUGs (a) and for highCA contigs
(b). Annotations with Gene Ontology (GO), Enzyme Commission (EC) and
Kyoto Encyclopedia of Genes and Genomes (KEGG) terms were inferred for
predicted proteins using annot8r (version 1.1.1)
\cite{schmid_annot8r:_2008}. For highCA contigs additional domain
annotations obtained with InterProScan \cite{pmid11590104} are also
enumerated.

\subsection*{Figure 2 - Comparing high level GO-slim annotations}

Comparing high level GO-slim annotations obtained through annot8r
(version 1.1.1) \cite{schmid_annot8r:_2008} for \textit{A. crassus} to
those for the model nematodes \textit{C. elegans} and
\textit{B. malayi} infered using the same pipeline. For GO categories
molecular function, cellular compartment and biological process the
number of terms in high level GO-slim categories is given. In the two
parasitic nematodes a higher degree of congruence in annotation
spectrum is observed (Spearman correlation coefficient 0.95) than in
comarison to the complete proteome of \textit{C. elegans} (0.90).

\subsection*{Figure 3 - Enrichment of Signal-positives for categories
  of evolutionary conservations}

Categories of evolutionary conservation recorded using the taxonomy of
BLAST-matches at two different bitscore thresholds (50 or 80) are
compared for the occurence of signal peptide cleavage sites and signal
anchor signatures, predicted using SignalP V4.0 \cite{pmid21959131}.
Contigs were categorised as conserved, novel in the kingdom Metazoa,
the phylum Nematoda or Spirurina \textit{sensu}
\cite{blaxter_molecular_1998}. TUGs without any match at a given
threshold were categorised as novel in \textit{A. crassus} (Ac). The
highest proportions of genes predicted to have secretory signal
peptides are observed in TUGs predicted to be part of gene families
that arose in the last common ancestor of Nematoda or to be novel to
\textit{A. crassus}.


\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\subsection*{Table 1 - Sampling, trimming and pre-assembly screening,
  library statistics}


<<echo=FALSE, results=tex>>=
## print(xtable(TRIM, display=rep("d", times=7)), floating=FALSE)
trim_mark <- transform(TRIM, lowqal = short+lowq+dust+shortq)[1:6, ]
trim_mark <- trim_mark[,c(1,7)]

trim_mark <- merge(pre.screen.tab,
                   trim_mark, by="row.names")

trim_mark$life.st <- c("adult female", "adult female", "L2 larvae",
                       "adult male", "adult female", "adult female")

trim_mark$source.p <- c("Europe Rhine", "Europe Poland", "Europe Rhine",
                       "Asia cultured", "Asia cultured", "Asia wild")

trim_mark$geolocation <- c("49.0262N; 8.310556E", "53.751959N; 21.730957E", "49.0262N; 8.310556E",
                           "22.6418N; 120.4440E", "22.6418N; 120.4440E", "22.5074N; 120.4220E")

trim_mark[7,] <- c("Eel", "", "", "", "", length(eel.clean), sum(nchar(eel.clean)),
                   length(eel.all), length(eel.all)-length(eel.clean), "liver", "Taiwan", "")

trim_mark <- trim_mark[, c(1, 10, 11, 12, 8, 9, 2, 4, 5, 3, 6, 7 )]
names(trim_mark)[c(1, 5)] <- c("library", "raw.reads")

trim_mark <- merge(trim_mark, colSums(all.counts), by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "mapping.unique"

sum.Ac <- colSums(all.counts[rownames(all.counts)%in%contig.df[contig.df$Ac, "contig"], ])
trim_mark <- merge(trim_mark, sum.Ac, by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "mapping.Ac"

sum.Ac.MN <- colSums(all.counts[rownames(all.counts)%in%contig.df[contig.df$AcMN, "contig"], ])
trim_mark <- merge(trim_mark, sum.Ac.MN, by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "mapping.MN"

sum.o.32 <- colSums(counts[rownames(counts)%in%contig.df[contig.df$AcMN, "contig"], ])
trim_mark <- merge(trim_mark, sum.o.32, by.x="library", by.y="row.names")
names(trim_mark)[length(trim_mark)] <- "over.32"

names(trim_mark) <- c("sequencing library",
                      "lifecycle stage",
                      "source population",
                      "geolocation", 
                      "raw reads",
                      "low quality reads",
                      "\\textit{A. crassus} rRNA reads",
                      "eel-host mRNA reads",
                      "eel-host rRNA reads",
                      "Cercozoa reads (rRNA)",
                      "valid reads",
                      "span of valid reads (in bases)",
                      "reads mapping (uniquely)",
                      "reads mapping to \\textit{A. crassus} contigs",
                      "reads mapping highCA contigs",
                      "reads mapping to contigs with count >32")

trim_mark[,5:16] <- lapply(trim_mark[,5:16], as.numeric)
total.trim <- colSums(trim_mark[,5:16])
total.trim <- c("total","", "","", total.trim)

trim_mark <- t(trim_mark)
trim_mark <- cbind(trim_mark, total.trim)
trim_mark.tab <- xtable(trim_mark)

align(trim_mark.tab) <- "p{3.5cm}p{1.5cm}p{1.5cm}p{1.5cm}p{1.5cm}p{1.5cm}p{1.5cm}p{1.5cm}"

print(trim_mark.tab, floating=FALSE, include.colnames=FALSE,
      hline.after=c(0,1, nrow(trim_mark)), sanitize.text.function = identity)
@

\newpage

\subsection*{Table 2 - Assembly classification and contig statistics}

<<echo=FALSE, results=tex>>=
T2 <- as.data.frame.array(tapply(contig.df$contig, contig.df$category%in%"MN", length))
rownames(T2) <- c("lowCA", "highCA")
names(T2) <- "total.contigs"

T2$rRNA.contigs <- table(contig.df$category%in%"MN", contig.df$contamination%in%"AcrRNA" &
                         !(contig.df$phylum.nr%in%c("Chordata") |
                          contig.df$phylum.nt%in%c("Chordata")))[, 2]

T2$fish.contigs <- table(contig.df$category%in%"MN",
                         contig.df$contamination%in%c("eelmRNA", "eelrRNA" ) |
                         contig.df$phylum.nr%in%c("Chordata") |
                         contig.df$phylum.nt%in%c("Chordata"))[, 2]

T2$xeno.contigs <- table(contig.df$category%in%"MN",
                         !(contig.df$contamination%in%c("eelmRNA", "eelrRNA", "AcrRNA") |
                           contig.df$phylum.nr%in%c("Chordata") |
                           contig.df$phylum.nt%in%c("Chordata")) & 
                         !(contig.df$kingdom.nt%in%c("Metazoa", "No_hit") &
                           contig.df$kingdom.nr%in%c("Metazoa", "No_hit")))[,2]
 
T2$remaining.contigs <- table(contig.df$category%in%"MN", contig.df$Ac)[, 2]
T2$remaining.span <- tapply(contig.df[contig.df$Ac, ]$seq,
                            contig.df[contig.df$Ac,]$category%in%"MN",
                            function (x) sum(nchar(as.character(x))))
T2$non.u.cov <- cov.mn
T2$cov <- cov.mn.u

T2$p4e <- table(c.df$category%in%"MN", gsub("p4e->", "", c.df$method))
T2$full.3p <- table(c.df$category%in%"MN", c.df$full.3p)[,2]
T2$full.5p <- table(c.df$category%in%"MN", c.df$full.5p)[,2]
T2$full.l <- table(c.df$category%in%"MN", c.df$full.length)[,2]

T2$GO <-  table(c.df$category%in%"MN", c.df$contig%in%GO.annot$pept_id)[, 2]
T2$EC <-  table(c.df$category%in%"MN", c.df$contig%in%EC.annot$pept_id)[, 2]
T2$KEGG <-  table(c.df$category%in%"MN", c.df$contig%in%KEGG.annot$pept_id)[, 2]
T2$IPR <-  table(c.df$category%in%"MN", c.df$contig%in%IPR.annot$pept_id)[, 2]
T2[1, "IPR"] <- "n/d"

T2$nem.blast <- table(c.df$category%in%"MN", c.df$phylum.nr%in%"Nematoda")[, 2]

T2$any.blast <- table(c.df$category%in%"MN", c.df$phylum.nr%in%"No_hit")[,1]

T2 <- as.data.frame(t(T2))
T2$"all TUGs" <- as.numeric(as.character(T2$lowCA))+as.numeric(as.character(T2$highCA))

T2["non.u.cov", "all TUGs"] <- cov.nu.total
T2["cov", "all TUGs"] <- cov.u.total

dig <- t(matrix(c(rep(0,24), rep(c(0, 3 , 3, 3), 2), rep(0,52)), nrow=4))

rownames(T2) <- c("total contigs",
                  "contigs hitting rRNA",
                  "contigs hitting eel-mRNA or Chordata",
                  "non-eukaryote contigs",
                  "contigs remaining",
                  "total span of remaining contigs (in bases)",
                  "non-unique mean base coverage of contigs",
                  "unique mean base coverage of contigs",
                  "protein predictions by BLAST similarity", 
                  "protein predictions by ESTscan",
                  "protein predictions by longest ORF",
                  "contigs without protein prediction",
                  "contigs with complete 3\' end",
                  "contig with complete 5\' end",
                  "full length contigs",
                  "contigs with GO- annotation",
                  "contigs with EC- annotation",
                  "contigs with KEGG- annotation",
                  "contigs with InerProScan- annotation",
                  "contigs with BLAST hit to nematode",
                  "contigs with any BLAST hit")

T2.tab <- xtable(T2[,c(2,1,3)], digits=dig)

print(T2.tab)

@ 

\newpage

\subsection*{Table 3 - Evolutionary conservation and novelty}

<<echo=FALSE, results=tex>>=
nov.tab <- rbind(summary.factor(contig.df[contig.df$Ac, "novel.50"]), 
                 summary.factor(contig.df[contig.df$Ac, "novel.80"]))
nov.tab.MN <- rbind(summary.factor(contig.df[contig.df$AcMN, "novel.50"]), 
                 summary.factor(contig.df[contig.df$AcMN, "novel.80"]))
nov.tab <- as.data.frame(rbind(nov.tab, nov.tab.MN))

nov.tab$"contig set" <- c("all TUGs", "all TUGs", "highCA", "highCA")
nov.tab$cutoff <- c("50", "80", "50", "80")


print(xtable(nov.tab[, c(6, 7, 1:5)]), floating=FALSE, include.rownames=FALSE)

@ 

\newpage

\subsection*{Table 4 - Overrepresentation of GO terms in positively
  selected \textit{A. crassus} TUGs}

<<echo=FALSE, results=tex>>=

GO.over.dn.ds$category <- ""
GO.over.dn.ds[c(1, GO.over.dn.ds.lines["MF"]+1, 
                GO.over.dn.ds.lines["MF"]+GO.over.dn.ds.lines["BP"]+1),
              "category"] <- c("molecular function", "biological process", "cellular compartment")
GO.over.dn.ds <- GO.over.dn.ds[, c(ncol(GO.over.dn.ds), 1:ncol(GO.over.dn.ds)-1)]

names(GO.over.dn.ds)[4] <- "Number annotated"
names(GO.over.dn.ds)[5] <- "Number dn/ds > 0.5"

GO.over.dn.ds.tab <- xtable(GO.over.dn.ds)
align(GO.over.dn.ds.tab) <- "p{1.7cm}p{1.7cm}lp{4.5cm}p{1cm}p{1cm}ll"

print(GO.over.dn.ds.tab, tabular.environment="longtable",
      floating=FALSE,
      hline.after=c(0, c(GO.over.dn.ds.lines["MF"],
        GO.over.dn.ds.lines["MF"]+GO.over.dn.ds.lines["BP"],
        GO.over.dn.ds.lines["MF"]+GO.over.dn.ds.lines["BP"]+ GO.over.dn.ds.lines["CC"])),
      include.rownames=FALSE)
@ 

\newpage

\subsection*{Table 5 - Measurements of multi-locus heterozygosity for
  single worms}

<<echo=FALSE, results=tex>>=

names(het.table) <- c("relative heterozygosity", "internal relatedness",
                      "homozygosity by loci", "standardised heterozygosity",
                      "informative SNPs")

het.table.tab <- xtable(het.table, digits=c(1, 2, 2, 2, 2, 0))
align(het.table.tab) <- "p{2.5cm}p{2cm}p{2cm}p{2cm}p{2cm}p{2cm}"

print(het.table.tab)
@ 

\newpage

\subsection*{Table 6 - Overrepresentation of GO-terms differentially
  expressed between male and female \textit{A. crassus}}

<<echo=FALSE, results=tex>>=

GO.over.sex$category <- ""
GO.over.sex[c(1, GO.over.sex.lines["MF"]+1, 
                GO.over.sex.lines["MF"]+GO.over.sex.lines["BP"]+1),
              "category"] <- c("molecular function", "biological process", "cellular compartment")
GO.over.sex <- GO.over.sex[, c(ncol(GO.over.sex), 1:ncol(GO.over.sex)-1)]

names(GO.over.sex)[4] <- "Number annotated"
names(GO.over.sex)[5] <- "Number significant"

GO.over.sex.tab <- xtable(GO.over.sex)

align(GO.over.sex.tab) <- "p{1.7cm}p{1.7cm}lp{4.5cm}p{1cm}p{1cm}ll"

print(GO.over.sex.tab, tabular.environment="longtable",
      floating=FALSE, 
      hline.after=c(0, c(GO.over.sex.lines["MF"],
        GO.over.sex.lines["MF"]+GO.over.sex.lines["BP"],
        GO.over.sex.lines["MF"]+GO.over.sex.lines["BP"]+ GO.over.sex.lines["CC"])),
        include.rownames=FALSE)
@ 

\newpage

\subsection*{Table 7 - Overrepresentation of GO-terms differentially
  expressed between Taiwanese and European \textit{A. crassus}}

<<echo=FALSE, results=tex>>=

GO.over.eel$category <- ""
GO.over.eel[c(1, GO.over.eel.lines["MF"]+1, 
                GO.over.eel.lines["MF"]+GO.over.eel.lines["BP"]+1),
              "category"] <- c("molecular function", "biological process", "cellular compartment")
GO.over.eel <- GO.over.eel[, c(ncol(GO.over.eel), 1:ncol(GO.over.eel)-1)]

names(GO.over.eel)[4] <- "Number annotated"
names(GO.over.eel)[5] <- "Number significant"


GO.over.eel.tab <- xtable(GO.over.eel)

align(GO.over.eel.tab) <- "p{1.7cm}p{1.7cm}lp{4.5cm}p{1cm}p{1cm}ll"

print(GO.over.eel.tab, tabular.environment="longtable",
      floating=FALSE, 
      hline.after=c(0, c(GO.over.eel.lines["MF"],
        GO.over.eel.lines["MF"]+GO.over.eel.lines["BP"],
        GO.over.eel.lines["MF"]+GO.over.eel.lines["BP"]+ GO.over.eel.lines["CC"])),
include.rownames=FALSE)

@ 

\normalsize

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Additional Files}

\subsubsection*{Additional text}

\textbf{Supporting\_information.pdf}: This document contains the 3
additional figures referenced in the main text and an additional text
describing the assembly process and evaluation of assembly quality in
further detail. This text contains additional 7 figures (additional
figures 4-10) and 3 tables.

\subsubsection*{Additional data}

\textbf{A.crassus\_transcriptome\_contig\_data.csv} All data computed
on the contig level, as described in the manuscript and additional
text including sequences (raw, coding, imputed, protein).

\textbf{A.crassus\_transcriptome\_screened\_SNPs.csv}: High quality
SNPs. The contig, the base relative to the start of the contig (base),
the reference base-call (from), the alternative base-call (to), the
number of reads supporting the reference (nfrom) and the alternative
(nto), the percentage of the alternate allele (perc), whether the SNP
is in the region of an ORF (inORF), the position in the Frame
(inFrame) and the effect of the SNP (effect; synonymous,
non-synonymous or nonsense) are given.

\textbf{A.crassus\_transcriptome\_originDE.csv}: Contigs
differentially expressed between European and Taiwanese
worms. Normalised counts and the natural logarithm of fold changes are
given.

\textbf{A.crassus\_transcriptome\_sexDE.csv}: Contigs differentially
expressed between male and female worms. Normalised counts and the
natural logarithm of fold changes are given.

\subsubsection*{Additional figure 11 (a-i)}
\textbf{Subgraphs of GO induced by the top 10 terms identified as
  enriched in different sets of genes}: Subgraphs of the GO ontology
categories induced by the top 10 terms identified as enriched in
different sets of genes. Boxes indicate the 10 most significant
terms. Box colour represents the relative significance, ranging from
dark red (most significant) to light yellow (least significant). In
each node the category-identifier, a (eventually truncated)
description of the term, the significance for enrichment and the
number of DE / total number of annotated gene is given. Black arrows
indicate an ``is-a'' relationship. GO ontology category and the set of
genes analysed for the enrichment are indicated in each figure.

\end{bmcformat}
\end{document}
