%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}


\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of the swimmbladder-nematode
    Anguillicola crassus: Resources for an alien parasite}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel G Heitlinger\correspondingauthor$^{1,2}$%
       \email{Emanuel G Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Horst Taraschewski$^1$%
       \email{Horst Taraschewski- dc20@rz.uni-karlsruhe.de}%
       and Mark Blaxter$^2$%
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute 1, University of Karlsruhe,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} 
  \paragraph*{Results:} 
  \paragraph*{Conclusions:}
  Yeh!
  
\end{abstract}


      \ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

\textit{Anguillicola crassus} Kuwahara, Niimi et Itagaki, 1974
\cite{kuwahara_Niimi_Itagaki_1974} is a nematode feeding on blood in
the swimmbladder of freshwater eels of the genus
\textit{Anguilla}. Originally endemic to East-Asian populations of the
Japanese Eel (\textit{Anguilla japonica}), \textit{A. crassus} has
attracted interest due to recent anthropogenic expansion of its
geographic- and host-range to Europe and the European eel
(\textit{Anguilla anguilla}). Soon after it had been recorded for the
first time in 1982 in North-West Germany \cite{fischer_teichwirt}, to
where it was most likely introduced by live-eel trade
\cite{koops_anguillicola-infestations_1989, koie_swimbladder_1991},
\textit{A. crassus} rapidly spread throughout populations of its newly
accquired host (for a review see \cite{kirk_impact_2003}). At the
present day it is found in all but the northernmost population of the
European eel in Iceland \cite{kristmundsson_parasite_2007}.

The impact of \textit{A. crassus} on the European eel has been a major
focus of research during the past decades. High prevalences of the
parasite of above 70\% (e.g. \cite{wrtz_distribution_1998}), as well
as high intesities of infections were reported, throughout the newly
colonized area \cite{lefebvre_anguillicolosis:_2004}.  Based on a
broad base of work on its epidemiology \textit{A. crassus} can be
regarded as a model for parasite introduction and spread
\cite{taraschewski_hosts_2007}.

As in the natural host in Asia prevalences and intesities are lower
\cite{mnderle_occurrence_2006}, high epidemiological parameters
were attributed to the inadequate immune-response of the European Eel
\cite{knopf_swimbladder_2006}. Interestingly the differences in the
two host also affect the size and life-history of the worm: In
European eels the nematodes are bigger and develop and reproduce
faster \cite{knopf_differences_2004}.  While the Japanese eel is
capable of killing larvae of the parasite after vaccination
\cite{knopf_vaccination_2008} or under high infection pressure
\cite{heitlinger_massive_2009}, only pathological effects such a
thikening of the swimmbladder wall \cite{wrtz_histopathological_2000}
have been found in the European eel. 

The genus \textit{Anguillicola} holds a phylogenetic position basal to
the Spirurina (clade III \textit{sensu} Blaxter
\cite{blaxter_molecular_1998}), one of 5 major clades of nematodes
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
exclusively exhibit a parasitic lifestyle and comprise improtant human
pathogens as well as prominent parasites of livestock (e.g. the
Filaroidea and Ascarididae). This phylogenetic position makes the
Anguillicoloidae an interesting system in the endeavour to understand
the emergence of parasitism in Spirurina and as an ``outgroup'' for
functional studies of parasitism in this clade. Some functionally
interesting genes in this respect are thought to be under diversifying
selection in an arms-race between host and
parasite\cite{zang_serine_2001}.

Recent advances in sequencing technology (often termed Next Generation
Sequencing; NGS), provide the opprotunity for rapid and cost-effective
generation of genome-scale data. The longer read length of
454-sequencing \cite{pmid16056220} compared to other NGS technologies,
allows \textit{de novo} assembly of Expressed Sequence Tags (ESTs) in
organisms lacking previouse genomic or transcriptomic data (for a
comprehensive list of studies using this approach before Oct 2010 see
\cite{pmid20950480}). Such transcriptomic datasetes are still less
expensive than genomic data-sets in terms sequencing costs and
analytical needs.

The difference of the immune attack on \textit{A. crassus} in the two
diffent hosts provides an opportunity to investigate the parasite's
response to different ``immune environments'' on a transcriptomic
basis.

In this study we report assembly of raw reads from cDNA libraries of
L2 larvae, four female adults and one male adult of
\textit{A. crassus} into contigs (continuous sequence, representing a
mRNA-transcript) of a reference transcriptome. We describe screening
for xenobiotics and host-contamination, prediction of protein
sequences and functional annotation of these contigs. Furthermore
combining ORF prediction and identification of SNPs allowed inferrence
of diversifying selection.

% Aditionally we analysed gene expression, by counting of raw
% reads from two European worms and two Taiwanese worms mapped to the
% assembled refference. We tested both positively selected and
% differentially expressed genes for overrepresentation of GO-terms and
% KEGG-parthways, with the aim to test for an enrichment of terms and
% parthways involved in host-parasite interaction.

As shorter read-length but higher throughput of the Illumina-Solexa
platform provides superior means for gene expression analyis
\cite{pmid21627854}, the transcriptome assembly presented here is
planned to be used as a reference in a future study using
gene-expression tags (SuperSAGE \cite{pmid20967605}) or RNA-seq
\cite{pmid19015660}. To test the suitability of the presented
transcriptome we mapped a pilot-set of expression-tags. This analysis
is used as an additional estimate for the correctness of the hypotesis
that each contig in the assembly constitutes.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}

<<load.libs, echo=FALSE, results=hide>>=
library(VennDiagram)
library(xtable)
library(reshape)
library(ggplot2)

library(GSEABase)
library(GOstats)
library(GO.db)

library(limma)
library(XML)
library(coin)
library(multcomp)

library(DESeq)
library(ShortRead)

source("/home/ele/thesis/454/common_R_functions.R")
@ 

 \subsection*{Sequencing, trimmng and pre-assembly screening}

<<trim, echo=FALSE, results=hide, cache=TRUE>>=
trimdat <- readLines("/home/ele/Data/454/trimmed/trimming.stats")
trimfile <-  grep("Trimming sff file *", trimdat, value=TRUE)
trimfile <- sapply(strsplit(trimfile, " |,"), function (x) basename(x[4]))
lib <- gsub(".sff'", "", trimfile)

raw_reads<- grep("Sequences analyzed:.*", trimdat, value=TRUE)
raw_reads <- as.numeric(gsub("Sequences analyzed: *", "", raw_reads))
short <- grep("by 'short':.*", trimdat, value=TRUE)
short <- as.numeric(gsub(" *by 'short': *", "", short))
lowq <- grep("by 'low_qual':.*", trimdat, value=TRUE)
lowq <- as.numeric(gsub(" *by 'low_qual': *", "", lowq))
dust <- grep("by 'dust':.*", trimdat, value=TRUE)
dust <- as.numeric(gsub(" *by 'dust': *", "", dust))
shortq <- grep("by 'shortq':.*", trimdat, value=TRUE)
shortq <- as.numeric(gsub(" *by 'shortq': *", "", shortq))
valid <- raw_reads-(short+lowq+dust+shortq)

TRIM <- data.frame(row.names=lib, raw_reads, short, lowq, dust, shortq, valid)
TRIM["total",] <- colSums(TRIM)

## subsitute the publication style library abrevations
rownames(TRIM) <- gsub("179F", "T1", rownames(TRIM))
rownames(TRIM) <- gsub("10F", "T2", rownames(TRIM))
rownames(TRIM) <- gsub("KS4F", "E1", rownames(TRIM))
rownames(TRIM) <- gsub("UW07F", "E2", rownames(TRIM))
rownames(TRIM) <- gsub("L2R3", "L2", rownames(TRIM))
rownames(TRIM) <- gsub("M175", "M", rownames(TRIM))

@ 

In total \Sexpr{TRIM["total","raw_reads"]} raw sequencing reads were
trimmed and quality screened. After this trimming
\Sexpr{TRIM["total","valid"]} sequences were regarded valid, table 1
gives an overview of trimming statistics for different
libraries. Notably the variablity in the number of sequences excluded
due to low quality and short quality-sequence in \texttt{seqclean} was
high indicating different properties of read-quality in the different
libraries. The \Sexpr{TRIM["total", "valid"]} reads regarded valid
after this trimming were further screened for rRNA and
host-contamination before assembly.

<<read.libs, echo=FALSE, results=hide, cache=TRUE>>=

RE <- list()
for (l in (c("10F", "179F", "M175", "L2R3", "KS4F", "UW07F"))){
   command <- paste("/home/ele/tools/Newbler2.6/bin/sffinfo -s /home/ele/Data/454/trimmed/",
                    l, ".sff.trimmed.sff;", sep="")
   connect <- pipe(command)
   RE[l] <- list(read.sequences(connect))
   closeAllConnections()
}
names <- gsub(" .*$", "", unlist(lapply(RE, names)))
lib <- melt(RE)

## subsitute the publication style library abrevations
lib$L1 <- gsub("179F", "T1", lib$L1)
lib$L1 <- gsub("10F", "T2", lib$L1)
lib$L1 <- gsub("KS4F", "E1", lib$L1)
lib$L1 <- gsub("UW07F", "E2", lib$L1)
lib$L1 <- gsub("L2R3", "L2", lib$L1)
lib$L1 <- gsub("M175", "M", lib$L1)

RE <- cbind(read=names, lib)
names(RE)[2:3] <- c("seq", "lib")
RE$seq <- as.character(RE$seq)
RE$length <- nchar(RE$seq)
@ 

\SweaveInput{../contamination/pre_assembly.Rnw}

<<count.screened, echo=FALSE, cach=TRUE>>=

pre.screen.tab <- table(R$lib, R$dbhit)
names(pre.screen.tab) <- c("A. crassus rRNA", "host", "host rRNA", "valid (assembled)")

nscreened <- nrow(R[R$dbhit%in%c("eelrRNA", "AcrRNA", "eelmRNA"), ])
nscreened.r <- nrow(R[R$dbhit%in%c("eelrRNA", "AcrRNA"),])
nscreened.m <- nrow(R[R$dbhit%in%"eelmRNA",])

AcrRNA_vs_nt <- read.blast.best("/home/ele/Data/454/pre_assembly_screening/AcrRNA_vs_nt.blt")
AcrRNA_cerco <- AcrRNA_vs_nt[grepl("Cerco|Flag", AcrRNA_vs_nt$V1), ]

cerco <- cbind( sequence.identifier=as.character(AcrRNA_cerco$V2), 
                sequence.identitiy=AcrRNA_cerco$V3,
                hsp.length=AcrRNA_cerco$V8-AcrRNA_cerco$V7)

@ 

In total \Sexpr{nscreened} reads were discarded prior to assembly,
\Sexpr{nscreened.r} due to \texttt{blast} hits (e-value cut of 1e-5,
hit longer than 80\% of the read-length and identity bigger 95\%) to
rRNA, \Sexpr{nscreened.m} to host-databases (same thresholds). Figure
(rRNA\_plots.png) and table 2 give an overview of this pre-assebmly
rRNA and host-screening in different libraries. During annotoation of
the rRNA-database we constructed for \textit{A.crassus} (assembled
from reads screened against general rRNA-databases) we found some
contigs with high similarity 18S sequences from cercozoan and other
flaggelate protists over the majority of their length. Table 3 gives
an overview of the hits in NCBI-nt for these 11 contigs assembled from
rRNA sequences. The origin of the rRNA-reads assembled in these
contigs (in screening-database assembly, not the main assebmly, where
they were screened out) showed that they originated nearly exclusively
from library L2.
 
Containing the highest number of raw reads, library E1 showed the
highest amount of low-quality and short sequences in trimming and a
high amount of rRNA and host-contamination, making it the library with
the lowest number of sequences used for assembly.

\subsection*{Assembly}

<<load.meth, echo=FALSE>>=
load("/home/ele/thesis/454/Method-assembly/Method.Rdata")

nMN <- nrow(contig.df[contig.df$category%in%"MN",])

nbad.c <- nrow(contig.df[contig.df$category%in%
                         c("M_n","N_n","N_1","M_1"),])

nSing <- nrow(contig.df[contig.df$category%in%0,])
@

For assembly we used a method proposed by \cite{pmid20950480}: We
combined two assemblies merging their contigs. For a detailed
description of the assembly method see the supplementary methods
file. In total \Sexpr{nMN} contigs were regarded good quality, as they
were supported by two assembly algorithms (in the remainder of the
manuscript ``the assembly'' referres to this ``second-order''-assembly
and quality contigs referres to the good-quality contig-category from
this). In additon an expanded dataset consisting of the afore
mentioned assembly plus the (\Sexpr{nbad.c}) contigs only supported by
one of the ``first-order'' assemblies and (\Sexpr{nSing}) singletons
was obtained (this is reffered to as the ``fullest assembly'', the
sequences in this dataset are referred to as Tentative Unique Genes;
TUGs). In the methods supplementary file we show that these datasets
are considered the most parsiomonous (having the smallest size) for
their quality (covering the largest amount of sequence in reference
transcritomes). In the high quality assembly parsimony and low
redundancy is prioritzsed, in the full assembly completeness.

<<mean.cov, echo=FALSE, cache=TRUE>>=

cov <- round(mean(con.pile[, "coverage"]),2)
cov.u <- round(mean(con.pile.uniq[, "uniq_coverage"]),2)
cov.mn <- round(mean(con.pile[con.pile$contig%in%
                                contig.df[contig.df$category=="MN", "contig"],
                                "coverage"]),2)
cov.mn.u <- round(mean(con.pile.uniq[con.pile.uniq$contig%in%
                                        contig.df[contig.df$category=="MN", "contig"],
                                        "uniq_coverage"]),2)

@ 

The mean per-base coverage of the TUGs was \Sexpr{cov} and the mean
per-base unique coverage was \Sexpr{cov.u}. The mean per-base coveage
of the quality contigs was lower with \Sexpr{cov.mn}, but the unique
coverage higher with \Sexpr{cov.mn.u}. This indicates a higher amount
of redundancy in the full assebmly set compared to the good-quality
set. The distribution of mean per base coverages for single contigs or
TUGs is given in Figure (coverage\_per\_contig.png)

Figure (coverage\_plots) shows the realtionship between, read-number
in the assemblies and coverage.

\subsection*{Marking possible host- and xenobiotic contamination}

\SweaveInput{../contamination/post_assembly.Rnw}

To elucidate the taxonimic origin of the assembled sequences, we used
extensive \texttt{blast}-queries.

\subsubsection*{Host-contamination approach}

In this evalutation we labeled all sequences hitting (e-value
cut-off of 1e-5) the same databases used for screening prior to
assembly (eel-mRNA, eelrRNA, \textit{A.crassus}rRNA) with the
respective database-hit, if the sequence did not produce a better hit
(based on bit-score comparison) against the nucleotide version of
nempep4 \cite{pmid21550347}.  

To make this evaluation more spqcific we only sequences hitting whith
the over 50\% of their length with an identity >70\% to lable the
TUGs as ``contamination'' accordig to the respective database-hit.

Two kinds sequences were labled ``valid'' \textit{A.crassus}
sequences: Valid due to evidence of nematode origing (hitting
nempep4), and valid due to missingness of evidence of host-origin.

<<post.con, echo=FALSE, cache=TRUE>>=
post.table <- rbind(table(contig.df$contamination),
                    rbind(tapply(contig.df$uniq_coverage, contig.df$contamination, mean)))

row.names(post.table) <- c("number",
                           "mean coverage")

nvalid <- nrow(contig.df[grepl("valid", contig.df$contamination) ,])

## Examples of selecting from the data

select_no_eel <- grepl("valid", contig.df$contamination) &
                            !contig.df$phylum.nt%in%"Chordata" &
                            !contig.df$phylum.nr%in%"Chordata"
select_no_eel_MN <- select_no_eel  & contig.df$category%in%"MN"

n_no_eel <- nrow(contig.df[select_no_eel,])
n_MNno_eel <- nrow(contig.df[select_no_eel_MN,])

select_AC <- grepl("valid", contig.df$contamination) &
                          contig.df$phylum.nt%in%c("Nematoda", "No_hit") &
                          contig.df$phylum.nr%in%c("Nematoda", "No_hit")
select_AC_MN <- select_AC & contig.df$category%in%"MN"

contig.df$Ac <- select_AC
contig.df$AcMN <- select_AC_MN

n_su_Ac <- nrow(contig.df[contig.df$Ac, ])
n_suMN_Ac <- nrow(contig.df[contig.df$AcMN, ])

@ 

Table 3 gives the numbers of TUGs labled as possible host or rRNA
contamination. The \Sexpr{nvalid} TUGs regarded to be of
\textit{A.crassus}-origin give a positive indication of the origin of
sequence data. However TUGs hitting \textit{A.crassus}-rRNA still
have a high coverage, indicating a noticable loss of raw-read data
(e.g. used in mapping) to additional rRNA-coverage. In contrast
TUGs labeled to be of host-origin had a slighly lower coverage than
TUGs labled valid, indicating an only moderate amount of raw-data
being ``lost'' to this kind of off-target data.

\subsubsection*{Off target data approach}

To elucitade possible xenobiotic sequences (comprising unexpected
off-target data) we searched our complete dataset against both NCBI-nt
and NCBI-nr. Figure (phylum\_plots) gives an overview of the number of
top-hits to different phyla sorted by kingdom.

For evey study of the biology of \textit{A.crassus} based on the
presented dataset we advise to use the taxonomic evaluation and
host-contamination screening at latest at the stage of interpretation
of results to avoid missleading conclusions based on differences in
off target data.

For example a study excluding putative host-derived sequence prior to
analysis, would have \Sexpr{n_no_eel} TUGs or \Sexpr{n_MNno_eel}
quality-contigs left for analysis, based on choosing only sequence not
labled ``Chordata'' (from nt- and nr-searches) and labled ``valid'' or
``valid-nempep'' (from the evaluation of host-contamination). A second
analysis aiming to exclude all data suspicious to be not of
\textit{A.crassus}-origin would choose \Sexpr{n_su_Ac} TUGs or
\Sexpr{n_suMN_Ac} quality-contigs, by using sequence labled
``Nematoda'' or ``no hit'' (from nt- and nr-searches) and ``valid'' or
``valid-nempep''. We use such restiricted dataset like in the second
example in later analyses to verify that results are not induced by a
possible contamination backround.

\subsection*{Evolutionary conservation}
\SweaveInput{../conservation/conservation.Rnw}

Based on taxonomically classified \texttt{blast}-results described
above we also determined evolutionary at 2 thresholds (bitscore of 50
and 80; roughly equivalent with an e-value of e-6 and e-10
respectively) at three taxonomic levels: We categorized a sequence if
it had a hit to ``nematoda'' and a second phylum as conserved across
the phylum level and to ``metazoa'' and a second kindom as conserved
across the kingdom level. As a third taxonomic level we defined the
Spirurina (Clade III \textit{sensu}
\cite{blaxter_molecular_1998}). Conversely from these data
``conserved'', ``novel in metazoa'', ``novel in nematoda'' and ``novel
in clade III'' categories could be computed. See table 5 for a summary
of these categories for the two thresholds.


\subsection*{Protein prediction}

\SweaveInput{../protein_prediction/imputed_from_p4EST.Rnw}

For in total
\Sexpr{nrow(contig.df[contig.df$method!="no-prediction",])#$} TUGs
a protein was predicted. A summary of protein prediction using the
different methods implemented in prot4EST is given in table 6.
 
The full open reading frame was obtained for in total
\Sexpr{nrow(imputed[imputed$full.length,])#$} TUGs, while for
\Sexpr{nrow(imputed[imputed$full.5p,])#$} the 5' end and for
\Sexpr{nrow(imputed[imputed$full.3p,])#$} the 3’ end was complete.  A
predicted ORF was defined at complete if it was starting with an
``atg''-start-codon, having at least 3 non-coding bases 5' of the
start-codon and having a ``tag'' (\Sexpr{stopcods["TAG"]}
observations), ``taa'' (\Sexpr{stopcods["TAA"]}) or ``tga''
(\Sexpr{stopcods["TGA"]}) stop-codon 3' of the ORF.

In additon to the protein-sequence used for annotation we obtained a
corrected nucleotide-sequence using the nucleotide equivalent of the
protein-prediction. This was necessary as \texttt{prot4Est} joins
high-scoring pairs from \texttt{blast}-searces introducing gaps (or
unkonwn fill-bases respectively) relative to the raw nucleotide
sequence if needed to obtain the correct frame for the ORF. In
\Sexpr{same.same[["FALSE"]]} TUGs the corrected sequence with the
imputed ORF was slightly changed compared to the raw sequence.
 
\subsection*{SNP analysis}
\label{sec:snp-analysis}

\SweaveInput{../snp_analysis/snp_analysis.Rnw}

<<snps.again, echo=FALSE, cache=TRUE>>=
varlog <- readLines("/home/ele/Data/454/mapping/all_vs_full_imputed_uq.varlog")
nRawSnps <- unlist(strsplit(varlog[length(varlog)], " "))[[1]]
nCov8bases <- nrow(imp.pile[imp.pile$imputed.coverage>7, ])
nbases <- nrow(imp.pile)
n.inOrf <- sum(contig.df$syn.sites, contig.df$nsyn.sites, na.rm=TRUE) 
n.outOrf <- n.inOrf-nbases

nReadSnps <- nrow(VAR)
nReadOrf <- tapply(VAR$inORF, VAR$inORF, length)
nReadFrame <- tapply(VAR$inFRAME, VAR$inFRAME, length)
tsv.read <- round(transversion.transition(VAR), 2)
tsv.read.orf <- round(do.call(rbind, by(VAR, VAR$inORF, transversion.transition)), 2)
read.dn.ds <- round(get.dn.ds(VAR), 2)

nScreeSnps <- nrow(VARq)
nScreeOrf <- tapply(VARq$inORF, VARq$inORF, length)
nScreeFrame <- tapply(VARq$inFRAME, VARq$inFRAME, length)
tsv.scree <- round(transversion.transition(VARq),2)
tsv.scree.orf <- round(do.call(rbind, by(VARq, VARq$inORF, transversion.transition)),2)
scree.dn.ds <- round(get.dn.ds(VARq),2)

nPercSnps <- nrow(VARqp)
nPercOrf <- tapply(VARqp$inORF, VARqp$inORF, length)
nPercFrame <- tapply(VARqp$inFRAME, VARqp$inFRAME, length)
tsv.perc <- round(transversion.transition(VARqp), 2)
tsv.perc.orf <- round(do.call(rbind, by(VARqp, VARqp$inORF, transversion.transition)), 2)
perc.dn.ds <- round(get.dn.ds(VARqp), 2)

snp.summary <- data.frame(c("No.SNPs", nReadSnps, nScreeSnps, nPercSnps),
                          c("in.ORF",nReadOrf[["TRUE"]], nScreeOrf[["TRUE"]], nPercOrf[["TRUE"]]),
                          check.names=FALSE)

FR <- rbind(nReadFrame, nScreeFrame, nPercFrame)
snp.summary <- cbind(snp.summary, rbind(colnames(FR), FR))

TS <- rbind(tsv.read[["ratioTS.TV"]], tsv.scree[["ratioTS.TV"]], tsv.perc[["ratioTS.TV"]])
snp.summary <- cbind(snp.summary, rbind("overall", TS))

TSf <- rbind(tsv.read.orf[["TRUE", "ratioTS.TV"]],
             tsv.scree.orf[["TRUE", "ratioTS.TV"]],
             tsv.perc.orf[["TRUE", "ratioTS.TV"]])

snp.summary <- cbind(snp.summary, rbind("ins.orf", TSf))

TSo <- rbind(tsv.read.orf[["FALSE", "ratioTS.TV"]],
             tsv.scree.orf[["FALSE", "ratioTS.TV"]],
             tsv.perc.orf[["FALSE", "ratioTS.TV"]])

snp.summary <- cbind(snp.summary, rbind("outs.orf", TSo))
snp.summary <- cbind(snp.summary, rbind("dn.ds", read.dn.ds, scree.dn.ds, perc.dn.ds))

names(snp.summary) <- c(" ", " ", "pos", "in", "codon", " ", "ti/tv ", " ", " ")

rownames(snp.summary) <- c("", "raw", "h.screened", "p.screened")

read.per.b <- nReadSnps/(nbases/1000)
scree.per.b <- nScreeSnps/(n.inOrf/1000)
perc.per.b <- nPercSnps/(n.inOrf/1000)

total.syn <- sum(contig.df[, "syn.sites"], na.rm=TRUE)
total.cov8.syn <- sum(contig.df[, "cov8.syn.sites"], na.rm=TRUE)

total.nsyn <- sum(contig.df[ , "nsyn.sites"], na.rm=TRUE )
total.cov8.nsyn <- sum(contig.df[ , "cov8.nsyn.sites"], na.rm=TRUE )

total.nSNP <- sum(contig.df$Nonsynonymous, na.rm=TRUE)
total.sSNP <- sum(contig.df$Synonymous, na.rm=TRUE)

s.per.s.base <- total.sSNP/(total.syn/1000)
s.per.s.cov8.base <- total.sSNP/(total.cov8.syn/1000)

n.per.n.base <- total.nSNP/(total.nsyn/1000)
n.per.n.cov8.base <- total.nSNP/(total.cov8.nsyn/1000)

@ 

In order to improve SNP-calling we screened a set of \Sexpr{nRawSnps}
candidate-SNPs predicted on in total \Sexpr{nbases} bases
(\Sexpr{nCov8bases} bases with a minimum coverage of more then 8-fold
available for SNP-calling). We first excluded SNPs predicted to be
present as more than 2 alleles and to undetermined (``N'') base in the
reference retaining \Sexpr{nReadSnps} SNPs. The ratio of transitions
(\Sexpr{tsv.read[[1]]}) to transversion (\Sexpr{tsv.read[[2]]}) in
this set was \Sexpr{round(tsv.read[[3]],2)} (ti/tv).

To further screen those SNPs we utilized observations made in
combination with protein prediction, i.e. with the inferred
open-reading frame. As noted above SNPs at unknown fill-bases had to
be disregarded, but the imputation of a continuous corrected ORF had
the advantage of making the coordinates for the ORF straight forward
to use.

\Sexpr{nReadOrf[["TRUE"]]} of the raw SNPs were predicted to be inside
an ORF, \Sexpr{nReadFrame[["1"]]} at a first positon in a codon,
\Sexpr{nReadFrame[["2"]]} on a second base and
\Sexpr{nReadFrame[["3"]]} on a third base in a codon.

As expected ti/tv inside ORFs was with
\Sexpr{tsv.read.orf["TRUE","ratioTS.TV"]} higher than outside ORFs
with \Sexpr{tsv.read.orf["FALSE","ratioTS.TV"]}.

To asses the effect of a mutation at a given base we used an idea of
Mark Blaxter (unpublished, some information at \cite{Marks_ontology}),
to classify every base as to its ``response'' to mutation. We used
this classification to compute the number of synonymous and
non-synonymous sites in a given TUG. In total \Sexpr{total.nsyn}
non-synonymous and \Sexpr{total.syn} sites were found in ORFs, of
these \Sexpr{total.cov8.nsyn} non-synonymous sites and
\Sexpr{total.cov8.syn} synonymous sites were covered at least 8-fold,
and thus available for SNP-calling.

For the raw set of candidate-SNPs the ratio of synonymous
polymorphisms per synonymous site to non-synonymous polymorphisms per
non-synonymous site (dn/ds) was \Sexpr{round(read.dn.ds, 2)}.

To improve SNP-calling with respect to the well-known homopolymer
issues in 454-sequencing\cite{pmid21685085}, we analysed the effect of
exclusion of SNPs in, or close to, homopolymer regions. We observed
changes in ti/tv and in dn/ds when SNPs were discareded due to
different sizes thresholds for homoploymer-runs and different
proximity thresholds (see figure snp\_ex\_parameter\_plots.png). Based
on this we decided to exclude SNPs with a homopolymer-run as long as
or longer than 4 bases inside a window of 11 bases (5 to bases to the
right, 5 to the left) around the SNP.

After these screening steps based on a sequence-features we
investigated the effect of data volume and mapping on the remaining
candidate SNPs. Based on a slight overrepresentation of SNPs on first-
and second-positions in the ORF and on a little bigger than average
excess of non-synonymous polymorphisms in SNPs with a percentage of
the minority allele smaller than 7\% (see figure
snp\_pos\_eff\_plots.png a and b) we examined exclusion of SNPs at
this percentage threshold. Evaluation of an exclusion of SNPs based on
coverage at the SNP-site, did not seem necessary (see figure
snp\_pos\_eff\_plots.png c and d).

Table 7 gives on overview of how the basic SNP-statistics described
above changed with screening of the candidate SNPs. The change of
ti/tv back to lower values, when SNPs were screeened based on a 7\%
coverage threshold, left the benefit of this screening-step
questionable.

However, calculating dn/ds on a per contig base, the screening based
on percentage threshold of the minority-allele showed it's benefits:
figure dn\_ds\_scales.png b shows how for the unscreened SNP-set dn/ds
scaled with the covereage of a contig. This correlation was not longer
present if the percentage screening was used (see figure
dn\_ds\_scales.png d, the linear model had a slightly negative
non-significant slope). For both the screened and unscreened sets of
SNPs there was a significant slope for the number of SNPs in a contig
predicting dn/ds (see figure dn\_ds\_scales.png a and c).

Figure dens\_dn\_ds.png gives the disribution of per-contig dn/ds for
the fully screened set of SNPs. The final numbers of SNPs per
kilo-base was \Sexpr{round(perc.per.b,2)},
\Sexpr{round(s.per.s.cov8.base,2)} synonymous SNPs per 1000 synonymous
bases and \Sexpr{round(n.per.n.cov8.base, 2)} non-synonymous SNPs per
1000 non-synonymous bases. For a total of
\Sexpr{nrow(subset(contig.df,!is.na(dn.ds)))} TUGs
(\Sexpr{nrow(subset(contig.df,!is.na(dn.ds)&category%in%"MN"))} high
  quality contigs) a value for dn/ds could be obtained, because at
  least one synonymous SNP was found.

  It should be noted that the overall dn/ds value of
  \Sexpr{perc.dn.ds} is different from the mean dn/ds value of contigs
  \Sexpr{mean(contig.df$dn.ds, na.rm=TRUE)#$}, as in the first the
  complete length of the sequence (also frome TUGs without synonymous
  SNP predicted) can be included.

\subsection*{Annotation}

\SweaveInput{../annotation/annotation.Rnw}
\SweaveInput{../annotation/read_signalp.Rnw}

<<annot, echo=FALSE, results=hide>>=
GOannotations <- unique(GO.annot$pept_id)
nGO <- length(GOannotations)

ECannotations <- unique(EC.annot$pept_id)
nEC <- length(ECannotations)

KEGGannotations <- unique(KEGG.annot$pept_id)
nKEGG <- length(KEGGannotations)

## IPRannotations <- unique(IPR.annot[grepl("GO:", IPR.annot$GO_descr), "pept_id"])
## nIPR <- length(IPRannotations)
## nIPRonly <- length(IPRannotations[!IPRannotations%in%GOannotations &
##                                   !IPRannotations%in%ECannotations &
##                                   !IPRannotations%in%KEGGannotations
##                                   ])

Allannotations <- unique(c(GOannotations, ECannotations, KEGGannotations))#,  IPRannotations))
nallannotations <- length(Allannotations)

venn.diagram(list(GO   = match(GOannotations, Allannotations),
                  EC   = match(ECannotations, Allannotations),
                  KEGG = match(KEGGannotations, Allannotations)),
                  ##                  IPR  = match(IPRannotations, Allannotations)),
                  filename = "/home/ele/thesis/454/figures/annotataionVenn.tiff")


Signn <- contig.df[contig.df$sigp.nn%in%"Y", "contig" ]
Sighmm <- contig.df[contig.df$sigp.hmm%in%"S", "contig" ]
Anchmm <- contig.df[contig.df$sigp.hmm%in%"A", "contig" ]

nSignn <- length(Signn)
nSighmm <- length(Sighmm)
nAnchmm <- length(Anchmm)

venn.diagram(list(nn.Signal   = match(Signn, contig.df$contig),
                  hmm.Signal   = match(Sighmm, contig.df$contig),
                  hmm.Anchor = match(Anchmm, contig.df$contig)),
                  filename = "/home/ele/thesis/454/figures/signalVenn.tiff")

@ 

We obtained annotations with GO-terms for \Sexpr{nGO}, with EC-numbers
for \Sexpr{nEC} and with KEGG-pathways for \Sexpr{nKEGG} TUGs using
the \texttt{Blast}-based program annot8r.

Figure (annotataionVenn.tiff) gives an overview of the
\Sexpr{nallannotations} TUGs annotations were found for with
different methods.

\SweaveInput{../annotation/brugia-compare.Rnw}

<<<<<<< HEAD
We compared GO-annotations for high-level GO-slim terms to the
annotations obtained the same way for all \textit{B. malayi}-proteins:
Our transcriptome for \textit{A. crassus} shows a remarkably similar
distribution of GO-terms to the full set of annotations for the genome
of the related nematode (see figure go\_bm\_com.png).

=======
>>>>>>> 0f447f595138cc7b21e7f97bd8049cfdf088d62f
In additon we inferred presence of signal peptide cleavage sites in
the predicted protein sequence: \texttt{SignalP} \cite{pmid17446895}
predicted \Sexpr{nSignn} signal peptides using neural networks and
\Sexpr{nSighmm} and \Sexpr{nAnchmm} signal peptides and signal anchors
respectively unsing hidden Markov models \cite{pmid9783217}.
<<<<<<< HEAD
=======
 

>>>>>>> 0f447f595138cc7b21e7f97bd8049cfdf088d62f

\subsection*{Consolidated results}

\SweaveInput{../snp_analysis/overrepresentation.Rnw}

<<nn.dn.ds, echo=FALSE, results=hide>>=

nn.box <- ggplot(dn.ds.df, aes(dn.ds.df$sigp.nn, dn.ds.df$dn.ds)) + 
  geom_boxplot() + 
  scale_y_log10("dn/ds") +
  scale_x_discrete("", 
                   breaks=levels(dn.ds.df$sigp.nn),
                   labels=c("No signal peptide",
                     "Signal peptide")) +
  opts(title = "SignalP neural networks prediction",
       axis.title.x = theme_text(size = 15), 
       axis.text.x = theme_text(size = 15))

  
hmm.box <- ggplot(dn.ds.df, aes(dn.ds.df$sigp.hmm, dn.ds.df$dn.ds)) + 
  geom_boxplot() + 
  scale_y_log10("dn/ds") +
  scale_x_discrete("", 
                   breaks=levels(dn.ds.df$sigp.hmm),
                   labels=c("Signal anchor",
                     "No signal peptide",
                     "Signal peptide")) +
  opts(title = "SignalP hmm prediction",
       axis.title.x = theme_text(size = 15), 
       axis.text.x = theme_text(size = 15))
dev.off()

png("../figures/sigp_dn_ds.png", width=2000, height=1000, res=144)

# Set up the page
grid.newpage()
pushViewport(viewport(layout = grid.layout(1, 2)))
vplayout <- function(x, y)
    viewport(layout.pos.row = x, layout.pos.col = y)

# Make each plot, in the correct location
print(nn.box, vp = vplayout(1, 1 ))
print(hmm.box, vp = vplayout(1, 2 ))

dev.off() 
@ 

\subsubsection*{Signal-positives have higher dn/ds}

TUGs predicted to contain signal peptide cleavage sites by the
neural-netowrks method in \texttt{SignalP} have higher dn.ds values
than TUGs without signal peptide cleavage sites (p =
\Sexpr{round(u.test[["p.value"]], 3)}; two sided U-test; see also
figure sigp\_dn\_ds.png a). TUGs predicted by the hmm method in
\texttt{SignalP} to contain a signal peptide have higher dn/ds than
TUGs predicted to contain no signal peptide or a signal anchor (p =
\Sexpr{round(t.test.hmm$p.value, 3)#$}; Two sided U-test; see also
figure sigp\_dn\_ds.png b). Although the differences were only
significant for the nn-predicted signal-peptides the results of the
hmm prediction show that the signal-peptides that are cleaved and
possibly secreted are making the difference. These differences were
consistent if only tested against a resticted dataset surely from
\textit{A.crassus} (see ).


\subsubsection*{Enrichement of GO-categories in high dn/ds}

In enrichment analyses, where the ``gene-universe'' to test against
has to be defined, possible off-target sequences have to be excluded
prior to analyses. Testing was performed against a universe of only
TUGs we were sure to be of \textit{A.crassus}-origin. We defined TUGs
with a dn.ds higher than 0.5 as divergently selected and tested each
node-term in the ontology for a overrepresentation in this set. Tables
X give an overview of enriched terms for different parts of the
ontology.

The terms ``amino acid transmembrane transporter activity'' and
``peptidase activity'' were among the overrepresented terms for
``molecular function''. Underreprsented were on the other hand terms
associated with ribosomal proteins and transcription (visible in
``molecular function'' and ``cellular compartment'') .

TODO: BP and CC overrepresentation. 

To make sure these inferrences were not biased by redundancy in the
fullest dataset we repeated the analysis using only the good-quality
contigs as gene-universe. The results showed to be consistent with
those obtained for the fullest datset (data not shown).

\subsubsection*{Novel in cladeIII have elevated dn/ds}

<<novel.dn.ds, echo=FALSE, results=hide>>=
novel.dn.ds.50 <- ggplot(subset(contig.df, !is.na(novel.50) & !is.na(dn.ds)),
                         aes(novel.50, dn.ds)) +
  geom_boxplot() +
  scale_y_log10("dn/ds") +
  scale_x_discrete("evolutionary conseravation at bitscore threshold of 50")

novel.dn.ds.80 <- ggplot(subset(contig.df, !is.na(novel.80) & !is.na(dn.ds)),
                                aes(novel.80, dn.ds)) +
  geom_boxplot() +
  scale_y_log10("dn/ds") +
  scale_x_discrete("evolutionary conseravation at bitsore threshold of 80")

# Set up the page

png("../figures/conservation_dn_ds.png", width=2000, height=1000, res=144)
grid.newpage()
pushViewport(viewport(layout = grid.layout(1, 2)))
vplayout <- function(x, y)
    viewport(layout.pos.row = x, layout.pos.col = y)

# Make each plot, in the correct location
print(novel.dn.ds.50, vp = vplayout(1, 1 ))
print(novel.dn.ds.80, vp = vplayout(1, 2 ))

dev.off() 

@ 

<<novel.dn.ds.test, echo=FALSE, results=hide, cache=TRUE>>=
dn.ds.df$novel.50 <- as.factor(dn.ds.df$novel.50)
dn.ds.df$novel.80 <- as.factor(dn.ds.df$novel.80)

novel.50.ndwd <- get.ndwd(dn.ds.df, dn.ds.df$dn.ds, dn.ds.df$novel.50)
novel.50.kw.ph <- pvalue(novel.50.ndwd, method = "single-step")

novel.80.ndwd <- get.ndwd(dn.ds.df, dn.ds.df$dn.ds, dn.ds.df$novel.80)
novel.80.kw.ph <- pvalue(novel.80.ndwd, method = "single-step")


u.novel.50 <- wilcox.test(dn.ds.df$dn.ds~
                          as.factor(dn.ds.df$novel.50%in%"novel.in.clade3"))
u.novel.80 <- wilcox.test(dn.ds.df$dn.ds~
                          as.factor(dn.ds.df$novel.80%in%"novel.in.clade3"))

@ 

Figure conservation\_dn\_ds.png shows differences in dn/ds across the
categories defined for evolutionary conservation. At a bitscore
threshold of 80 sequences novel in clade III had a significantly
higher dn/ds than other sequences (p =
\Sexpr{round(u.novel.80[["p.value"]],3)}; two sided U-test). At a
bitscore thershold of 50 results were non-significant but showed the
same trend.

\subsubsection*{Sequences novel in nematodes are enriched for
  Signal-positives}

<<novel.nn, echo=FALSE, results=hide>>=

nn.50 <- table(contig.df$novel.50, contig.df$sigp.nn)
nn.80 <- table(contig.df$novel.80, contig.df$sigp.nn)

hmm.50 <- table(contig.df$novel.50, contig.df$sigp.hmm)
hmm.80 <- table(contig.df$novel.80, contig.df$sigp.hmm)


nn.50.p <- ggplot(subset(contig.df, !is.na(novel.50) & !is.na(sigp.nn)),
              aes(novel.50, fill=sigp.nn)) +
       geom_bar(position="fill") + coord_polar(theta="y") +
  scale_y_continuous("evolutionary conseravation categories at bitsore threshold of 50") +
  scale_x_discrete("") +
  opts(title="proportion of sequences in SignalP-nn category")


nn.80.p <- ggplot(subset(contig.df, !is.na(novel.80) & !is.na(sigp.nn)),
              aes(novel.80, fill=sigp.nn)) +
  geom_bar(position="fill") + coord_polar(theta="y") +
  scale_y_continuous("evolutionary conseravation categories at bitsore threshold of 80") +
  scale_x_discrete("") +
  opts(title="proportion of sequences in SignalP-nn category")

hmm.50.p <- ggplot(subset(contig.df, !is.na(novel.50) & !is.na(sigp.hmm)),
                   aes(novel.50, fill=sigp.hmm)) +
  geom_bar(position="fill") + coord_polar(theta="y") +
  scale_y_continuous("evolutionary conseravation categories at bitsore threshold of 50") +
  scale_x_discrete("") +
  opts(title="proportion of sequences in SignalP-hmm category")

hmm.80.p <- ggplot(subset(contig.df, !is.na(novel.80) & !is.na(sigp.hmm)),
                   aes(novel.80, fill=sigp.hmm)) +
  geom_bar(position="fill") + coord_polar(theta="y") +
  scale_y_continuous("evolutionary conseravation categories at bitsore threshold of 80") +
  scale_x_discrete("") +
  opts(title="proportion of sequences in SignalP-hmm category")

png("../figures/signal_novel.png", width=2000, height=2000, res=144)
grid.newpage()
pushViewport(viewport(layout = grid.layout(2, 2)))
vplayout <- function(x, y)
    viewport(layout.pos.row = x, layout.pos.col = y)

# Make each plot, in the correct location
print(nn.50.p, vp = vplayout(1, 1 ))
print(nn.80.p, vp = vplayout(1, 2 ))
print(hmm.50.p, vp = vplayout(2, 1 ))
print(hmm.80.p, vp = vplayout(2, 2 ))

dev.off() 
@ 

Figure signal\_novel.png gives the proportions of
\texttt{SignalP}-predictions for each category of evolutionary
conservation. Generally - across bitscore thresholds and
\texttt{SignalP} prediction methods - sequences novel in nematodes
containt the highest proportion of signal-positives followed by
sequences novel in clade III or in metazoa. Conserved sequences
contain the lowest proportion of signal-positives.

\subsection*{Differential expression}
                
\SweaveInput{../gene_expression/gene_expression.Rnw}

Using methods developed for sequencing data, we analysed
gene-expression inferred from mapping. Of the \Sexpr{n454.map} reads
mapping to the fullest assembly \Sexpr{n454.Umap} mapped uniquely
(with teir best hit) and were counted on a per library base.

\subsection*{Comparison with tag-sequencing pilot dataset}

\SweaveInput{../solexa_compare/solexa_compare.Rnw}

\Sexpr{nMapped} of \Sexpr{nTags} (\Sexpr{nUniq} unique) NlaIII-tags
mapped to the fullest assembly. Only \Sexpr{nUMapped}
(\Sexpr{nUniqUMapped} uniqe) tags did not map to any sequence in the
fullest assembly.

Table 9a gives correlations coefficients between tag-counts and
454-libraries. Perason-correlations-coefficients between 454-libraries
were generally low, indicating a high proportion of noise or
biological differences between samples. Correlation between
expression-tags and 454-read counts were even lower. However when only
analysing counts to good-quality contigs, correlation coefficients
improved both between libraries and between 454-libraries and
solexa-tags (see table 9b). No further improvements were made, when
counts were limited to contigs surely \textit{A.crassus} (see table
9c). Correlations between library T2 and other 454-libraries, as well
as with solexa-tag counts were lower than between other libraries.  

To gain power in statistical analysis we limited the set of
gene-objects analysed for differential expression to the good-quality
contigs.

\subsubsection*{Differential expression between male and female worms}

Despite the lack of replicates for male worms we were able to identify
\Sexpr{length(sig.mf.contigs)} sequences being significatnly
over-exprssed im male worms. In fact all these TUGs were nearly
exclusively expressed in males. 

\subsubsection*{Differential expression between adults and L2-larvae}

For the L2-library we changed our approach and used gene-expression
analysis rather to highlight the off-target data in this lirary. For
this reason we used counts for the fullest assembly.

\Sexpr{length(sig.ad.contigs)} sequences being expressed exclusively
in L2 library were strongly enriched in sequences being labled as
possible off-target data in taxonimic classification. From these
sequences only \Sexpr{l2.met.hits} had best hits to metazoa and only
\Sexpr{l2.nem.hits} to nematoda.

\subsubsection*{Differential expression between worms from the European
  and Japanese eel}

None of the TUGs in the present evaluation showed significant
differential expression between worms from the European and Japanese
Eel. Diagnostic plots provided by \texttt{DESeq} made clear, that both
depth of sequencing and number of replicates have to be higher
contrasting these conditions.

However, comparing expression-analysis on the full dataset to analysis
limited to the high quality of relieable \textit{A.crassus}-contigs it
was clear that the quality-dataset reduces within-condition variance
and results were closer to significance: The lowest adjusted p-values
for the cleaned dataset were around 0.4, while on the full dataset
only adjusted p-values above 0.8 could be obtained.

\subsection*{Secreted immunomodulatory molecules}

We further highlight a list of genes with similarities to known
immunumodulators. Should I???

<<save, echo=FALSE>>=
## for a more readable version of contig.df
readable <- !names(contig.df)%in%c("seq", "imp", "ontology", "coding")
save.image("paper.Rdata")
write.csv(contig.df, "../A_crassus_contigs_full.csv")
write.csv(contig.df[,readable], "../A_crassus_contigs_readable.csv")

@ 

\section*{Discussion}

We are providing transcriptome-data for the parasite
\textit{A.crassus}, enabeling a broad spectrum of molecuar research on
this ecologically and economically important species.

We emphasise the importance of screening for xenobiotics. We consider
this aspect important in any deep transcriptome project. First the
depth of sequencing is leading to the generation of large amounts of
off-target data from a ``metatranscriptomic community'' associated with a
target organism. Second due to the abundance of laboratory
contaminations and the possiblity of cross-contamination if libraries
are sequenced only on a subset of a picotiter-plate (i.e. without the
use of barcodes distinguishing between samples \cite{pmid20137071})
non-biological contamination can be introduced. 
However, in the context of a parasite (or an infected host) the
screening for off-target data and contamiation becomes even more
important: Correct inferrence of biological origin for a given contig
constitues a prerequesite for most downsteam analysis or the
interpretation of results.

Corss-contamination from different compartments of a picolitre-plate
was ruled out by our sequence provider, using Multiplex Indexes (MID)
for one library and similarity searches to neighbouring lanes for the
other libraries.

For the remaining off-target and contamination problem we archived
seperation of sequences in two steps, one before assembly, one
afterwards. Both screening-steps had to rely soly on sequence
comparison. The screening-step before assembly has to employ lower
stingency as sequence comparisons on sequence as short as reads are
less informative than on longer contig-sequence. In our case of
\textit{A. crassus}, neither of the two host species has genomic data
available for use in similarity searches. A publicly available
transcriptome-dataset for European eel \cite{pmid21080939} in addition
to a unpublished dataset for the same species was augmented with a
dataset generated from the Japanese eel sequenced for the purpose of
screening \textit{A.crassus}-sequences in the present project. The
pre-assembly screening had the rationale of faciliating the assembly
process reducing the amount of divergent sequence from two
host-species and the amount of extensively covered rRNA sequence. In
our sequencing we were not able to reproducibly alleviate the rRNA
coverage. This has probably been due to the fact that extraction of
total-RNA from worms filled with host blood resulted in low amounts of
starting material, and amplification using standard kits did not allow
to reproducabely alleviate rRNA abundance. As the same problems
existed in preparation of liver tissue of the host species it seems
likely that the blood of eels contains substances limiting the success
of specific amplification protocols. In fact it is known that
compounds like hemoglogin can inhibit PCR reactions \cite{pmid9327537}
and reverse transcription \cite{pmid16109794}.

Although raw reads with rRNA hits were screened out prior to assembly,
it was still possible to gain insights from these off-target data, as
we assembled and annotated screening databases. Some of the rRNA data
especially from the L2 library showed high similarity to flaggelate
eukaryotes. It could be possibly derived from an unknown protist
living in the swimbladder of eels (possibly as a commensal of
\textit{A.crassus}), from where the L2 larvae for RNA-preparation were
washed out. This seems worth further investigation, especially as it
has been controversal wether encapsulated objects in the swimbladder
of eels could be attributed soly to \textit{A.crassus}
\cite{heitlinger_massive_2009} or to opportunist coinfections. 

A second examination of seqeunce origin was obtained after
assembly. It can employ higher stringency do to the longer sequence
length in an assebled dataset and because we did not discard any
off-target data, but marked TUGs for careful selection prior to
special analyses or differential interpretation afterwards. Such
attempts to employ taxonomic screening were used before in a
transcriptome projects for the garter snake \cite{pmid21138572} and a
study on lake sturgeon even evaluated horizontal gene-transfer, when
xenobiotic sequences were found \cite{pmid20386959} (with a negative
result). A study describing a custom pipline for
transcriptome-assembly from pyrosequencing reads \cite{pmid20034392}
suggested the use of \texttt{EST3} \cite{pmid17218127}, to inferre
sequence-origin based on nucleotide frequency. However, we were not
able to use this approach successfully, probably due to the fact that
xenobiotic sequence in our dataset stems from multiple sources with
different gc-content and codon-usage.

Data from our L2 larvae library showed its anormaly later in
gene-expression analysis, when off-target xenobiotic data was found to
be responsible for the differences to other libraries. This makes this
off-target data an interesting starting point for future investigation
of the species community living in infected swimbladders of eels.

Compared to other NGS transcriptome sequencing projects, our assembly
approach generated a smaller number of contigs. Projects using the
\texttt{mira} assembler often report substantially more contigs for
data-sets of similar size (see e.g. \cite{pmid21364769}), comparable
to the \texttt{mira} sub-assebly in our appoach. We demonstrated that
our assembly approach generated a set less redundant and more complete
sequences compared to the two ``out of the box'' approaches tested.

Protein prediction showed a trend towards completeness and elevated
coverage on the 3’ end of transcripts, as a result of RNA preparation
using oligo-dT primers, an effect that seems to be ubiquitous in deep
transcriptome sequencing projects (e.g. \cite{pmid20331785}). The low
number of complete ORFs and especially the low number of start-codons
(and corresponding Methoinin amino acids) in predicted proteins seems
cumbersome but cross-validation in combination with SNP-calling
demonstrated the overall correctness of predicted proteins.

We were able to demonstrate, that screening of SNPs in or adjacent to
hompolymer regions ``improved'' overall measurements on SNP-quality:

First the ratio of transitions to transversions (ti/tv) increased.
Such an increase is explainable by the removal of ``noise'' associated
with common homopolymer-errors \cite{pmid21685085}. Assuming that
errors would be independent of transversion-transition bias erroneous
SNPs would have a ti/tv of 0.5 and thereby lower the overall
value. Other explanations for these observations are hard to find so
it can be concluded that removing noise from homoploymer
sequencing-error ti/tv increases.  The value of \Sexpr{tsv.perc[,3]}
(\Sexpr{tsv.perc.orf["FALSE",3]} outside,
\Sexpr{tsv.perc.orf["TRUE",3]} inside orfs) is in good agreement with
the overall ti/tv of humans (2.16 \cite{pmid21169219}) or
\textit{Drosophila} (2.07 \cite{pmid21143862}). 

The ratio of non-synonymos SNPs per non-synonymous site to synonymous
SNPs per synonymous site (dn/ds) decreased with removal of SNPs
adjacent to homoploymer regions from \Sexpr{read.dn.ds} to
\Sexpr{perc.dn.ds} after full screening. Similar to ti/tv it the most
plausible explantion is the removal of error, as unbiased error would
lead to a dn/ds of 1. While dn/ds is not unproblematic to interpret
within populations \cite{pmid19081788}, assuming negative (purifying)
selection on most protein-coding genes lower values seem more
plausible, also in comparison with other studies (see further text).

We used a threshold value for the minorty allesl of 7\% for exclusion
of SNPs, this corresponds to the ca. 10 ``haploid equivalents'' (5
individual worms plus an neglegible amount of L2 larvae - in the L2
library and within the female adult worms - bearing possibly
additional diversity). It is hard to explain, that ti/tv decreased in
this filtering step, while dn/ds still further decreased.

The benefit of this screening was mainly a reduction of non-synonymous
SNPs in high coverage contigs. When it was applyied dn/ds did not
scale with coverage. Working with an estimate of dn/ds independent of
coverage, efforts to control for sampling a biased by samling depth
(i.e. coverage) like developed \cite{pmid18590545} and used
\cite{pmid20478048} could be avoided.  

When the whole of coding sequences are studied, of which only a small
subset of sites can be under diversifying selection, dn/ds of ~0.5 has
been suggested as threshold for assuming diversifying selection
\cite{pmid15579698} instead of the classical threshold of 1
\cite{pmid6449605}. In the transcripts from the female reproductive
tract of \textit{Drosophila} dn/ds was 0.15 \cite{pmid15579698} and in
th 0.21 male reproductive tract \cite{pmid11404480} (although for ESTs
specific to the male accesory gland were shown to have a higher dn/ds
of 0.47). Pyrosequencing studies found dn/ds to be between 0.13 and
0.27 (depending on tissue type genes were mainly expresse in) in the
Zebra finch transcriptome \cite{pmid20359325}, 0.12 in the
transcriptome of \textit{Tigriopus californicus} \cite{pmid21199025}
and 0.3 in the parasitic nematode \textit{Ancylostoma caninum}
\cite{pmid20470405}. In comparison with these results even our
estimate after screening seems high (although it should be noted, that
the latter studies report a mean dn/ds over contigs - the
\textit{A. canium} doesn't make clear what exactly is reported - and
therefore the value has to be compared to our mean dn/ds over contigs
of \Sexpr{round(mean(contig.df$dn.ds,na.rm=TRUE),2)#$}) and further
investigation using deeper sequencing of more individuals on the
solexa GAII platform will be used to fully exclude the possiblily of
this reslut being induced by sequencing error. Moreover such an
experiment should try to test that divergence between populations is
leading to positive selection on only the possibly diverging European
populations. For such a study the set of SNPs found here are
invaluable, as it can be used to define a gold standard set of SNPs
found with both technologies.

We were able to obtain high-quality annotations for a large set of
TUGs. Comparison with protein sequence derived from \textit{B. malayi}
showed a a remarkable degree of agreement regarding the occurence of
terms. This implies, that our transcriptome-dataset is a
representative subset of a nematode-parasite
genome. 

Overrepresentation of GO-term in genes under diversifying selection
(at a threshold of dn/ds>0.5, as established above) highlighted many
interesting gene-products:

In the molecular function category two amino acid transmembrane
transporters (``Contig5699'' and ``Contig866'') - the only contigs
with this annotation (or anotation, which is an offspring-term of
this) and a dn/ds obtained - were found to have a dn/ds>0.5. Such
transporter are thought to be important in the survival of parsites in
a host \cite{pmid18272295}.

Enrichment in the category ``peptidase activity'' highlighted twelve
peptidases (from 43 with a dn/ds obtained). All twelve have orthologs
in \textit{B.malayi} and \textit{C.elgans} and are conserved across
kingdoms. Despite their conservation peptidases are thought to have
have accquired new and prominent roles in host-parsite interaction
compared to free living organisms: In \textit{A.crassus} a
trypsin-like proteinase has been identified thought to be utilized by
the tissue-dwelling L3 stage to penetrate host tissue and an aspartyl
proteinase thought to be a digestive enzyme in adults
\cite{polzer_identification_1993}.

The under-represetation of ribosomal proteins (term ``structural
constituent of ribosome'') in disruplingly selected contigs is in good
agreement with the notion that ribosomas proteins are extremely
conserved across kingdoms \cite{pmid9664699} and should be under under
strong negative selection. 

The additional prediction of signal sites for cleavage allowed
interpretation and cross-validation of the results from SNP-calling:
The detection of signal-peptides secretion using \textit{in silico}
analysis of ESTs has been used to highlight candidate genes for
example in \textit{Nippostrongylus brasiliensis}
\cite{harcus_signal_2004} and in a large scale anaylsis across all
nematode \cite{nagaraj_needles_2008} ESTs. Proteomic analysis in
\textit{Brugia malayi} \cite{pmid19352421,pmid18958170} and
\textit{Heligmosomoides polygyrus} \cite{pmid21722761} was able to
find evidence for excretion for some of the protein-products and to
highlight additional candidate genes.

We found an elevated dn/ds for signal-positves. These result could be
explained follow the logic of signal-positives being more likely to be
secreted to the host-parasite interface and proteins involved in
host-parasite interaction being more likely to be under disruptive
selection. Signal-positive TUGs with high dn/ds constitute another set
of genes worth further examination in future studies.

TUGs predicted to be novel in the phylum nematoda contained the
highest proprotion of signal-positives. A interpretation of this
findings could be a confirmation of a study on \textit{Nippostrongylus
  brasiliensis} \cite{harcus_signal_2004}, where signal positives were
reported as less conserved. In the present study we did not aim to
identify ``novelity to \textit{A.crassus}'' as we believe in a deep
sequencing project he absence of sequence similarity could be
attributed to erroneous sequence instead of true novelity, and thereby
blur analysis. However novelity in nematodes and to a lesser extend
novelity in Spirurina seems to support the notion, that - if not
diversified within nematoda to an extend leading to a complete loss of
similarity, like suggested in the mentioned study - signal positives
in nematodes could have taken a divergent evlutionary path from their
orthologs in other phyla.

It was within our expectation, that expression analysis failed to give
conclusive results, as the prestent dataset is not fully adequate for
this kind of analysis: First we did not include replictes for
libraries of male adults as well as for L2-larvae. Second one of the
replicates for female worms (library E1) resulted in a low amount of
sequence mappable to protein-coding (non-rRNA) genes.
However some of the results are still valueable:

DESeq was able to report genes significantly differening in expression
between male and female worms and betweeen the L2 library and the all
other worms. This was possible for male worms as well as for
L2-larvae, were no replicated samples were obtained, due due the
special features of this package \cite{pmid20979621}. However only
over-expression in non-repeated samples can be detected, as obviousely
lack of expression in one sample can't validate

Comparisons were lacking significance, as methods are designed for
deeper sequencing and more importantly more replicates would be
needed. Differences between the L2-library and other libraries were
mainly due to off-target data, and TUGs soly found in the L2 library
are ... 



%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}

  
%%%%%%%%%%%%%%%%%%
\section*{Methods}
\subsection*{Worm samples, RNA extraction, cDNA synthesis and
  Sequencing}
\textit{A.crassus} from Japanese eels were sampled from Kao-Ping river
and an adjacent aquaculture in Taiwan as described in
\cite{heitlinger_massive_2009}. Worms from the European eel were
sampled in Sniardwy Lake, Poland (53.751959N ,21.730957E) and from the
Linkenheimer Altrhein, Germany (49.0262N; 8.310556E). After
determination of the sex of adult nematodes, all worms were stored in
RNA-later (Quiagen, Hilden, Germany) until extraction of RNA. RNA was
extracted from:

\begin{itemize}
\item one worm from a  cultured Taiwanese eel (sample T1)
\item one worm from a wild Taiwanese eel (sample T2)
\item one worm from an eel from the German sampling site (E1)
\item one worm an eel from Polish sampling site (E2)
\end{itemize}

In addition RNA was extracted from L2-Larve from the German sampling
site (sample L2) and a from a male worm from the Taiwanese aquaculture
(sample M). RNA was reverse transcribed and amplified into cDNA using
the MINT-cDNA synthesis kit (Evrogen, Moscow, Russia).

For host-contamination screening a liver-sample from an uninfected
Japanese eel was prepared using the methods as described above for
\textit{A.crassus} samples.

A emulsion PCR was performed for each cDNA library according to the
manufacturer's potocol (Roche/454 Life Sciences). For library E1 a
Multiplex Index (MID) (Roche/454 Life Sciences) was used in
preparation of the sequencing adapter. The libraries were sequenced in
different runs of the Roche/454 Genome Sequencer FLX System: Library
T2 on an eighth plate of the instrument usning standard
``FLX-chemistry'', the remaining libraries on a eighth of a plate
using ``FLX-Titanium-chemistry''.

\subsection*{Trimming, quality control and assembly}

Raw sequences were extracted in fasta format (with the corresponding
qualities files) using sffinfo (Roche/454) and screened for adapter
sequences of the MINT-amplification-kit using cross-match \cite{PHRAP}
(with parameters -minscore 20 and -minmatch 10). Seqclean
\cite{tgicl_pertea} was used to screen poly-A-tails, low quality,
repetitive and short (<100 bases) sequences. In addition all reads
were \texttt{blast}ed (1e-5 -F F) against the following databases:

\begin{itemize}
\item a combined eel-mRNA database consisting of an assembly of
  sequences from the liver of the Japanese eel sequenced for this
  purpose (as described above), a sequence assembly of unpublished
  (sanger-) ESTs (made available to us by Gordon Cramb; University of
  St Andrews) and from EeelBase \cite{pmid21080939} a publically
  availble transcriptome database for the European eel.
\item a eel-rRNA database from a rRNA screening of the above and
  assembly together with publically available rRNA-sequences.
\item an \textit{A.crassus} rRNA-database from screening of our
  dataset against nematode-rRNA, and assembly of these rRNA
  reads. This database notably also contained xenobiotic rRNA
  sequences.
\end{itemize}

Reads mapping to one of the databases with more than 80\% of their
length and 95\% identity were removed from the dataset. Screenig and
trimming information was written back into sff-format using
\texttt{sfffile} (Roch/454).

We used an approach proposed by Kumar and Blaxter\cite{pmid20950480},
combining assemblies from the \texttt{mira} \cite{miraEST} and
\texttt{newbler} \cite{pmid16056220}. Briefly the two assemblies are
combined into one using Cap3\cite{Cap3_Huang} and only contigs
supported by both assemblers are regarded good quaility. For further
details see the supplementary methods.

\subsection*{Post assembly classification and taxonomic assesment}

After assembly contigs were assesed a second time for
host-contamination and other xenobiotics:

The contigs were \texttt{blast}ed (with a cut-off 1e-5) against the
same databases used prior to assembly (Eel-mRNA, Eel-rRNA,
\textit{A.crassus}-rRNA and additionally against the nucleotide
version of nempep4 \cite{parkinson_nembase:resource_2004,
  pmid21550347}, determining the best hit across databases. These best
hits across databases were screened and only such hits involving more
then 50\% of the 

Additionally \texttt{blast} (\texttt{blastn} e-value cut-off 1e-5)
against NCBI-nt and (\texttt{blastx} e-value cut-off 1e-5) against
NCBI-nt was used to determine taxon-membership of the top hit at the
family, phylum and kingdom rank.

\subsection*{Protein prediction and annotation}

Proteins were predicted using the \texttt{Prot4EST} (version 3.0b)
\cite{wasmuth_prot4est:_2004}: First \texttt{blast} searches against a
rRNA-database, a mitochondrial database and against
uniref100\cite{pmid18836194} were preformed. Then results were used to
predict proteins directly (joining single high scoring pairs, and
thereby intorducing gaps and ambiguous bases if needed). Secondly
using the codon-usage from \texttt{blast}-predictions a simulated
transcriptome was generated, reverse translating the
\textit{B. malayi} proteom, as training-data-set for
\texttt{ESTscan}’s\cite{estscan} hidden Markov models.  If both
\texttt{blast}-based prediction and \texttt{ESTscan} failed, simply
the longest ORF is inferred.

\texttt{Blast}-based annotations were inferred using Annot8r (version
1.1.1) \cite{schmid_annot8r:_2008}: Searches were performed against
all sequences in uniref100 \cite{pmid18836194} being annotated with
GO-terms, EC-numbers and KEGG-parthways. Up to 10 (possibly
contradictory) annotations based on a bitscore cut-off of 55 were
obtained for each annotated database.  For comparison annotations were
obtained the same way for all \textit{Brugia malayi} proteins in
uniref100.

% Additionally domain-based annotation was
% obtained using InterProScan (iprscan command-line tool, version 4.6)
% \cite{pmid11590104}.

\texttt{SignalP V3.0} \cite{pmid17446895} was used to predict signal
peptide cleavage sites and signal anchor signatures.

\subsection*{SNP analysis}

As protein-prediction inferres gaps (e.g from sequencing errors) to
predict the most likely protein, not only start- and end-coordinates
of open reading frames (ORFs) had to be extracted from the output of
\texttt{Prot4EST}. We did this in a custom \texttt{perl}-script using
a \texttt{blast}-search with the nucleotide equivalent of the protein
as query and the raw sequence as subject. We obtain the
hit-coordinates as ORF-coordinates and imputed the
\texttt{blast}-query as corrected ORF-sequencs.

We mapped the raw reads against the the complete unigene set, with the
imputed sequences for those contigs with proteins predicted, using
ssaha2 (with parameters -kmer 13 -skip 3 -seeds 6 -score 100 -cmatch
10 -ckmer 6 -output sam -best 1).

\texttt{pileup}-files were produced using \texttt{samtools}
\cite{journals/bioinformatics/LiHWFRHMAD09}, discarding sequences
mapping to multiple regions with the best hit. VarScan
\cite{pmid19542151} (pileup2snp) was used with default parameters on
\texttt{pileup}-files. This output was further screened as described
in the results part of the manuscript.

% \subsection*{Respiration analysis}

% We extracted contigs annotated with ``GO:0009060 aerobic respiration''
% and ``GO:0009061 anaerobic respiration''

\subsection*{Gene-expression analysis}

For NlaIII-tag-sequencing total RNA was prepared as described above
from a worm from the Polish sampling site. A sequence-tag libray was
created following the protocol supplied by Illmina for this method.
Briefly after synthesis of cDNA on oligo(dt)-beads, this cDNA is
digested with the enzyme NlaIII (restriction site ``CATG''). After
ligation of an adaptor containing its restriction site the enzyme MmeI
cuts 17 bases downstream of its binding site generating a sequence tag
of in total 21 bases.

For 454 reads, read counts were obtained from the mapping to imputed
sequence described above. Tag-sequences were mapped using
\texttt{BWA}\cite{pmid19451168}. And read counts extracted using
\texttt{Samtools}.

The R-package DESeq\cite{pmid20979621} was used to normalize for
library-size and analyse statistical significance of differential
expression.

\subsection*{General coding methods}

The bulk of analysis (unless otherwise cited) presented in this paper
was carried out in R \cite{R_project} using custom scripts. We used a
method provided in the R-packages
Sweave\cite{lmucs-papers:Leisch:2002} and Weaver\cite{weaver} for
``reproducible research'' combining R and \TeX code in a single
file. All intermediate data files needed to compile the present
manuscript from data-sources are provided upon request. For
visualistaion we used the R-packages lattice\cite{lattice-book} and
ggplot2\cite{ggplot-book}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Competing interests}
The authors declare no competing interests.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgments}
\ifthenelse{\boolean{publ}}{\small}{} We thank Stephen Bridgett from
gene-pool sequencing service for general help with raw data and for
cross-contamination screening of libraries.

The work of EGH is funded by
Volkswagen Foundation, "F\"{o}rderinitiative Evolutionsbiologie".
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article} % Style BST file
  \bibliography{/home/ele/bibtex/master,/home/ele/bibtex/master2}
} % Bibliography file (usually '*.bib' )

%%%%%%%%%%%


\ifthenelse{\boolean{publ}}{\end{multicols}}{}
\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \subsection*{Figure 1 - rRNA-screening statistics}
 


  \subsection*{Figure 2 - Contig-coverage}

  \subsection*{Figure 3 - }

  \subsection*{Figure 4 - }

\subsection*{Figure 5 - }

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
  \subsection*{Table 1 - Trimming statistics}

<<echo=FALSE, results=tex>>=
print(xtable(TRIM, display=rep("d", times=7)), floating=FALSE)
@ 

\subsection*{Table 2 - Pre-assembly screening statistics}
<<echo=FALSE, results=tex>>=
print(xtable(pre.screen.tab), floating=FALSE)
@   

\subsection*{Table 3 - \texttt{Blast}-hits to protozoan rRNA in
  pre-assembly screening}
<<echo=FALSE, results=tex>>=
print(xtable(cerco), include.rownames=FALSE, floating=FALSE)
@   

\subsection*{Table 4 - Post-assembly host-screening }
  
<<echo=FALSE, results=tex>>=
digs <- matrix(c(0, 2), nrow=2 , ncol=6)
print(xtable(post.table, digits=digs), floating=FALSE)
@

\subsection*{Table 5 - Evolutionary conservation }
<<echo=FALSE, results=tex>>=
nov.tab <- rbind(summary.factor(contig.df$novel.50), 
                 summary.factor(contig.df$novel.80))[, c(1,3,4,2)]
rownames(nov.tab) <- c("bit.threshold.50", "bit.threshold.80")
print(xtable(nov.tab), floating=FALSE)

@ 

\subsection*{Table 6 - Protein prediction statistics}

<<echo=FALSE, results=tex>>=
print(xtable(imp.sum),
      floating=FALSE)
@


\subsection*{Table 7 - SNP summry statistics}

<<echo=FALSE, results=tex>>=

print(xtable(snp.summary),
      hline.after = c(-1,1),
      floating=FALSE)

@ 


\subsection*{Table 8 - GO-terms in positively selected}

<<echo=FALSE, results=tex>>=
mf <- mf.dn.ds.AC[, 5:8]
bp <- bp.dn.ds.AC[, 5:8]
cc <- cc.dn.ds.AC[, 5:8]
nhlines <- c(0, nrow(mf), nrow(mf) + nrow(bp), nrow(mf) + nrow(bp) + nrow(cc))

onto.tab <- xtable(rbind(mf, bp, cc))

print(onto.tab, tabular.environment="longtable",
      floating=FALSE,
      hline.after=nhlines, include.rownames=FALSE)
@ 

\subsection*{Table 9 - Correlation between read-counts in
  454-libraries and solexa-tags}

\newpage

\subsubsection*{Table 9 a - analysing all TUGs}

<<echo=FALSE, results=tex>>=

print(xtable(cor.count.tab, digits=3), floating=FALSE)

@ 
\subsubsection*{Table 9 b - analysing good-category contigs only}

<<echo=FALSE, results=tex>>=

print(xtable(cor.count.tab.MN, digits=3), floating=FALSE)

@ 

\subsubsection*{Table 9 c - analysing good-category contigs surely from \textit{A.crassus} only}

<<echo=FALSE, results=tex>>=

print(xtable(cor.count.tab.MN.AC, digits=3), floating=FALSE)

@ 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Additional Files}

File A\_crassus\_contigs\_full.csv lists all data computed on the contig
level, including sequences (raw, coding, imputed). File
A\_crassus\_contigs\_readable.csv lists only the metadata not including
sequences.

 

\end{bmcformat}
\end{document}
