%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    

% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{multirow}
\usepackage{longtable}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}


\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\usepackage{/usr/share/R/texmf/tex/latex/Sweave}
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \title{The transcriptome of the swimbladder-nematode
    Anguillicola crassus: Resources for an alien parasite}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Emanuel G Heitlinger\correspondingauthor$^{1,2}$%
       \email{Emanuel G Heitlinger\correspondingauthor - emanuelheitlinger@gmail.com}%
       Stephen Bridgett$^{3}$%
       \email{Stephen Bridgett- sbridget@staffmail.ed.ac.uk}%
       Anna Montazam$^{3}$%
       \email{Anna Montazam- Anna.Montazam@ed.ac.uk}%
       Horst Taraschewski$^1$%
       \email{Horst Taraschewski- dc20@rz.uni-karlsruhe.de}%
       and Mark Blaxter$^2$%
       \email{Mark Blaxter - mark.blaxter@ed.ac.uk}%
     }%
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \address{%
        \iid(1)Department of Ecology and Parasitology, Zoological
        Institute 1, University of Karlsruhe,%
        Kornblumenstrasse 13, Karlsruhe, Germany\\
        \iid(2)Institute of Evolutionary Biology, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
        \iid(3)The GenePool Sequencing Service, The Ashworth laboratories, The University of Edinburgh, King's Buildings Campus, Edinburgh, UK
      }%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
  \paragraph*{Background:} 
  \paragraph*{Results:} 
  \paragraph*{Conclusions:}
  Yeh!
  
\end{abstract}


      \ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
 

The nematode Anguillicola crassus Kuwahara, Niimi et Itagaki, 1974
\cite{kuwahara_Niimi_Itagaki_1974} is a parasite of freshwater eels of
the genus Anguilla, and adults localise to the swim bladder where they
feed on blood. Larvae are transmitted via crustacean intermediate
hosts. Originally endemic to East-Asian populations of the Japanese
eel (Anguilla japonica), A. crassus has attracted interest due to
recent anthropogenic expansion of its geographic and host ranges to
Europe and the European eel (Anguilla anguilla). Recorded for the
first time in 1982 in North-West Germany \cite{fischer_teichwirt},
where it was most likely introduced through the live-eel trade
\cite{koops_anguillicola-infestations_1989, koie_swimbladder_1991},
A. crassus has spread rapidly through populations of its newly
acquired host \cite{kirk_impact_2003}. At the present day it is found
in all An. anguilla populations except those in Iceland
\cite{kristmundsson_parasite_2007}. A. crassus can be regarded as a
model for invasive parasite introduction and spread
\cite{taraschewski_hosts_2007}.

A. crassus has a major impact on An. anguilla populations. In its
natural host in Asia infection prevalence and mean intensity of
infection are lower than in Europe \cite{mnderle_occurrence_2006},
where high prevalence (above 70\% \cite{wrtz_distribution_1998}) and
high infection intesities have been reported throughout the newly
colonized area \cite{lefebvre_anguillicolosis:_2004}. The virulence of
A .crassus in this new host has been attributed to an inadequate
immune response in An. anguilla \cite{knopf_swimbladder_2006}. While
the An.  japonica is capable of killing larvae of the parasite after
vaccination \cite{knopf_vaccination_2008} or under high infection
pressure \cite{heitlinger_massive_2009}, responses in An. anguilla
have hallmarks of pathology, including thickening of the swim bladder
wall \cite{wrtz_histopathological_2000}.  Interestingly host also
affects the adult size and life-history of the nematodes: In European
eels the nematodes are bigger and develop and reproduce faster
\cite{knopf_differences_2004}.


The genus Anguillicola is placed in the nematode suborder Spirurina
(clade III sensu \cite{blaxter_molecular_1998})
\cite{nadler_molecular_2007, wijov_evolutionary_2006}. The Spirurina
are exclusively parasitic and include important human pathogens (the
causative agents of filariases and ascariasis) as well as prominent
veterinary parasites. Molecular phylogenetic analyses place
Anguillicola in a clade of spirurine nematodes (Spirurina B of
[Laetsch et al submitted]) that have an freshwater or marine
intermediate host, but infect a wide range of carnivorous definitive
hosts. Spirurina B is sister to the main Spirurina C, including the
agents of filariases and ascariasis), and thus A. crassus may be used
as an outgroup taxon to understand the evolution of parasitic
phenotypes in these species.

Recent advances in sequencing technology (often termed Next Generation
Sequencing; NGS), provide the opportunity for rapid and cost-effective
generation of genome-scale data. The Roche 454 platform
\cite{pmid16056220} offers longer reads than other NGS technologies,
and thus is suited to de novo assembly of genome-scale data in
previously understudied species. Roche 454 data has particular
application in transcriptomics \cite{pmid20950480}. The difference in
the biology of A. crassus in An. japonica (coevolved) and An. anguilla
(recently captured) eel hosts likely results from an interaction
between different host and parasite responses, underpinned by
definitive differences in host genetics, and possible genetic
differentiation between the invading European and endemic Asian
parasites. As part of aprogramme to understand the invasiveness of
A. crassus in An. anguilla, we are investigating differences in gene
expression and genetic distinction between invading European and
endemic Asian A. crassus exposed to the two different host
species. Here we report on the generation of a reference transcriptome
for A. crassus based on Roche 454 data, and explore patterns of gene
expression and diversity.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results}


\subsection*{Sampling A. crassus}

One female worm and one male worm were sampled from an aquaculture
with height infection loads in Taiwan. An additonal female worm was
sampled from a stream with low infection pressure adjacent to the
aquaculture. All these worms were parsitising endemic
\textit{An. japonica}. A female worm and pool of L2 larval stages were
sampled from \textit{An. anguilla} in the river Rhein, one female worm
from a lake in Poland. All adult worms were filled with large amounts
of host-bood, therefore we anticipated abundand host-contamination in
sequencing and decided to sequence a liver sample of an unifected
\textit{An. japonica} for screening.


 \subsection*{Sequencing, trimming and pre-assembly screening}






A total of 756363 raw sequencing reads were
generated for \textit{A. crassus} (Table 1). These were trimmed for
base call quality, and filtered by length to give
585949 high-quality reads (spanning
100491819 bases). In the eel data-set
from 159370 raw reads 135072 were
assembled after basic quality screening.

We then screened the \textit{A. crassus} reads for contamination by
host (30071 matched previously
sequenced eel genes in our own \textit{An. anguilla} 454
transcriptome, which was partitioned in 10639
mRNA and 53 rRNA TUGs after
the nematode (181783 reads matched large
or small subunit nuclear or mitochondrial ribosomal RNA sequences of
A. crassus) (Table 1). In addition to fish mRNAs, we identified (and
removed) 5286 reads in the
library derived from the L2 nematodes that had significant similarity
to cercozoan (likely parasite) ribosomal RNA genes (Table 1).

\subsection*{Assembly}



We assembled the remaining 353055 reads (spanning
100491819 bases) using the combined assembler strategy
\cite{pmid20950480} and Roche 454 GSassembler (version 2.6) and MIRA
(version 3.21) \cite{miraEST}. From this we derived 13851
contigs that were supported by both assembly algorithms,
3745 contigs only supported by one of the assembly
algorithms and 22591 singletons that were not assembled by
either approach (Table 2). When scored by matches to known genes, the
contigs supported by both assemblers are of the highest credibility,
and this set is thus termed the high credibility assembly
(highCA). Those with evidence from only one assembler and the
singletons are of lower credibility (lowCA). These datasets are the
most parsimonious (having the smallest size) for their quality
(covering the largest amount of sequence in reference
transcritomes). In the highCA parsimony and low redundancy is
prioritized, while in the complete assembly (highCA plus lowCA)
completeness is proiritized. The 40187 sequences (contig consensuses
and singletons) in the complete assembly are referred to below as
tentatively unique genes (TUGs).




We screened the complete assembly for residual host contamination, and
identified
40187
  TUGs that had higher, significant similarity to eel (and chordate)
  sequences (our 454 ESTs and EMBLBank Chordata proteins) than to
  nematode sequences \cite{pmid21550347}.

  Given our prior identification of cercozoan ribosomal RNAs, we also
  screened the complete assembly for contamination with other
  transcriptomes, and found 365 TUGs with hits
  to fungi (e.g Ajellomycetaceae, 53 hits),
  672 TUGs whith hits to plants and
  2002 hits to Protists (e.g. Trypanosomatidae,
  26 hits and Vahlkampfiidae, 38 hits), Bacteria (mostly
  Proteobacteria, 484 hits) and Viruses (see also additional figure
  phylum\_plots.png).

  No hits were found to Wolbachia or related Bacteria known as
  symbionts of Ecdyosozoans.

  Our assembly thus has 32518 TUGs,
  spanning 154052 bases
  (of which 11371 are
  highCA-derived, and span
  154052 bases) that are
  likely to derive from of \textit{A. crassus}.

\subsection*{Protein prediction}

%% read P4EST output processed with
%% coordinates_from_p4e.pl

%% SNP calling from VARSCAN output




%%% annotation.Rnw --- 

%% Author: emanuelheitlinger@gmail.com
